<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="generator" content="pandoc" />
    <meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

    <meta
      name="author"
      content="Brendon Smith br3ndonland Source code on GitHub license"
    />

    <title>R sample code</title>

    <script src="site_libs/header-attrs-2.3/header-attrs.js"></script>
    <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
    <script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
    <script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
    <script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
    <script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
    <link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
    <script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
    <script src="site_libs/navigation-1.1/tabsets.js"></script>
    <script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>

    <style type="text/css">
      code {
        white-space: pre;
      }
    </style>
    <style type="text/css" data-origin="pandoc">
      pre > code.sourceCode {
        white-space: pre;
        position: relative;
      }
      pre > code.sourceCode > span {
        display: inline-block;
        line-height: 1.25;
      }
      pre > code.sourceCode > span:empty {
        height: 1.2em;
      }
      code.sourceCode > span {
        color: inherit;
        text-decoration: inherit;
      }
      div.sourceCode {
        margin: 1em 0;
      }
      pre.sourceCode {
        margin: 0;
      }
      @media screen {
        div.sourceCode {
          overflow: auto;
        }
      }
      @media print {
        pre > code.sourceCode {
          white-space: pre-wrap;
        }
        pre > code.sourceCode > span {
          text-indent: -5em;
          padding-left: 5em;
        }
      }
      pre.numberSource code {
        counter-reset: source-line 0;
      }
      pre.numberSource code > span {
        position: relative;
        left: -4em;
        counter-increment: source-line;
      }
      pre.numberSource code > span > a:first-child::before {
        content: counter(source-line);
        position: relative;
        left: -1em;
        text-align: right;
        vertical-align: baseline;
        border: none;
        display: inline-block;
        -webkit-touch-callout: none;
        -webkit-user-select: none;
        -khtml-user-select: none;
        -moz-user-select: none;
        -ms-user-select: none;
        user-select: none;
        padding: 0 4px;
        width: 4em;
        color: #aaaaaa;
      }
      pre.numberSource {
        margin-left: 3em;
        border-left: 1px solid #aaaaaa;
        padding-left: 4px;
      }
      div.sourceCode {
      }
      @media screen {
        pre > code.sourceCode > span > a:first-child::before {
          text-decoration: underline;
        }
      }
      code span.al {
        color: #ff0000;
        font-weight: bold;
      } /* Alert */
      code span.an {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Annotation */
      code span.at {
        color: #7d9029;
      } /* Attribute */
      code span.bn {
        color: #40a070;
      } /* BaseN */
      code span.bu {
      } /* BuiltIn */
      code span.cf {
        color: #007020;
        font-weight: bold;
      } /* ControlFlow */
      code span.ch {
        color: #4070a0;
      } /* Char */
      code span.cn {
        color: #880000;
      } /* Constant */
      code span.co {
        color: #60a0b0;
        font-style: italic;
      } /* Comment */
      code span.cv {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* CommentVar */
      code span.do {
        color: #ba2121;
        font-style: italic;
      } /* Documentation */
      code span.dt {
        color: #902000;
      } /* DataType */
      code span.dv {
        color: #40a070;
      } /* DecVal */
      code span.er {
        color: #ff0000;
        font-weight: bold;
      } /* Error */
      code span.ex {
      } /* Extension */
      code span.fl {
        color: #40a070;
      } /* Float */
      code span.fu {
        color: #06287e;
      } /* Function */
      code span.im {
      } /* Import */
      code span.in {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Information */
      code span.kw {
        color: #007020;
        font-weight: bold;
      } /* Keyword */
      code span.op {
        color: #666666;
      } /* Operator */
      code span.ot {
        color: #007020;
      } /* Other */
      code span.pp {
        color: #bc7a00;
      } /* Preprocessor */
      code span.sc {
        color: #4070a0;
      } /* SpecialChar */
      code span.ss {
        color: #bb6688;
      } /* SpecialString */
      code span.st {
        color: #4070a0;
      } /* String */
      code span.va {
        color: #19177c;
      } /* Variable */
      code span.vs {
        color: #4070a0;
      } /* VerbatimString */
      code span.wa {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Warning */
    </style>
    <script>
      // apply pandoc div.sourceCode style to pre.sourceCode instead
      ;(function () {
        var sheets = document.styleSheets
        for (var i = 0; i < sheets.length; i++) {
          if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue
          try {
            var rules = sheets[i].cssRules
          } catch (e) {
            continue
          }
          for (var j = 0; j < rules.length; j++) {
            var rule = rules[j]
            // check if there is a div.sourceCode rule
            if (
              rule.type !== rule.STYLE_RULE ||
              rule.selectorText !== "div.sourceCode"
            )
              continue
            var style = rule.style.cssText
            // check if color or background-color is set
            if (rule.style.color === "" && rule.style.backgroundColor === "")
              continue
            // replace div.sourceCode by a pre.sourceCode rule
            sheets[i].deleteRule(j)
            sheets[i].insertRule("pre.sourceCode{" + style + "}", j)
          }
        }
      })()
    </script>
    <style type="text/css">
      pre:not([class]) {
        background-color: white;
      }
    </style>

    <style type="text/css">
      h1 {
        font-size: 34px;
      }
      h1.title {
        font-size: 38px;
      }
      h2 {
        font-size: 30px;
      }
      h3 {
        font-size: 24px;
      }
      h4 {
        font-size: 18px;
      }
      h5 {
        font-size: 16px;
      }
      h6 {
        font-size: 12px;
      }
      .table th:not([align]) {
        text-align: left;
      }
    </style>

    <style type="text/css">
      .main-container {
        max-width: 940px;
        margin-left: auto;
        margin-right: auto;
      }
      code {
        color: inherit;
        background-color: rgba(0, 0, 0, 0.04);
      }
      img {
        max-width: 100%;
      }
      .tabbed-pane {
        padding-top: 12px;
      }
      .html-widget {
        margin-bottom: 20px;
      }
      button.code-folding-btn:focus {
        outline: none;
      }
      summary {
        display: list-item;
      }
    </style>

    <style type="text/css">
      /* padding for bootstrap navbar */
      body {
        padding-top: 51px;
        padding-bottom: 40px;
      }
      /* offset scroll position for anchor links (for fixed navbar)  */
      .section h1 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h2 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h3 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h4 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h5 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h6 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .dropdown-submenu {
        position: relative;
      }
      .dropdown-submenu > .dropdown-menu {
        top: 0;
        left: 100%;
        margin-top: -6px;
        margin-left: -1px;
        border-radius: 0 6px 6px 6px;
      }
      .dropdown-submenu:hover > .dropdown-menu {
        display: block;
      }
      .dropdown-submenu > a:after {
        display: block;
        content: " ";
        float: right;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        border-width: 5px 0 5px 5px;
        border-left-color: #cccccc;
        margin-top: 5px;
        margin-right: -10px;
      }
      .dropdown-submenu:hover > a:after {
        border-left-color: #ffffff;
      }
      .dropdown-submenu.pull-left {
        float: none;
      }
      .dropdown-submenu.pull-left > .dropdown-menu {
        left: -100%;
        margin-left: 10px;
        border-radius: 6px 0 6px 6px;
      }
    </style>

    <script>
      // manage active state of menu based on current page
      $(document).ready(function () {
        // active menu anchor
        href = window.location.pathname
        href = href.substr(href.lastIndexOf("/") + 1)
        if (href === "") href = "index.html"
        var menuAnchor = $('a[href="' + href + '"]')

        // mark it active
        menuAnchor.parent().addClass("active")

        // if it's got a parent navbar menu mark it active as well
        menuAnchor.closest("li.dropdown").addClass("active")
      })
    </script>

    <!-- tabsets -->

    <style type="text/css">
      .tabset-dropdown > .nav-tabs {
        display: inline-table;
        max-height: 500px;
        min-height: 44px;
        overflow-y: auto;
        background: white;
        border: 1px solid #ddd;
        border-radius: 4px;
      }

      .tabset-dropdown > .nav-tabs > li.active:before {
        content: "";
        font-family: "Glyphicons Halflings";
        display: inline-block;
        padding: 10px;
        border-right: 1px solid #ddd;
      }

      .tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
        content: "&#xe258;";
        border: none;
      }

      .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
        content: "";
        font-family: "Glyphicons Halflings";
        display: inline-block;
        padding: 10px;
        border-right: 1px solid #ddd;
      }

      .tabset-dropdown > .nav-tabs > li.active {
        display: block;
      }

      .tabset-dropdown > .nav-tabs > li > a,
      .tabset-dropdown > .nav-tabs > li > a:focus,
      .tabset-dropdown > .nav-tabs > li > a:hover {
        border: none;
        display: inline-block;
        border-radius: 4px;
        background-color: transparent;
      }

      .tabset-dropdown > .nav-tabs.nav-tabs-open > li {
        display: block;
        float: none;
      }

      .tabset-dropdown > .nav-tabs > li {
        display: none;
      }
    </style>

    <!-- code folding -->

    <style type="text/css">
      #TOC {
        margin: 25px 0px 20px 0px;
      }
      @media (max-width: 768px) {
        #TOC {
          position: relative;
          width: 100%;
        }
      }

      @media print {
        .toc-content {
          /* see https://github.com/w3c/csswg-drafts/issues/4434 */
          float: right;
        }
      }

      .toc-content {
        padding-left: 30px;
        padding-right: 40px;
      }

      div.main-container {
        max-width: 1200px;
      }

      div.tocify {
        width: 20%;
        max-width: 260px;
        max-height: 85%;
      }

      @media (min-width: 768px) and (max-width: 991px) {
        div.tocify {
          width: 25%;
        }
      }

      @media (max-width: 767px) {
        div.tocify {
          width: 100%;
          max-width: none;
        }
      }

      .tocify ul,
      .tocify li {
        line-height: 20px;
      }

      .tocify-subheader .tocify-item {
        font-size: 0.9em;
      }

      .tocify .list-group-item {
        border-radius: 0px;
      }
    </style>
  </head>

  <body>
    <div class="container-fluid main-container">
      <!-- setup 3col/9col grid for toc_float and main content  -->
      <div class="row-fluid">
        <div class="col-xs-12 col-sm-4 col-md-3">
          <div id="TOC" class="tocify"></div>
        </div>

        <div class="toc-content col-xs-12 col-sm-8 col-md-9">
          <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
            <div class="container">
              <div class="navbar-header">
                <button
                  type="button"
                  class="navbar-toggle collapsed"
                  data-toggle="collapse"
                  data-target="#navbar"
                >
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="index.html">R guide</a>
              </div>
              <div id="navbar" class="navbar-collapse collapse">
                <ul class="nav navbar-nav">
                  <li>
                    <a href="/R-guide/">Guide</a>
                  </li>
                  <li>
                    <a href="/R-guide/code">Sample code</a>
                  </li>
                  <li>
                    <a href="https://github.com/br3ndonland/R-guide">GitHub</a>
                  </li>
                </ul>
                <ul class="nav navbar-nav navbar-right"></ul>
              </div>
              <!--/.nav-collapse -->
            </div>
            <!--/.container -->
          </div>
          <!--/.navbar -->

          <div class="fluid-row" id="header">
            <h1 class="title toc-ignore">R sample code</h1>
            <h4 class="author">
              Brendon Smith<br /><a href="https://github.com/br3ndonland"
                >br3ndonland</a
              ><br />Source code on
              <a href="https://github.com/br3ndonland/R-guide">GitHub</a
              ><br /><a href="https://choosealicense.com/"
                ><img
                  src="https://img.shields.io/badge/license-MIT-blue.svg?longCache=true&amp;style=for-the-badge"
                  alt="license"
              /></a>
            </h4>
          </div>

          <div id="data-structures" class="section level2">
            <h2>Data structures</h2>
            <div id="data-frame" class="section level3">
              <h3>Data frame</h3>
              <div id="example-from-cookbook-for-r" class="section level4">
                <h4>
                  Example from
                  <a
                    href="http://www.cookbook-r.com/Manipulating_data/Reordering_the_columns_in_a_data_frame/"
                    >Cookbook for R</a
                  >
                </h4>
                <div id="sample-data-frame" class="section level5">
                  <h5>Sample data frame</h5>
                  <div class="sourceCode" id="cb1">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a>sample_data &lt;-</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="st">  </span><span class="kw">data.frame</span> (</span>
<span id="cb1-3"><a href="#cb1-3"></a>    <span class="dt">id =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,</span>
<span id="cb1-4"><a href="#cb1-4"></a>    <span class="dt">weight =</span> <span class="kw">c</span>(<span class="dv">20</span>, <span class="dv">27</span>, <span class="dv">24</span>),</span>
<span id="cb1-5"><a href="#cb1-5"></a>    <span class="dt">size =</span> <span class="kw">c</span>(<span class="st">&#39;small&#39;</span>, <span class="st">&#39;large&#39;</span>, <span class="st">&#39;medium&#39;</span>)</span>
<span id="cb1-6"><a href="#cb1-6"></a>  )</span>
<span id="cb1-7"><a href="#cb1-7"></a>sample_data</span></code></pre>
                  </div>
                  <pre><code>##   id weight   size
## 1  1     20  small
## 2  2     27  large
## 3  3     24 medium</code></pre>
                </div>
                <div
                  id="sort-data-frame-by-column-number"
                  class="section level5"
                >
                  <h5>Sort data frame by column number</h5>
                  <div class="sourceCode" id="cb3">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>sample_data[<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">2</span>)]</span></code></pre>
                  </div>
                  <pre><code>##   id   size weight
## 1  1  small     20
## 2  2  large     27
## 3  3 medium     24</code></pre>
                  <div class="sourceCode" id="cb5">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># To actually change &#39;data&#39;, run:</span></span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="co"># data &lt;- data[c(1,3,2)]</span></span></code></pre>
                  </div>
                </div>
                <div id="sort-data-frame-by-column-name" class="section level5">
                  <h5>Sort data frame by column name</h5>
                  <div class="sourceCode" id="cb6">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>sample_data[<span class="kw">c</span>(<span class="st">&#39;size&#39;</span>, <span class="st">&#39;id&#39;</span>, <span class="st">&#39;weight&#39;</span>)]</span></code></pre>
                  </div>
                  <pre><code>##     size id weight
## 1  small  1     20
## 2  large  2     27
## 3 medium  3     24</code></pre>
                </div>
              </div>
              <div id="example-from-stack-overflow" class="section level4">
                <h4>
                  Example from
                  <a
                    href="http://stackoverflow.com/questions/1296646/how-to-sort-a-dataframe-by-columns-in-r"
                    >Stack Overflow</a
                  >
                </h4>
                <div id="sample-data-frame-1" class="section level5">
                  <h5>Sample data frame</h5>
                  <div class="sourceCode" id="cb8">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>dd &lt;-</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="st">  </span><span class="kw">data.frame</span>(</span>
<span id="cb8-3"><a href="#cb8-3"></a>    <span class="dt">b =</span> <span class="kw">factor</span>(</span>
<span id="cb8-4"><a href="#cb8-4"></a>      <span class="kw">c</span>(<span class="st">&#39;Hi&#39;</span>, <span class="st">&#39;Med&#39;</span>, <span class="st">&#39;Hi&#39;</span>, <span class="st">&#39;Low&#39;</span>),</span>
<span id="cb8-5"><a href="#cb8-5"></a>      <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&#39;Low&#39;</span>, <span class="st">&#39;Med&#39;</span>, <span class="st">&#39;Hi&#39;</span>),</span>
<span id="cb8-6"><a href="#cb8-6"></a>      <span class="dt">ordered =</span> <span class="ot">TRUE</span></span>
<span id="cb8-7"><a href="#cb8-7"></a>    ),</span>
<span id="cb8-8"><a href="#cb8-8"></a>    <span class="dt">x =</span> <span class="kw">c</span>(<span class="st">&#39;A&#39;</span>, <span class="st">&#39;D&#39;</span>, <span class="st">&#39;A&#39;</span>, <span class="st">&#39;C&#39;</span>),</span>
<span id="cb8-9"><a href="#cb8-9"></a>    <span class="dt">y =</span> <span class="kw">c</span>(<span class="dv">8</span>, <span class="dv">3</span>, <span class="dv">9</span>, <span class="dv">9</span>),</span>
<span id="cb8-10"><a href="#cb8-10"></a>    <span class="dt">z =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">2</span>)</span>
<span id="cb8-11"><a href="#cb8-11"></a>  )</span>
<span id="cb8-12"><a href="#cb8-12"></a>dd</span></code></pre>
                  </div>
                  <pre><code>##     b x y z
## 1  Hi A 8 1
## 2 Med D 3 1
## 3  Hi A 9 1
## 4 Low C 9 2</code></pre>
                </div>
                <div id="sort-data-frame-by-column" class="section level5">
                  <h5>Sort data frame by column</h5>
                  <p>
                    Sorting the frame by column z (descending) then by column
                    b(ascending)
                  </p>
                  <div class="sourceCode" id="cb10">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>dd[<span class="kw">with</span>(dd, <span class="kw">order</span>(<span class="op">-</span>z, b)),]</span></code></pre>
                  </div>
                  <pre><code>##     b x y z
## 4 Low C 9 2
## 2 Med D 3 1
## 1  Hi A 8 1
## 3  Hi A 9 1</code></pre>
                </div>
                <div
                  id="sort-data-frame-by-column-index"
                  class="section level5"
                >
                  <h5>Sort data frame by column index</h5>
                  <blockquote>
                    <p>
                      Edit some 2+ years later: It was just asked how to do this
                      by column index. The answer is to simply pass the desired
                      sorting column(s) to the order() function rather than
                      using the name of the column (and with() for easier/more
                      direct access).
                    </p>
                  </blockquote>
                  <div class="sourceCode" id="cb12">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>dd[<span class="kw">order</span>(<span class="op">-</span>dd[, <span class="dv">4</span>], dd[, <span class="dv">1</span>]),]</span></code></pre>
                  </div>
                  <pre><code>##     b x y z
## 4 Low C 9 2
## 2 Med D 3 1
## 1  Hi A 8 1
## 3  Hi A 9 1</code></pre>
                </div>
              </div>
            </div>
          </div>
          <div id="statistical-models" class="section level2">
            <h2>Statistical models</h2>
            <div id="t-test" class="section level3">
              <h3>T-test</h3>
              <div id="atheroma-thickness" class="section level4">
                <h4>Atheroma thickness</h4>
                <p>
                  Atheroma thickness data from Smith BW J Food Res 2013 (doi:
                  <a href="https://dx.doi.org/10.5539%2Fjfr.v2n1p168"
                    >10.5539/jfr.v2n1p168</a
                  >).
                </p>
                <div id="import-dataset" class="section level5">
                  <h5>Import dataset</h5>
                  <div class="sourceCode" id="cb14">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>jfr2013_atheroma &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;data/t_test_jfr2013_atheroma.csv&#39;</span>,</span>
<span id="cb14-2"><a href="#cb14-2"></a>                               <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb14-3"><a href="#cb14-3"></a>                               <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span>)</span>
<span id="cb14-4"><a href="#cb14-4"></a><span class="kw">attach</span>(jfr2013_atheroma)</span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="kw">names</span>(jfr2013_atheroma)</span></code></pre>
                  </div>
                  <pre><code>## [1] &quot;diet&quot;      &quot;thickness&quot;</code></pre>
                </div>
                <div id="perform-t-test" class="section level5">
                  <h5>Perform t-test</h5>
                  <ul>
                    <li>
                      The default assumes unequal variance and applies the
                      Satterthwaite/Welch df modification.
                    </li>
                    <li>
                      var.equal = TRUE is used to specify equal variances and a
                      pooled variance estimate.
                    </li>
                  </ul>
                  <div class="sourceCode" id="cb16">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="kw">t.test</span>(</span>
<span id="cb16-2"><a href="#cb16-2"></a>  jfr2013_atheroma<span class="op">$</span>thickness <span class="op">~</span><span class="st"> </span>jfr2013_atheroma<span class="op">$</span>diet,</span>
<span id="cb16-3"><a href="#cb16-3"></a>  <span class="dt">paired =</span> <span class="ot">FALSE</span>,</span>
<span id="cb16-4"><a href="#cb16-4"></a>  <span class="dt">alternative =</span> <span class="st">&#39;two.sided&#39;</span>,</span>
<span id="cb16-5"><a href="#cb16-5"></a>  <span class="dt">var.equal =</span> <span class="ot">TRUE</span>,</span>
<span id="cb16-6"><a href="#cb16-6"></a>  <span class="dt">conf.level =</span> <span class="fl">0.95</span></span>
<span id="cb16-7"><a href="#cb16-7"></a>)</span></code></pre>
                  </div>
                  <pre><code>##
##  Two Sample t-test
##
## data:  jfr2013_atheroma$thickness by jfr2013_atheroma$diet
## t = 1.4364, df = 15, p-value = 0.1714
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -43.34361 222.51028
## sample estimates:
## mean in group 0.1 mean in group 0.4
##          356.2500          266.6667</code></pre>
                </div>
                <div id="test-assumptions" class="section level5">
                  <h5>Test assumptions</h5>
                  <p>
                    The t test uses raw data, not residuals, so assumptions are
                    tested using raw data.
                  </p>
                  <div id="normality" class="section level6">
                    <h6>Normality</h6>
                    <ul>
                      <li>
                        The RColorBrewer package is loaded for additional
                        colors, and a palette is added to R. Use brackets after
                        vector name for specific colors when creating plots (for
                        example: <code>col=set1[4:14]</code>).
                      </li>
                      <li>
                        Legend can also be manually positioned with xy
                        coordinates (numbers from 0,0 at bottom left of whole
                        plot to 1,1 at top right)
                      </li>
                    </ul>
                    <div class="sourceCode" id="cb18">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># Shapiro-Wilk W test</span></span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="kw">shapiro.test</span>(thickness)</span></code></pre>
                    </div>
                    <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  thickness
## W = 0.91231, p-value = 0.1094</code></pre>
                    <div class="sourceCode" id="cb20">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="co"># Normal Quantile-Quantile Plot</span></span>
<span id="cb20-2"><a href="#cb20-2"></a><span class="co"># Set graphical parameters</span></span>
<span id="cb20-3"><a href="#cb20-3"></a><span class="kw">par</span>(</span>
<span id="cb20-4"><a href="#cb20-4"></a>  <span class="dt">cex =</span> <span class="dv">1</span>,</span>
<span id="cb20-5"><a href="#cb20-5"></a>  <span class="dt">cex.main =</span> <span class="dv">3</span>,</span>
<span id="cb20-6"><a href="#cb20-6"></a>  <span class="dt">cex.lab =</span> <span class="dv">2</span>,</span>
<span id="cb20-7"><a href="#cb20-7"></a>  <span class="dt">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb20-8"><a href="#cb20-8"></a>  <span class="dt">cex.axis =</span> <span class="fl">1.5</span>,</span>
<span id="cb20-9"><a href="#cb20-9"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb20-10"><a href="#cb20-10"></a>  <span class="dt">lend =</span> <span class="dv">0</span>,</span>
<span id="cb20-11"><a href="#cb20-11"></a>  <span class="dt">ljoin =</span> <span class="dv">0</span>,</span>
<span id="cb20-12"><a href="#cb20-12"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb20-13"><a href="#cb20-13"></a>  <span class="dt">pch =</span> <span class="dv">1</span>,</span>
<span id="cb20-14"><a href="#cb20-14"></a>  <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb20-15"><a href="#cb20-15"></a>)</span>
<span id="cb20-16"><a href="#cb20-16"></a></span>
<span id="cb20-17"><a href="#cb20-17"></a><span class="co"># Load RColorBrewer and set palette</span></span>
<span id="cb20-18"><a href="#cb20-18"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb20-19"><a href="#cb20-19"></a>set1 &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&#39;Set1&#39;</span>)</span>
<span id="cb20-20"><a href="#cb20-20"></a></span>
<span id="cb20-21"><a href="#cb20-21"></a><span class="co"># Create vector for categories</span></span>
<span id="cb20-22"><a href="#cb20-22"></a>categories &lt;-<span class="st"> </span><span class="kw">factor</span>(jfr2013_atheroma<span class="op">$</span>diet)</span>
<span id="cb20-23"><a href="#cb20-23"></a></span>
<span id="cb20-24"><a href="#cb20-24"></a><span class="co"># Create plot</span></span>
<span id="cb20-25"><a href="#cb20-25"></a><span class="kw">qqnorm</span>(</span>
<span id="cb20-26"><a href="#cb20-26"></a>  jfr2013_atheroma<span class="op">$</span>thickness,</span>
<span id="cb20-27"><a href="#cb20-27"></a>  <span class="dt">ylab =</span> <span class="kw">expression</span>(<span class="kw">bold</span>(<span class="kw">paste</span>(</span>
<span id="cb20-28"><a href="#cb20-28"></a>    <span class="st">&#39;Thickness, &#39;</span>, mu, <span class="st">&#39;m&#39;</span></span>
<span id="cb20-29"><a href="#cb20-29"></a>  ))),</span>
<span id="cb20-30"><a href="#cb20-30"></a>  <span class="dt">axes =</span> <span class="ot">FALSE</span>,</span>
<span id="cb20-31"><a href="#cb20-31"></a>  <span class="dt">col =</span> set1[categories],</span>
<span id="cb20-32"><a href="#cb20-32"></a>  <span class="dt">lwd =</span> <span class="dv">4</span>,</span>
<span id="cb20-33"><a href="#cb20-33"></a>  <span class="dt">cex =</span> <span class="dv">2</span></span>
<span id="cb20-34"><a href="#cb20-34"></a>)</span>
<span id="cb20-35"><a href="#cb20-35"></a><span class="co"># Add normal line</span></span>
<span id="cb20-36"><a href="#cb20-36"></a><span class="kw">qqline</span>(jfr2013_atheroma<span class="op">$</span>thickness, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb20-37"><a href="#cb20-37"></a><span class="co"># Add border to complete axes</span></span>
<span id="cb20-38"><a href="#cb20-38"></a><span class="kw">box</span>(<span class="dt">which =</span> <span class="st">&#39;plot&#39;</span>, <span class="dt">bty =</span> <span class="st">&#39;o&#39;</span>, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb20-39"><a href="#cb20-39"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>) <span class="co"># x axis</span></span>
<span id="cb20-40"><a href="#cb20-40"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>) <span class="co"># y axis</span></span>
<span id="cb20-41"><a href="#cb20-41"></a><span class="co"># Add legend</span></span>
<span id="cb20-42"><a href="#cb20-42"></a><span class="kw">legend</span>(</span>
<span id="cb20-43"><a href="#cb20-43"></a>  <span class="st">&#39;topleft&#39;</span>,</span>
<span id="cb20-44"><a href="#cb20-44"></a>  <span class="dt">legend =</span> <span class="kw">levels</span>(categories),</span>
<span id="cb20-45"><a href="#cb20-45"></a>  <span class="dt">box.lwd =</span> <span class="dv">3</span>,</span>
<span id="cb20-46"><a href="#cb20-46"></a>  <span class="dt">pch =</span> <span class="fl">1.5</span>,</span>
<span id="cb20-47"><a href="#cb20-47"></a>  <span class="dt">cex =</span> <span class="fl">1.75</span>,</span>
<span id="cb20-48"><a href="#cb20-48"></a>  <span class="dt">pt.cex =</span> <span class="dv">2</span>,</span>
<span id="cb20-49"><a href="#cb20-49"></a>  <span class="dt">pt.lwd =</span> <span class="dv">4</span>,</span>
<span id="cb20-50"><a href="#cb20-50"></a>  <span class="dt">col =</span> set1</span>
<span id="cb20-51"><a href="#cb20-51"></a>)</span></code></pre>
                    </div>
                    <p>
                      <img
                        src="code_files/figure-html/code-ttest-atheroma-03-assumptions-01-1.png"
                        width="672"
                      />
                    </p>
                  </div>
                  <div id="equality-of-variances" class="section level6">
                    <h6>Equality of variances</h6>
                    <div class="sourceCode" id="cb21">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="kw">var.test</span>(jfr2013_atheroma<span class="op">$</span>thickness <span class="op">~</span><span class="st"> </span>jfr2013_atheroma<span class="op">$</span>diet)</span></code></pre>
                    </div>
                    <pre><code>##
##  F test to compare two variances
##
## data:  jfr2013_atheroma$thickness by jfr2013_atheroma$diet
## F = 1.8725, num df = 7, denom df = 8, p-value = 0.3983
## alternative hypothesis: true ratio of variances is not equal to 1
## 95 percent confidence interval:
##  0.4134808 9.1738855
## sample estimates:
## ratio of variances
##           1.872473</code></pre>
                  </div>
                </div>
                <div id="boxplot" class="section level5">
                  <h5>Boxplot</h5>
                  <ul>
                    <li>
                      <strong>Parameters (<code>par</code>):</strong> Graphical
                      parameters need to be set again because of code chunk
                      separation in RMarkdown. Normally they would only be set
                      once per R session.
                    </li>
                    <li>
                      <strong>X axis:</strong> The default axes will usually
                      detect categorical data and label axes accordingly, but
                      categories will need to be specified for axes created
                      separately. It is easiest to create a separate vector with
                      the categories, and then reference it when creating the
                      plot. Labels can also be specified with
                      <code>labels=c('group1', 'group2')</code>.
                    </li>
                    <li>
                      <strong>Y axis:</strong> Note that it does not always
                      start at zero, but encompasses the data range with some
                      padding.
                    </li>
                  </ul>
                  <div class="sourceCode" id="cb23">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="co"># Set graphical parameters</span></span>
<span id="cb23-2"><a href="#cb23-2"></a><span class="kw">par</span>(</span>
<span id="cb23-3"><a href="#cb23-3"></a>  <span class="dt">cex =</span> <span class="dv">1</span>,</span>
<span id="cb23-4"><a href="#cb23-4"></a>  <span class="dt">cex.main =</span> <span class="dv">3</span>,</span>
<span id="cb23-5"><a href="#cb23-5"></a>  <span class="dt">cex.lab =</span> <span class="dv">2</span>,</span>
<span id="cb23-6"><a href="#cb23-6"></a>  <span class="dt">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb23-7"><a href="#cb23-7"></a>  <span class="dt">cex.axis =</span> <span class="fl">1.5</span>,</span>
<span id="cb23-8"><a href="#cb23-8"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb23-9"><a href="#cb23-9"></a>  <span class="dt">lend =</span> <span class="dv">0</span>,</span>
<span id="cb23-10"><a href="#cb23-10"></a>  <span class="dt">ljoin =</span> <span class="dv">0</span>,</span>
<span id="cb23-11"><a href="#cb23-11"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb23-12"><a href="#cb23-12"></a>  <span class="dt">pch =</span> <span class="dv">1</span>,</span>
<span id="cb23-13"><a href="#cb23-13"></a>  <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb23-14"><a href="#cb23-14"></a>)</span>
<span id="cb23-15"><a href="#cb23-15"></a></span>
<span id="cb23-16"><a href="#cb23-16"></a><span class="co"># Create boxplot</span></span>
<span id="cb23-17"><a href="#cb23-17"></a><span class="kw">boxplot</span>(</span>
<span id="cb23-18"><a href="#cb23-18"></a>  jfr2013_atheroma<span class="op">$</span>thickness <span class="op">~</span><span class="st"> </span>jfr2013_atheroma<span class="op">$</span>diet,</span>
<span id="cb23-19"><a href="#cb23-19"></a>  <span class="dt">main =</span> <span class="st">&#39;Atheroma Thickness&#39;</span>,</span>
<span id="cb23-20"><a href="#cb23-20"></a>  <span class="dt">xlab =</span> <span class="st">&#39;Diet&#39;</span>,</span>
<span id="cb23-21"><a href="#cb23-21"></a>  <span class="dt">ylab =</span> <span class="kw">expression</span>(<span class="kw">bold</span>(<span class="kw">paste</span>(</span>
<span id="cb23-22"><a href="#cb23-22"></a>    <span class="st">&#39;Thickness, &#39;</span>, mu, <span class="st">&#39;m&#39;</span></span>
<span id="cb23-23"><a href="#cb23-23"></a>  ))),</span>
<span id="cb23-24"><a href="#cb23-24"></a>  <span class="dt">axes =</span> <span class="ot">FALSE</span>,</span>
<span id="cb23-25"><a href="#cb23-25"></a>  <span class="dt">lwd =</span> <span class="dv">4</span>,</span>
<span id="cb23-26"><a href="#cb23-26"></a>  <span class="dt">medlwd =</span> <span class="dv">7</span>,</span>
<span id="cb23-27"><a href="#cb23-27"></a>  <span class="dt">outlwd =</span> <span class="dv">4</span>,</span>
<span id="cb23-28"><a href="#cb23-28"></a>  <span class="dt">cex =</span> <span class="dv">2</span>,</span>
<span id="cb23-29"><a href="#cb23-29"></a>  <span class="dt">col =</span> set1</span>
<span id="cb23-30"><a href="#cb23-30"></a>)</span>
<span id="cb23-31"><a href="#cb23-31"></a></span>
<span id="cb23-32"><a href="#cb23-32"></a><span class="co"># Add border to complete axes</span></span>
<span id="cb23-33"><a href="#cb23-33"></a><span class="kw">box</span>(<span class="dt">which =</span> <span class="st">&#39;plot&#39;</span>, <span class="dt">bty =</span> <span class="st">&#39;l&#39;</span>, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb23-34"><a href="#cb23-34"></a></span>
<span id="cb23-35"><a href="#cb23-35"></a><span class="co"># Create vector with means of each group</span></span>
<span id="cb23-36"><a href="#cb23-36"></a>means &lt;-<span class="st"> </span><span class="kw">aggregate</span>(thickness <span class="op">~</span><span class="st"> </span>diet, jfr2013_atheroma, mean)</span>
<span id="cb23-37"><a href="#cb23-37"></a><span class="co"># Add points for group means to boxplot</span></span>
<span id="cb23-38"><a href="#cb23-38"></a><span class="kw">points</span>(</span>
<span id="cb23-39"><a href="#cb23-39"></a>  means<span class="op">$</span>thickness,</span>
<span id="cb23-40"><a href="#cb23-40"></a>  <span class="dt">cex =</span> <span class="dv">2</span>,</span>
<span id="cb23-41"><a href="#cb23-41"></a>  <span class="dt">pch =</span> <span class="dv">23</span>,</span>
<span id="cb23-42"><a href="#cb23-42"></a>  <span class="dt">bg =</span> <span class="dv">0</span>,</span>
<span id="cb23-43"><a href="#cb23-43"></a>  <span class="dt">lwd =</span> <span class="dv">4</span></span>
<span id="cb23-44"><a href="#cb23-44"></a>)</span>
<span id="cb23-45"><a href="#cb23-45"></a></span>
<span id="cb23-46"><a href="#cb23-46"></a><span class="co"># X axis</span></span>
<span id="cb23-47"><a href="#cb23-47"></a><span class="kw">axis</span>(</span>
<span id="cb23-48"><a href="#cb23-48"></a>  <span class="dv">1</span>,</span>
<span id="cb23-49"><a href="#cb23-49"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb23-50"><a href="#cb23-50"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb23-51"><a href="#cb23-51"></a>  <span class="dt">labels =</span> <span class="kw">levels</span>(categories),</span>
<span id="cb23-52"><a href="#cb23-52"></a>  <span class="dt">at =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)</span>
<span id="cb23-53"><a href="#cb23-53"></a>)</span>
<span id="cb23-54"><a href="#cb23-54"></a></span>
<span id="cb23-55"><a href="#cb23-55"></a><span class="co"># Y axis</span></span>
<span id="cb23-56"><a href="#cb23-56"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-ttest-atheroma-04-plots-1.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
              <div id="aorta-hsp70" class="section level4">
                <h4>Aorta Hsp70</h4>
                <p>
                  Aorta Hsp70 data from Smith BW J Ultrasound Med 2015 (doi:
                  <a href="https://doi.org/10.7863/ultra.34.7.1209"
                    >10.7863/ultra.34.7.1209</a
                  >).
                </p>
                <div id="import-dataset-1" class="section level5">
                  <h5>Import dataset</h5>
                  <div class="sourceCode" id="cb24">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>AortaHsp70 &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;data/t_test_jum2015_hsp70.csv&#39;</span>,</span>
<span id="cb24-2"><a href="#cb24-2"></a>                         <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb24-3"><a href="#cb24-3"></a>                         <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span>)</span>
<span id="cb24-4"><a href="#cb24-4"></a><span class="kw">attach</span>(AortaHsp70)</span>
<span id="cb24-5"><a href="#cb24-5"></a><span class="kw">names</span>(AortaHsp70)</span></code></pre>
                  </div>
                  <pre><code>## [1] &quot;exposure&quot; &quot;Hsp70&quot;</code></pre>
                </div>
                <div id="perform-t-test-1" class="section level5">
                  <h5>Perform t-test</h5>
                  <ul>
                    <li>
                      The default assumes unequal variance and applies the
                      Satterthwaite/Welch df modification.
                    </li>
                    <li>
                      var.equal = TRUE is used to specify equal variances and a
                      pooled variance estimate.
                    </li>
                  </ul>
                  <div class="sourceCode" id="cb26">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="kw">t.test</span>(</span>
<span id="cb26-2"><a href="#cb26-2"></a>  AortaHsp70<span class="op">$</span>Hsp70 <span class="op">~</span><span class="st"> </span>AortaHsp70<span class="op">$</span>exposure,</span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="dt">paired =</span> <span class="ot">FALSE</span>,</span>
<span id="cb26-4"><a href="#cb26-4"></a>  <span class="dt">alternative =</span> <span class="st">&#39;two.sided&#39;</span>,</span>
<span id="cb26-5"><a href="#cb26-5"></a>  <span class="dt">var.equal =</span> <span class="ot">TRUE</span>,</span>
<span id="cb26-6"><a href="#cb26-6"></a>  <span class="dt">conf.level =</span> <span class="fl">0.95</span></span>
<span id="cb26-7"><a href="#cb26-7"></a>)</span></code></pre>
                  </div>
                  <pre><code>##
##  Two Sample t-test
##
## data:  AortaHsp70$Hsp70 by AortaHsp70$exposure
## t = -0.91363, df = 22, p-value = 0.3708
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.11935215  0.04635215
## sample estimates:
## mean in group Definity   mean in group Saline
##                 0.2575                 0.2940</code></pre>
                </div>
                <div id="test-assumptions-1" class="section level5">
                  <h5>Test assumptions</h5>
                  <p>
                    The t test uses raw data, not residuals, so assumptions are
                    tested using raw data.
                  </p>
                  <div id="normality-1" class="section level6">
                    <h6>Normality</h6>
                    <ul>
                      <li>
                        The RColorBrewer package is loaded for additional
                        colors, and a palette is added to R. Use brackets after
                        vector name for specific colors when creating plots (for
                        example: <code>col=set1[4:14]</code>).
                      </li>
                      <li>
                        Legend can also be manually positioned with xy
                        coordinates (numbers from 0,0 at bottom left of whole
                        plot to 1,1 at top right)
                      </li>
                    </ul>
                    <div class="sourceCode" id="cb28">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="co"># Shapiro-Wilk W test</span></span>
<span id="cb28-2"><a href="#cb28-2"></a><span class="kw">shapiro.test</span>(AortaHsp70<span class="op">$</span>Hsp70)</span></code></pre>
                    </div>
                    <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  AortaHsp70$Hsp70
## W = 0.90895, p-value = 0.03347</code></pre>
                    <div class="sourceCode" id="cb30">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="co"># Normal Quantile-Quantile Plot</span></span>
<span id="cb30-2"><a href="#cb30-2"></a><span class="co"># Set graphical parameters</span></span>
<span id="cb30-3"><a href="#cb30-3"></a><span class="kw">par</span>(</span>
<span id="cb30-4"><a href="#cb30-4"></a>  <span class="dt">cex =</span> <span class="dv">1</span>,</span>
<span id="cb30-5"><a href="#cb30-5"></a>  <span class="dt">cex.main =</span> <span class="dv">3</span>,</span>
<span id="cb30-6"><a href="#cb30-6"></a>  <span class="dt">cex.lab =</span> <span class="dv">2</span>,</span>
<span id="cb30-7"><a href="#cb30-7"></a>  <span class="dt">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb30-8"><a href="#cb30-8"></a>  <span class="dt">cex.axis =</span> <span class="fl">1.5</span>,</span>
<span id="cb30-9"><a href="#cb30-9"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb30-10"><a href="#cb30-10"></a>  <span class="dt">lend =</span> <span class="dv">0</span>,</span>
<span id="cb30-11"><a href="#cb30-11"></a>  <span class="dt">ljoin =</span> <span class="dv">0</span>,</span>
<span id="cb30-12"><a href="#cb30-12"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb30-13"><a href="#cb30-13"></a>  <span class="dt">pch =</span> <span class="dv">1</span>,</span>
<span id="cb30-14"><a href="#cb30-14"></a>  <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb30-15"><a href="#cb30-15"></a>)</span>
<span id="cb30-16"><a href="#cb30-16"></a></span>
<span id="cb30-17"><a href="#cb30-17"></a><span class="co"># Load RColorBrewer and set palette</span></span>
<span id="cb30-18"><a href="#cb30-18"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb30-19"><a href="#cb30-19"></a>set1 &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&#39;Set1&#39;</span>)</span>
<span id="cb30-20"><a href="#cb30-20"></a></span>
<span id="cb30-21"><a href="#cb30-21"></a><span class="co"># Create vector for categories</span></span>
<span id="cb30-22"><a href="#cb30-22"></a>categories &lt;-<span class="st"> </span><span class="kw">factor</span>(AortaHsp70<span class="op">$</span>exposure)</span>
<span id="cb30-23"><a href="#cb30-23"></a></span>
<span id="cb30-24"><a href="#cb30-24"></a><span class="co"># Create plot</span></span>
<span id="cb30-25"><a href="#cb30-25"></a><span class="kw">qqnorm</span>(</span>
<span id="cb30-26"><a href="#cb30-26"></a>  Hsp70,</span>
<span id="cb30-27"><a href="#cb30-27"></a>  <span class="dt">ylab =</span> <span class="kw">expression</span>(<span class="kw">bold</span>(<span class="kw">paste</span>(</span>
<span id="cb30-28"><a href="#cb30-28"></a>    <span class="st">&#39;Hsp70/&#39;</span>, beta, <span class="st">&#39;-actin&#39;</span></span>
<span id="cb30-29"><a href="#cb30-29"></a>  ))),</span>
<span id="cb30-30"><a href="#cb30-30"></a>  <span class="dt">axes =</span> <span class="ot">FALSE</span>,</span>
<span id="cb30-31"><a href="#cb30-31"></a>  <span class="dt">col =</span> set1[categories],</span>
<span id="cb30-32"><a href="#cb30-32"></a>  <span class="dt">lwd =</span> <span class="dv">4</span>,</span>
<span id="cb30-33"><a href="#cb30-33"></a>  <span class="dt">cex =</span> <span class="dv">2</span></span>
<span id="cb30-34"><a href="#cb30-34"></a>)</span>
<span id="cb30-35"><a href="#cb30-35"></a><span class="co"># Add normal line</span></span>
<span id="cb30-36"><a href="#cb30-36"></a><span class="kw">qqline</span>(AortaHsp70<span class="op">$</span>Hsp70, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb30-37"><a href="#cb30-37"></a><span class="co"># Add border to complete axes</span></span>
<span id="cb30-38"><a href="#cb30-38"></a><span class="kw">box</span>(<span class="dt">which =</span> <span class="st">&#39;plot&#39;</span>, <span class="dt">bty =</span> <span class="st">&#39;o&#39;</span>, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb30-39"><a href="#cb30-39"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>) <span class="co"># x axis</span></span>
<span id="cb30-40"><a href="#cb30-40"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>) <span class="co"># y axis</span></span>
<span id="cb30-41"><a href="#cb30-41"></a><span class="co"># Add legend</span></span>
<span id="cb30-42"><a href="#cb30-42"></a><span class="kw">legend</span>(</span>
<span id="cb30-43"><a href="#cb30-43"></a>  <span class="st">&#39;topleft&#39;</span>,</span>
<span id="cb30-44"><a href="#cb30-44"></a>  <span class="dt">legend =</span> <span class="kw">levels</span>(categories),</span>
<span id="cb30-45"><a href="#cb30-45"></a>  <span class="dt">box.lwd =</span> <span class="dv">3</span>,</span>
<span id="cb30-46"><a href="#cb30-46"></a>  <span class="dt">pch =</span> <span class="fl">1.5</span>,</span>
<span id="cb30-47"><a href="#cb30-47"></a>  <span class="dt">cex =</span> <span class="fl">1.75</span>,</span>
<span id="cb30-48"><a href="#cb30-48"></a>  <span class="dt">pt.cex =</span> <span class="dv">2</span>,</span>
<span id="cb30-49"><a href="#cb30-49"></a>  <span class="dt">pt.lwd =</span> <span class="dv">4</span>,</span>
<span id="cb30-50"><a href="#cb30-50"></a>  <span class="dt">col =</span> set1</span>
<span id="cb30-51"><a href="#cb30-51"></a>)</span></code></pre>
                    </div>
                    <p>
                      <img
                        src="code_files/figure-html/code-ttest-hsp70-03-assumptions-01-1.png"
                        width="672"
                      />
                    </p>
                  </div>
                  <div id="equality-of-variances-1" class="section level6">
                    <h6>Equality of variances</h6>
                    <div class="sourceCode" id="cb31">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a><span class="kw">var.test</span>(AortaHsp70<span class="op">$</span>Hsp70 <span class="op">~</span><span class="st"> </span>AortaHsp70<span class="op">$</span>exposure)</span></code></pre>
                    </div>
                    <pre><code>##
##  F test to compare two variances
##
## data:  AortaHsp70$Hsp70 by AortaHsp70$exposure
## F = 0.38929, num df = 11, denom df = 11, p-value = 0.1328
## alternative hypothesis: true ratio of variances is not equal to 1
## 95 percent confidence interval:
##  0.1120669 1.3522650
## sample estimates:
## ratio of variances
##          0.3892868</code></pre>
                  </div>
                </div>
                <div id="boxplot-1" class="section level5">
                  <h5>Boxplot</h5>
                  <ul>
                    <li>
                      <strong>Parameters (<code>par</code>):</strong> Graphical
                      parameters need to be set again because of code chunk
                      separation in RMarkdown. Normally they would only be set
                      once per R session.
                    </li>
                    <li>
                      <strong>X axis:</strong> The default axes will usually
                      detect categorical data and label axes accordingly, but
                      categories will need to be specified for axes created
                      separately. It is easiest to create a separate vector with
                      the categories, and then reference it when creating the
                      plot. Labels can also be specified with
                      <code>labels=c('group1', 'group2')</code>.
                    </li>
                    <li>
                      <strong>Y axis:</strong> Note that it does not always
                      start at zero, but encompasses the data range with some
                      padding.
                    </li>
                  </ul>
                  <div class="sourceCode" id="cb33">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="co"># Set graphical parameters</span></span>
<span id="cb33-2"><a href="#cb33-2"></a><span class="kw">par</span>(</span>
<span id="cb33-3"><a href="#cb33-3"></a>  <span class="dt">cex =</span> <span class="dv">1</span>,</span>
<span id="cb33-4"><a href="#cb33-4"></a>  <span class="dt">cex.main =</span> <span class="dv">3</span>,</span>
<span id="cb33-5"><a href="#cb33-5"></a>  <span class="dt">cex.lab =</span> <span class="dv">2</span>,</span>
<span id="cb33-6"><a href="#cb33-6"></a>  <span class="dt">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb33-7"><a href="#cb33-7"></a>  <span class="dt">cex.axis =</span> <span class="fl">1.5</span>,</span>
<span id="cb33-8"><a href="#cb33-8"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb33-9"><a href="#cb33-9"></a>  <span class="dt">lend =</span> <span class="dv">0</span>,</span>
<span id="cb33-10"><a href="#cb33-10"></a>  <span class="dt">ljoin =</span> <span class="dv">0</span>,</span>
<span id="cb33-11"><a href="#cb33-11"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb33-12"><a href="#cb33-12"></a>  <span class="dt">pch =</span> <span class="dv">1</span>,</span>
<span id="cb33-13"><a href="#cb33-13"></a>  <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb33-14"><a href="#cb33-14"></a>)</span>
<span id="cb33-15"><a href="#cb33-15"></a></span>
<span id="cb33-16"><a href="#cb33-16"></a><span class="co"># Create boxplot</span></span>
<span id="cb33-17"><a href="#cb33-17"></a><span class="kw">boxplot</span>(</span>
<span id="cb33-18"><a href="#cb33-18"></a>  AortaHsp70<span class="op">$</span>Hsp70 <span class="op">~</span><span class="st"> </span>AortaHsp70<span class="op">$</span>exposure,</span>
<span id="cb33-19"><a href="#cb33-19"></a>  <span class="dt">main =</span> <span class="st">&#39;Aorta Hsp70&#39;</span>,</span>
<span id="cb33-20"><a href="#cb33-20"></a>  <span class="dt">xlab =</span> <span class="st">&#39;Exposure&#39;</span>,</span>
<span id="cb33-21"><a href="#cb33-21"></a>  <span class="dt">ylab =</span> <span class="kw">expression</span>(<span class="kw">bold</span>(<span class="kw">paste</span>(</span>
<span id="cb33-22"><a href="#cb33-22"></a>    <span class="st">&#39;Hsp70/&#39;</span>, beta, <span class="st">&#39;-actin&#39;</span></span>
<span id="cb33-23"><a href="#cb33-23"></a>  ))),</span>
<span id="cb33-24"><a href="#cb33-24"></a>  <span class="dt">axes =</span> <span class="ot">FALSE</span>,</span>
<span id="cb33-25"><a href="#cb33-25"></a>  <span class="dt">lwd =</span> <span class="dv">4</span>,</span>
<span id="cb33-26"><a href="#cb33-26"></a>  <span class="dt">medlwd =</span> <span class="dv">7</span>,</span>
<span id="cb33-27"><a href="#cb33-27"></a>  <span class="dt">outlwd =</span> <span class="dv">4</span>,</span>
<span id="cb33-28"><a href="#cb33-28"></a>  <span class="dt">cex =</span> <span class="dv">2</span>,</span>
<span id="cb33-29"><a href="#cb33-29"></a>  <span class="dt">col =</span> set1</span>
<span id="cb33-30"><a href="#cb33-30"></a>)</span>
<span id="cb33-31"><a href="#cb33-31"></a></span>
<span id="cb33-32"><a href="#cb33-32"></a><span class="co"># Add border to complete axes</span></span>
<span id="cb33-33"><a href="#cb33-33"></a><span class="kw">box</span>(<span class="dt">which =</span> <span class="st">&#39;plot&#39;</span>, <span class="dt">bty =</span> <span class="st">&#39;l&#39;</span>, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb33-34"><a href="#cb33-34"></a></span>
<span id="cb33-35"><a href="#cb33-35"></a><span class="co"># Create vector with means of each group</span></span>
<span id="cb33-36"><a href="#cb33-36"></a>means &lt;-<span class="st"> </span><span class="kw">aggregate</span>(Hsp70 <span class="op">~</span><span class="st"> </span>exposure, AortaHsp70, mean)</span>
<span id="cb33-37"><a href="#cb33-37"></a><span class="co"># Add points for group means to boxplot</span></span>
<span id="cb33-38"><a href="#cb33-38"></a><span class="kw">points</span>(</span>
<span id="cb33-39"><a href="#cb33-39"></a>  means<span class="op">$</span>Hsp70,</span>
<span id="cb33-40"><a href="#cb33-40"></a>  <span class="dt">cex =</span> <span class="dv">2</span>,</span>
<span id="cb33-41"><a href="#cb33-41"></a>  <span class="dt">pch =</span> <span class="dv">23</span>,</span>
<span id="cb33-42"><a href="#cb33-42"></a>  <span class="dt">bg =</span> <span class="dv">0</span>,</span>
<span id="cb33-43"><a href="#cb33-43"></a>  <span class="dt">lwd =</span> <span class="dv">4</span></span>
<span id="cb33-44"><a href="#cb33-44"></a>)</span>
<span id="cb33-45"><a href="#cb33-45"></a></span>
<span id="cb33-46"><a href="#cb33-46"></a><span class="co"># X axis</span></span>
<span id="cb33-47"><a href="#cb33-47"></a><span class="kw">axis</span>(</span>
<span id="cb33-48"><a href="#cb33-48"></a>  <span class="dv">1</span>,</span>
<span id="cb33-49"><a href="#cb33-49"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb33-50"><a href="#cb33-50"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb33-51"><a href="#cb33-51"></a>  <span class="dt">labels =</span> <span class="kw">levels</span>(categories),</span>
<span id="cb33-52"><a href="#cb33-52"></a>  <span class="dt">at =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)</span>
<span id="cb33-53"><a href="#cb33-53"></a>)</span>
<span id="cb33-54"><a href="#cb33-54"></a></span>
<span id="cb33-55"><a href="#cb33-55"></a><span class="co"># Y axis</span></span>
<span id="cb33-56"><a href="#cb33-56"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-ttest-hsp70-04-plots-1.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
            </div>
            <div id="linear-regression" class="section level3">
              <h3>Linear regression</h3>
              <div id="simple-linear-regression-aztcpm" class="section level4">
                <h4>Simple linear regression aztcpm</h4>
                <p>
                  <code>aztcpm</code> dataset from University of Illinois
                  statistics course
                  <a
                    href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lab-02.url"
                    >PATH 591 lab 02</a
                  >
                  and
                  <a
                    href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-04A-annotated.url"
                    >PATH 591 lecture 04 A</a
                  >
                </p>
                <div id="intro" class="section level5">
                  <h5>Intro</h5>
                  <blockquote>
                    <p>
                      These data are from an in-vitro study of the effect of AZT
                      dose on DNA replication (CPM) for FeLV (feline leukemia
                      virus). (The experimental unit is the cell culture
                      sample). For the linear regression analysis, the dependent
                      variable is CPM (counts per minute, indicating quantity of
                      Feline Leukemia Virus DNA in a cell culture) and the
                      independent (predictor) variable is DOSE (dose of AZT).
                    </p>
                  </blockquote>
                </div>
                <div id="import-dataset-2" class="section level5">
                  <h5>Import dataset</h5>
                  <div class="sourceCode" id="cb34">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>aztcpm &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;data/lm_aztcpm.csv&#39;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span>)</span>
<span id="cb34-2"><a href="#cb34-2"></a><span class="kw">attach</span>(aztcpm)</span>
<span id="cb34-3"><a href="#cb34-3"></a><span class="kw">names</span>(aztcpm)</span></code></pre>
                  </div>
                  <pre><code>## [1] &quot;DOSE&quot; &quot;CPM&quot;</code></pre>
                </div>
                <div id="perform-linear-regression" class="section level5">
                  <h5>Perform linear regression</h5>
                  <div class="sourceCode" id="cb36">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>lm1 &lt;-<span class="st"> </span><span class="kw">lm</span>(CPM <span class="op">~</span><span class="st"> </span>DOSE)</span></code></pre>
                  </div>
                </div>
                <div
                  id="perform-regression-diagnostics-and-test-assumptions"
                  class="section level5"
                >
                  <h5>Perform regression diagnostics and test assumptions</h5>
                  <div class="sourceCode" id="cb37">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb37-2"><a href="#cb37-2"></a>gvlmalm1 &lt;-<span class="st"> </span><span class="kw">gvlma</span>(lm1)</span>
<span id="cb37-3"><a href="#cb37-3"></a><span class="kw">summary</span>(gvlmalm1)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = CPM ~ DOSE)
##
## Residuals:
##    Min     1Q Median     3Q    Max
##  -8712  -5708  -2483   3224  21640
##
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept) 39810.58    1711.46  23.261  &lt; 2e-16 ***
## DOSE         -813.61      89.45  -9.096 2.93e-10 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 8215 on 31 degrees of freedom
## Multiple R-squared:  0.7274, Adjusted R-squared:  0.7186
## F-statistic: 82.73 on 1 and 31 DF,  p-value: 2.931e-10
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = lm1)
##
##                     Value   p-value                   Decision
## Global Stat        32.863 1.274e-06 Assumptions NOT satisfied!
## Skewness            9.168 2.463e-03 Assumptions NOT satisfied!
## Kurtosis            1.413 2.346e-01    Assumptions acceptable.
## Link Function      12.218 4.733e-04 Assumptions NOT satisfied!
## Heteroscedasticity 10.064 1.512e-03 Assumptions NOT satisfied!</code></pre>
                  <div class="sourceCode" id="cb39">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(lm1))</span></code></pre>
                  </div>
                  <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(lm1)
## W = 0.85782, p-value = 0.0005083</code></pre>
                  <div class="sourceCode" id="cb41">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a><span class="kw">cor.test</span>(<span class="kw">fitted</span>(lm1), <span class="kw">abs</span>(<span class="kw">rstandard</span>(lm1)), <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(fitted(lm1), abs(rstandard(lm1)), method = &quot;spearman&quot;): Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  fitted(lm1) and abs(rstandard(lm1))
## S = 6143.9, p-value = 0.8827
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##         rho
## -0.02671635</code></pre>
                  <div class="sourceCode" id="cb44">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a><span class="kw">library</span>(car)</span></code></pre>
                  </div>
                  <pre><code>## Loading required package: carData</code></pre>
                  <div class="sourceCode" id="cb46">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a><span class="kw">durbinWatsonTest</span>(lm1)</span></code></pre>
                  </div>
                  <pre><code>##  lag Autocorrelation D-W Statistic p-value
##    1        0.797716     0.1772224       0
##  Alternative hypothesis: rho != 0</code></pre>
                  <div class="sourceCode" id="cb48">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a><span class="kw">outlierTest</span>(lm1)</span></code></pre>
                  </div>
                  <pre><code>## No Studentized residuals with Bonferroni p &lt; 0.05
## Largest |rstudent|:
##    rstudent unadjusted p-value Bonferroni p
## 33 3.027239          0.0050324      0.16607</code></pre>
                </div>
                <div id="generate-plots" class="section level5">
                  <h5>Generate plots</h5>
                  <div class="sourceCode" id="cb50">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a><span class="kw">plot</span>(gvlmalm1, <span class="dt">onepage =</span> <span class="ot">FALSE</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-4.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-5.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-6.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb51">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a><span class="kw">plot</span>(lm1, <span class="dt">which =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-7.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-8.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-9.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb52">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a><span class="kw">acf</span>(<span class="kw">rstandard</span>(lm1), <span class="dt">main =</span> <span class="st">&#39;Autocorrelation plot&#39;</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-aztcpm-04-plots-10.png"
                      width="672"
                    />
                  </p>
                </div>
                <div id="log-transform" class="section level5">
                  <h5>Log transform</h5>
                  <div class="sourceCode" id="cb53">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a>lmlog &lt;-<span class="st"> </span><span class="kw">update</span>(lm1, <span class="kw">log</span>(CPM) <span class="op">~</span><span class="st"> </span>DOSE)</span></code></pre>
                  </div>
                </div>
                <div
                  id="perform-regression-diagnostics-and-test-assumptions-1"
                  class="section level5"
                >
                  <h5>Perform regression diagnostics and test assumptions</h5>
                  <p>
                    Log transformation improves compliance with linear model
                    assumptions. Some serial correlation remains, which is
                    evident in plots of residuals vs fitted values and in the
                    significant result of the Durbin-Watson D test, but it is
                    within acceptable limits of the global validation of linear
                    model assumptions.
                  </p>
                  <div class="sourceCode" id="cb54">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb54-2"><a href="#cb54-2"></a>gvlmalog &lt;-<span class="st"> </span><span class="kw">gvlma</span>(lmlog)</span>
<span id="cb54-3"><a href="#cb54-3"></a><span class="kw">summary</span>(gvlmalog)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = log(CPM) ~ DOSE)
##
## Residuals:
##      Min       1Q   Median       3Q      Max
## -0.57193 -0.11370 -0.02660  0.08659  0.46827
##
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept) 10.66086    0.04420  241.20   &lt;2e-16 ***
## DOSE        -0.05144    0.00231  -22.27   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 0.2122 on 31 degrees of freedom
## Multiple R-squared:  0.9412, Adjusted R-squared:  0.9393
## F-statistic: 495.8 on 1 and 31 DF,  p-value: &lt; 2.2e-16
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = lmlog)
##
##                       Value p-value                Decision
## Global Stat        3.952318 0.41250 Assumptions acceptable.
## Skewness           0.056280 0.81248 Assumptions acceptable.
## Kurtosis           0.676082 0.41094 Assumptions acceptable.
## Link Function      0.004124 0.94880 Assumptions acceptable.
## Heteroscedasticity 3.215833 0.07293 Assumptions acceptable.</code></pre>
                  <div class="sourceCode" id="cb56">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(lmlog))</span></code></pre>
                  </div>
                  <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(lmlog)
## W = 0.94082, p-value = 0.07176</code></pre>
                  <div class="sourceCode" id="cb58">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a><span class="kw">cor.test</span>(<span class="kw">fitted</span>(lmlog), <span class="kw">abs</span>(<span class="kw">rstandard</span>(lmlog)), <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(fitted(lmlog), abs(rstandard(lmlog)), method = &quot;spearman&quot;): Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  fitted(lmlog) and abs(rstandard(lmlog))
## S = 6684, p-value = 0.5168
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##        rho
## -0.1169866</code></pre>
                  <div class="sourceCode" id="cb61">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a><span class="kw">library</span>(car)</span>
<span id="cb61-2"><a href="#cb61-2"></a><span class="kw">durbinWatsonTest</span>(lmlog)</span></code></pre>
                  </div>
                  <pre><code>##  lag Autocorrelation D-W Statistic p-value
##    1       0.4659162     0.7381887       0
##  Alternative hypothesis: rho != 0</code></pre>
                  <div class="sourceCode" id="cb63">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1"></a><span class="kw">outlierTest</span>(lmlog)</span></code></pre>
                  </div>
                  <pre><code>##    rstudent unadjusted p-value Bonferroni p
## 1 -3.574629          0.0012106      0.03995</code></pre>
                </div>
                <div id="generate-plots-1" class="section level5">
                  <h5>Generate plots</h5>
                  <div class="sourceCode" id="cb65">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1"></a><span class="kw">plot</span>(gvlmalog, <span class="dt">onepage =</span> <span class="ot">FALSE</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-4.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-5.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-6.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb66">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1"></a><span class="kw">plot</span>(lmlog, <span class="dt">which =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-7.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-8.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-9.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb67">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1"></a><span class="kw">acf</span>(<span class="kw">rstandard</span>(lmlog), <span class="dt">main =</span> <span class="st">&#39;Autocorrelation plot&#39;</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-aztcpm-07-transform-plots-10.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
              <div
                id="simple-linear-regression-jfs-data"
                class="section level4"
              >
                <h4>Simple linear regression jfs data</h4>
                <p>
                  Feed intake vs total cholesterol data from Smith BW J Food Sci
                  2015 (doi:
                  <a href="https://doi.org/10.1111/1750-3841.12968"
                    >10.1111/1750-3841.12968</a
                  >).
                </p>
                <div id="import-dataset-3" class="section level5">
                  <h5>Import dataset</h5>
                  <div class="sourceCode" id="cb68">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1"></a>feedtc &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;data/lm_jfs2015_feed_vs_tc.csv&#39;</span>,</span>
<span id="cb68-2"><a href="#cb68-2"></a>                     <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb68-3"><a href="#cb68-3"></a>                     <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span>)</span>
<span id="cb68-4"><a href="#cb68-4"></a><span class="kw">attach</span>(feedtc)</span>
<span id="cb68-5"><a href="#cb68-5"></a><span class="kw">names</span>(feedtc)</span></code></pre>
                  </div>
                  <pre><code>## [1] &quot;id&quot;   &quot;diet&quot; &quot;tc&quot;   &quot;feed&quot;</code></pre>
                </div>
                <div id="perform-linear-regression-1" class="section level5">
                  <h5>Perform linear regression</h5>
                  <div class="sourceCode" id="cb70">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1"></a>feedtclm1 &lt;-<span class="st"> </span><span class="kw">lm</span>(feedtc<span class="op">$</span>tc <span class="op">~</span><span class="st"> </span>feedtc<span class="op">$</span>feed)</span></code></pre>
                  </div>
                </div>
                <div
                  id="perform-regression-diagnostics-and-test-assumptions-2"
                  class="section level5"
                >
                  <h5>Perform regression diagnostics and test assumptions</h5>
                  <p>
                    The dataset contains some potential outliers (M1356 and
                    M1376), and a violation of the link function according to
                    gvlma.
                  </p>
                  <ul>
                    <li>
                      Outliers
                      <ul>
                        <li>
                          The major outlier is M1376, which returns a Cook’s D
                          value of 0.43. This may or may not be an outlier,
                          depending on the definition. It is just under the
                          critical F value (F0.50(1,40)=0.463).
                        </li>
                        <li>
                          The Bonferroni outlier test does not identify any
                          outliers either.
                        </li>
                        <li>
                          However, according to R in Action 8.4.3, Cook’s D
                          values greater than 4/(n-k-1), where n is the sample
                          size and k is the number of predictor variables,
                          indicate influential observations (though he usually
                          uses a cutoff of 1). M1376 would be considered an
                          outlier by this definition.
                        </li>
                      </ul>
                    </li>
                    <li>
                      Link function
                      <ul>
                        <li>
                          The link function violation is difficult to address.
                          The dataset conforms to a normal distribution as
                          evidenced by the normal probability plot and
                          Shapiro-Wilk W test, so it would not seem like a
                          concern, but it is enough to violate gvlma. However,
                          this doesn’t seem like a situation where it would be
                          neccessary to switch to another model (such as a
                          generalized linear mixed model) and specify a
                          different link function.
                        </li>
                        <li>
                          The Spearman rank order correlation between absolute
                          values of residuals and predicted values returns a
                          significant result, but heteroscedasticity is within
                          acceptable limits according to gvlma and by visual
                          inspection of plots of residuals vs. fitted values and
                          autocorrelation.
                        </li>
                      </ul>
                    </li>
                  </ul>
                  <div class="sourceCode" id="cb71">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb71-2"><a href="#cb71-2"></a>feedtcgvlmalm1 &lt;-<span class="st"> </span><span class="kw">gvlma</span>(feedtclm1)</span>
<span id="cb71-3"><a href="#cb71-3"></a><span class="kw">summary</span>(feedtcgvlmalm1)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = feedtc$tc ~ feedtc$feed)
##
## Residuals:
##     Min      1Q  Median      3Q     Max
## -465.42 -208.18   11.74  168.20  449.91
##
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)  -470.33     232.65  -2.022   0.0489 *
## feedtc$feed    77.39      14.94   5.179 4.58e-06 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 239.6 on 47 degrees of freedom
## Multiple R-squared:  0.3633, Adjusted R-squared:  0.3498
## F-statistic: 26.82 on 1 and 47 DF,  p-value: 4.578e-06
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = feedtclm1)
##
##                       Value   p-value                   Decision
## Global Stat        15.17502 0.0043516 Assumptions NOT satisfied!
## Skewness            0.03692 0.8476301    Assumptions acceptable.
## Kurtosis            1.62599 0.2022584    Assumptions acceptable.
## Link Function      13.31102 0.0002639 Assumptions NOT satisfied!
## Heteroscedasticity  0.20109 0.6538402    Assumptions acceptable.</code></pre>
                  <div class="sourceCode" id="cb73">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(feedtclm1))</span></code></pre>
                  </div>
                  <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(feedtclm1)
## W = 0.97342, p-value = 0.3295</code></pre>
                  <div class="sourceCode" id="cb75">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1"></a><span class="kw">cor.test</span>(<span class="kw">fitted</span>(feedtclm1), <span class="kw">abs</span>(<span class="kw">rstandard</span>(feedtclm1)), <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(fitted(feedtclm1), abs(rstandard(feedtclm1)), : Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  fitted(feedtclm1) and abs(rstandard(feedtclm1))
## S = 25966, p-value = 0.02278
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##        rho
## -0.3248125</code></pre>
                  <div class="sourceCode" id="cb78">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1"></a><span class="kw">library</span>(car)</span>
<span id="cb78-2"><a href="#cb78-2"></a><span class="kw">durbinWatsonTest</span>(feedtclm1)</span></code></pre>
                  </div>
                  <pre><code>##  lag Autocorrelation D-W Statistic p-value
##    1       0.4305706      1.065764       0
##  Alternative hypothesis: rho != 0</code></pre>
                  <div class="sourceCode" id="cb80">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1"></a><span class="kw">outlierTest</span>(feedtclm1)</span></code></pre>
                  </div>
                  <pre><code>## No Studentized residuals with Bonferroni p &lt; 0.05
## Largest |rstudent|:
##     rstudent unadjusted p-value Bonferroni p
## 45 -2.208409           0.032238           NA</code></pre>
                </div>
                <div id="generate-plots-2" class="section level5">
                  <h5>Generate plots</h5>
                  <div class="sourceCode" id="cb82">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1"></a><span class="kw">plot</span>(feedtcgvlmalm1, <span class="dt">onepage =</span> <span class="ot">FALSE</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-4.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-5.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-6.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb83">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1"></a><span class="kw">plot</span>(feedtclm1, <span class="dt">which =</span> <span class="dv">3</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-7.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb84">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1"></a>cookscritical &lt;-<span class="st"> </span><span class="dv">4</span> <span class="op">/</span><span class="st"> </span>(<span class="kw">nrow</span>(feedtc) <span class="op">-</span><span class="st"> </span><span class="kw">length</span>(feedtclm1<span class="op">$</span>coefficients) <span class="op">-</span><span class="st"> </span><span class="dv">2</span>)</span>
<span id="cb84-2"><a href="#cb84-2"></a><span class="kw">plot</span>(feedtclm1, <span class="dt">which =</span> <span class="dv">4</span>, <span class="dt">cook.levels =</span> cookscritical)</span>
<span id="cb84-3"><a href="#cb84-3"></a><span class="kw">abline</span>(<span class="dt">h =</span> cookscritical, <span class="dt">lty =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-8.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb85">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1"></a><span class="kw">plot</span>(feedtclm1, <span class="dt">which =</span> <span class="dv">5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-9.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb86">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1"></a><span class="kw">acf</span>(<span class="kw">rstandard</span>(feedtclm1), <span class="dt">main =</span> <span class="st">&#39;Autocorrelation plot&#39;</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-jfs-feedtc-04-plots-10.png"
                      width="672"
                    />
                  </p>
                </div>
                <div id="calculate-spearman-correlation" class="section level5">
                  <h5>Calculate Spearman correlation</h5>
                  <p>
                    The Spearman rank order correlation between the independent
                    and dependent variable is another type of correlation that
                    is more widely valid, especially in situations where F test
                    assumptions are violated.
                  </p>
                  <div class="sourceCode" id="cb87">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1"></a><span class="kw">cor.test</span>(feedtc<span class="op">$</span>tc, feedtc<span class="op">$</span>feed, <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(feedtc$tc, feedtc$feed, method = &quot;spearman&quot;): Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  feedtc$tc and feedtc$feed
## S = 8028.4, p-value = 8.003e-06
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##       rho
## 0.5903873</code></pre>
                </div>
              </div>
              <div
                id="multiple-linear-regression-and-anova-example"
                class="section level4"
              >
                <h4>Multiple linear regression and ANOVA example</h4>
                <p>
                  This dataset from Doug Simpson at University of Illinois
                  contains information on the number of calories in various
                  beers, along with their percent alcohol and brewery.
                </p>
                <p>
                  <code>PercentAlcohol</code> is a quantitative continuous
                  independent variable and Brewery is a categorical independent
                  variable.
                </p>
                <div id="import-dataset-4" class="section level5">
                  <h5>Import dataset</h5>
                  <div class="sourceCode" id="cb90">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1"></a>beer &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;data/anova_mlr_beer.csv&#39;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span>)</span>
<span id="cb90-2"><a href="#cb90-2"></a><span class="kw">attach</span>(beer)</span></code></pre>
                  </div>
                </div>
                <div
                  id="perform-multiple-linear-regression-in-lm"
                  class="section level5"
                >
                  <h5>Perform multiple linear regression in lm</h5>
                  <div class="sourceCode" id="cb91">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1"></a>beerlm1 &lt;-<span class="st"> </span><span class="kw">lm</span>(beer<span class="op">$</span>Calories <span class="op">~</span><span class="st"> </span>beer<span class="op">$</span>PercentAlcohol <span class="op">+</span><span class="st"> </span>beer<span class="op">$</span>Brewery)</span>
<span id="cb91-2"><a href="#cb91-2"></a><span class="kw">anova</span>(beerlm1)</span></code></pre>
                  </div>
                  <pre><code>## Analysis of Variance Table
##
## Response: beer$Calories
##                     Df  Sum Sq Mean Sq  F value  Pr(&gt;F)
## beer$PercentAlcohol  1 17914.2 17914.2 102.3698 7.9e-13 ***
## beer$Brewery         8  4424.9   553.1   3.1608 0.00682 **
## Residuals           42  7349.8   175.0
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                </div>
                <div
                  id="perform-regression-diagnostics-and-test-assumptions-3"
                  class="section level5"
                >
                  <h5>Perform regression diagnostics and test assumptions</h5>
                  <div class="sourceCode" id="cb93">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb93-2"><a href="#cb93-2"></a>beergvlmalm1 &lt;-<span class="st"> </span><span class="kw">gvlma</span>(beerlm1)</span>
<span id="cb93-3"><a href="#cb93-3"></a><span class="kw">summary</span>(beergvlmalm1)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = beer$Calories ~ beer$PercentAlcohol + beer$Brewery)
##
## Residuals:
##      Min       1Q   Median       3Q      Max
## -28.0179  -6.4610   0.6312  10.2197  25.3521
##
## Coefficients:
##                                Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)                     -36.937     17.061  -2.165  0.03612 *
## beer$PercentAlcohol              33.812      3.439   9.831 1.86e-12 ***
## beer$BreweryAnchor               24.258     14.152   1.714  0.09389 .
## beer$BreweryAnheuser Busch        2.150      6.400   0.336  0.73858
## beer$BreweryD. G. Yuengling      20.088      7.825   2.567  0.01391 *
## beer$BreweryG. Heileman          17.115     10.722   1.596  0.11795
## beer$BreweryHigh Falls Brewing   17.996      8.382   2.147  0.03762 *
## beer$BreweryLeinenkugel          24.470      7.077   3.458  0.00126 **
## beer$BreweryMiller               10.801      6.206   1.740  0.08911 .
## beer$BreweryPabst                29.021     10.607   2.736  0.00908 **
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 13.23 on 42 degrees of freedom
## Multiple R-squared:  0.7524, Adjusted R-squared:  0.6994
## F-statistic: 14.18 on 9 and 42 DF,  p-value: 3.745e-10
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = beerlm1)
##
##                      Value   p-value                   Decision
## Global Stat        16.8557 0.0020618 Assumptions NOT satisfied!
## Skewness            1.0664 0.3017523    Assumptions acceptable.
## Kurtosis            0.3225 0.5700957    Assumptions acceptable.
## Link Function      14.4847 0.0001413 Assumptions NOT satisfied!
## Heteroscedasticity  0.9820 0.3216947    Assumptions acceptable.</code></pre>
                  <div class="sourceCode" id="cb95">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(beerlm1))</span></code></pre>
                  </div>
                  <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(beerlm1)
## W = 0.97308, p-value = 0.2957</code></pre>
                  <div class="sourceCode" id="cb97">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1"></a><span class="kw">cor.test</span>(<span class="kw">fitted</span>(beerlm1), <span class="kw">abs</span>(<span class="kw">rstandard</span>(beerlm1)), <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(fitted(beerlm1), abs(rstandard(beerlm1)), method = &quot;spearman&quot;): Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  fitted(beerlm1) and abs(rstandard(beerlm1))
## S = 19184, p-value = 0.3561
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##       rho
## 0.1319358</code></pre>
                  <div class="sourceCode" id="cb100">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1"></a><span class="kw">library</span>(car)</span>
<span id="cb100-2"><a href="#cb100-2"></a><span class="kw">durbinWatsonTest</span>(beerlm1)</span></code></pre>
                  </div>
                  <pre><code>##  lag Autocorrelation D-W Statistic p-value
##    1      0.07781896      1.843075   0.168
##  Alternative hypothesis: rho != 0</code></pre>
                  <div class="sourceCode" id="cb102">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1"></a><span class="kw">leveneTest</span>(beer<span class="op">$</span>Calories, beer<span class="op">$</span>Brewery)</span></code></pre>
                  </div>
                  <pre><code>## Warning in leveneTest.default(beer$Calories, beer$Brewery): beer$Brewery coerced to factor.</code></pre>
                  <pre><code>## Levene&#39;s Test for Homogeneity of Variance (center = median)
##       Df F value Pr(&gt;F)
## group  8  0.8577 0.5587
##       43</code></pre>
                </div>
                <div id="generate-plots-3" class="section level5">
                  <h5>Generate plots</h5>
                  <div class="sourceCode" id="cb105">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1"></a><span class="kw">plot</span>(beergvlmalm1, <span class="dt">onepage =</span> <span class="ot">FALSE</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-4.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-5.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-6.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-7.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-8.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-9.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-10.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-11.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-12.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-13.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-14.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb106">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1"></a><span class="kw">plot</span>(beerlm1, <span class="dt">which =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">5</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning: not plotting observations with leverage one:
##   1</code></pre>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-15.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-16.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-04-plots-17.png"
                      width="672"
                    />
                  </p>
                </div>
                <div
                  id="perform-anova-in-aov-for-comparison"
                  class="section level5"
                >
                  <h5>Perform ANOVA in aov for comparison</h5>
                  <div class="sourceCode" id="cb108">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1"></a>beeraov1 &lt;-<span class="st"> </span><span class="kw">aov</span>(beer<span class="op">$</span>Calories <span class="op">~</span><span class="st"> </span>beer<span class="op">$</span>PercentAlcohol <span class="op">+</span><span class="st"> </span>beer<span class="op">$</span>Brewery)</span>
<span id="cb108-2"><a href="#cb108-2"></a><span class="kw">summary</span>(beeraov1)</span></code></pre>
                  </div>
                  <pre><code>##                     Df Sum Sq Mean Sq F value  Pr(&gt;F)
## beer$PercentAlcohol  1  17914   17914 102.370 7.9e-13 ***
## beer$Brewery         8   4425     553   3.161 0.00682 **
## Residuals           42   7350     175
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                  <div class="sourceCode" id="cb110">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1"></a><span class="kw">plot</span>(beeraov1)</span></code></pre>
                  </div>
                  <pre><code>## Warning: not plotting observations with leverage one:
##   1</code></pre>
                  <p>
                    <img
                      src="code_files/figure-html/code-lm-simple-beer-05-aov-comparison-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-05-aov-comparison-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-05-aov-comparison-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-lm-simple-beer-05-aov-comparison-4.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
            </div>
            <div id="anova" class="section level3">
              <h3>ANOVA</h3>
              <div id="one-way-anova-example" class="section level4">
                <h4>One-way ANOVA example</h4>
                <p>
                  Gonadal adipose weights from Smith BW J Food Sci 2015 (doi:
                  <a href="https://doi.org/10.1111/1750-3841.12968"
                    >10.1111/1750-3841.12968</a
                  >).
                </p>
                <div id="import-dataset-5" class="section level5">
                  <h5>Import dataset</h5>
                  <div class="sourceCode" id="cb112">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1"></a>adipose_weights &lt;-</span>
<span id="cb112-2"><a href="#cb112-2"></a><span class="st">  </span><span class="kw">read.table</span>(<span class="st">&#39;data/anova_one_way_jfs2015_adipose.csv&#39;</span>,</span>
<span id="cb112-3"><a href="#cb112-3"></a>             <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb112-4"><a href="#cb112-4"></a>             <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span>)</span>
<span id="cb112-5"><a href="#cb112-5"></a><span class="kw">attach</span>(adipose_weights)</span>
<span id="cb112-6"><a href="#cb112-6"></a><span class="kw">names</span>(adipose_weights)</span></code></pre>
                  </div>
                  <pre><code>## [1] &quot;id&quot;      &quot;diet&quot;    &quot;adipose&quot;</code></pre>
                </div>
                <div id="perform-anova-in-aov" class="section level5">
                  <h5>Perform ANOVA in aov</h5>
                  <div class="sourceCode" id="cb114">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1"></a><span class="co"># Convert independent variable to a factor for use later by glht()</span></span>
<span id="cb114-2"><a href="#cb114-2"></a>adipose_weights<span class="op">$</span>diet &lt;-<span class="st"> </span><span class="kw">as.factor</span>(adipose_weights<span class="op">$</span>diet)</span>
<span id="cb114-3"><a href="#cb114-3"></a><span class="co"># Perform ANOVA</span></span>
<span id="cb114-4"><a href="#cb114-4"></a>adiposeaov1 &lt;-<span class="st"> </span><span class="kw">aov</span>(adipose <span class="op">~</span><span class="st"> </span>diet, <span class="dt">data =</span> adipose_weights)</span>
<span id="cb114-5"><a href="#cb114-5"></a><span class="kw">summary</span>(adiposeaov1)</span></code></pre>
                  </div>
                  <pre><code>##             Df   Sum Sq   Mean Sq F value   Pr(&gt;F)
## diet         4 0.002509 0.0006271   9.949 9.35e-06 ***
## Residuals   42 0.002648 0.0000630
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 2 observations deleted due to missingness</code></pre>
                </div>
                <div id="perform-multiple-comparisons" class="section level5">
                  <h5>Perform multiple comparisons</h5>
                  <ul>
                    <li>
                      The RColorBrewer package is loaded for additional colors,
                      and a palette is added to R. Use brackets after vector
                      name for specific colors when creating plots (for example:
                      <code>col=set1[4:14]</code>).
                    </li>
                    <li>
                      I chose the <code>Pastel1</code> palette. “I’m a very
                      vivacious man… Earth tones are so passe. I’m a pastel
                      man.” Larry David, <em>Curb Your Enthusiasm</em> S01E05
                      “Interior Decorator”
                    </li>
                  </ul>
                  <div class="sourceCode" id="cb116">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1"></a><span class="co"># Standard Tukey Honest Significant Differences</span></span>
<span id="cb116-2"><a href="#cb116-2"></a><span class="kw">TukeyHSD</span>(adiposeaov1)</span></code></pre>
                  </div>
                  <pre><code>##   Tukey multiple comparisons of means
##     95% family-wise confidence level
##
## Fit: aov(formula = adipose ~ diet, data = adipose_weights)
##
## $diet
##                     diff          lwr         upr     p adj
## WD-AIN       0.013272442  0.002278112 0.024266773 0.0109966
## WDSG-AIN     0.007630272 -0.003102252 0.018362795 0.2716464
## WDTP-AIN     0.018925192  0.008192669 0.029657715 0.0000924
## WDTPSG-AIN   0.020266814  0.009534291 0.030999337 0.0000292
## WDSG-WD     -0.005642171 -0.016038167 0.004753826 0.5389261
## WDTP-WD      0.005652750 -0.004743247 0.016048746 0.5371109
## WDTPSG-WD    0.006994372 -0.003401625 0.017390368 0.3242717
## WDTP-WDSG    0.011294920  0.001176200 0.021413640 0.0219233
## WDTPSG-WDSG  0.012636543  0.002517823 0.022755262 0.0079286
## WDTPSG-WDTP  0.001341622 -0.008777098 0.011460342 0.9955000</code></pre>
                  <div class="sourceCode" id="cb118">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1"></a><span class="kw">par</span>(<span class="dt">las =</span> <span class="dv">2</span>, <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">8</span>, <span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb118-2"><a href="#cb118-2"></a><span class="kw">plot</span>(<span class="kw">TukeyHSD</span>(adiposeaov1))</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-adipose-weights-02-run-multcomp-1.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb119">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1"></a><span class="co"># More comprehensive method in multcomp package</span></span>
<span id="cb119-2"><a href="#cb119-2"></a><span class="kw">par</span>(<span class="dt">las =</span> <span class="dv">1</span>, <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">2</span>))</span>
<span id="cb119-3"><a href="#cb119-3"></a><span class="kw">library</span>(multcomp)</span>
<span id="cb119-4"><a href="#cb119-4"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb119-5"><a href="#cb119-5"></a>tuk &lt;-<span class="st"> </span><span class="kw">glht</span>(adiposeaov1, <span class="dt">linfct =</span> <span class="kw">mcp</span>(<span class="dt">diet =</span> <span class="st">&#39;Tukey&#39;</span>))</span>
<span id="cb119-6"><a href="#cb119-6"></a>means &lt;-<span class="st"> </span><span class="kw">aggregate</span>(adipose <span class="op">~</span><span class="st"> </span>diet, adipose_weights, mean)</span>
<span id="cb119-7"><a href="#cb119-7"></a><span class="kw">plot</span>(<span class="kw">cld</span>(tuk, <span class="dt">level =</span> <span class="fl">.05</span>), <span class="dt">col =</span> <span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&#39;Pastel1&#39;</span>))</span>
<span id="cb119-8"><a href="#cb119-8"></a><span class="kw">points</span>(means, <span class="dt">pch =</span> <span class="dv">23</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-adipose-weights-02-run-multcomp-2.png"
                      width="672"
                    />
                  </p>
                </div>
                <div id="test-assumptions-2" class="section level5">
                  <h5>Test assumptions</h5>
                  <div class="sourceCode" id="cb120">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1"></a><span class="kw">plot</span>(adiposeaov1)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-adipose-weights-03-assumptions-plot-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-adipose-weights-03-assumptions-plot-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-adipose-weights-03-assumptions-plot-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-adipose-weights-03-assumptions-plot-4.png"
                      width="672"
                    />
                  </p>
                  <div id="normality-2" class="section level6">
                    <h6>Normality</h6>
                    <div class="sourceCode" id="cb121">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1"></a><span class="co"># Shapiro-Wilk W test</span></span>
<span id="cb121-2"><a href="#cb121-2"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(adiposeaov1))</span></code></pre>
                    </div>
                    <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(adiposeaov1)
## W = 0.97688, p-value = 0.4704</code></pre>
                  </div>
                  <div id="equality-of-variances-2" class="section level6">
                    <h6>Equality of variances</h6>
                    <div class="sourceCode" id="cb123">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1"></a><span class="kw">library</span>(car)</span>
<span id="cb123-2"><a href="#cb123-2"></a><span class="kw">leveneTest</span>(adipose_weights<span class="op">$</span>adipose, adipose_weights<span class="op">$</span>diet)</span></code></pre>
                    </div>
                    <pre><code>## Levene&#39;s Test for Homogeneity of Variance (center = median)
##       Df F value Pr(&gt;F)
## group  4  0.5104 0.7284
##       42</code></pre>
                  </div>
                </div>
                <div
                  id="perform-anova-in-lm-for-comparison"
                  class="section level5"
                >
                  <h5>Perform ANOVA in lm for comparison</h5>
                  <div class="sourceCode" id="cb125">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1"></a>adiposelm1 &lt;-<span class="st"> </span><span class="kw">lm</span>(adipose_weights<span class="op">$</span>adipose <span class="op">~</span><span class="st"> </span>adipose_weights<span class="op">$</span>diet)</span>
<span id="cb125-2"><a href="#cb125-2"></a><span class="kw">anova</span>(adiposelm1)</span></code></pre>
                  </div>
                  <pre><code>## Analysis of Variance Table
##
## Response: adipose_weights$adipose
##                      Df    Sum Sq    Mean Sq F value    Pr(&gt;F)
## adipose_weights$diet  4 0.0025086 0.00062714  9.9489 9.349e-06 ***
## Residuals            42 0.0026475 0.00006304
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                </div>
                <div
                  id="perform-regression-diagnostics-and-test-assumptions-4"
                  class="section level5"
                >
                  <h5>Perform regression diagnostics and test assumptions</h5>
                  <div class="sourceCode" id="cb127">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb127-2"><a href="#cb127-2"></a>adiposegvlmalm1 &lt;-<span class="st"> </span><span class="kw">gvlma</span>(adiposelm1)</span>
<span id="cb127-3"><a href="#cb127-3"></a><span class="kw">summary</span>(adiposegvlmalm1)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = adipose_weights$adipose ~ adipose_weights$diet)
##
## Residuals:
##        Min         1Q     Median         3Q        Max
## -0.0177688 -0.0054835 -0.0009354  0.0053295  0.0181588
##
## Coefficients:
##                            Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)                0.018048   0.002807   6.430 9.57e-08 ***
## adipose_weights$dietWD     0.013272   0.003858   3.440  0.00133 **
## adipose_weights$dietWDSG   0.007630   0.003766   2.026  0.04914 *
## adipose_weights$dietWDTP   0.018925   0.003766   5.025 9.81e-06 ***
## adipose_weights$dietWDTPSG 0.020267   0.003766   5.381 3.06e-06 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 0.00794 on 42 degrees of freedom
##   (2 observations deleted due to missingness)
## Multiple R-squared:  0.4865, Adjusted R-squared:  0.4376
## F-statistic: 9.949 on 4 and 42 DF,  p-value: 9.349e-06
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = adiposelm1)
##
##                        Value p-value                Decision
## Global Stat        1.372e+00  0.8491 Assumptions acceptable.
## Skewness           1.174e+00  0.2786 Assumptions acceptable.
## Kurtosis           3.779e-02  0.8459 Assumptions acceptable.
## Link Function      3.133e-15  1.0000 Assumptions acceptable.
## Heteroscedasticity 1.598e-01  0.6893 Assumptions acceptable.</code></pre>
                  <div class="sourceCode" id="cb129">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(adiposelm1))</span></code></pre>
                  </div>
                  <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(adiposelm1)
## W = 0.97688, p-value = 0.4704</code></pre>
                  <div class="sourceCode" id="cb131">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1"></a><span class="kw">cor.test</span>(<span class="kw">fitted</span>(adiposelm1), <span class="kw">abs</span>(<span class="kw">rstandard</span>(adiposelm1)), <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(fitted(adiposelm1), abs(rstandard(adiposelm1)), : Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  fitted(adiposelm1) and abs(rstandard(adiposelm1))
## S = 14364, p-value = 0.2547
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##       rho
## 0.1695033</code></pre>
                  <div class="sourceCode" id="cb134">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1"></a><span class="kw">library</span>(car)</span>
<span id="cb134-2"><a href="#cb134-2"></a><span class="kw">durbinWatsonTest</span>(adiposelm1)</span></code></pre>
                  </div>
                  <pre><code>##  lag Autocorrelation D-W Statistic p-value
##    1      -0.1135751      2.212731    0.88
##  Alternative hypothesis: rho != 0</code></pre>
                  <div class="sourceCode" id="cb136">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1"></a><span class="kw">outlierTest</span>(adiposelm1)</span></code></pre>
                  </div>
                  <pre><code>## No Studentized residuals with Bonferroni p &lt; 0.05
## Largest |rstudent|:
##    rstudent unadjusted p-value Bonferroni p
## 13 2.584726           0.013403      0.62992</code></pre>
                  <div class="sourceCode" id="cb138">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1"></a><span class="kw">leveneTest</span>(adipose_weights<span class="op">$</span>adipose, adipose_weights<span class="op">$</span>diet)</span></code></pre>
                  </div>
                  <pre><code>## Levene&#39;s Test for Homogeneity of Variance (center = median)
##       Df F value Pr(&gt;F)
## group  4  0.5104 0.7284
##       42</code></pre>
                </div>
              </div>
              <div id="two-way-factorial-anova-example" class="section level4">
                <h4>Two-way factorial ANOVA example</h4>
                <p>R in Action 9.5, ToothGrowth dataset</p>
                <blockquote>
                  <p>
                    This example uses the ToothGrowth dataset in the base
                    installation to demonstrate a two-way between-groups ANOVA.
                    Sixty guinea pigs are randomly assigned to receive one of
                    three levels of ascorbic acid (0.5, 1, or 2mg), and one of
                    two delivery methods (orange juice or Vitamin C), under the
                    restriction that each treatment combination has 10 guinea
                    pigs. The dependent variable is tooth length.
                  </p>
                </blockquote>
                <div id="load-dataset" class="section level5">
                  <h5>Load dataset</h5>
                  <div class="sourceCode" id="cb140">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1"></a><span class="kw">attach</span>(ToothGrowth)</span>
<span id="cb140-2"><a href="#cb140-2"></a><span class="kw">table</span>(supp, dose)</span></code></pre>
                  </div>
                  <pre><code>##     dose
## supp 0.5  1  2
##   OJ  10 10 10
##   VC  10 10 10</code></pre>
                  <div class="sourceCode" id="cb142">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1"></a><span class="kw">aggregate</span>(len, <span class="dt">by =</span> <span class="kw">list</span>(supp, dose), <span class="dt">FUN =</span> mean)</span></code></pre>
                  </div>
                  <pre><code>##   Group.1 Group.2     x
## 1      OJ     0.5 13.23
## 2      VC     0.5  7.98
## 3      OJ     1.0 22.70
## 4      VC     1.0 16.77
## 5      OJ     2.0 26.06
## 6      VC     2.0 26.14</code></pre>
                  <div class="sourceCode" id="cb144">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1"></a><span class="kw">aggregate</span>(len, <span class="dt">by =</span> <span class="kw">list</span>(supp, dose), <span class="dt">FUN =</span> sd)</span></code></pre>
                  </div>
                  <pre><code>##   Group.1 Group.2        x
## 1      OJ     0.5 4.459709
## 2      VC     0.5 2.746634
## 3      OJ     1.0 3.910953
## 4      VC     1.0 2.515309
## 5      OJ     2.0 2.655058
## 6      VC     2.0 4.797731</code></pre>
                </div>
                <div
                  id="perform-factorial-analysis-of-variance"
                  class="section level5"
                >
                  <h5>Perform factorial analysis of variance</h5>
                  <div class="sourceCode" id="cb146">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1"></a>fit &lt;-<span class="st"> </span><span class="kw">lm</span>(len <span class="op">~</span><span class="st"> </span>supp <span class="op">*</span><span class="st"> </span>dose)</span>
<span id="cb146-2"><a href="#cb146-2"></a><span class="kw">anova</span>(fit)</span></code></pre>
                  </div>
                  <pre><code>## Analysis of Variance Table
##
## Response: len
##           Df  Sum Sq Mean Sq  F value    Pr(&gt;F)
## supp       1  205.35  205.35  12.3170 0.0008936 ***
## dose       1 2224.30 2224.30 133.4151 &lt; 2.2e-16 ***
## supp:dose  1   88.92   88.92   5.3335 0.0246314 *
## Residuals 56  933.63   16.67
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                </div>
                <div
                  id="perform-regression-diagnostics-and-test-assumptions-5"
                  class="section level5"
                >
                  <h5>Perform regression diagnostics and test assumptions</h5>
                  <div class="sourceCode" id="cb148">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb148-2"><a href="#cb148-2"></a>gvlmafit &lt;-<span class="st"> </span><span class="kw">gvlma</span>(fit)</span>
<span id="cb148-3"><a href="#cb148-3"></a><span class="kw">summary</span>(gvlmafit)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = len ~ supp * dose)
##
## Residuals:
##     Min      1Q  Median      3Q     Max
## -8.2264 -2.8462  0.0504  2.2893  7.9386
##
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)   11.550      1.581   7.304 1.09e-09 ***
## suppVC        -8.255      2.236  -3.691 0.000507 ***
## dose           7.811      1.195   6.534 2.03e-08 ***
## suppVC:dose    3.904      1.691   2.309 0.024631 *
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 4.083 on 56 degrees of freedom
## Multiple R-squared:  0.7296, Adjusted R-squared:  0.7151
## F-statistic: 50.36 on 3 and 56 DF,  p-value: 6.521e-16
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = fit)
##
##                      Value  p-value                   Decision
## Global Stat        10.6646 0.030604 Assumptions NOT satisfied!
## Skewness            0.0934 0.759902    Assumptions acceptable.
## Kurtosis            1.0752 0.299778    Assumptions acceptable.
## Link Function       8.7812 0.003043 Assumptions NOT satisfied!
## Heteroscedasticity  0.7148 0.397854    Assumptions acceptable.</code></pre>
                  <div class="sourceCode" id="cb150">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(fit))</span></code></pre>
                  </div>
                  <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(fit)
## W = 0.98147, p-value = 0.4936</code></pre>
                  <div class="sourceCode" id="cb152">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1"></a><span class="kw">cor.test</span>(<span class="kw">fitted</span>(fit), <span class="kw">abs</span>(<span class="kw">rstandard</span>(fit)), <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(fitted(fit), abs(rstandard(fit)), method = &quot;spearman&quot;): Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  fitted(fit) and abs(rstandard(fit))
## S = 35096, p-value = 0.8506
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##        rho
## 0.02483306</code></pre>
                  <div class="sourceCode" id="cb155">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1"></a><span class="kw">library</span>(car)</span>
<span id="cb155-2"><a href="#cb155-2"></a><span class="kw">durbinWatsonTest</span>(fit)</span></code></pre>
                  </div>
                  <pre><code>##  lag Autocorrelation D-W Statistic p-value
##    1       0.1333076       1.68846   0.086
##  Alternative hypothesis: rho != 0</code></pre>
                  <div class="sourceCode" id="cb157">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1"></a><span class="kw">outlierTest</span>(fit)</span></code></pre>
                  </div>
                  <pre><code>## No Studentized residuals with Bonferroni p &lt; 0.05
## Largest |rstudent|:
##     rstudent unadjusted p-value Bonferroni p
## 22 -2.185495           0.033129           NA</code></pre>
                </div>
                <div id="generate-plots-4" class="section level5">
                  <h5>Generate plots</h5>
                  <div class="sourceCode" id="cb159">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1"></a><span class="kw">plot</span>(gvlmafit, <span class="dt">onepage =</span> <span class="ot">FALSE</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-4.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-5.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-6.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-7.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-8.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb160">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1"></a><span class="kw">plot</span>(fit, <span class="dt">which =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-9.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-10.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-11.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb161">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1"></a><span class="kw">interaction.plot</span>(</span>
<span id="cb161-2"><a href="#cb161-2"></a>  dose,</span>
<span id="cb161-3"><a href="#cb161-3"></a>  supp,</span>
<span id="cb161-4"><a href="#cb161-4"></a>  len,</span>
<span id="cb161-5"><a href="#cb161-5"></a>  <span class="dt">type =</span> <span class="st">&#39;b&#39;</span>,</span>
<span id="cb161-6"><a href="#cb161-6"></a>  <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&#39;red&#39;</span>, <span class="st">&#39;blue&#39;</span>),</span>
<span id="cb161-7"><a href="#cb161-7"></a>  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">16</span>, <span class="dv">18</span>),</span>
<span id="cb161-8"><a href="#cb161-8"></a>  <span class="dt">main =</span> <span class="st">&#39;Interaction between Dose and Supplement Type&#39;</span></span>
<span id="cb161-9"><a href="#cb161-9"></a>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-tooth-04-plots-12.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
              <div
                id="two-way-factorial-anova-example-1"
                class="section level4"
              >
                <h4>Two-way factorial ANOVA example</h4>
                <p>
                  Atheroma thickness data from Smith BW et al, J Ultrasound Med
                  2012 (doi:
                  <a href="https://doi.org/10.7863/jum.2012.31.5.711"
                    >10.7863/jum.2012.31.5.711</a
                  >).
                </p>
                <div id="import-dataset-6" class="section level5">
                  <h5>Import dataset</h5>
                  <div class="sourceCode" id="cb162">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1"></a>fall2009thickness &lt;-</span>
<span id="cb162-2"><a href="#cb162-2"></a><span class="st">  </span><span class="kw">read.table</span>(</span>
<span id="cb162-3"><a href="#cb162-3"></a>    <span class="st">&#39;data/anova_factorial_jum2012_atheroma_thickness.csv&#39;</span>,</span>
<span id="cb162-4"><a href="#cb162-4"></a>    <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb162-5"><a href="#cb162-5"></a>    <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span></span>
<span id="cb162-6"><a href="#cb162-6"></a>  )</span>
<span id="cb162-7"><a href="#cb162-7"></a><span class="kw">attach</span>(fall2009thickness)</span>
<span id="cb162-8"><a href="#cb162-8"></a><span class="kw">head</span>(fall2009thickness)</span></code></pre>
                  </div>
                  <pre><code>##     id      uca mpa thickness
## 1 L604 Definity   0        35
## 2 L611 Definity   0        10
## 3 L614 Definity   0        70
## 4 L617 Definity   0        60
## 5 L624 Definity   0       100
## 6 L593   Saline   0        50</code></pre>
                </div>
                <div
                  id="perform-factorial-analysis-of-variance-1"
                  class="section level5"
                >
                  <h5>Perform factorial analysis of variance</h5>
                  <div class="sourceCode" id="cb164">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1"></a>thick_fit &lt;-<span class="st"> </span><span class="kw">lm</span>(thickness <span class="op">~</span><span class="st"> </span>uca <span class="op">*</span><span class="st"> </span>mpa, <span class="dt">data =</span> fall2009thickness)</span>
<span id="cb164-2"><a href="#cb164-2"></a><span class="kw">anova</span>(thick_fit)</span></code></pre>
                  </div>
                  <pre><code>## Analysis of Variance Table
##
## Response: thickness
##           Df Sum Sq Mean Sq F value  Pr(&gt;F)
## uca        1   9619  9618.9  4.6436 0.03775 *
## mpa        1   5255  5255.4  2.5371 0.11971
## uca:mpa    1   3285  3285.1  1.5859 0.21580
## Residuals 37  76643  2071.4
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                </div>
                <div
                  id="perform-regression-diagnostics-and-test-assumptions-6"
                  class="section level5"
                >
                  <h5>Perform regression diagnostics and test assumptions</h5>
                  <div class="sourceCode" id="cb166">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb166-2"><a href="#cb166-2"></a>gvlmathick_fit &lt;-<span class="st"> </span><span class="kw">gvlma</span>(thick_fit)</span>
<span id="cb166-3"><a href="#cb166-3"></a><span class="kw">summary</span>(gvlmathick_fit)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = thickness ~ uca * mpa, data = fall2009thickness)
##
## Residuals:
##    Min     1Q Median     3Q    Max
## -92.81 -26.34 -11.70  13.76 141.39
##
## Coefficients:
##               Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)     49.026     16.802   2.918  0.00596 **
## ucaSaline        7.320     23.838   0.307  0.76053
## mpa              3.608     12.762   0.283  0.77898
## ucaSaline:mpa   23.278     18.484   1.259  0.21580
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 45.51 on 37 degrees of freedom
## Multiple R-squared:  0.1915, Adjusted R-squared:  0.126
## F-statistic: 2.922 on 3 and 37 DF,  p-value: 0.04662
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = thick_fit)
##
##                      Value  p-value                   Decision
## Global Stat        15.0581 0.004582 Assumptions NOT satisfied!
## Skewness            5.2041 0.022534 Assumptions NOT satisfied!
## Kurtosis            4.1197 0.042388 Assumptions NOT satisfied!
## Link Function       0.9562 0.328138    Assumptions acceptable.
## Heteroscedasticity  4.7782 0.028823 Assumptions NOT satisfied!</code></pre>
                  <div class="sourceCode" id="cb168">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(thick_fit))</span></code></pre>
                  </div>
                  <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(thick_fit)
## W = 0.95015, p-value = 0.07084</code></pre>
                  <div class="sourceCode" id="cb170">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1"></a><span class="kw">cor.test</span>(<span class="kw">fitted</span>(thick_fit), <span class="kw">abs</span>(<span class="kw">rstandard</span>(thick_fit)), <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(fitted(thick_fit), abs(rstandard(thick_fit)), : Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  fitted(thick_fit) and abs(rstandard(thick_fit))
## S = 9194.5, p-value = 0.2121
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##      rho
## 0.199083</code></pre>
                  <div class="sourceCode" id="cb173">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1"></a><span class="kw">library</span>(car)</span>
<span id="cb173-2"><a href="#cb173-2"></a><span class="kw">durbinWatsonTest</span>(thick_fit)</span></code></pre>
                  </div>
                  <pre><code>##  lag Autocorrelation D-W Statistic p-value
##    1      -0.3832382       2.76177   0.014
##  Alternative hypothesis: rho != 0</code></pre>
                  <div class="sourceCode" id="cb175">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1"></a><span class="kw">outlierTest</span>(thick_fit)</span></code></pre>
                  </div>
                  <pre><code>##    rstudent unadjusted p-value Bonferroni p
## 27 3.699668         0.00071694     0.029395</code></pre>
                </div>
                <div id="generate-plots-5" class="section level5">
                  <h5>Generate plots</h5>
                  <div class="sourceCode" id="cb177">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb177-1"><a href="#cb177-1"></a><span class="kw">plot</span>(gvlmathick_fit, <span class="dt">onepage =</span> <span class="ot">FALSE</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-4.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-5.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-6.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-7.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-8.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb178">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1"></a><span class="kw">plot</span>(thick_fit, <span class="dt">which =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-9.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-10.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-11.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb179">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1"></a><span class="kw">interaction.plot</span>(</span>
<span id="cb179-2"><a href="#cb179-2"></a>  uca,</span>
<span id="cb179-3"><a href="#cb179-3"></a>  mpa,</span>
<span id="cb179-4"><a href="#cb179-4"></a>  thickness,</span>
<span id="cb179-5"><a href="#cb179-5"></a>  <span class="dt">type =</span> <span class="st">&#39;b&#39;</span>,</span>
<span id="cb179-6"><a href="#cb179-6"></a>  <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&#39;red&#39;</span>, <span class="st">&#39;blue&#39;</span>),</span>
<span id="cb179-7"><a href="#cb179-7"></a>  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">16</span>, <span class="dv">18</span>),</span>
<span id="cb179-8"><a href="#cb179-8"></a>  <span class="dt">main =</span> <span class="st">&#39;Interaction between contrast agent and pressure level&#39;</span></span>
<span id="cb179-9"><a href="#cb179-9"></a>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-04-plots-12.png"
                      width="672"
                    />
                  </p>
                </div>
                <div
                  id="log-transform-to-correct-non-normality-heteroscedasticity-and-outlier"
                  class="section level5"
                >
                  <h5>
                    Log transform to correct non-normality, heteroscedasticity,
                    and outlier
                  </h5>
                  <div class="sourceCode" id="cb180">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1"></a>log_thick_fit &lt;-<span class="st"> </span><span class="kw">lm</span>(<span class="kw">log</span>(thickness) <span class="op">~</span><span class="st"> </span>uca <span class="op">*</span><span class="st"> </span>mpa, <span class="dt">data =</span> fall2009thickness)</span>
<span id="cb180-2"><a href="#cb180-2"></a><span class="kw">anova</span>(log_thick_fit)</span></code></pre>
                  </div>
                  <pre><code>## Analysis of Variance Table
##
## Response: log(thickness)
##           Df  Sum Sq Mean Sq F value  Pr(&gt;F)
## uca        1  2.4372 2.43722  4.1806 0.04805 *
## mpa        1  0.6951 0.69511  1.1923 0.28192
## uca:mpa    1  0.3084 0.30843  0.5291 0.47158
## Residuals 37 21.5703 0.58298
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                </div>
                <div
                  id="perform-regression-diagnostics-and-test-assumptions-7"
                  class="section level5"
                >
                  <h5>Perform regression diagnostics and test assumptions</h5>
                  <div class="sourceCode" id="cb182">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb182-2"><a href="#cb182-2"></a>gvlma_log_thick_fit &lt;-<span class="st"> </span><span class="kw">gvlma</span>(log_thick_fit)</span>
<span id="cb182-3"><a href="#cb182-3"></a><span class="kw">summary</span>(gvlma_log_thick_fit)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = log(thickness) ~ uca * mpa, data = fall2009thickness)
##
## Residuals:
##      Min       1Q   Median       3Q      Max
## -1.51653 -0.50113  0.09291  0.44825  1.18425
##
## Coefficients:
##               Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)    3.64610    0.28187  12.935 2.67e-15 ***
## ucaSaline      0.26316    0.39991   0.658    0.515
## mpa            0.06159    0.21410   0.288    0.775
## ucaSaline:mpa  0.22555    0.31009   0.727    0.472
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 0.7635 on 37 degrees of freedom
## Multiple R-squared:  0.1376, Adjusted R-squared:  0.06764
## F-statistic: 1.967 on 3 and 37 DF,  p-value: 0.1358
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = log_thick_fit)
##
##                     Value p-value                Decision
## Global Stat        3.4821  0.4806 Assumptions acceptable.
## Skewness           2.1479  0.1428 Assumptions acceptable.
## Kurtosis           0.5245  0.4689 Assumptions acceptable.
## Link Function      0.6395  0.4239 Assumptions acceptable.
## Heteroscedasticity 0.1701  0.6800 Assumptions acceptable.</code></pre>
                  <div class="sourceCode" id="cb184">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(log_thick_fit))</span></code></pre>
                  </div>
                  <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(log_thick_fit)
## W = 0.93909, p-value = 0.02924</code></pre>
                  <div class="sourceCode" id="cb186">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb186-1"><a href="#cb186-1"></a><span class="kw">cor.test</span>(<span class="kw">fitted</span>(log_thick_fit),</span>
<span id="cb186-2"><a href="#cb186-2"></a>         <span class="kw">abs</span>(<span class="kw">rstandard</span>(log_thick_fit)),</span>
<span id="cb186-3"><a href="#cb186-3"></a>         <span class="dt">method =</span> <span class="st">&#39;spearman&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## Warning in cor.test.default(fitted(log_thick_fit), abs(rstandard(log_thick_fit)), : Cannot compute exact p-value with ties</code></pre>
                  <pre><code>##
##  Spearman&#39;s rank correlation rho
##
## data:  fitted(log_thick_fit) and abs(rstandard(log_thick_fit))
## S = 12485, p-value = 0.5861
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##         rho
## -0.08758111</code></pre>
                  <div class="sourceCode" id="cb189">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1"></a><span class="kw">library</span>(car)</span>
<span id="cb189-2"><a href="#cb189-2"></a><span class="kw">durbinWatsonTest</span>(log_thick_fit)</span></code></pre>
                  </div>
                  <pre><code>##  lag Autocorrelation D-W Statistic p-value
##    1      -0.3411439      2.681506   0.072
##  Alternative hypothesis: rho != 0</code></pre>
                  <div class="sourceCode" id="cb191">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1"></a><span class="kw">outlierTest</span>(log_thick_fit)</span></code></pre>
                  </div>
                  <pre><code>## No Studentized residuals with Bonferroni p &lt; 0.05
## Largest |rstudent|:
##     rstudent unadjusted p-value Bonferroni p
## 37 -2.276215           0.028882           NA</code></pre>
                </div>
                <div id="generate-plots-6" class="section level5">
                  <h5>Generate plots</h5>
                  <div class="sourceCode" id="cb193">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb193-1"><a href="#cb193-1"></a><span class="kw">plot</span>(gvlma_log_thick_fit, <span class="dt">onepage =</span> <span class="ot">FALSE</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-4.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-5.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-6.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-7.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-8.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb194">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb194-1"><a href="#cb194-1"></a><span class="kw">plot</span>(log_thick_fit, <span class="dt">which =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-9.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-10.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-11.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb195">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb195-1"><a href="#cb195-1"></a><span class="kw">interaction.plot</span>(</span>
<span id="cb195-2"><a href="#cb195-2"></a>  uca,</span>
<span id="cb195-3"><a href="#cb195-3"></a>  mpa,</span>
<span id="cb195-4"><a href="#cb195-4"></a>  thickness,</span>
<span id="cb195-5"><a href="#cb195-5"></a>  <span class="dt">type =</span> <span class="st">&#39;b&#39;</span>,</span>
<span id="cb195-6"><a href="#cb195-6"></a>  <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&#39;red&#39;</span>, <span class="st">&#39;blue&#39;</span>),</span>
<span id="cb195-7"><a href="#cb195-7"></a>  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">16</span>, <span class="dv">18</span>),</span>
<span id="cb195-8"><a href="#cb195-8"></a>  <span class="dt">main =</span> <span class="st">&#39;Interaction between contrast agent and pressure level&#39;</span></span>
<span id="cb195-9"><a href="#cb195-9"></a>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-anova-2way-fall2009thickness-07-transform-plots-12.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
              <div id="anova-compact-letter-display" class="section level4">
                <h4>ANOVA Compact Letter Display</h4>
                <p>
                  Compact Letter Display (CLD) for ANOVA multiple comparisons
                  with the <code>multcomp</code> package
                </p>
                <div class="sourceCode" id="cb196">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb196-1"><a href="#cb196-1"></a><span class="co"># multiple comparison procedures</span></span>
<span id="cb196-2"><a href="#cb196-2"></a><span class="co"># set up a one-way ANOVA</span></span>
<span id="cb196-3"><a href="#cb196-3"></a><span class="kw">library</span>(multcomp)</span>
<span id="cb196-4"><a href="#cb196-4"></a><span class="kw">data</span>(warpbreaks)</span>
<span id="cb196-5"><a href="#cb196-5"></a>amod &lt;-<span class="st"> </span><span class="kw">aov</span>(breaks <span class="op">~</span><span class="st"> </span>tension, <span class="dt">data =</span> warpbreaks)</span>
<span id="cb196-6"><a href="#cb196-6"></a><span class="co"># specify all pair-wise comparisons among levels of variable &#39;tension&#39;</span></span>
<span id="cb196-7"><a href="#cb196-7"></a>tuk &lt;-<span class="st"> </span><span class="kw">glht</span>(amod, <span class="dt">linfct =</span> <span class="kw">mcp</span>(<span class="dt">tension =</span> <span class="st">&#39;Tukey&#39;</span>))</span>
<span id="cb196-8"><a href="#cb196-8"></a><span class="co"># extract information</span></span>
<span id="cb196-9"><a href="#cb196-9"></a>tuk.cld &lt;-<span class="st"> </span><span class="kw">cld</span>(tuk)</span>
<span id="cb196-10"><a href="#cb196-10"></a><span class="co"># use sufficiently large upper margin</span></span>
<span id="cb196-11"><a href="#cb196-11"></a>old.par &lt;-<span class="st"> </span><span class="kw">par</span>(<span class="dt">mai =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">1</span>))</span>
<span id="cb196-12"><a href="#cb196-12"></a><span class="co"># plot</span></span>
<span id="cb196-13"><a href="#cb196-13"></a><span class="kw">plot</span>(tuk.cld)</span></code></pre>
                </div>
                <p>
                  <img
                    src="code_files/figure-html/code-anova-multcomp-cld-1.png"
                    width="672"
                  />
                </p>
                <div class="sourceCode" id="cb197">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1"></a><span class="kw">par</span>(old.par)</span></code></pre>
                </div>
              </div>
            </div>
            <div id="repeated-measures-anova" class="section level3">
              <h3>Repeated measures ANOVA</h3>
              <div id="repeated-measures-anova-example" class="section level4">
                <h4>Repeated measures ANOVA example</h4>
                <p>R in Action 2nd ed, 9.6, CO2 dataset</p>
                <p>
                  <strong
                    >note that the code from the first edition may return
                    errors</strong
                  >
                </p>
                <blockquote>
                  <p>
                    In addition to the base installation, we’ll be using the
                    <code>car</code>, <code>gplots</code>, <code>HH</code>,
                    <code>rrcov</code>, and <code>mvoutlier</code> packages in
                    our examples. Be sure to install them before trying out the
                    sample code.
                  </p>
                  <p>
                    The CO2 dataset included in the base installation contains
                    the results of a study of cold tolerance in Northern and
                    Southern plants of the grass species Echinochloa crus-galli
                    (Potvin, Lechowicz, &amp; Tardif, 1990). The photosynthetic
                    rates of chilled plants were compared with the
                    photosynthetic rates of nonchilled plants at several ambient
                    CO2 concentrations. Half the plants were from Quebec, and
                    half were from Mississippi.
                  </p>
                  <p>
                    In this example, we’ll focus on chilled plants. The
                    dependent variable is carbon dioxide uptake (uptake) in
                    ml/L, and the independent variables are Type (Quebec versus
                    Mississippi) and ambient CO2 concentration (conc) with seven
                    levels (ranging from 95 to 1000 umol/m^2 sec). Type is a
                    between-groups factor, and conc is a within- groups factor.
                    Type is already stored as a factor, but you’ll need to
                    convert conc to a factor before continuing. The analysis is
                    presented in the next listing.
                  </p>
                </blockquote>
                <div class="sourceCode" id="cb198">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1"></a>CO2<span class="op">$</span>conc &lt;-<span class="st"> </span><span class="kw">factor</span>(CO2<span class="op">$</span>conc)</span>
<span id="cb198-2"><a href="#cb198-2"></a>w1b1 &lt;-<span class="st"> </span><span class="kw">subset</span>(CO2, Treatment <span class="op">==</span><span class="st"> &#39;chilled&#39;</span>)</span>
<span id="cb198-3"><a href="#cb198-3"></a>co2_fit &lt;-<span class="st"> </span><span class="kw">aov</span>(uptake <span class="op">~</span><span class="st"> </span>conc <span class="op">*</span><span class="st"> </span>Type <span class="op">+</span><span class="st"> </span><span class="kw">Error</span>(Plant <span class="op">/</span><span class="st"> </span>(conc)), w1b1)</span>
<span id="cb198-4"><a href="#cb198-4"></a><span class="kw">summary</span>(co2_fit)</span></code></pre>
                </div>
                <pre><code>##
## Error: Plant
##           Df Sum Sq Mean Sq F value  Pr(&gt;F)
## Type       1 2667.2  2667.2   60.41 0.00148 **
## Residuals  4  176.6    44.1
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Error: Plant:conc
##           Df Sum Sq Mean Sq F value   Pr(&gt;F)
## conc       6 1472.4  245.40   52.52 1.26e-12 ***
## conc:Type  6  428.8   71.47   15.30 3.75e-07 ***
## Residuals 24  112.1    4.67
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                <div class="sourceCode" id="cb200">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1"></a><span class="kw">par</span>(<span class="dt">las =</span> <span class="dv">2</span>)</span>
<span id="cb200-2"><a href="#cb200-2"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb200-3"><a href="#cb200-3"></a><span class="kw">with</span>(</span>
<span id="cb200-4"><a href="#cb200-4"></a>  w1b1,</span>
<span id="cb200-5"><a href="#cb200-5"></a>  <span class="kw">interaction.plot</span>(</span>
<span id="cb200-6"><a href="#cb200-6"></a>    conc,</span>
<span id="cb200-7"><a href="#cb200-7"></a>    Type,</span>
<span id="cb200-8"><a href="#cb200-8"></a>    uptake,</span>
<span id="cb200-9"><a href="#cb200-9"></a>    <span class="dt">type =</span> <span class="st">&#39;b&#39;</span>,</span>
<span id="cb200-10"><a href="#cb200-10"></a>    <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&#39;red&#39;</span>, <span class="st">&#39;blue&#39;</span>),</span>
<span id="cb200-11"><a href="#cb200-11"></a>    <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">16</span>, <span class="dv">18</span>),</span>
<span id="cb200-12"><a href="#cb200-12"></a>    <span class="dt">main =</span> <span class="st">&#39;Interaction Plot for Plant Type and Concentration&#39;</span></span>
<span id="cb200-13"><a href="#cb200-13"></a>  )</span>
<span id="cb200-14"><a href="#cb200-14"></a>)</span></code></pre>
                </div>
                <p>
                  <img
                    src="code_files/figure-html/code-anova-rm-co2-02-plot-1.png"
                    width="672"
                  />
                </p>
                <div class="sourceCode" id="cb201">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1"></a><span class="kw">boxplot</span>(</span>
<span id="cb201-2"><a href="#cb201-2"></a>  uptake <span class="op">~</span><span class="st"> </span>Type <span class="op">*</span><span class="st"> </span>conc,</span>
<span id="cb201-3"><a href="#cb201-3"></a>  <span class="dt">data =</span> w1b1,</span>
<span id="cb201-4"><a href="#cb201-4"></a>  <span class="dt">col =</span> (<span class="kw">c</span>(<span class="st">&#39;gold&#39;</span>, <span class="st">&#39;green&#39;</span>)),</span>
<span id="cb201-5"><a href="#cb201-5"></a>  <span class="dt">main =</span> <span class="st">&#39;Chilled Quebec and Mississippi Plants&#39;</span>,</span>
<span id="cb201-6"><a href="#cb201-6"></a>  <span class="dt">ylab =</span> <span class="st">&#39;Carbon dioxide uptake rate (umol/m^2 sec)&#39;</span></span>
<span id="cb201-7"><a href="#cb201-7"></a>)</span></code></pre>
                </div>
                <p>
                  <img
                    src="code_files/figure-html/code-anova-rm-co2-02-plot-2.png"
                    width="672"
                  />
                </p>
                <blockquote>
                  <p>
                    <strong>The many approaches to mixed-model designs</strong>
                  </p>
                  <p>
                    The CO2 example in this section was analyzed using a
                    traditional repeated measures ANOVA. The approach assumes
                    that the covariance matrix for any within-groups factor
                    follows a specified form known as sphericity. Specifically,
                    it assumes that the vari- ances of the differences between
                    any two levels of the within-groups factor are equal. In
                    real-world data, it’s unlikely that this assumption will be
                    met. This has led to a number of alternative approaches,
                    including the following:
                  </p>
                  <ul>
                    <li>
                      Using the lmer() function in the lme4 package to fit
                      linear mixed models (Bates, 2005)
                    </li>
                    <li>
                      Using the Anova() function in the car package to adjust
                      traditional test statis- tics to account for lack of
                      sphericity (for example, the Geisser–Greenhouse
                      correction)
                    </li>
                    <li>
                      Using the gls() function in the nlme package to fit
                      generalized least squares models with specified
                      variance-covariance structures (UCLA, 2009)
                    </li>
                    <li>
                      Using multivariate analysis of variance to model repeated
                      measured data (Hand, 1987)
                    </li>
                  </ul>
                  <p>
                    Coverage of these approaches is beyond the scope of this
                    text. If you’re interested in learning more, check out
                    Pinheiro and Bates (2000) and Zuur et al. (2009).
                  </p>
                </blockquote>
              </div>
            </div>
            <div id="ancova" class="section level3">
              <h3>ANCOVA</h3>
              <div id="one-way-ancova-example" class="section level4">
                <h4>One-way ANCOVA example</h4>
                <p>R in Action 9.4, litter dataset from multcomp package</p>
                <blockquote>
                  <p>
                    Pregnant mice were divided into four treatment groups; each
                    group received a different dose of a drug (0, 5, 50, or
                    500). The mean post-birth weight for each litter was the
                    dependent variable and gestation time was included as a
                    covariate. The analysis is given in the following listing.
                  </p>
                </blockquote>
                <div id="load-dataset-1" class="section level5">
                  <h5>Load dataset</h5>
                  <div class="sourceCode" id="cb202">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1"></a><span class="kw">library</span>(multcomp)</span>
<span id="cb202-2"><a href="#cb202-2"></a><span class="kw">attach</span>(litter)</span>
<span id="cb202-3"><a href="#cb202-3"></a><span class="kw">table</span>(dose)</span></code></pre>
                  </div>
                  <pre><code>## dose
##   0   5  50 500
##  20  19  18  17</code></pre>
                  <div class="sourceCode" id="cb204">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1"></a><span class="kw">aggregate</span>(weight, <span class="dt">by =</span> <span class="kw">list</span>(dose), <span class="dt">FUN =</span> mean)</span></code></pre>
                  </div>
                  <pre><code>##   Group.1        x
## 1       0 32.30850
## 2       5 29.30842
## 3      50 29.86611
## 4     500 29.64647</code></pre>
                </div>
                <div id="perform-analysis-of-covariance" class="section level5">
                  <h5>Perform analysis of covariance</h5>
                  <div class="sourceCode" id="cb206">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1"></a>litter_fit &lt;-<span class="st"> </span><span class="kw">aov</span>(weight <span class="op">~</span><span class="st"> </span>gesttime <span class="op">+</span><span class="st"> </span>dose)</span>
<span id="cb206-2"><a href="#cb206-2"></a><span class="kw">summary</span>(litter_fit)</span></code></pre>
                  </div>
                  <pre><code>##             Df Sum Sq Mean Sq F value  Pr(&gt;F)
## gesttime     1  134.3  134.30   8.049 0.00597 **
## dose         3  137.1   45.71   2.739 0.04988 *
## Residuals   69 1151.3   16.69
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                  <div
                    id="calculate-adjusted-group-means"
                    class="section level6"
                  >
                    <h6>Calculate adjusted group means</h6>
                    <blockquote>
                      <p>
                        Because you’re using a covariate, you may want to obtain
                        adjusted group means—that is, the group means obtained
                        after partialing out the effects of the covariate. You
                        can use the effect() function in the effects library to
                        calculate adjusted means:
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb208">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb208-1"><a href="#cb208-1"></a><span class="kw">library</span>(effects)</span></code></pre>
                    </div>
                    <pre><code>## Registered S3 methods overwritten by &#39;lme4&#39;:
##   method                          from
##   cooks.distance.influence.merMod car
##   influence.merMod                car
##   dfbeta.influence.merMod         car
##   dfbetas.influence.merMod        car</code></pre>
                    <pre><code>## lattice theme set by effectsTheme()
## See ?effectsTheme for details.</code></pre>
                    <div class="sourceCode" id="cb211">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1"></a><span class="kw">effect</span>(<span class="st">&#39;dose&#39;</span>, litter_fit)</span></code></pre>
                    </div>
                    <pre><code>##
##  dose effect
## dose
##        0        5       50      500
## 32.35367 28.87672 30.56614 29.33460</code></pre>
                    <blockquote>
                      <p>
                        In this case, the adjusted means are similar to the
                        unadjusted means produced by the aggregate() function,
                        but this won’t always be the case. The effects package
                        provides a powerful method of obtaining adjusted means
                        for complex research designs and presenting them
                        visually. See the package documentation on CRAN for more
                        details.
                      </p>
                    </blockquote>
                  </div>
                  <div id="multiple-comparisons" class="section level6">
                    <h6>Multiple comparisons</h6>
                    <div class="sourceCode" id="cb213">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1"></a><span class="kw">library</span>(multcomp)</span>
<span id="cb213-2"><a href="#cb213-2"></a>litter_contrast &lt;-<span class="st"> </span><span class="kw">rbind</span>(<span class="st">&#39;no drug vs. drug&#39;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="dv">3</span>,<span class="op">-</span><span class="dv">1</span>,<span class="op">-</span><span class="dv">1</span>,<span class="op">-</span><span class="dv">1</span>))</span>
<span id="cb213-3"><a href="#cb213-3"></a><span class="kw">summary</span>(<span class="kw">glht</span>(litter_fit, <span class="dt">linfct =</span> <span class="kw">mcp</span>(<span class="dt">dose =</span> litter_contrast)))</span></code></pre>
                    </div>
                    <pre><code>##
##   Simultaneous Tests for General Linear Hypotheses
##
## Multiple Comparisons of Means: User-defined Contrasts
##
##
## Fit: aov(formula = weight ~ gesttime + dose)
##
## Linear Hypotheses:
##                       Estimate Std. Error t value Pr(&gt;|t|)
## no drug vs. drug == 0    8.284      3.209   2.581    0.012 *
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
                    <blockquote>
                      <p>
                        The contrast c(3, -1, -1, -1) specifies a comparison of
                        the first group with the average of the other three. The
                        hypothesis is tested with a t statistic (2.581 in this
                        case), which is significant at the p &lt; .05 level.
                        Therefore, you can conclude that the no-drug group has a
                        higher birth weight than drug conditions. Other
                        contrasts can be added to the rbind() function (see
                        help(glht) for details).
                      </p>
                    </blockquote>
                  </div>
                </div>
                <div id="assumptions" class="section level5">
                  <h5>Assumptions</h5>
                  <div class="sourceCode" id="cb215">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1"></a><span class="co"># global validation of linear model assumptions</span></span>
<span id="cb215-2"><a href="#cb215-2"></a><span class="kw">library</span>(gvlma)</span>
<span id="cb215-3"><a href="#cb215-3"></a>gvlmalitter &lt;-<span class="st"> </span><span class="kw">gvlma</span>(<span class="kw">lm</span>(weight <span class="op">~</span><span class="st"> </span>gesttime <span class="op">+</span><span class="st"> </span>dose, <span class="dt">data =</span> litter))</span>
<span id="cb215-4"><a href="#cb215-4"></a><span class="kw">summary</span>(gvlmalitter)</span></code></pre>
                  </div>
                  <pre><code>##
## Call:
## lm(formula = weight ~ gesttime + dose, data = litter)
##
## Residuals:
##      Min       1Q   Median       3Q      Max
## -11.5649  -2.0068   0.1476   3.0755   7.3023
##
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)  -45.366     24.984  -1.816  0.07374 .
## gesttime       3.519      1.131   3.111  0.00271 **
## dose5         -3.477      1.318  -2.639  0.01027 *
## dose50        -1.788      1.344  -1.330  0.18780
## dose500       -3.019      1.352  -2.232  0.02883 *
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
##
## Residual standard error: 4.085 on 69 degrees of freedom
## Multiple R-squared:  0.1908, Adjusted R-squared:  0.1439
## F-statistic: 4.067 on 4 and 69 DF,  p-value: 0.005105
##
##
## ASSESSMENT OF THE LINEAR MODEL ASSUMPTIONS
## USING THE GLOBAL TEST ON 4 DEGREES-OF-FREEDOM:
## Level of Significance =  0.05
##
## Call:
##  gvlma(x = lm(weight ~ gesttime + dose, data = litter))
##
##                        Value p-value                Decision
## Global Stat        8.3233466 0.08043 Assumptions acceptable.
## Skewness           3.0816555 0.07918 Assumptions acceptable.
## Kurtosis           0.0002286 0.98794 Assumptions acceptable.
## Link Function      3.3202398 0.06843 Assumptions acceptable.
## Heteroscedasticity 1.9212226 0.16572 Assumptions acceptable.</code></pre>
                  <div class="sourceCode" id="cb217">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb217-1"><a href="#cb217-1"></a><span class="kw">plot</span>(litter_fit)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-ancova-litter-04-plot-1.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-ancova-litter-04-plot-2.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-ancova-litter-04-plot-3.png"
                      width="672"
                    /><img
                      src="code_files/figure-html/code-ancova-litter-04-plot-4.png"
                      width="672"
                    />
                  </p>
                  <div id="normality-3" class="section level6">
                    <h6>Normality</h6>
                    <div class="sourceCode" id="cb218">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb218-1"><a href="#cb218-1"></a><span class="co"># Shapiro-Wilk W test</span></span>
<span id="cb218-2"><a href="#cb218-2"></a><span class="kw">shapiro.test</span>(<span class="kw">rstandard</span>(litter_fit))</span></code></pre>
                    </div>
                    <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  rstandard(litter_fit)
## W = 0.97617, p-value = 0.175</code></pre>
                  </div>
                  <div id="equality-of-variances-3" class="section level6">
                    <h6>Equality of variances</h6>
                    <p>
                      Note that the Brown-Forsythe test returns a significant
                      result, indicating heterogeneity of residual variances.
                      However, this is for a one-way model without the
                      <code>gesttime</code> covariate. Including the covariate
                      accounts for significant residual variance and satisfies
                      all gvlma assumptions.
                    </p>
                    <div class="sourceCode" id="cb220">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb220-1"><a href="#cb220-1"></a><span class="kw">library</span>(car)</span>
<span id="cb220-2"><a href="#cb220-2"></a><span class="kw">leveneTest</span>(weight, dose, <span class="dt">data =</span> litter)</span></code></pre>
                    </div>
                    <pre><code>## Levene&#39;s Test for Homogeneity of Variance (center = median: litter)
##       Df F value  Pr(&gt;F)
## group  3  3.1788 0.02925 *
##       70
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                  </div>
                </div>
              </div>
            </div>
            <div id="manova" class="section level3">
              <h3>MANOVA</h3>
              <p>MANOVA = Multivariate Analysis of Variance</p>
              <div id="manova-example" class="section level4">
                <h4>MANOVA example</h4>
                <p>Ultrasound data</p>
                <div id="import-dataset-7" class="section level5">
                  <h5>Import dataset</h5>
                  <div class="sourceCode" id="cb222">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb222-1"><a href="#cb222-1"></a>hr &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;data/manova_hr.csv&#39;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span>)</span>
<span id="cb222-2"><a href="#cb222-2"></a><span class="kw">attach</span>(hr)</span>
<span id="cb222-3"><a href="#cb222-3"></a><span class="kw">names</span>(hr)</span></code></pre>
                  </div>
                  <pre><code>## [1] &quot;US&quot;   &quot;USC&quot;  &quot;V&quot;    &quot;VC&quot;   &quot;Time&quot;</code></pre>
                </div>
                <div id="perform-multivariate-anova" class="section level5">
                  <h5>Perform multivariate ANOVA</h5>
                  <div class="sourceCode" id="cb224">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb224-1"><a href="#cb224-1"></a>skull.manova1 =<span class="st"> </span><span class="kw">manova</span>(<span class="kw">cbind</span>(US, USC, V, VC) <span class="op">~</span><span class="st"> </span><span class="kw">as.factor</span>(Time), <span class="dt">data =</span> hr)</span>
<span id="cb224-2"><a href="#cb224-2"></a><span class="kw">summary</span>(skull.manova1)</span></code></pre>
                  </div>
                  <pre><code>##                 Df  Pillai approx F num Df den Df Pr(&gt;F)
## as.factor(Time)  2 0.65326   1.2127      8     20 0.3412
## Residuals       12</code></pre>
                  <div class="sourceCode" id="cb226">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb226-1"><a href="#cb226-1"></a><span class="kw">summary</span>(skull.manova1, <span class="dt">test =</span> <span class="st">&#39;Wilks&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>##                 Df   Wilks approx F num Df den Df Pr(&gt;F)
## as.factor(Time)  2 0.35085   1.5486      8     18 0.2094
## Residuals       12</code></pre>
                  <div class="sourceCode" id="cb228">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb228-1"><a href="#cb228-1"></a><span class="kw">summary</span>(skull.manova1, <span class="dt">test =</span> <span class="st">&#39;Hotelling-Lawley&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>##                 Df Hotelling-Lawley approx F num Df den Df Pr(&gt;F)
## as.factor(Time)  2           1.8385   1.8385      8     16 0.1427
## Residuals       12</code></pre>
                  <div class="sourceCode" id="cb230">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1"></a><span class="kw">summary</span>(skull.manova1, <span class="dt">test =</span> <span class="st">&#39;Roy&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>##                 Df    Roy approx F num Df den Df  Pr(&gt;F)
## as.factor(Time)  2 1.8321   4.5803      4     10 0.02324 *
## Residuals       12
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
                </div>
              </div>
            </div>
            <div id="fishers-exact" class="section level3">
              <h3>Fisher’s exact</h3>
              <div
                id="fishers-exact-test-arthritis-example"
                class="section level4"
              >
                <h4>Fisher’s exact test arthritis example</h4>
                <p>Arthritis data from vcd package</p>
                <div id="load-dataset-2" class="section level5">
                  <h5>Load dataset</h5>
                  <div class="sourceCode" id="cb232">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb232-1"><a href="#cb232-1"></a><span class="co"># see R in Action section 7.2 for walkthrough</span></span>
<span id="cb232-2"><a href="#cb232-2"></a><span class="kw">library</span>(vcd)</span></code></pre>
                  </div>
                  <pre><code>## Loading required package: grid</code></pre>
                  <div class="sourceCode" id="cb234">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1"></a><span class="kw">summary</span>(Arthritis)</span></code></pre>
                  </div>
                  <pre><code>##        ID          Treatment      Sex          Age          Improved
##  Min.   : 1.00   Placebo:43   Female:59   Min.   :23.00   None  :42
##  1st Qu.:21.75   Treated:41   Male  :25   1st Qu.:46.00   Some  :14
##  Median :42.50                            Median :57.00   Marked:28
##  Mean   :42.50                            Mean   :53.36
##  3rd Qu.:63.25                            3rd Qu.:63.00
##  Max.   :84.00                            Max.   :74.00</code></pre>
                  <div class="sourceCode" id="cb236">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1"></a><span class="kw">attach</span>(Arthritis)</span>
<span id="cb236-2"><a href="#cb236-2"></a>arthritis_table &lt;-<span class="st"> </span><span class="kw">xtabs</span>( <span class="op">~</span><span class="st"> </span>Treatment <span class="op">+</span><span class="st"> </span>Improved, <span class="dt">data =</span> Arthritis)</span>
<span id="cb236-3"><a href="#cb236-3"></a>arthritis_table</span></code></pre>
                  </div>
                  <pre><code>##          Improved
## Treatment None Some Marked
##   Placebo   29    7      7
##   Treated   13    7     21</code></pre>
                </div>
                <div id="perform-fishers-exact-test" class="section level5">
                  <h5>Perform fisher’s exact test</h5>
                  <div class="sourceCode" id="cb238">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1"></a><span class="kw">fisher.test</span>(arthritis_table)</span></code></pre>
                  </div>
                  <pre><code>##
##  Fisher&#39;s Exact Test for Count Data
##
## data:  arthritis_table
## p-value = 0.001393
## alternative hypothesis: two.sided</code></pre>
                </div>
              </div>
              <div
                id="fishers-exact-test-prostate-cancer-example"
                class="section level4"
              >
                <h4>Fisher’s exact test prostate cancer example</h4>
                <p>
                  Prostate cancer incidence data from TRAMP Tomato Soy paper
                </p>
                <p>
                  Zuniga KE, Clinton SK, Erdman JW. The interactions of dietary
                  tomato powder and soy germ on prostate carcinogenesis in the
                  TRAMP model. Cancer Prev. Res. (2013). DOI:
                  <a href="https://doi.org/10.1158/1940-6207.CAPR-12-0443"
                    >10.1158/1940-6207.CAPR-12-0443</a
                  >
                </p>
                <div id="create-data-frame" class="section level5">
                  <h5>Create data frame</h5>
                  <p>Data also available in fishers_exact_cpr_2013.csv</p>
                  <div class="sourceCode" id="cb240">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1"></a><span class="co"># tomato+soy vs control</span></span>
<span id="cb240-2"><a href="#cb240-2"></a>diet &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;tomsoy&#39;</span>, <span class="st">&#39;tomsoy&#39;</span>, <span class="st">&#39;control&#39;</span>, <span class="st">&#39;control&#39;</span>)</span>
<span id="cb240-3"><a href="#cb240-3"></a>cancer &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;cancer&#39;</span>, <span class="st">&#39;nocancer&#39;</span>, <span class="st">&#39;cancer&#39;</span>, <span class="st">&#39;nocancer&#39;</span>)</span>
<span id="cb240-4"><a href="#cb240-4"></a>count &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">12</span>, <span class="dv">15</span>, <span class="dv">29</span>, <span class="dv">0</span>)</span>
<span id="cb240-5"><a href="#cb240-5"></a>PCaTomatoSoy &lt;-<span class="st"> </span><span class="kw">data.frame</span>(diet, cancer, count)</span>
<span id="cb240-6"><a href="#cb240-6"></a>PCaTomatoSoy</span></code></pre>
                  </div>
                  <pre><code>##      diet   cancer count
## 1  tomsoy   cancer    12
## 2  tomsoy nocancer    15
## 3 control   cancer    29
## 4 control nocancer     0</code></pre>
                </div>
                <div
                  id="create-contingency-table-from-data-frame"
                  class="section level5"
                >
                  <h5>Create contingency table from data frame</h5>
                  <div class="sourceCode" id="cb242">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1"></a>cancertable &lt;-<span class="st"> </span><span class="kw">xtabs</span>(count <span class="op">~</span><span class="st"> </span>diet <span class="op">+</span><span class="st"> </span>cancer, <span class="dt">data =</span> PCaTomatoSoy)</span>
<span id="cb242-2"><a href="#cb242-2"></a>cancertable</span></code></pre>
                  </div>
                  <pre><code>##          cancer
## diet      cancer nocancer
##   control     29        0
##   tomsoy      12       15</code></pre>
                </div>
                <div id="perform-fishers-exact-test-1" class="section level5">
                  <h5>Perform Fisher’s exact test</h5>
                  <div class="sourceCode" id="cb244">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1"></a><span class="kw">fisher.test</span>(cancertable)</span></code></pre>
                  </div>
                  <pre><code>##
##  Fisher&#39;s Exact Test for Count Data
##
## data:  cancertable
## p-value = 1.07e-06
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  6.891611      Inf
## sample estimates:
## odds ratio
##        Inf</code></pre>
                </div>
              </div>
            </div>
            <div id="wilcoxon" class="section level3">
              <h3>Wilcoxon</h3>
              <div id="wilcoxon-test-example" class="section level4">
                <h4>Wilcoxon test example</h4>
                <p>UScrime dataset from MASS package</p>
                <p>See R in Action section 7.5.1 for walkthrough.</p>
                <div id="load-dataset-3" class="section level5">
                  <h5>Load dataset</h5>
                  <div class="sourceCode" id="cb246">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb246-1"><a href="#cb246-1"></a><span class="kw">library</span>(MASS)</span>
<span id="cb246-2"><a href="#cb246-2"></a><span class="kw">head</span>(UScrime)</span></code></pre>
                  </div>
                  <pre><code>##     M So  Ed Po1 Po2  LF  M.F Pop  NW  U1 U2 GDP Ineq     Prob    Time    y
## 1 151  1  91  58  56 510  950  33 301 108 41 394  261 0.084602 26.2011  791
## 2 143  0 113 103  95 583 1012  13 102  96 36 557  194 0.029599 25.2999 1635
## 3 142  1  89  45  44 533  969  18 219  94 33 318  250 0.083401 24.3006  578
## 4 136  0 121 149 141 577  994 157  80 102 39 673  167 0.015801 29.9012 1969
## 5 141  0 121 109 101 591  985  18  30  91 20 578  174 0.041399 21.2998 1234
## 6 121  0 110 118 115 547  964  25  44  84 29 689  126 0.034201 20.9995  682</code></pre>
                  <div class="sourceCode" id="cb248">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1"></a><span class="kw">with</span>(UScrime, <span class="kw">by</span>(Prob, So, median))</span></code></pre>
                  </div>
                  <pre><code>## So: 0
## [1] 0.038201
## ---------------------------------------------------------------------------------------------------------------
## So: 1
## [1] 0.055552</code></pre>
                </div>
                <div id="perform-wilcoxon-test" class="section level5">
                  <h5>Perform Wilcoxon test</h5>
                  <div class="sourceCode" id="cb250">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb250-1"><a href="#cb250-1"></a><span class="kw">wilcox.test</span>(</span>
<span id="cb250-2"><a href="#cb250-2"></a>  Prob <span class="op">~</span><span class="st"> </span>So,</span>
<span id="cb250-3"><a href="#cb250-3"></a>  <span class="dt">data =</span> UScrime,</span>
<span id="cb250-4"><a href="#cb250-4"></a>  <span class="dt">paired =</span> <span class="ot">FALSE</span>,</span>
<span id="cb250-5"><a href="#cb250-5"></a>  <span class="dt">alternative =</span> <span class="st">&#39;two.sided&#39;</span>,</span>
<span id="cb250-6"><a href="#cb250-6"></a>  <span class="dt">var.equal =</span> <span class="ot">TRUE</span>,</span>
<span id="cb250-7"><a href="#cb250-7"></a>  <span class="dt">conf.level =</span> <span class="fl">0.95</span></span>
<span id="cb250-8"><a href="#cb250-8"></a>)</span></code></pre>
                  </div>
                  <pre><code>##
##  Wilcoxon rank sum exact test
##
## data:  Prob by So
## W = 81, p-value = 8.488e-05
## alternative hypothesis: true location shift is not equal to 0</code></pre>
                </div>
                <div id="test-assumptions-3" class="section level5">
                  <h5>Test assumptions</h5>
                  <div id="equality-of-variances-4" class="section level6">
                    <h6>Equality of variances</h6>
                    <p>
                      Note that, although the Wilcoxon rank sum test does not
                      require normality, it does require homogeneity of
                      variances and independence of random samples.
                    </p>
                    <div class="sourceCode" id="cb252">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb252-1"><a href="#cb252-1"></a><span class="kw">var.test</span>(Prob <span class="op">~</span><span class="st"> </span>So, <span class="dt">data =</span> UScrime)</span></code></pre>
                    </div>
                    <pre><code>##
##  F test to compare two variances
##
## data:  Prob by So
## F = 0.624, num df = 30, denom df = 15, p-value = 0.2646
## alternative hypothesis: true ratio of variances is not equal to 1
## 95 percent confidence interval:
##  0.2360299 1.4396653
## sample estimates:
## ratio of variances
##          0.6240006</code></pre>
                  </div>
                </div>
              </div>
            </div>
            <div id="kruskal-wallis" class="section level3">
              <h3>Kruskal-Wallis</h3>
              <div id="kruskal-wallis-example" class="section level4">
                <h4>Kruskal-Wallis example</h4>
                <p>
                  Cardiac magnesium levels from Smith BW J Food Res 2013 (doi:
                  <a href="https://doi.org/10.5539/jfr.v2n1p168"
                    >10.5539/jfr.v2n1p168</a
                  >).
                </p>
                <div id="import-dataset-8" class="section level5">
                  <h5>Import dataset</h5>
                  <ul>
                    <li>
                      In RStudio, tools-&gt;import dataset can also be used.
                    </li>
                    <li>
                      Dataset is attached so variables can be named without
                      reference to the original dataset.
                    </li>
                  </ul>
                  <div class="sourceCode" id="cb254">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb254-1"><a href="#cb254-1"></a>cardiacMg &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;data/kruskal_wallis_jfr2013_mg_cardiac.csv&#39;</span>,</span>
<span id="cb254-2"><a href="#cb254-2"></a>                        <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb254-3"><a href="#cb254-3"></a>                        <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span>)</span>
<span id="cb254-4"><a href="#cb254-4"></a><span class="kw">attach</span>(cardiacMg)</span></code></pre>
                  </div>
                </div>
                <div id="perform-kruskal-wallis-test" class="section level5">
                  <h5>Perform Kruskal-Wallis test</h5>
                  <div class="sourceCode" id="cb255">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1"></a><span class="co"># Convert independent variable to a factor for use later by PMCMRplus</span></span>
<span id="cb255-2"><a href="#cb255-2"></a>cardiacMg<span class="op">$</span>diet =<span class="st"> </span><span class="kw">as.factor</span>(cardiacMg<span class="op">$</span>diet)</span>
<span id="cb255-3"><a href="#cb255-3"></a><span class="co"># Perform Kruskal-Wallis test</span></span>
<span id="cb255-4"><a href="#cb255-4"></a><span class="kw">kruskal.test</span>(mg <span class="op">~</span><span class="st"> </span>diet, <span class="dt">data =</span> cardiacMg)</span></code></pre>
                  </div>
                  <pre><code>##
##  Kruskal-Wallis rank sum test
##
## data:  mg by diet
## Kruskal-Wallis chi-squared = 6.7757, df = 2, p-value = 0.03378</code></pre>
                  <div id="multiple-comparisons-1" class="section level6">
                    <h6>Multiple comparisons</h6>
                    <ul>
                      <li>
                        Pairwise Wilcoxon testing with experiment-wide error
                        correction
                      </li>
                      <li>
                        Nemenyi test (pairwise test for multiple comparisons of
                        mean rank sums). Not corrected for multiple comparisons.
                      </li>
                      <li>Dunnett non-parametric multiple comparisons</li>
                    </ul>
                    <div class="sourceCode" id="cb257">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb257-1"><a href="#cb257-1"></a><span class="co"># Pairwise Wilcoxon</span></span>
<span id="cb257-2"><a href="#cb257-2"></a><span class="kw">pairwise.wilcox.test</span>(cardiacMg<span class="op">$</span>mg, cardiacMg<span class="op">$</span>diet, <span class="dt">p.adjust.method =</span> <span class="st">&#39;holm&#39;</span>)</span></code></pre>
                    </div>
                    <pre><code>## Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute exact p-value with ties

## Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute exact p-value with ties

## Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute exact p-value with ties</code></pre>
                    <pre><code>##
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction
##
## data:  cardiacMg$mg and cardiacMg$diet
##
##      0.1   0.4
## 0.4  0.885 -
## chow 0.063 0.048
##
## P value adjustment method: holm</code></pre>
                    <div class="sourceCode" id="cb260">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1"></a><span class="co"># Nemenyi</span></span>
<span id="cb260-2"><a href="#cb260-2"></a><span class="kw">library</span>(PMCMRplus)</span>
<span id="cb260-3"><a href="#cb260-3"></a><span class="kw">kwAllPairsNemenyiTest</span>(cardiacMg<span class="op">$</span>mg, cardiacMg<span class="op">$</span>diet, <span class="dt">method =</span> <span class="st">&#39;Tukey&#39;</span>)</span></code></pre>
                    </div>
                    <pre><code>## Warning in kwAllPairsNemenyiTest.default(cardiacMg$mg, cardiacMg$diet, method = &quot;Tukey&quot;): Ties are present, p-values are not corrected.</code></pre>
                    <pre><code>##
##  Pairwise comparisons using Tukey-Kramer-Nemenyi all-pairs test with Tukey-Dist approximation</code></pre>
                    <pre><code>## data: cardiacMg$mg and cardiacMg$diet</code></pre>
                    <pre><code>##      0.1   0.4
## 0.4  0.992 -
## chow 0.039 0.044</code></pre>
                    <pre><code>##
## P value adjustment method: single-step</code></pre>
                    <pre><code>## alternative hypothesis: two.sided</code></pre>
                    <div class="sourceCode" id="cb267">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb267-1"><a href="#cb267-1"></a><span class="co"># Dunnett</span></span>
<span id="cb267-2"><a href="#cb267-2"></a><span class="kw">library</span>(nparcomp)</span>
<span id="cb267-3"><a href="#cb267-3"></a><span class="kw">nparcomp</span>(mg <span class="op">~</span><span class="st"> </span>diet,</span>
<span id="cb267-4"><a href="#cb267-4"></a>         <span class="dt">data =</span> cardiacMg,</span>
<span id="cb267-5"><a href="#cb267-5"></a>         <span class="dt">type =</span> <span class="st">&#39;Dunnett&#39;</span>,</span>
<span id="cb267-6"><a href="#cb267-6"></a>         <span class="dt">control =</span> <span class="st">&#39;chow&#39;</span>)</span></code></pre>
                    </div>
                    <pre><code>##
##  #------Nonparametric Multiple Comparisons for relative contrast effects-----#
##
##  - Alternative Hypothesis:  True relative contrast effect p is not equal to 1/2
##  - Type of Contrast : Dunnett
##  - Confidence level: 95 %
##  - Method = Logit - Transformation
##  - Estimation Method: Pairwise rankings
##
##  #---------------------------Interpretation----------------------------------#
##  p(a,b) &gt; 1/2 : b tends to be larger than a
##  #---------------------------------------------------------------------------#
## </code></pre>
                    <pre><code>## $Data.Info
##   Sample Size
## 1    0.1    8
## 2    0.4    9
## 3   chow    3
##
## $Contrast
##            0.1 0.4 chow
## 0.1 - chow   1   0   -1
## 0.4 - chow   0   1   -1
##
## $Analysis
##        Comparison Estimator Lower Upper Statistic    p.Value
## 1 p( chow , 0.1 )     0.042 0.002 0.541 -2.124749 0.06608568
## 2 p( chow , 0.4 )     0.001 0.000 0.005 -9.757859 0.00000000
##
## $Overall
##   Quantile p.Value
## 1 2.236422       0
##
## $input
## $input$formula
## mg ~ diet
##
## $input$data
##    diet    mg
## 1  chow 0.199
## 2  chow 0.363
## 3  chow 0.363
## 4   0.1 0.321
## 5   0.1 0.141
## 6   0.1 0.144
## 7   0.1 0.145
## 8   0.1 0.140
## 9   0.1 0.157
## 10  0.1 0.137
## 11  0.1 0.144
## 12  0.4 0.140
## 13  0.4 0.160
## 14  0.4 0.138
## 15  0.4 0.145
## 16  0.4 0.150
## 17  0.4 0.146
## 18  0.4 0.144
## 19  0.4 0.149
## 20  0.4 0.139
##
## $input$type
## [1] &quot;Dunnett&quot;
##
## $input$conf.level
## [1] 0.95
##
## $input$alternative
## [1] &quot;two.sided&quot; &quot;less&quot;      &quot;greater&quot;
##
## $input$asy.method
## [1] &quot;logit&quot;  &quot;probit&quot; &quot;normal&quot; &quot;mult.t&quot;
##
## $input$plot.simci
## [1] FALSE
##
## $input$control
## [1] &quot;chow&quot;
##
## $input$info
## [1] TRUE
##
## $input$rounds
## [1] 3
##
## $input$contrast.matrix
## NULL
##
## $input$correlation
## [1] FALSE
##
## $input$weight.matrix
## [1] FALSE
##
##
## $text.Output
## [1] &quot;True relative contrast effect p is not equal to 1/2&quot;
##
## $connames
## [1] &quot;p( chow , 0.1 )&quot; &quot;p( chow , 0.4 )&quot;
##
## $AsyMethod
## [1] &quot;Logit - Transformation&quot;
##
## attr(,&quot;class&quot;)
## [1] &quot;nparcomp&quot;</code></pre>
                  </div>
                </div>
                <div id="test-assumptions-4" class="section level5">
                  <h5>Test assumptions</h5>
                  <div id="normality-4" class="section level6">
                    <h6>Normality</h6>
                    <ul>
                      <li>
                        The RColorBrewer package is loaded for additional
                        colors, and a palette is added to R. Use brackets after
                        vector name for specific colors when creating plots (for
                        example: <code>col=set1[4:14]</code>).
                      </li>
                      <li>
                        Legend can also be manually positioned with xy
                        coordinates (numbers from 0,0 at bottom left of plot to
                        1,1 at top right)
                      </li>
                      <li>
                        As demonstrated by the tests below, these data fail the
                        assumption of normality, and are not amenable to
                        transformation, hence the use of non-parametric testing.
                      </li>
                    </ul>
                    <div class="sourceCode" id="cb270">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb270-1"><a href="#cb270-1"></a><span class="co"># Shapiro-Wilk W test</span></span>
<span id="cb270-2"><a href="#cb270-2"></a><span class="kw">shapiro.test</span>(cardiacMg<span class="op">$</span>mg)</span></code></pre>
                    </div>
                    <pre><code>##
##  Shapiro-Wilk normality test
##
## data:  cardiacMg$mg
## W = 0.55747, p-value = 1.101e-06</code></pre>
                    <div class="sourceCode" id="cb272">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb272-1"><a href="#cb272-1"></a><span class="co"># Normal Quantile-Quantile Plot</span></span>
<span id="cb272-2"><a href="#cb272-2"></a><span class="co"># Set graphical parameters</span></span>
<span id="cb272-3"><a href="#cb272-3"></a><span class="kw">par</span>(</span>
<span id="cb272-4"><a href="#cb272-4"></a>  <span class="dt">cex =</span> <span class="dv">1</span>,</span>
<span id="cb272-5"><a href="#cb272-5"></a>  <span class="dt">cex.main =</span> <span class="dv">3</span>,</span>
<span id="cb272-6"><a href="#cb272-6"></a>  <span class="dt">cex.lab =</span> <span class="dv">2</span>,</span>
<span id="cb272-7"><a href="#cb272-7"></a>  <span class="dt">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb272-8"><a href="#cb272-8"></a>  <span class="dt">cex.axis =</span> <span class="fl">1.5</span>,</span>
<span id="cb272-9"><a href="#cb272-9"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb272-10"><a href="#cb272-10"></a>  <span class="dt">lend =</span> <span class="dv">0</span>,</span>
<span id="cb272-11"><a href="#cb272-11"></a>  <span class="dt">ljoin =</span> <span class="dv">0</span>,</span>
<span id="cb272-12"><a href="#cb272-12"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb272-13"><a href="#cb272-13"></a>  <span class="dt">pch =</span> <span class="dv">1</span>,</span>
<span id="cb272-14"><a href="#cb272-14"></a>  <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb272-15"><a href="#cb272-15"></a>)</span>
<span id="cb272-16"><a href="#cb272-16"></a></span>
<span id="cb272-17"><a href="#cb272-17"></a><span class="co"># Load RColorBrewer and set palette</span></span>
<span id="cb272-18"><a href="#cb272-18"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb272-19"><a href="#cb272-19"></a>set1 &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&#39;Set1&#39;</span>)</span>
<span id="cb272-20"><a href="#cb272-20"></a></span>
<span id="cb272-21"><a href="#cb272-21"></a><span class="co"># Create vector for categories</span></span>
<span id="cb272-22"><a href="#cb272-22"></a>categories &lt;-<span class="st"> </span><span class="kw">factor</span>(cardiacMg<span class="op">$</span>diet)</span>
<span id="cb272-23"><a href="#cb272-23"></a></span>
<span id="cb272-24"><a href="#cb272-24"></a><span class="co"># Create plot</span></span>
<span id="cb272-25"><a href="#cb272-25"></a><span class="kw">qqnorm</span>(</span>
<span id="cb272-26"><a href="#cb272-26"></a>  mg,</span>
<span id="cb272-27"><a href="#cb272-27"></a>  <span class="dt">ylab =</span> <span class="st">&#39;Mg, mg/g&#39;</span>,</span>
<span id="cb272-28"><a href="#cb272-28"></a>  <span class="dt">axes =</span> <span class="ot">FALSE</span>,</span>
<span id="cb272-29"><a href="#cb272-29"></a>  <span class="dt">col =</span> set1[categories],</span>
<span id="cb272-30"><a href="#cb272-30"></a>  <span class="dt">lwd =</span> <span class="dv">4</span>,</span>
<span id="cb272-31"><a href="#cb272-31"></a>  <span class="dt">cex =</span> <span class="dv">2</span></span>
<span id="cb272-32"><a href="#cb272-32"></a>)</span>
<span id="cb272-33"><a href="#cb272-33"></a><span class="co"># Add normal line</span></span>
<span id="cb272-34"><a href="#cb272-34"></a><span class="kw">qqline</span>(cardiacMg<span class="op">$</span>mg, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb272-35"><a href="#cb272-35"></a><span class="co"># Add border to complete axes</span></span>
<span id="cb272-36"><a href="#cb272-36"></a><span class="kw">box</span>(<span class="dt">which =</span> <span class="st">&#39;plot&#39;</span>, <span class="dt">bty =</span> <span class="st">&#39;o&#39;</span>, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb272-37"><a href="#cb272-37"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>)  <span class="co"># x axis</span></span>
<span id="cb272-38"><a href="#cb272-38"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>)  <span class="co"># y axis</span></span>
<span id="cb272-39"><a href="#cb272-39"></a><span class="co"># Add legend</span></span>
<span id="cb272-40"><a href="#cb272-40"></a><span class="kw">legend</span>(</span>
<span id="cb272-41"><a href="#cb272-41"></a>  <span class="st">&#39;topleft&#39;</span>,</span>
<span id="cb272-42"><a href="#cb272-42"></a>  <span class="dt">legend =</span> <span class="kw">levels</span>(categories),</span>
<span id="cb272-43"><a href="#cb272-43"></a>  <span class="dt">box.lwd =</span> <span class="dv">3</span>,</span>
<span id="cb272-44"><a href="#cb272-44"></a>  <span class="dt">pch =</span> <span class="fl">1.5</span>,</span>
<span id="cb272-45"><a href="#cb272-45"></a>  <span class="dt">cex =</span> <span class="fl">1.75</span>,</span>
<span id="cb272-46"><a href="#cb272-46"></a>  <span class="dt">pt.cex =</span> <span class="dv">2</span>,</span>
<span id="cb272-47"><a href="#cb272-47"></a>  <span class="dt">pt.lwd =</span> <span class="dv">4</span>,</span>
<span id="cb272-48"><a href="#cb272-48"></a>  <span class="dt">col =</span> set1</span>
<span id="cb272-49"><a href="#cb272-49"></a>)</span></code></pre>
                    </div>
                    <p>
                      <img
                        src="code_files/figure-html/code-kruskal-cardiacmg-03-assumptions-normality-1.png"
                        width="672"
                      />
                    </p>
                  </div>
                  <div id="equality-of-variances-5" class="section level6">
                    <h6>Equality of variances</h6>
                    <ul>
                      <li>
                        This is a brown-forsythe test that evaluates deviations
                        from group medians, and thus is robust to violations of
                        normality.
                      </li>
                      <li>Fligner test included for comparison.</li>
                    </ul>
                    <div class="sourceCode" id="cb273">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb273-1"><a href="#cb273-1"></a><span class="kw">library</span>(lawstat)</span>
<span id="cb273-2"><a href="#cb273-2"></a><span class="kw">levene.test</span>(cardiacMg<span class="op">$</span>mg, cardiacMg<span class="op">$</span>diet)</span></code></pre>
                    </div>
                    <pre><code>##
##  Modified robust Brown-Forsythe Levene-type test based on the absolute deviations from the median
##
## data:  cardiacMg$mg
## Test Statistic = 1.1301, p-value = 0.3461</code></pre>
                    <div class="sourceCode" id="cb275">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb275-1"><a href="#cb275-1"></a><span class="co"># Fligner test</span></span>
<span id="cb275-2"><a href="#cb275-2"></a><span class="kw">fligner.test</span>(cardiacMg<span class="op">$</span>mg <span class="op">~</span><span class="st"> </span>cardiacMg<span class="op">$</span>diet)</span></code></pre>
                    </div>
                    <pre><code>##
##  Fligner-Killeen test of homogeneity of variances
##
## data:  cardiacMg$mg by cardiacMg$diet
## Fligner-Killeen:med chi-squared = 0.11837, df = 2, p-value = 0.9425</code></pre>
                  </div>
                </div>
                <div id="generate-boxplot" class="section level5">
                  <h5>Generate boxplot</h5>
                  <ul>
                    <li>
                      <strong>Parameters (<code>par</code>):</strong> Graphical
                      parameters need to be set again because of code chunk
                      separation in RMarkdown. Normally they would only be set
                      once per R session.
                    </li>
                    <li>
                      <strong>X axis:</strong> The default axes will usually
                      detect categorical data and label axes accordingly, but
                      categories will need to be specified for axes created
                      separately. It is easiest to create a separate vector with
                      the categories, and then reference it when creating the
                      plot. Labels can also be specified with
                      <code>labels=c('group1', 'group2')</code>.
                    </li>
                    <li>
                      <strong>Y axis:</strong> Note that it does not always
                      start at zero, but encompasses the data range with some
                      padding.
                    </li>
                  </ul>
                  <div class="sourceCode" id="cb277">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb277-1"><a href="#cb277-1"></a><span class="co"># Set graphical parameters</span></span>
<span id="cb277-2"><a href="#cb277-2"></a><span class="kw">par</span>(</span>
<span id="cb277-3"><a href="#cb277-3"></a>  <span class="dt">cex =</span> <span class="dv">1</span>,</span>
<span id="cb277-4"><a href="#cb277-4"></a>  <span class="dt">cex.main =</span> <span class="dv">3</span>,</span>
<span id="cb277-5"><a href="#cb277-5"></a>  <span class="dt">cex.lab =</span> <span class="dv">2</span>,</span>
<span id="cb277-6"><a href="#cb277-6"></a>  <span class="dt">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb277-7"><a href="#cb277-7"></a>  <span class="dt">cex.axis =</span> <span class="fl">1.5</span>,</span>
<span id="cb277-8"><a href="#cb277-8"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb277-9"><a href="#cb277-9"></a>  <span class="dt">lend =</span> <span class="dv">0</span>,</span>
<span id="cb277-10"><a href="#cb277-10"></a>  <span class="dt">ljoin =</span> <span class="dv">0</span>,</span>
<span id="cb277-11"><a href="#cb277-11"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb277-12"><a href="#cb277-12"></a>  <span class="dt">pch =</span> <span class="dv">1</span>,</span>
<span id="cb277-13"><a href="#cb277-13"></a>  <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb277-14"><a href="#cb277-14"></a>)</span>
<span id="cb277-15"><a href="#cb277-15"></a></span>
<span id="cb277-16"><a href="#cb277-16"></a><span class="co"># Create boxplot</span></span>
<span id="cb277-17"><a href="#cb277-17"></a><span class="kw">boxplot</span>(</span>
<span id="cb277-18"><a href="#cb277-18"></a>  cardiacMg<span class="op">$</span>mg <span class="op">~</span><span class="st"> </span>cardiacMg<span class="op">$</span>diet,</span>
<span id="cb277-19"><a href="#cb277-19"></a>  <span class="dt">main =</span> <span class="st">&#39;Cardiac mg&#39;</span>,</span>
<span id="cb277-20"><a href="#cb277-20"></a>  <span class="dt">xlab =</span> <span class="st">&#39;diet&#39;</span>,</span>
<span id="cb277-21"><a href="#cb277-21"></a>  <span class="dt">ylab =</span> <span class="st">&#39;Mg, mg/g&#39;</span>,</span>
<span id="cb277-22"><a href="#cb277-22"></a>  <span class="dt">axes =</span> <span class="ot">FALSE</span>,</span>
<span id="cb277-23"><a href="#cb277-23"></a>  <span class="dt">lwd =</span> <span class="dv">4</span>,</span>
<span id="cb277-24"><a href="#cb277-24"></a>  <span class="dt">medlwd =</span> <span class="dv">7</span>,</span>
<span id="cb277-25"><a href="#cb277-25"></a>  <span class="dt">outlwd =</span> <span class="dv">4</span>,</span>
<span id="cb277-26"><a href="#cb277-26"></a>  <span class="dt">cex =</span> <span class="dv">2</span>,</span>
<span id="cb277-27"><a href="#cb277-27"></a>  <span class="dt">col =</span> set1</span>
<span id="cb277-28"><a href="#cb277-28"></a>)</span>
<span id="cb277-29"><a href="#cb277-29"></a></span>
<span id="cb277-30"><a href="#cb277-30"></a><span class="co"># Add border to complete axes</span></span>
<span id="cb277-31"><a href="#cb277-31"></a><span class="kw">box</span>(<span class="dt">which =</span> <span class="st">&#39;plot&#39;</span>, <span class="dt">bty =</span> <span class="st">&#39;l&#39;</span>, <span class="dt">lwd =</span> <span class="dv">5</span>)</span>
<span id="cb277-32"><a href="#cb277-32"></a></span>
<span id="cb277-33"><a href="#cb277-33"></a><span class="co"># Create vector with means of each group</span></span>
<span id="cb277-34"><a href="#cb277-34"></a>means &lt;-<span class="st"> </span><span class="kw">aggregate</span>(mg <span class="op">~</span><span class="st"> </span>diet, <span class="dt">data =</span> cardiacMg, mean)</span>
<span id="cb277-35"><a href="#cb277-35"></a><span class="co"># Add points for group means to boxplot</span></span>
<span id="cb277-36"><a href="#cb277-36"></a><span class="kw">points</span>(</span>
<span id="cb277-37"><a href="#cb277-37"></a>  means<span class="op">$</span>mg,</span>
<span id="cb277-38"><a href="#cb277-38"></a>  <span class="dt">cex =</span> <span class="dv">2</span>,</span>
<span id="cb277-39"><a href="#cb277-39"></a>  <span class="dt">pch =</span> <span class="dv">23</span>,</span>
<span id="cb277-40"><a href="#cb277-40"></a>  <span class="dt">bg =</span> <span class="dv">0</span>,</span>
<span id="cb277-41"><a href="#cb277-41"></a>  <span class="dt">lwd =</span> <span class="dv">4</span></span>
<span id="cb277-42"><a href="#cb277-42"></a>)</span>
<span id="cb277-43"><a href="#cb277-43"></a></span>
<span id="cb277-44"><a href="#cb277-44"></a><span class="co"># X axis</span></span>
<span id="cb277-45"><a href="#cb277-45"></a><span class="kw">axis</span>(</span>
<span id="cb277-46"><a href="#cb277-46"></a>  <span class="dv">1</span>,</span>
<span id="cb277-47"><a href="#cb277-47"></a>  <span class="dt">lwd =</span> <span class="dv">5</span>,</span>
<span id="cb277-48"><a href="#cb277-48"></a>  <span class="dt">tcl =</span> <span class="fl">-0.5</span>,</span>
<span id="cb277-49"><a href="#cb277-49"></a>  <span class="dt">labels =</span> <span class="kw">levels</span>(categories),</span>
<span id="cb277-50"><a href="#cb277-50"></a>  <span class="dt">at =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)</span>
<span id="cb277-51"><a href="#cb277-51"></a>)</span>
<span id="cb277-52"><a href="#cb277-52"></a></span>
<span id="cb277-53"><a href="#cb277-53"></a><span class="co"># Y axis</span></span>
<span id="cb277-54"><a href="#cb277-54"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">lwd =</span> <span class="dv">5</span>, <span class="dt">tcl =</span> <span class="fl">-0.5</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-kruskal-cardiacmg-04-plot-1.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div id="plots" class="section level2">
            <h2>Plots</h2>
            <div id="general-examples" class="section level3">
              <h3>General examples</h3>
              <div id="ggplot2-qplot-example" class="section level4">
                <h4>ggplot2 qplot example</h4>
                <p>Example from ggplot2 documentation p. 135</p>
                <div id="intro-1" class="section level5">
                  <h5>Intro</h5>
                  <p>
                    This file demonstrates some of the capabilities in the
                    ggplot2 package. The mtcars dataset is included with R. The
                    beginning of the dataset is shown with
                    <code>head(mtcars)</code> to verify.
                  </p>
                </div>
                <div id="import-data" class="section level5">
                  <h5>Import data</h5>
                  <div class="sourceCode" id="cb278">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb278-1"><a href="#cb278-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb278-2"><a href="#cb278-2"></a><span class="kw">head</span>(mtcars)</span></code></pre>
                  </div>
                  <pre><code>##                    mpg cyl disp  hp drat    wt  qsec vs am gear carb
## Mazda RX4         21.0   6  160 110 3.90 2.620 16.46  0  1    4    4
## Mazda RX4 Wag     21.0   6  160 110 3.90 2.875 17.02  0  1    4    4
## Datsun 710        22.8   4  108  93 3.85 2.320 18.61  1  1    4    1
## Hornet 4 Drive    21.4   6  258 110 3.08 3.215 19.44  1  0    3    1
## Hornet Sportabout 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2
## Valiant           18.1   6  225 105 2.76 3.460 20.22  1  0    3    1</code></pre>
                </div>
                <div id="plot-data" class="section level5">
                  <h5>Plot data</h5>
                  <div class="sourceCode" id="cb280">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb280-1"><a href="#cb280-1"></a><span class="co"># from ggplot2 documentation p.135</span></span>
<span id="cb280-2"><a href="#cb280-2"></a><span class="co"># Use data from data.frame</span></span>
<span id="cb280-3"><a href="#cb280-3"></a><span class="kw">qplot</span>(mpg, wt, <span class="dt">data =</span> mtcars)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-1.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb281">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb281-1"><a href="#cb281-1"></a><span class="kw">qplot</span>(mpg, wt, <span class="dt">data =</span> mtcars, <span class="dt">colour =</span> cyl)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-2.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb282">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb282-1"><a href="#cb282-1"></a><span class="kw">qplot</span>(mpg, wt, <span class="dt">data =</span> mtcars, <span class="dt">size =</span> cyl)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-3.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb283">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb283-1"><a href="#cb283-1"></a><span class="kw">qplot</span>(mpg, wt, <span class="dt">data =</span> mtcars, <span class="dt">facets =</span> vs <span class="op">~</span><span class="st"> </span>am)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-4.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb284">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb284-1"><a href="#cb284-1"></a><span class="co"># It will use data from local environment</span></span>
<span id="cb284-2"><a href="#cb284-2"></a>hp &lt;-<span class="st"> </span>mtcars<span class="op">$</span>hp</span>
<span id="cb284-3"><a href="#cb284-3"></a>wt &lt;-<span class="st"> </span>mtcars<span class="op">$</span>wt</span>
<span id="cb284-4"><a href="#cb284-4"></a>cyl &lt;-<span class="st"> </span>mtcars<span class="op">$</span>cyl</span>
<span id="cb284-5"><a href="#cb284-5"></a>vs &lt;-<span class="st"> </span>mtcars<span class="op">$</span>vs</span>
<span id="cb284-6"><a href="#cb284-6"></a>am &lt;-<span class="st"> </span>mtcars<span class="op">$</span>am</span>
<span id="cb284-7"><a href="#cb284-7"></a><span class="kw">qplot</span>(hp, wt)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-5.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb285">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb285-1"><a href="#cb285-1"></a><span class="kw">qplot</span>(hp, wt, <span class="dt">colour =</span> cyl)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-6.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb286">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb286-1"><a href="#cb286-1"></a><span class="kw">qplot</span>(hp, wt, <span class="dt">size =</span> cyl)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-7.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb287">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb287-1"><a href="#cb287-1"></a><span class="kw">qplot</span>(hp, wt, <span class="dt">facets =</span> vs <span class="op">~</span><span class="st"> </span>am)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-8.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb288">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb288-1"><a href="#cb288-1"></a><span class="kw">qplot</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">rnorm</span>(<span class="dv">10</span>), <span class="dt">colour =</span> <span class="kw">runif</span>(<span class="dv">10</span>))</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-9.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb289">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb289-1"><a href="#cb289-1"></a><span class="kw">qplot</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, letters[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-10.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb290">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb290-1"><a href="#cb290-1"></a>mod &lt;-<span class="st"> </span><span class="kw">lm</span>(mpg <span class="op">~</span><span class="st"> </span>wt, <span class="dt">data =</span> mtcars)</span>
<span id="cb290-2"><a href="#cb290-2"></a><span class="kw">qplot</span>(<span class="kw">resid</span>(mod), <span class="kw">fitted</span>(mod))</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-11.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb291">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb291-1"><a href="#cb291-1"></a><span class="kw">qplot</span>(<span class="kw">resid</span>(mod), <span class="kw">fitted</span>(mod), <span class="dt">facets =</span> . <span class="op">~</span><span class="st"> </span>vs)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-12.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb292">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb292-1"><a href="#cb292-1"></a>f &lt;-<span class="st"> </span><span class="cf">function</span>() {</span>
<span id="cb292-2"><a href="#cb292-2"></a>  a &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">10</span></span>
<span id="cb292-3"><a href="#cb292-3"></a>  b &lt;-<span class="st"> </span>a <span class="op">^</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb292-4"><a href="#cb292-4"></a>  <span class="kw">qplot</span>(a, b)</span>
<span id="cb292-5"><a href="#cb292-5"></a>}</span>
<span id="cb292-6"><a href="#cb292-6"></a><span class="kw">f</span>()</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-13.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb293">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb293-1"><a href="#cb293-1"></a><span class="co"># qplot will attempt to guess what geom you want depending on the input</span></span>
<span id="cb293-2"><a href="#cb293-2"></a><span class="co"># both x and y supplied = scatterplot</span></span>
<span id="cb293-3"><a href="#cb293-3"></a><span class="kw">qplot</span>(mpg, wt, <span class="dt">data =</span> mtcars)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-14.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb294">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb294-1"><a href="#cb294-1"></a><span class="co"># just x supplied = histogram</span></span>
<span id="cb294-2"><a href="#cb294-2"></a><span class="kw">qplot</span>(mpg, <span class="dt">data =</span> mtcars)</span></code></pre>
                  </div>
                  <pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-15.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb296">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb296-1"><a href="#cb296-1"></a><span class="co"># just y supplied = scatterplot, with x = seq_along(y)</span></span>
<span id="cb296-2"><a href="#cb296-2"></a><span class="kw">qplot</span>(<span class="dt">y =</span> mpg, <span class="dt">data =</span> mtcars)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-16.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb297">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb297-1"><a href="#cb297-1"></a><span class="co"># Use different geoms</span></span>
<span id="cb297-2"><a href="#cb297-2"></a><span class="kw">qplot</span>(mpg, wt, <span class="dt">data =</span> mtcars, <span class="dt">geom =</span> <span class="st">&#39;path&#39;</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-17.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb298">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb298-1"><a href="#cb298-1"></a><span class="kw">qplot</span>(<span class="kw">factor</span>(cyl),</span>
<span id="cb298-2"><a href="#cb298-2"></a>      wt,</span>
<span id="cb298-3"><a href="#cb298-3"></a>      <span class="dt">data =</span> mtcars,</span>
<span id="cb298-4"><a href="#cb298-4"></a>      <span class="dt">geom =</span> <span class="kw">c</span>(<span class="st">&#39;boxplot&#39;</span>, <span class="st">&#39;jitter&#39;</span>))</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-18.png"
                      width="672"
                    />
                  </p>
                  <div class="sourceCode" id="cb299">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb299-1"><a href="#cb299-1"></a><span class="kw">qplot</span>(mpg, <span class="dt">data =</span> mtcars, <span class="dt">geom =</span> <span class="st">&#39;dotplot&#39;</span>)</span></code></pre>
                  </div>
                  <pre><code>## `stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-ggplot2-02-plot-19.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
              <div
                id="creating-rotated-axis-labels-in-r"
                class="section level4"
              >
                <h4>Creating rotated axis labels in R</h4>
                <p>
                  7.27
                  <a href="http://www.hep.by/gnu/r-patched/r-faq/R-FAQ_78.html"
                    >How can I create rotated axis labels?</a
                  >
                </p>
                <div id="intro-2" class="section level5">
                  <h5>Intro</h5>
                  <p>
                    To rotate axis labels (using base graphics), you need to use
                    text(), rather than mtext(), as the latter does not support
                    par(‘srt’).
                  </p>
                </div>
                <div id="example-plot" class="section level5">
                  <h5>Example plot</h5>
                  <div class="sourceCode" id="cb301">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb301-1"><a href="#cb301-1"></a><span class="co"># Increase bottom margin to make room for rotated labels</span></span>
<span id="cb301-2"><a href="#cb301-2"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">7</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="fl">0.1</span>)</span>
<span id="cb301-3"><a href="#cb301-3"></a><span class="co"># Create plot with no x axis and no x axis label</span></span>
<span id="cb301-4"><a href="#cb301-4"></a><span class="kw">plot</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">8</span>, <span class="dt">xaxt =</span> <span class="st">&#39;n&#39;</span>,  <span class="dt">xlab =</span> <span class="st">&#39;&#39;</span>)</span>
<span id="cb301-5"><a href="#cb301-5"></a><span class="co"># Set up x axis with tick marks alone</span></span>
<span id="cb301-6"><a href="#cb301-6"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">labels =</span> <span class="ot">FALSE</span>)</span>
<span id="cb301-7"><a href="#cb301-7"></a><span class="co"># Create some text labels</span></span>
<span id="cb301-8"><a href="#cb301-8"></a>labels &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&#39;Label&#39;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>, <span class="dt">sep =</span> <span class="st">&#39; &#39;</span>)</span>
<span id="cb301-9"><a href="#cb301-9"></a><span class="co"># Plot x axis labels at default tick marks</span></span>
<span id="cb301-10"><a href="#cb301-10"></a><span class="kw">text</span>(</span>
<span id="cb301-11"><a href="#cb301-11"></a>  <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>,</span>
<span id="cb301-12"><a href="#cb301-12"></a>  <span class="kw">par</span>(<span class="st">&#39;usr&#39;</span>)[<span class="dv">3</span>] <span class="op">-</span><span class="st"> </span><span class="fl">0.25</span>,</span>
<span id="cb301-13"><a href="#cb301-13"></a>  <span class="dt">srt =</span> <span class="dv">45</span>,</span>
<span id="cb301-14"><a href="#cb301-14"></a>  <span class="dt">adj =</span> <span class="dv">1</span>,</span>
<span id="cb301-15"><a href="#cb301-15"></a>  <span class="dt">labels =</span> labels,</span>
<span id="cb301-16"><a href="#cb301-16"></a>  <span class="dt">xpd =</span> <span class="ot">TRUE</span></span>
<span id="cb301-17"><a href="#cb301-17"></a>)</span>
<span id="cb301-18"><a href="#cb301-18"></a><span class="co"># Plot x axis label at line 6 (of 7)</span></span>
<span id="cb301-19"><a href="#cb301-19"></a><span class="kw">mtext</span>(<span class="dv">1</span>, <span class="dt">text =</span> <span class="st">&#39;X Axis Label&#39;</span>, <span class="dt">line =</span> <span class="dv">6</span>)</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-rotated-axis-labels-01-1.png"
                      width="672"
                    />
                  </p>
                </div>
                <div id="comments" class="section level5">
                  <h5>Comments</h5>
                  <ul>
                    <li>
                      When plotting the x axis labels, we use srt = 45 for text
                      rotation angle, adj = 1 to place the right end of text at
                      the tick marks, and xpd = TRUE to allow for text outside
                      the plot region. You can adjust the value of the 0.25
                      offset as required to move the axis labels up or down
                      relative to the x axis. See ?par for more information.
                    </li>
                    <li>
                      Also see Figure 1 and associated code in Paul Murrell
                      (2003), ‘Integrating grid Graphics Output with Base
                      Graphics Output’, R News, 3/2, 7-12.
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div id="volcano-plots" class="section level3">
              <h3>Volcano plots</h3>
              <div id="getting-genetics-done-example" class="section level4">
                <h4>Getting Genetics Done example</h4>
                <p>
                  Volcano plot example from
                  <a
                    href="http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html"
                    >Getting Genetics Done</a
                  >, with data from
                  <a
                    href="https://gist.githubusercontent.com/stephenturner/806e31fce55a8b7175af/raw/9479acb809fae09aa50ea7df54a1199b3f1ffa11/results.txt"
                    >this GitHub Gist</a
                  >.
                </p>
                <div id="import-dataset-from-github" class="section level5">
                  <h5>Import dataset from GitHub</h5>
                  <div class="sourceCode" id="cb302">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb302-1"><a href="#cb302-1"></a>res &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;data/volcano_ggd_data.txt&#39;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb302-2"><a href="#cb302-2"></a><span class="kw">head</span>(res)</span></code></pre>
                  </div>
                  <pre><code>##      Gene log2FoldChange    pvalue      padj
## 1    DOK6         0.5100 1.861e-08 0.0003053
## 2    TBX5        -2.1290 5.655e-08 0.0004191
## 3 SLC32A1         0.9003 7.664e-08 0.0004191
## 4  IFITM1        -1.6870 3.735e-06 0.0068090
## 5   NUP93         0.3659 3.373e-06 0.0068090
## 6 EMILIN2         1.5340 2.976e-06 0.0068090</code></pre>
                </div>
                <div id="make-a-basic-volcano-plot" class="section level5">
                  <h5>Make a basic volcano plot</h5>
                  <p>
                    The <code>-log10(pvalue)</code> is used so p values can be
                    plotted as whole numbers. Note that I was not able to
                    separate each section into different R markdown code chunks.
                    I needed the generation of points to be continuous with
                    generation of the original plot.
                  </p>
                  <div class="sourceCode" id="cb304">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb304-1"><a href="#cb304-1"></a><span class="kw">with</span>(res,</span>
<span id="cb304-2"><a href="#cb304-2"></a>     <span class="kw">plot</span>(</span>
<span id="cb304-3"><a href="#cb304-3"></a>       log2FoldChange,</span>
<span id="cb304-4"><a href="#cb304-4"></a>       <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb304-5"><a href="#cb304-5"></a>       <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb304-6"><a href="#cb304-6"></a>       <span class="dt">main =</span> <span class="st">&#39;Volcano plot&#39;</span>,</span>
<span id="cb304-7"><a href="#cb304-7"></a>       <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">2.5</span>, <span class="dv">2</span>)</span>
<span id="cb304-8"><a href="#cb304-8"></a>     ))</span>
<span id="cb304-9"><a href="#cb304-9"></a><span class="co"># Add colored points:</span></span>
<span id="cb304-10"><a href="#cb304-10"></a><span class="co"># red if padj&lt;0.05</span></span>
<span id="cb304-11"><a href="#cb304-11"></a><span class="co"># orange if log2FC&gt;1</span></span>
<span id="cb304-12"><a href="#cb304-12"></a><span class="co"># green if both</span></span>
<span id="cb304-13"><a href="#cb304-13"></a><span class="kw">with</span>(<span class="kw">subset</span>(res, padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span>),</span>
<span id="cb304-14"><a href="#cb304-14"></a>     <span class="kw">points</span>(</span>
<span id="cb304-15"><a href="#cb304-15"></a>       log2FoldChange,</span>
<span id="cb304-16"><a href="#cb304-16"></a>       <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb304-17"><a href="#cb304-17"></a>       <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb304-18"><a href="#cb304-18"></a>       <span class="dt">col =</span> <span class="st">&#39;red&#39;</span></span>
<span id="cb304-19"><a href="#cb304-19"></a>     ))</span>
<span id="cb304-20"><a href="#cb304-20"></a><span class="kw">with</span>(<span class="kw">subset</span>(res, <span class="kw">abs</span>(log2FoldChange) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb304-21"><a href="#cb304-21"></a>     <span class="kw">points</span>(</span>
<span id="cb304-22"><a href="#cb304-22"></a>       log2FoldChange,</span>
<span id="cb304-23"><a href="#cb304-23"></a>       <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb304-24"><a href="#cb304-24"></a>       <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb304-25"><a href="#cb304-25"></a>       <span class="dt">col =</span> <span class="st">&#39;orange&#39;</span></span>
<span id="cb304-26"><a href="#cb304-26"></a>     ))</span>
<span id="cb304-27"><a href="#cb304-27"></a><span class="kw">with</span>(<span class="kw">subset</span>(res, padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb304-28"><a href="#cb304-28"></a>     <span class="kw">points</span>(</span>
<span id="cb304-29"><a href="#cb304-29"></a>       log2FoldChange,</span>
<span id="cb304-30"><a href="#cb304-30"></a>       <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb304-31"><a href="#cb304-31"></a>       <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb304-32"><a href="#cb304-32"></a>       <span class="dt">col =</span> <span class="st">&#39;green&#39;</span></span>
<span id="cb304-33"><a href="#cb304-33"></a>     ))</span>
<span id="cb304-34"><a href="#cb304-34"></a></span>
<span id="cb304-35"><a href="#cb304-35"></a><span class="co"># Label points with the textxy function from the calibrate package</span></span>
<span id="cb304-36"><a href="#cb304-36"></a><span class="kw">library</span>(calibrate)</span>
<span id="cb304-37"><a href="#cb304-37"></a><span class="kw">with</span>(<span class="kw">subset</span>(res, padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb304-38"><a href="#cb304-38"></a>     <span class="kw">textxy</span>(</span>
<span id="cb304-39"><a href="#cb304-39"></a>       log2FoldChange,</span>
<span id="cb304-40"><a href="#cb304-40"></a>       <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb304-41"><a href="#cb304-41"></a>       <span class="dt">labs =</span> Gene,</span>
<span id="cb304-42"><a href="#cb304-42"></a>       <span class="dt">cex =</span> <span class="fl">.8</span></span>
<span id="cb304-43"><a href="#cb304-43"></a>     ))</span></code></pre>
                  </div>
                  <p>
                    <img
                      src="code_files/figure-html/code-plot-volcano-ggd-02-plot-1.png"
                      width="672"
                    />
                  </p>
                </div>
              </div>
              <div id="nrf1-proteomics-volcano-plot" class="section level4">
                <h4>Nrf1 proteomics volcano plot</h4>
                <p>
                  Experimental data example from
                  <a href="https://github.com/br3ndonland/R-proteomics-Nrf1"
                    >Nrf1 proteomics project</a
                  >. The goal of this experiment was to identify a molecular
                  complex associated with Nrf1, a protein molecule our research
                  group was studying. Complement C1q proteins were identified as
                  potentially interacting with Nrf1.
                </p>
                <div id="import-data-1" class="section level5">
                  <h5>Import data</h5>
                  <div class="sourceCode" id="cb305">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb305-1"><a href="#cb305-1"></a>proteomics_data &lt;-</span>
<span id="cb305-2"><a href="#cb305-2"></a><span class="st">  </span><span class="kw">read.table</span>(</span>
<span id="cb305-3"><a href="#cb305-3"></a>    <span class="st">&#39;https://raw.githubusercontent.com/br3ndonland/R-proteomics-Nrf1/master/data/R-proteomics-Nrf1.csv&#39;</span>,</span>
<span id="cb305-4"><a href="#cb305-4"></a>    <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb305-5"><a href="#cb305-5"></a>    <span class="dt">sep =</span> <span class="st">&#39;,&#39;</span></span>
<span id="cb305-6"><a href="#cb305-6"></a>  )</span>
<span id="cb305-7"><a href="#cb305-7"></a><span class="kw">attach</span>(proteomics_data)</span>
<span id="cb305-8"><a href="#cb305-8"></a><span class="kw">head</span>(proteomics_data)</span></code></pre>
                  </div>
                  <pre><code>##       Gene log2_FC_HAchol_HAchow log2_FC_HAchow_lacZ     pvalue pvaluechow log2_FC_HAbort_HAchow pvaluebort
## 1 Serpinb5            0.09538888          -1.3634997 0.74081917 0.39977101             1.7303640 0.31471162
## 2      Sfn            0.41354494          -1.0428302 0.40347276 0.20906780             2.5999482 0.01377441
## 3     Eno1            0.63929006          -0.6662456 0.22617602 0.32692726             0.8034121 0.06946092
## 4     C1qc            1.25725953           0.0891710 0.01419030 0.86011207             0.0946041 0.84034899
## 5     C1qa            1.16884182           0.0303050 0.01219113 0.95321354             0.1015774 0.80231428
## 6    Gstm1            0.19700734          -0.8650671 0.65225515 0.09657287             0.4450431 0.21202782</code></pre>
                </div>
                <div id="generate-plots-7" class="section level5">
                  <h5>Generate plots</h5>
                  <ul>
                    <li>
                      Each point is a protein.
                      <ul>
                        <li>Red if p&lt;0.05 for the comparison shown</li>
                        <li>Orange if [log2 fold change]&gt;1</li>
                        <li>Green if both</li>
                      </ul>
                    </li>
                    <li>
                      log2 transformation is used to normalize positive and
                      negative fold changes.
                    </li>
                    <li>
                      -log10(pvalue) is used so p values can be plotted as whole
                      numbers.
                    </li>
                  </ul>
                  <div id="ha-cholesterol-plot" class="section level6">
                    <h6>HA cholesterol plot</h6>
                    <p>
                      This plot compares chow-fed mice with mice fed the Paigen
                      diet to promote accumulation of cholesterol in the liver,
                      and demonstrates the increased abundance of Complement
                      C1qA, C1qB, and C1qC, in mice fed the Paigen diet.
                    </p>
                    <div class="sourceCode" id="cb307">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb307-1"><a href="#cb307-1"></a><span class="co"># bold axis titles</span></span>
<span id="cb307-2"><a href="#cb307-2"></a><span class="kw">par</span>(<span class="dt">font.lab =</span> <span class="dv">2</span>)</span>
<span id="cb307-3"><a href="#cb307-3"></a></span>
<span id="cb307-4"><a href="#cb307-4"></a><span class="co"># create plot</span></span>
<span id="cb307-5"><a href="#cb307-5"></a><span class="kw">with</span>(</span>
<span id="cb307-6"><a href="#cb307-6"></a>  proteomics_data,</span>
<span id="cb307-7"><a href="#cb307-7"></a>  <span class="kw">plot</span>(</span>
<span id="cb307-8"><a href="#cb307-8"></a>    log2_FC_HAchol_HAchow,</span>
<span id="cb307-9"><a href="#cb307-9"></a>    <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb307-10"><a href="#cb307-10"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb307-11"><a href="#cb307-11"></a>    <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">2</span>, <span class="dv">2</span>),</span>
<span id="cb307-12"><a href="#cb307-12"></a>    <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">3</span>),</span>
<span id="cb307-13"><a href="#cb307-13"></a>    <span class="dt">yaxp =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">3</span>, <span class="dv">3</span>),</span>
<span id="cb307-14"><a href="#cb307-14"></a>    <span class="dt">main =</span> <span class="st">&#39;Cholesterol-induced Nrf1 interacting proteins&#39;</span>,</span>
<span id="cb307-15"><a href="#cb307-15"></a>    <span class="dt">xlab =</span> <span class="st">&#39;log2(FC HA chol/HA chow)&#39;</span>,</span>
<span id="cb307-16"><a href="#cb307-16"></a>    <span class="dt">ylab =</span> <span class="st">&#39;-log10(pvalue HA chol vs HA chow)&#39;</span></span>
<span id="cb307-17"><a href="#cb307-17"></a>  )</span>
<span id="cb307-18"><a href="#cb307-18"></a>)</span>
<span id="cb307-19"><a href="#cb307-19"></a></span>
<span id="cb307-20"><a href="#cb307-20"></a><span class="co"># Add colored points:</span></span>
<span id="cb307-21"><a href="#cb307-21"></a><span class="co"># red if pvalue&lt;0.05, orange if [log2FC]&gt;1, green if both)</span></span>
<span id="cb307-22"><a href="#cb307-22"></a><span class="kw">with</span>(</span>
<span id="cb307-23"><a href="#cb307-23"></a>  <span class="kw">subset</span>(proteomics_data, pvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span>),</span>
<span id="cb307-24"><a href="#cb307-24"></a>  <span class="kw">points</span>(</span>
<span id="cb307-25"><a href="#cb307-25"></a>    log2_FC_HAchol_HAchow,</span>
<span id="cb307-26"><a href="#cb307-26"></a>    <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb307-27"><a href="#cb307-27"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb307-28"><a href="#cb307-28"></a>    <span class="dt">col =</span> <span class="st">&#39;red&#39;</span></span>
<span id="cb307-29"><a href="#cb307-29"></a>  )</span>
<span id="cb307-30"><a href="#cb307-30"></a>)</span>
<span id="cb307-31"><a href="#cb307-31"></a></span>
<span id="cb307-32"><a href="#cb307-32"></a><span class="kw">with</span>(</span>
<span id="cb307-33"><a href="#cb307-33"></a>  <span class="kw">subset</span>(proteomics_data, <span class="kw">abs</span>(log2_FC_HAchol_HAchow) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb307-34"><a href="#cb307-34"></a>  <span class="kw">points</span>(</span>
<span id="cb307-35"><a href="#cb307-35"></a>    log2_FC_HAchol_HAchow,</span>
<span id="cb307-36"><a href="#cb307-36"></a>    <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb307-37"><a href="#cb307-37"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb307-38"><a href="#cb307-38"></a>    <span class="dt">col =</span> <span class="st">&#39;orange&#39;</span></span>
<span id="cb307-39"><a href="#cb307-39"></a>  )</span>
<span id="cb307-40"><a href="#cb307-40"></a>)</span>
<span id="cb307-41"><a href="#cb307-41"></a></span>
<span id="cb307-42"><a href="#cb307-42"></a><span class="kw">with</span>(</span>
<span id="cb307-43"><a href="#cb307-43"></a>  <span class="kw">subset</span>(proteomics_data, pvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span></span>
<span id="cb307-44"><a href="#cb307-44"></a><span class="st">           </span><span class="kw">abs</span>(log2_FC_HAchol_HAchow) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb307-45"><a href="#cb307-45"></a>  <span class="kw">points</span>(</span>
<span id="cb307-46"><a href="#cb307-46"></a>    log2_FC_HAchol_HAchow,</span>
<span id="cb307-47"><a href="#cb307-47"></a>    <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb307-48"><a href="#cb307-48"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb307-49"><a href="#cb307-49"></a>    <span class="dt">col =</span> <span class="st">&#39;green&#39;</span></span>
<span id="cb307-50"><a href="#cb307-50"></a>  )</span>
<span id="cb307-51"><a href="#cb307-51"></a>)</span>
<span id="cb307-52"><a href="#cb307-52"></a></span>
<span id="cb307-53"><a href="#cb307-53"></a><span class="co"># Label points with the textxy function from the calibrate package</span></span>
<span id="cb307-54"><a href="#cb307-54"></a><span class="kw">library</span>(calibrate)</span>
<span id="cb307-55"><a href="#cb307-55"></a><span class="kw">with</span>(</span>
<span id="cb307-56"><a href="#cb307-56"></a>  <span class="kw">subset</span>(proteomics_data, pvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span></span>
<span id="cb307-57"><a href="#cb307-57"></a><span class="st">           </span><span class="kw">abs</span>(log2_FC_HAchol_HAchow) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb307-58"><a href="#cb307-58"></a>  <span class="kw">textxy</span>(</span>
<span id="cb307-59"><a href="#cb307-59"></a>    log2_FC_HAchol_HAchow,</span>
<span id="cb307-60"><a href="#cb307-60"></a>    <span class="op">-</span><span class="kw">log10</span>(pvalue),</span>
<span id="cb307-61"><a href="#cb307-61"></a>    <span class="dt">labs =</span> Gene,</span>
<span id="cb307-62"><a href="#cb307-62"></a>    <span class="dt">cex =</span> <span class="dv">1</span></span>
<span id="cb307-63"><a href="#cb307-63"></a>  )</span>
<span id="cb307-64"><a href="#cb307-64"></a>)</span></code></pre>
                    </div>
                    <p>
                      <img
                        src="code_files/figure-html/code-plot-volcano-nrf1-02-plot-hachol-1.png"
                        width="672"
                      />
                    </p>
                  </div>
                  <div id="ha-bortezomib-plot" class="section level6">
                    <h6>HA Bortezomib plot</h6>
                    <p>Bortezomib treatment alters the proteome.</p>
                    <div class="sourceCode" id="cb308">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb308-1"><a href="#cb308-1"></a><span class="co"># bold axis titles</span></span>
<span id="cb308-2"><a href="#cb308-2"></a><span class="kw">par</span>(<span class="dt">font.lab =</span> <span class="dv">2</span>)</span>
<span id="cb308-3"><a href="#cb308-3"></a></span>
<span id="cb308-4"><a href="#cb308-4"></a><span class="co"># create plot</span></span>
<span id="cb308-5"><a href="#cb308-5"></a><span class="kw">with</span>(</span>
<span id="cb308-6"><a href="#cb308-6"></a>  proteomics_data,</span>
<span id="cb308-7"><a href="#cb308-7"></a>  <span class="kw">plot</span>(</span>
<span id="cb308-8"><a href="#cb308-8"></a>    log2_FC_HAbort_HAchow,</span>
<span id="cb308-9"><a href="#cb308-9"></a>    <span class="op">-</span><span class="kw">log10</span>(pvaluebort),</span>
<span id="cb308-10"><a href="#cb308-10"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb308-11"><a href="#cb308-11"></a>    <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">3</span>, <span class="dv">3</span>),</span>
<span id="cb308-12"><a href="#cb308-12"></a>    <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4</span>),</span>
<span id="cb308-13"><a href="#cb308-13"></a>    <span class="dt">main =</span> <span class="st">&#39;Nrf1 AP-MS volcano plot Bortezomib vs chow&#39;</span>,</span>
<span id="cb308-14"><a href="#cb308-14"></a>    <span class="dt">xlab =</span> <span class="st">&#39;log2(FC HA bort/HA chow)&#39;</span>,</span>
<span id="cb308-15"><a href="#cb308-15"></a>    <span class="dt">ylab =</span> <span class="st">&#39;-log10(pvalue HA bort vs HA chow)&#39;</span></span>
<span id="cb308-16"><a href="#cb308-16"></a>  )</span>
<span id="cb308-17"><a href="#cb308-17"></a>)</span>
<span id="cb308-18"><a href="#cb308-18"></a><span class="co"># Add colored points:</span></span>
<span id="cb308-19"><a href="#cb308-19"></a><span class="co"># red if pvalue&lt;0.05, orange if log2FC&gt;1, green if both)</span></span>
<span id="cb308-20"><a href="#cb308-20"></a><span class="kw">with</span>(</span>
<span id="cb308-21"><a href="#cb308-21"></a>  <span class="kw">subset</span>(proteomics_data, pvaluebort <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span>),</span>
<span id="cb308-22"><a href="#cb308-22"></a>  <span class="kw">points</span>(</span>
<span id="cb308-23"><a href="#cb308-23"></a>    log2_FC_HAbort_HAchow,</span>
<span id="cb308-24"><a href="#cb308-24"></a>    <span class="op">-</span><span class="kw">log10</span>(pvaluebort),</span>
<span id="cb308-25"><a href="#cb308-25"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb308-26"><a href="#cb308-26"></a>    <span class="dt">col =</span> <span class="st">&#39;red&#39;</span></span>
<span id="cb308-27"><a href="#cb308-27"></a>  )</span>
<span id="cb308-28"><a href="#cb308-28"></a>)</span>
<span id="cb308-29"><a href="#cb308-29"></a></span>
<span id="cb308-30"><a href="#cb308-30"></a><span class="kw">with</span>(</span>
<span id="cb308-31"><a href="#cb308-31"></a>  <span class="kw">subset</span>(proteomics_data, <span class="kw">abs</span>(log2_FC_HAbort_HAchow) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb308-32"><a href="#cb308-32"></a>  <span class="kw">points</span>(</span>
<span id="cb308-33"><a href="#cb308-33"></a>    log2_FC_HAbort_HAchow,</span>
<span id="cb308-34"><a href="#cb308-34"></a>    <span class="op">-</span><span class="kw">log10</span>(pvaluebort),</span>
<span id="cb308-35"><a href="#cb308-35"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb308-36"><a href="#cb308-36"></a>    <span class="dt">col =</span> <span class="st">&#39;orange&#39;</span></span>
<span id="cb308-37"><a href="#cb308-37"></a>  )</span>
<span id="cb308-38"><a href="#cb308-38"></a>)</span>
<span id="cb308-39"><a href="#cb308-39"></a></span>
<span id="cb308-40"><a href="#cb308-40"></a><span class="kw">with</span>(</span>
<span id="cb308-41"><a href="#cb308-41"></a>  <span class="kw">subset</span>(</span>
<span id="cb308-42"><a href="#cb308-42"></a>    proteomics_data,</span>
<span id="cb308-43"><a href="#cb308-43"></a>    pvaluebort <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2_FC_HAbort_HAchow) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb308-44"><a href="#cb308-44"></a>  ),</span>
<span id="cb308-45"><a href="#cb308-45"></a>  <span class="kw">points</span>(</span>
<span id="cb308-46"><a href="#cb308-46"></a>    log2_FC_HAbort_HAchow,</span>
<span id="cb308-47"><a href="#cb308-47"></a>    <span class="op">-</span><span class="kw">log10</span>(pvaluebort),</span>
<span id="cb308-48"><a href="#cb308-48"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb308-49"><a href="#cb308-49"></a>    <span class="dt">col =</span> <span class="st">&#39;green&#39;</span></span>
<span id="cb308-50"><a href="#cb308-50"></a>  )</span>
<span id="cb308-51"><a href="#cb308-51"></a>)</span>
<span id="cb308-52"><a href="#cb308-52"></a></span>
<span id="cb308-53"><a href="#cb308-53"></a><span class="co"># Label points with the textxy function from the calibrate package</span></span>
<span id="cb308-54"><a href="#cb308-54"></a><span class="kw">library</span>(calibrate)</span>
<span id="cb308-55"><a href="#cb308-55"></a><span class="kw">with</span>(</span>
<span id="cb308-56"><a href="#cb308-56"></a>  <span class="kw">subset</span>(</span>
<span id="cb308-57"><a href="#cb308-57"></a>    proteomics_data,</span>
<span id="cb308-58"><a href="#cb308-58"></a>    pvaluebort <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2_FC_HAbort_HAchow) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb308-59"><a href="#cb308-59"></a>  ),</span>
<span id="cb308-60"><a href="#cb308-60"></a>  <span class="kw">textxy</span>(</span>
<span id="cb308-61"><a href="#cb308-61"></a>    log2_FC_HAbort_HAchow,</span>
<span id="cb308-62"><a href="#cb308-62"></a>    <span class="op">-</span><span class="kw">log10</span>(pvaluebort),</span>
<span id="cb308-63"><a href="#cb308-63"></a>    <span class="dt">labs =</span> Gene,</span>
<span id="cb308-64"><a href="#cb308-64"></a>    <span class="dt">cex =</span> <span class="dv">1</span></span>
<span id="cb308-65"><a href="#cb308-65"></a>  )</span>
<span id="cb308-66"><a href="#cb308-66"></a>)</span></code></pre>
                    </div>
                    <p>
                      <img
                        src="code_files/figure-html/code-plot-volcano-nrf1-02-plot-bort-1.png"
                        width="672"
                      />
                    </p>
                  </div>
                  <div id="ha-vs-lacz-plot" class="section level6">
                    <h6>HA vs lacZ plot</h6>
                    <p>
                      The proteome was not significantly different with or
                      without the HA tag, indicating issues with the HA
                      immunoprecipitation. Note that the code for labeling
                      points appears to be separated from the rest of the code
                      for the plot, possibly because there were no points
                      labeled.
                    </p>
                    <div class="sourceCode" id="cb309">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb309-1"><a href="#cb309-1"></a><span class="co"># bold axis titles</span></span>
<span id="cb309-2"><a href="#cb309-2"></a><span class="kw">par</span>(<span class="dt">font.lab =</span> <span class="dv">2</span>)</span>
<span id="cb309-3"><a href="#cb309-3"></a></span>
<span id="cb309-4"><a href="#cb309-4"></a><span class="co"># create plot</span></span>
<span id="cb309-5"><a href="#cb309-5"></a><span class="kw">with</span>(</span>
<span id="cb309-6"><a href="#cb309-6"></a>  proteomics_data,</span>
<span id="cb309-7"><a href="#cb309-7"></a>  <span class="kw">plot</span>(</span>
<span id="cb309-8"><a href="#cb309-8"></a>    log2_FC_HAchow_lacZ,<span class="op">-</span><span class="kw">log10</span>(pvaluechow),</span>
<span id="cb309-9"><a href="#cb309-9"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb309-10"><a href="#cb309-10"></a>    <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">2</span>, <span class="dv">2</span>),</span>
<span id="cb309-11"><a href="#cb309-11"></a>    <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">3</span>),</span>
<span id="cb309-12"><a href="#cb309-12"></a>    <span class="dt">yaxp =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">3</span>, <span class="dv">3</span>),</span>
<span id="cb309-13"><a href="#cb309-13"></a>    <span class="dt">main =</span> <span class="st">&#39;Nrf1 AP-MS volcano plot HA vs lacZ&#39;</span>,</span>
<span id="cb309-14"><a href="#cb309-14"></a>    <span class="dt">xlab =</span> <span class="st">&#39;log2(FC HA chow/lacZ chow)&#39;</span>,</span>
<span id="cb309-15"><a href="#cb309-15"></a>    <span class="dt">ylab =</span> <span class="st">&#39;-log10(pvalue HA chow vs lacZ chow)&#39;</span></span>
<span id="cb309-16"><a href="#cb309-16"></a>  )</span>
<span id="cb309-17"><a href="#cb309-17"></a>)</span>
<span id="cb309-18"><a href="#cb309-18"></a></span>
<span id="cb309-19"><a href="#cb309-19"></a><span class="co"># Add colored points:</span></span>
<span id="cb309-20"><a href="#cb309-20"></a><span class="co"># red if pvalue&lt;0.05, orange if [log2FC]&gt;1, green if both)</span></span>
<span id="cb309-21"><a href="#cb309-21"></a><span class="kw">with</span>(</span>
<span id="cb309-22"><a href="#cb309-22"></a>  <span class="kw">subset</span>(proteomics_data, pvaluechow <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span>),</span>
<span id="cb309-23"><a href="#cb309-23"></a>  <span class="kw">points</span>(</span>
<span id="cb309-24"><a href="#cb309-24"></a>    log2_FC_HAchow_lacZ,<span class="op">-</span><span class="kw">log10</span>(pvaluechow),</span>
<span id="cb309-25"><a href="#cb309-25"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb309-26"><a href="#cb309-26"></a>    <span class="dt">col =</span> <span class="st">&#39;red&#39;</span></span>
<span id="cb309-27"><a href="#cb309-27"></a>  )</span>
<span id="cb309-28"><a href="#cb309-28"></a>)</span>
<span id="cb309-29"><a href="#cb309-29"></a></span>
<span id="cb309-30"><a href="#cb309-30"></a><span class="kw">with</span>(</span>
<span id="cb309-31"><a href="#cb309-31"></a>  <span class="kw">subset</span>(proteomics_data, <span class="kw">abs</span>(log2_FC_HAchow_lacZ) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb309-32"><a href="#cb309-32"></a>  <span class="kw">points</span>(</span>
<span id="cb309-33"><a href="#cb309-33"></a>    log2_FC_HAchow_lacZ,<span class="op">-</span><span class="kw">log10</span>(pvaluechow),</span>
<span id="cb309-34"><a href="#cb309-34"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb309-35"><a href="#cb309-35"></a>    <span class="dt">col =</span> <span class="st">&#39;orange&#39;</span></span>
<span id="cb309-36"><a href="#cb309-36"></a>  )</span>
<span id="cb309-37"><a href="#cb309-37"></a>)</span>
<span id="cb309-38"><a href="#cb309-38"></a></span>
<span id="cb309-39"><a href="#cb309-39"></a><span class="kw">with</span>(</span>
<span id="cb309-40"><a href="#cb309-40"></a>  <span class="kw">subset</span>(proteomics_data, pvaluechow <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span></span>
<span id="cb309-41"><a href="#cb309-41"></a><span class="st">           </span><span class="kw">abs</span>(log2_FC_HAchow_lacZ) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb309-42"><a href="#cb309-42"></a>  <span class="kw">points</span>(</span>
<span id="cb309-43"><a href="#cb309-43"></a>    log2_FC_HAchow_lacZ,<span class="op">-</span><span class="kw">log10</span>(pvaluechow),</span>
<span id="cb309-44"><a href="#cb309-44"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb309-45"><a href="#cb309-45"></a>    <span class="dt">col =</span> <span class="st">&#39;green&#39;</span></span>
<span id="cb309-46"><a href="#cb309-46"></a>  )</span>
<span id="cb309-47"><a href="#cb309-47"></a>)</span></code></pre>
                    </div>
                    <p>
                      <img
                        src="code_files/figure-html/code-plot-volcano-nrf1-02-plot-halacz-1.png"
                        width="672"
                      />
                    </p>
                    <div class="sourceCode" id="cb310">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb310-1"><a href="#cb310-1"></a><span class="co"># Label points with the textxy function from the calibrate package</span></span>
<span id="cb310-2"><a href="#cb310-2"></a><span class="kw">library</span>(calibrate)</span>
<span id="cb310-3"><a href="#cb310-3"></a><span class="kw">with</span>(</span>
<span id="cb310-4"><a href="#cb310-4"></a>  <span class="kw">subset</span>(proteomics_data, pvaluechow <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span></span>
<span id="cb310-5"><a href="#cb310-5"></a><span class="st">           </span><span class="kw">abs</span>(log2_FC_HAchow_lacZ) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb310-6"><a href="#cb310-6"></a>  <span class="kw">textxy</span>(</span>
<span id="cb310-7"><a href="#cb310-7"></a>    log2_FC_HAchow_lacZ,<span class="op">-</span><span class="kw">log10</span>(pvaluechow),</span>
<span id="cb310-8"><a href="#cb310-8"></a>    <span class="dt">labs =</span> Gene,</span>
<span id="cb310-9"><a href="#cb310-9"></a>    <span class="dt">cex =</span> <span class="dv">1</span></span>
<span id="cb310-10"><a href="#cb310-10"></a>  )</span>
<span id="cb310-11"><a href="#cb310-11"></a>)</span></code></pre>
                    </div>
                  </div>
                  <div id="scatter-plot" class="section level6">
                    <h6>Scatter plot</h6>
                    <p>
                      The scatter plot is an interesting way to visualize two
                      between-group comparisons, but the volcano plot is more
                      useful for this dataset.
                    </p>
                    <p>To set size of points according to p value:</p>
                    <pre><code>dotsize &lt;- -log10(pvalue)
head(dotsize)</code></pre>
                    <p>
                      then put <code>cex=dotsize</code> inside the
                      <code>plot()</code> and <code>points()</code> functions.
                    </p>
                    <div class="sourceCode" id="cb312">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb312-1"><a href="#cb312-1"></a><span class="co"># set size of points according to p value</span></span>
<span id="cb312-2"><a href="#cb312-2"></a>dotsize &lt;-<span class="st"> </span><span class="op">-</span><span class="kw">log10</span>(pvalue)</span>
<span id="cb312-3"><a href="#cb312-3"></a></span>
<span id="cb312-4"><a href="#cb312-4"></a><span class="co"># bold axis titles</span></span>
<span id="cb312-5"><a href="#cb312-5"></a><span class="kw">par</span>(<span class="dt">font.lab =</span> <span class="dv">2</span>)</span>
<span id="cb312-6"><a href="#cb312-6"></a></span>
<span id="cb312-7"><a href="#cb312-7"></a><span class="co"># create plot</span></span>
<span id="cb312-8"><a href="#cb312-8"></a><span class="kw">with</span>(</span>
<span id="cb312-9"><a href="#cb312-9"></a>  proteomics_data,</span>
<span id="cb312-10"><a href="#cb312-10"></a>  <span class="kw">plot</span>(</span>
<span id="cb312-11"><a href="#cb312-11"></a>    log2_FC_HAchol_HAchow,</span>
<span id="cb312-12"><a href="#cb312-12"></a>    log2_FC_HAbort_HAchow,</span>
<span id="cb312-13"><a href="#cb312-13"></a>    <span class="dt">pch =</span> <span class="dv">20</span>,</span>
<span id="cb312-14"><a href="#cb312-14"></a>    <span class="dt">main =</span> <span class="st">&#39;Nrf1 AP-MS scatter plot&#39;</span>,</span>
<span id="cb312-15"><a href="#cb312-15"></a>    <span class="dt">xlab =</span> <span class="st">&#39;log2(FC HA chol/HA chow)&#39;</span>,</span>
<span id="cb312-16"><a href="#cb312-16"></a>    <span class="dt">ylab =</span> <span class="st">&#39;log2(FC HA bort/HA chow)&#39;</span>,</span>
<span id="cb312-17"><a href="#cb312-17"></a>    <span class="dt">cex =</span> dotsize</span>
<span id="cb312-18"><a href="#cb312-18"></a>  )</span>
<span id="cb312-19"><a href="#cb312-19"></a>)</span>
<span id="cb312-20"><a href="#cb312-20"></a></span>
<span id="cb312-21"><a href="#cb312-21"></a><span class="co"># Label points with the textxy function from the calibrate package</span></span>
<span id="cb312-22"><a href="#cb312-22"></a><span class="kw">library</span>(calibrate)</span>
<span id="cb312-23"><a href="#cb312-23"></a><span class="kw">with</span>(</span>
<span id="cb312-24"><a href="#cb312-24"></a>  <span class="kw">subset</span>(proteomics_data, pvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span></span>
<span id="cb312-25"><a href="#cb312-25"></a><span class="st">           </span><span class="kw">abs</span>(log2_FC_HAchol_HAchow) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb312-26"><a href="#cb312-26"></a>  <span class="kw">textxy</span>(log2_FC_HAchol_HAchow,</span>
<span id="cb312-27"><a href="#cb312-27"></a>         log2_FC_HAbort_HAchow,</span>
<span id="cb312-28"><a href="#cb312-28"></a>         <span class="dt">labs =</span> Gene)</span>
<span id="cb312-29"><a href="#cb312-29"></a>)</span></code></pre>
                    </div>
                    <p>
                      <img
                        src="code_files/figure-html/code-plot-volcano-nrf1-02-plot-scatter-1.png"
                        width="672"
                      />
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>

    <script>
      // add bootstrap table styles to pandoc tables
      function bootstrapStylePandocTables() {
        $("tr.header")
          .parent("thead")
          .parent("table")
          .addClass("table table-condensed")
      }
      $(document).ready(function () {
        bootstrapStylePandocTables()
      })
    </script>

    <!-- tabsets -->

    <script>
      $(document).ready(function () {
        window.buildTabsets("TOC")
      })

      $(document).ready(function () {
        $(".tabset-dropdown > .nav-tabs > li").click(function () {
          $(this).parent().toggleClass("nav-tabs-open")
        })
      })
    </script>

    <!-- code folding -->

    <script>
      $(document).ready(function () {
        // move toc-ignore selectors from section div to header
        $("div.section.toc-ignore")
          .removeClass("toc-ignore")
          .children("h1,h2,h3,h4,h5")
          .addClass("toc-ignore")

        // establish options
        var options = {
          selectors: "h1,h2,h3",
          theme: "bootstrap3",
          context: ".toc-content",
          hashGenerator: function (text) {
            return text.replace(/[.\\/?&!#<>]/g, "").replace(/\s/g, "_")
          },
          ignoreSelector: ".toc-ignore",
          scrollTo: 0,
        }
        options.showAndHide = true
        options.smoothScroll = true

        // tocify
        var toc = $("#TOC").tocify(options).data("toc-tocify")
      })
    </script>

    <!-- dynamically load mathjax for compatibility with self-contained -->
    <script>
      ;(function () {
        var script = document.createElement("script")
        script.type = "text/javascript"
        script.src =
          "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"
        document.getElementsByTagName("head")[0].appendChild(script)
      })()
    </script>
  </body>
</html>
