<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="generator" content="pandoc" />
    <meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

    <meta
      name="author"
      content="Brendon Smith br3ndonland Source code on GitHub license"
    />

    <title>R guide</title>

    <script src="site_libs/header-attrs-2.3/header-attrs.js"></script>
    <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
    <script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
    <script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
    <script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
    <script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
    <link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
    <script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
    <script src="site_libs/navigation-1.1/tabsets.js"></script>
    <script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>

    <style type="text/css">
      code {
        white-space: pre;
      }
    </style>
    <style type="text/css" data-origin="pandoc">
      pre > code.sourceCode {
        white-space: pre;
        position: relative;
      }
      pre > code.sourceCode > span {
        display: inline-block;
        line-height: 1.25;
      }
      pre > code.sourceCode > span:empty {
        height: 1.2em;
      }
      code.sourceCode > span {
        color: inherit;
        text-decoration: inherit;
      }
      div.sourceCode {
        margin: 1em 0;
      }
      pre.sourceCode {
        margin: 0;
      }
      @media screen {
        div.sourceCode {
          overflow: auto;
        }
      }
      @media print {
        pre > code.sourceCode {
          white-space: pre-wrap;
        }
        pre > code.sourceCode > span {
          text-indent: -5em;
          padding-left: 5em;
        }
      }
      pre.numberSource code {
        counter-reset: source-line 0;
      }
      pre.numberSource code > span {
        position: relative;
        left: -4em;
        counter-increment: source-line;
      }
      pre.numberSource code > span > a:first-child::before {
        content: counter(source-line);
        position: relative;
        left: -1em;
        text-align: right;
        vertical-align: baseline;
        border: none;
        display: inline-block;
        -webkit-touch-callout: none;
        -webkit-user-select: none;
        -khtml-user-select: none;
        -moz-user-select: none;
        -ms-user-select: none;
        user-select: none;
        padding: 0 4px;
        width: 4em;
        color: #aaaaaa;
      }
      pre.numberSource {
        margin-left: 3em;
        border-left: 1px solid #aaaaaa;
        padding-left: 4px;
      }
      div.sourceCode {
      }
      @media screen {
        pre > code.sourceCode > span > a:first-child::before {
          text-decoration: underline;
        }
      }
      code span.al {
        color: #ff0000;
        font-weight: bold;
      } /* Alert */
      code span.an {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Annotation */
      code span.at {
        color: #7d9029;
      } /* Attribute */
      code span.bn {
        color: #40a070;
      } /* BaseN */
      code span.bu {
      } /* BuiltIn */
      code span.cf {
        color: #007020;
        font-weight: bold;
      } /* ControlFlow */
      code span.ch {
        color: #4070a0;
      } /* Char */
      code span.cn {
        color: #880000;
      } /* Constant */
      code span.co {
        color: #60a0b0;
        font-style: italic;
      } /* Comment */
      code span.cv {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* CommentVar */
      code span.do {
        color: #ba2121;
        font-style: italic;
      } /* Documentation */
      code span.dt {
        color: #902000;
      } /* DataType */
      code span.dv {
        color: #40a070;
      } /* DecVal */
      code span.er {
        color: #ff0000;
        font-weight: bold;
      } /* Error */
      code span.ex {
      } /* Extension */
      code span.fl {
        color: #40a070;
      } /* Float */
      code span.fu {
        color: #06287e;
      } /* Function */
      code span.im {
      } /* Import */
      code span.in {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Information */
      code span.kw {
        color: #007020;
        font-weight: bold;
      } /* Keyword */
      code span.op {
        color: #666666;
      } /* Operator */
      code span.ot {
        color: #007020;
      } /* Other */
      code span.pp {
        color: #bc7a00;
      } /* Preprocessor */
      code span.sc {
        color: #4070a0;
      } /* SpecialChar */
      code span.ss {
        color: #bb6688;
      } /* SpecialString */
      code span.st {
        color: #4070a0;
      } /* String */
      code span.va {
        color: #19177c;
      } /* Variable */
      code span.vs {
        color: #4070a0;
      } /* VerbatimString */
      code span.wa {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Warning */
    </style>
    <script>
      // apply pandoc div.sourceCode style to pre.sourceCode instead
      ;(function () {
        var sheets = document.styleSheets
        for (var i = 0; i < sheets.length; i++) {
          if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue
          try {
            var rules = sheets[i].cssRules
          } catch (e) {
            continue
          }
          for (var j = 0; j < rules.length; j++) {
            var rule = rules[j]
            // check if there is a div.sourceCode rule
            if (
              rule.type !== rule.STYLE_RULE ||
              rule.selectorText !== "div.sourceCode"
            )
              continue
            var style = rule.style.cssText
            // check if color or background-color is set
            if (rule.style.color === "" && rule.style.backgroundColor === "")
              continue
            // replace div.sourceCode by a pre.sourceCode rule
            sheets[i].deleteRule(j)
            sheets[i].insertRule("pre.sourceCode{" + style + "}", j)
          }
        }
      })()
    </script>
    <style type="text/css">
      pre:not([class]) {
        background-color: white;
      }
    </style>

    <style type="text/css">
      h1 {
        font-size: 34px;
      }
      h1.title {
        font-size: 38px;
      }
      h2 {
        font-size: 30px;
      }
      h3 {
        font-size: 24px;
      }
      h4 {
        font-size: 18px;
      }
      h5 {
        font-size: 16px;
      }
      h6 {
        font-size: 12px;
      }
      .table th:not([align]) {
        text-align: left;
      }
    </style>

    <style type="text/css">
      .main-container {
        max-width: 940px;
        margin-left: auto;
        margin-right: auto;
      }
      code {
        color: inherit;
        background-color: rgba(0, 0, 0, 0.04);
      }
      img {
        max-width: 100%;
      }
      .tabbed-pane {
        padding-top: 12px;
      }
      .html-widget {
        margin-bottom: 20px;
      }
      button.code-folding-btn:focus {
        outline: none;
      }
      summary {
        display: list-item;
      }
    </style>

    <style type="text/css">
      /* padding for bootstrap navbar */
      body {
        padding-top: 51px;
        padding-bottom: 40px;
      }
      /* offset scroll position for anchor links (for fixed navbar)  */
      .section h1 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h2 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h3 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h4 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h5 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .section h6 {
        padding-top: 56px;
        margin-top: -56px;
      }
      .dropdown-submenu {
        position: relative;
      }
      .dropdown-submenu > .dropdown-menu {
        top: 0;
        left: 100%;
        margin-top: -6px;
        margin-left: -1px;
        border-radius: 0 6px 6px 6px;
      }
      .dropdown-submenu:hover > .dropdown-menu {
        display: block;
      }
      .dropdown-submenu > a:after {
        display: block;
        content: " ";
        float: right;
        width: 0;
        height: 0;
        border-color: transparent;
        border-style: solid;
        border-width: 5px 0 5px 5px;
        border-left-color: #cccccc;
        margin-top: 5px;
        margin-right: -10px;
      }
      .dropdown-submenu:hover > a:after {
        border-left-color: #ffffff;
      }
      .dropdown-submenu.pull-left {
        float: none;
      }
      .dropdown-submenu.pull-left > .dropdown-menu {
        left: -100%;
        margin-left: 10px;
        border-radius: 6px 0 6px 6px;
      }
    </style>

    <script>
      // manage active state of menu based on current page
      $(document).ready(function () {
        // active menu anchor
        href = window.location.pathname
        href = href.substr(href.lastIndexOf("/") + 1)
        if (href === "") href = "index.html"
        var menuAnchor = $('a[href="' + href + '"]')

        // mark it active
        menuAnchor.parent().addClass("active")

        // if it's got a parent navbar menu mark it active as well
        menuAnchor.closest("li.dropdown").addClass("active")
      })
    </script>

    <!-- tabsets -->

    <style type="text/css">
      .tabset-dropdown > .nav-tabs {
        display: inline-table;
        max-height: 500px;
        min-height: 44px;
        overflow-y: auto;
        background: white;
        border: 1px solid #ddd;
        border-radius: 4px;
      }

      .tabset-dropdown > .nav-tabs > li.active:before {
        content: "";
        font-family: "Glyphicons Halflings";
        display: inline-block;
        padding: 10px;
        border-right: 1px solid #ddd;
      }

      .tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
        content: "&#xe258;";
        border: none;
      }

      .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
        content: "";
        font-family: "Glyphicons Halflings";
        display: inline-block;
        padding: 10px;
        border-right: 1px solid #ddd;
      }

      .tabset-dropdown > .nav-tabs > li.active {
        display: block;
      }

      .tabset-dropdown > .nav-tabs > li > a,
      .tabset-dropdown > .nav-tabs > li > a:focus,
      .tabset-dropdown > .nav-tabs > li > a:hover {
        border: none;
        display: inline-block;
        border-radius: 4px;
        background-color: transparent;
      }

      .tabset-dropdown > .nav-tabs.nav-tabs-open > li {
        display: block;
        float: none;
      }

      .tabset-dropdown > .nav-tabs > li {
        display: none;
      }
    </style>

    <!-- code folding -->

    <style type="text/css">
      #TOC {
        margin: 25px 0px 20px 0px;
      }
      @media (max-width: 768px) {
        #TOC {
          position: relative;
          width: 100%;
        }
      }

      @media print {
        .toc-content {
          /* see https://github.com/w3c/csswg-drafts/issues/4434 */
          float: right;
        }
      }

      .toc-content {
        padding-left: 30px;
        padding-right: 40px;
      }

      div.main-container {
        max-width: 1200px;
      }

      div.tocify {
        width: 20%;
        max-width: 260px;
        max-height: 85%;
      }

      @media (min-width: 768px) and (max-width: 991px) {
        div.tocify {
          width: 25%;
        }
      }

      @media (max-width: 767px) {
        div.tocify {
          width: 100%;
          max-width: none;
        }
      }

      .tocify ul,
      .tocify li {
        line-height: 20px;
      }

      .tocify-subheader .tocify-item {
        font-size: 0.9em;
      }

      .tocify .list-group-item {
        border-radius: 0px;
      }
    </style>
  </head>

  <body>
    <div class="container-fluid main-container">
      <!-- setup 3col/9col grid for toc_float and main content  -->
      <div class="row-fluid">
        <div class="col-xs-12 col-sm-4 col-md-3">
          <div id="TOC" class="tocify"></div>
        </div>

        <div class="toc-content col-xs-12 col-sm-8 col-md-9">
          <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
            <div class="container">
              <div class="navbar-header">
                <button
                  type="button"
                  class="navbar-toggle collapsed"
                  data-toggle="collapse"
                  data-target="#navbar"
                >
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="index.html">R guide</a>
              </div>
              <div id="navbar" class="navbar-collapse collapse">
                <ul class="nav navbar-nav">
                  <li>
                    <a href="/R-guide/">Guide</a>
                  </li>
                  <li>
                    <a href="/R-guide/code">Sample code</a>
                  </li>
                  <li>
                    <a href="https://github.com/br3ndonland/R-guide">GitHub</a>
                  </li>
                </ul>
                <ul class="nav navbar-nav navbar-right"></ul>
              </div>
              <!--/.nav-collapse -->
            </div>
            <!--/.container -->
          </div>
          <!--/.navbar -->

          <div class="fluid-row" id="header">
            <h1 class="title toc-ignore">R guide</h1>
            <h3 class="subtitle">
              A quick reference guide and sample code for statistical
              programming in R
            </h3>
            <h4 class="author">
              Brendon Smith<br /><a href="https://github.com/br3ndonland"
                >br3ndonland</a
              ><br />Source code on
              <a href="https://github.com/br3ndonland/R-guide">GitHub</a
              ><br /><a href="https://choosealicense.com/"
                ><img
                  src="https://img.shields.io/badge/license-MIT-blue.svg?longCache=true&amp;style=for-the-badge"
                  alt="license"
              /></a>
            </h4>
          </div>

          <div id="intro" class="section level2">
            <h2>Intro</h2>
            <p>
              <strong
                >R is an interpreted computer language used primarily for
                statistical analysis.</strong
              >
            </p>
            <p>
              From the <a href="http://cran.rstudio.com/faqs.html">R FAQ</a>:
            </p>
            <blockquote>
              <p>
                R is a system for statistical computation and graphics. It
                consists of a language plus a run-time environment with
                graphics, a debugger, access to certain system functions, and
                the ability to run programs stored in script files.
              </p>
              <p>
                The design of R has been heavily influenced by two existing
                languages: Becker, Chambers &amp; Wilks’ S (see
                <a
                  href="http://cran.rstudio.com/doc/FAQ/R-FAQ.html#What-is-S_003f"
                  >What is S?</a
                >) and Sussman’s
                <a href="http://www.cs.indiana.edu/scheme-repository/home.html"
                  >Scheme</a
                >. Whereas the resulting language is very similar in appearance
                to S, the underlying implementation and semantics are derived
                from Scheme. See
                <a
                  href="http://cran.rstudio.com/doc/FAQ/R-FAQ.html#What-are-the-differences-between-R-and-S_003f"
                  >What are the differences between R and S?</a
                >, for further details.
              </p>
              <p>
                The core of R is an <em>interpreted computer language</em> which
                allows branching and looping as well as modular programming
                using functions. Most of the user-visible functions in R are
                written in R. It is possible for the user to interface to
                procedures written in the C, C++, or FORTRAN languages for
                efficiency. The R distribution contains functionality for a
                large number of statistical procedures. Among these are: linear
                and generalized linear models, nonlinear regression models, time
                series analysis, classical parametric and nonparametric tests,
                clustering and smoothing. There is also a large set of functions
                which provide a flexible graphical environment for creating
                various kinds of data presentations. Additional modules (‘add-on
                packages’) are available for a variety of specific purposes (see
                <a
                  href="http://cran.rstudio.com/doc/FAQ/R-FAQ.html#R-Add_002dOn-Packages"
                  >R Add-On Packages</a
                >).
              </p>
              <p>
                The name is partly based on the (first) names of the first two R
                authors (Robert Gentleman and Ross Ihaka), and partly a play on
                the name of the Bell Labs language ‘S’, which stood for System.’
              </p>
            </blockquote>
          </div>
          <div id="general-r-resources" class="section level2">
            <h2>General R resources</h2>
            <ul>
              <li>
                Read the <a href="https://cran.rstudio.com/faqs.html">R FAQ</a>.
                <em>GNU General Public License.</em>
              </li>
              <li>
                <a href="https://www.rdocumentation.org/">R documentation</a>
                provides a comprehensive listing of R packages that are
                available in CRAN, BioConductor and GitHub.
              </li>
              <li>
                <strong
                  ><a
                    href="https://www.manning.com/books/r-in-action-second-edition"
                    >R in Action</a
                  ></strong
                >
                by Robert Kabacoff is an excellent reference guide. If I had to
                recommend one single resource, I would direct you here. Second
                edition
                <em
                  >copyright 2015 by Manning Publications Co. All rights
                  reserved.</em
                >
              </li>
              <li>
                <a href="http://www.statmethods.net/">Quick-R</a> by Robert
                Kabacoff. Great quick reference.
                <em>Copyright Robert I. Kabacoff, PhD.</em>
              </li>
              <li>
                <a href="http://www.cookbook-r.com/">Cookbook for R</a> by
                Winston Chang.
                <em>CC BY-SA 3.0 license (freely shared and adaptable)</em>
              </li>
              <li>
                <a
                  href="http://proquest.safaribooksonline.com/book/programming/r/9781449363086"
                  >R Graphics Cookbook</a
                >
                by Winston Chang (mostly for ggplot2).
                <em
                  >CC0 1.0 license (public domain dedication, no copyright).</em
                >
              </li>
              <li>
                <a
                  href="https://www.amazon.com/R-Book-Michael-J-Crawley/dp/0470973927"
                  >The R book</a
                >
                by Crawley
              </li>
              <li>
                <a
                  href="https://cran.r-project.org/doc/contrib/Verzani-SimpleR.pdf"
                  >simpleR PDF by Verzani</a
                >.
                <em>Copyright John Verzani, 2001-2. All rights reserved.</em>
                This book is frequently referenced, but outdated (‘Appendix:
                Teaching Tricks’ recommends distributing R code to students on
                ‘floppies.’)
              </li>
              <li>
                <a
                  href="https://www.amazon.com/Linear-Models-Chapman-Statistical-Science/dp/1439887330/ref=dp_ob_title_bk"
                  >Linear models in R</a
                >
                by Faraway.
                <em>Copyright 2014 by Taylor &amp; Francis Group, LLC.</em>
              </li>
            </ul>
            <p><a href="#top">back to top</a></p>
          </div>
          <div id="setup" class="section level2">
            <h2>Setup</h2>
            <div id="installation" class="section level3">
              <h3>Installation</h3>
              <ul>
                <li>
                  <p>
                    <strong
                      >I highly recommend using a package manager to install R
                      and RStudio.</strong
                    >
                    Otherwise, <a href="https://cran.rstudio.com/">R</a> and
                    <a href="https://www.rstudio.com">RStudio</a> must be
                    downloaded and installed separately. R cannot be updated
                    from within RStudio, and installing a new R version can
                    require reinstallation of packages. It’s difficult to
                    manage.
                  </p>
                </li>
                <li>
                  <p>
                    Install <a href="https://brew.sh/">Homebrew</a>. Homebrew
                    includes
                    <a href="https://caskroom.github.io/">Homebrew-Cask</a> to
                    manage other macOS applications.
                  </p>
                  <ul>
                    <li>
                      <p>
                        I previously used
                        <a href="https://www.anaconda.com/">Anaconda</a> to
                        manage my Python and R distributions, and now use
                        Homebrew. I switched because Anaconda is a very large
                        installation, and not as flexible or general as
                        Homebrew. Conda and brew are sort of like competing
                        package managers.
                      </p>
                    </li>
                    <li>
                      <details>
                        <p>
                          <summary>
                            <em>Click here for Anaconda info</em>
                          </summary>
                        </p>
                        <h4 id="anaconda">Anaconda</h4>
                        <ul>
                          <li>
                            I install Anaconda by
                            <a href="https://www.anaconda.com/download/"
                              >direct download</a
                            >. I have also tried installing via Homebrew Cask,
                            but didn’t get the correct command prompt or path.
                          </li>
                          <li>
                            Anaconda contains the
                            <a href="https://conda.io/docs/">Conda</a> and
                            <a href="https://pypi.python.org/pypi">PyPI</a>
                            (pip) package managers. Packages can be installed
                            from <code>pip</code>, and will integrate into the
                            Anaconda environment.
                          </li>
                          <li>
                            <a href="https://conda.io/docs/commands"
                              >Conda commands</a
                            >:
                            <ul>
                              <li>
                                Check installed packages with
                                <code>$ conda list</code> and
                                <code>$ pip list</code>.
                              </li>
                              <li>
                                Find packages with <code>$ conda search</code>.
                              </li>
                              <li>
                                Update conda with
                                <code
                                  >$ conda update --prefix &lt;PATH&gt;
                                  anaconda</code
                                >
                              </li>
                              <li>
                                Update packages with
                                <code>$ conda update --all</code>
                              </li>
                            </ul>
                          </li>
                        </ul>
                        <h5 id="additions-to-the-standard-anaconda-install">
                          Additions to the standard Anaconda install
                        </h5>
                        <ul>
                          <li>
                            R and RStudio
                            <ul>
                              <li>
                                <p>
                                  When I was working with R, I preferred to
                                  install R and RStudio via Anaconda for easier
                                  version and package management. Otherwise,
                                  <a href="https://cran.rstudio.com/">R</a> and
                                  <a href="https://www.rstudio.com">RStudio</a>
                                  must be downloaded and installed separately. R
                                  cannot be updated from within RStudio, and
                                  installing a new R version can require
                                  reinstallation of packages. It’s just too
                                  difficult to manage.
                                </p>
                                <div class="sourceCode" id="cb1">
                                  <pre
                                    class="sourceCode sh"
                                  ><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a>$ <span class="ex">conda</span> install -c r r-essentials</span>
<span id="cb1-2"><a href="#cb1-2"></a>$ <span class="ex">conda</span> install rstudio</span></code></pre>
                                </div>
                              </li>
                            </ul>
                          </li>
                        </ul>
                      </details>
                    </li>
                  </ul>
                </li>
                <li>
                  <p>Install R and RStudio via Homebrew</p>
                  <div class="sourceCode" id="cb2">
                    <pre
                      class="sourceCode sh"
                    ><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">brew</span> install r</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="ex">brew</span> cask install r-app</span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="ex">brew</span> cask install rstudio</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    Update Homebrew regularly with <code>brew update</code> and
                    <code>brew upgrade</code>
                  </p>
                </li>
                <li>
                  <p>
                    Check health of installation with <code>brew doctor</code>
                  </p>
                </li>
                <li>
                  <p>
                    See <a href="https://docs.brew.sh">docs</a> for further
                    info.
                  </p>
                </li>
              </ul>
            </div>
            <div id="my-rstudio-setup" class="section level3">
              <h3>My RStudio setup</h3>
              <ul>
                <li>
                  Preferences
                  <ul>
                    <li>
                      Appearance
                      <ul>
                        <li>Modern RStudio theme</li>
                        <li>Material editor theme</li>
                        <li><a href="https://dank.sh">Dank Mono</a> font</li>
                      </ul>
                    </li>
                    <li>
                      Code
                      <ul>
                        <li>
                          Editing
                          <ul>
                            <li>Insert spaces for tab</li>
                            <li>Tab width: 2</li>
                          </ul>
                        </li>
                        <li>
                          Saving
                          <ul>
                            <li>
                              Strip trailing horizontal whitespace when saving
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li>
                      R Markdown
                      <ul>
                        <li>Show document outline by default</li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li>
                  Information on other RStudio options can be found in the
                  <a
                    href="https://support.rstudio.com/hc/en-us/articles/200549016-Customizing-RStudio"
                    >RStudio documentation</a
                  >.
                </li>
              </ul>
              <p><a href="#top">back to top</a></p>
            </div>
          </div>
          <div id="working-with-r-files" class="section level2">
            <h2>Working with R files</h2>
            <p>
              I highly recommend working with a notebook file format. Jupyter
              Notebook and R Markdown are the most popular formats.
            </p>
            <div id="jupyter-notebook" class="section level3">
              <h3>Jupyter Notebook</h3>
              <ul>
                <li>
                  Jupyter Notebook is a file format most widely used for
                  scientific computing. It supports many programming languages,
                  including Python and R. Jupyter Notebook files enable creation
                  of reproducible computational narratives containing Markdown
                  text interspersed with functional code chunks.
                </li>
                <li>
                  Jupyter Notebook files can be run with
                  <a href="http://jupyterlab.readthedocs.io/en/latest/"
                    >JupyterLab</a
                  >.
                  <ul>
                    <li>
                      JupyterLab is a development environment produced by
                      <a href="http://jupyter.org/">Project Jupyter</a>.
                    </li>
                    <li>
                      JupyterLab can run Jupyter Notebooks, R Markdown,
                      Markdown, and many other file formats.
                    </li>
                    <li>
                      Install Jupyter with Homebrew or the Python
                      <code>pip</code> module manager. I previously used
                      <a href="https://www.anaconda.com/">Anaconda</a> to manage
                      my Python and R distributions, and now use Homebrew. I
                      switched because Anaconda is a very large installation,
                      and not as flexible or general as Homebrew.
                    </li>
                    <li>
                      Launch from the command line with
                      <code>jupyter notebook</code> or
                    </li>
                  </ul>
                </li>
                <li>
                  Google provides a cloud-based Jupyter Notebook environment
                  called
                  <a href="https://colab.research.google.com">Colaboratory</a>.
                </li>
                <li>
                  See my
                  <a href="https://github.com/br3ndonland/R-proteomics-Nrf1"
                    >R-proteomics-Nrf1 repo</a
                  >
                  for an example of notebook formats.
                </li>
              </ul>
            </div>
            <div id="r-markdown" class="section level3">
              <h3>R Markdown</h3>
              <ul>
                <li>
                  <strong
                    >R Markdown is a document creation package based on
                    Markdown, a syntax for easy generation of HTML files, and
                    Pandoc, a document output and conversion system.</strong
                  >
                </li>
                <li>
                  Markdown resources
                  <ul>
                    <li>
                      I have provided a guide to Markdown syntax on
                      <a
                        href="https://github.com/br3ndonland/udacity-google/blob/master/markdown-guide.md"
                        >GitHub</a
                      >.
                    </li>
                    <li>
                      <a href="https://www.markdownguide.org/">MarkdownGuide</a>
                    </li>
                    <li>
                      <a
                        href="https://guides.github.com/features/mastering-markdown/"
                        >GitHub-Flavored Markdown</a
                      >
                    </li>
                  </ul>
                </li>
                <li>
                  R Markdown resources
                  <ul>
                    <li>
                      <a href="http://rmarkdown.rstudio.com/"
                        >RStudio R Markdown page</a
                      >
                    </li>
                    <li>
                      <a href="https://bookdown.org/yihui/rmarkdown/"
                        >R Markdown: The Definitive Guide by Yihui Xie</a
                      >
                    </li>
                  </ul>
                </li>
                <li>
                  Pandoc resources
                  <ul>
                    <li>
                      <a href="https://pandoc.org/MANUAL.html"
                        >Pandoc user’s guide</a
                      >
                    </li>
                  </ul>
                </li>
                <li>
                  To create an R Markdown document in RStudio, File -&gt; New
                  File -&gt; R Markdown.
                </li>
                <li>
                  An RMarkdown file contains three types of data:
                  <ul>
                    <li>
                      YAML front matter header at the top of the file to specify
                      output methods
                    </li>
                    <li>Markdown-formatted text</li>
                    <li>Functional code chunks.</li>
                  </ul>
                </li>
                <li>
                  When you click the <strong>Knit</strong> button the code will
                  run and the output will be created.
                </li>
                <li>
                  R Markdown has started offering a Notebook output format that
                  is similar to the Jupyter Notebook. See the
                  <a href="#output-formats">output formats section</a>.
                </li>
              </ul>
              <div id="yaml-front-matter-header" class="section level4">
                <h4>YAML front matter header</h4>
                <ul>
                  <li>
                    The header specifies a document title and subtitle, table of
                    contents, and output options.
                  </li>
                </ul>
              </div>
              <div id="functional-code-chunks" class="section level4">
                <h4>Functional code chunks</h4>
                <ul>
                  <li>
                    In addition to the standard Markdown inline code chunks and
                    Markdown fenced code chunks, RMarkdown includes functional
                    language-specific code chunks that allow code to be run from
                    within the Markdown document.
                  </li>
                  <li>
                    RMarkdown code chunks are inserted with a modified Markdown
                    code fence format:
                    <ul>
                      <li>
                        Normal Markdown code chunks are enclosed within triple
                        backticks, with the language name after the first set of
                        backticks.
                      </li>
                      <li>
                        Functional R code is specified with
                        <code>{r}</code> after the first set of backticks. R
                        will run this code when knitting the RMarkdown document.
                        If you want to include a standard Markdown code chunk
                        without running the code in R, leave out the curly
                        braces.
                      </li>
                      <li>
                        Conclude the code chunk with another set of triple
                        backticks.
                      </li>
                    </ul>
                  </li>
                  <li>
                    See the
                    <a href="http://rmarkdown.rstudio.com/lesson-3.html"
                      >R Markdown code chunks page</a
                    >
                    for more info.
                  </li>
                </ul>
                <p>Example:</p>
                <div class="sourceCode" id="cb3">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">summary</span>(cars)</span></code></pre>
                </div>
                <pre><code>##      speed           dist
##  Min.   : 4.0   Min.   :  2.00
##  1st Qu.:12.0   1st Qu.: 26.00
##  Median :15.0   Median : 36.00
##  Mean   :15.4   Mean   : 42.98
##  3rd Qu.:19.0   3rd Qu.: 56.00
##  Max.   :25.0   Max.   :120.00</code></pre>
                <ul>
                  <li>
                    <p>
                      Change size of figures (at beginning of r code chunk in
                      brackets):
                    </p>
                    <pre
                      class="text"
                    ><code>{r, out.width = &#39;1000px&#39;, out.height = &#39;2000px&#39;}</code></pre>
                  </li>
                  <li>
                    <p>
                      Prevent printing of code in HTML output:
                      <code>echo=FALSE</code>
                    </p>
                    <p>
                      <img
                        src="index_files/figure-html/index-rmarkdown-intro-pressure-plot-1.png"
                        width="672"
                      />
                    </p>
                  </li>
                </ul>
              </div>
              <div id="output-formats" class="section level4">
                <h4>Output formats</h4>
                <ul>
                  <li>
                    There are
                    <a href="http://rmarkdown.rstudio.com/formats.html"
                      >many formats</a
                    >
                    available. RMarkdown documents can easily be interconverted
                    among the formats.
                  </li>
                  <li>Only one output format can be specified at a time.</li>
                  <li>
                    When a report is knit, it will appear in the RStudio viewer,
                    and will automatically be saved to the same directory as the
                    .rmd file.
                  </li>
                  <li>
                    The
                    <a
                      href="https://rmarkdown.rstudio.com/github_document_format.html"
                      ><code>github_document</code> output format</a
                    >
                    outputs a GitHub-friendly Markdown file. It is
                    RStudio-specific, not a Pandoc standard, and provides some
                    RStudio-specific benefits like local HTML preview.
                    <ul>
                      <li>
                        <p>
                          At the time of this writing, the markdown output
                          formats (either <code>github_document</code> or
                          <code>md_document: variant: gfm</code>) don’t seem to
                          be outputting a table of contents, even when
                          <code>toc: true</code> is included. The solution is to
                          include a separate header file containing the header
                          and TOC.
                        </p>
                        <div class="sourceCode" id="cb6">
                          <pre
                            class="sourceCode yaml"
                          ><code class="sourceCode yaml"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">output</span><span class="kw">:</span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="at">  </span><span class="fu">github_document</span><span class="kw">:</span></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="at">    </span><span class="fu">includes</span><span class="kw">:</span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="at">      </span><span class="fu">in_header</span><span class="kw">:</span><span class="at"> R-guide-header.md</span></span></code></pre>
                        </div>
                      </li>
                    </ul>
                  </li>
                  <li>
                    The
                    <a href="https://bookdown.org/yihui/rmarkdown/notebook.html"
                      >R Notebook</a
                    >
                    format is similar to a Jupyter Notebook. Code chunks can be
                    executed independently and interactively, rather than having
                    to knit the entire workbook. I haven’t found it to be
                    useful, and would prefer to use a Jupyter Notebook.
                  </li>
                  <li>
                    R Markdown can build
                    <a href="https://bookdown.org/yihui/rmarkdown/websites.html"
                      >websites</a
                    >.
                    <ul>
                      <li>
                        This website was generated with the simple
                        <a
                          href="https://bookdown.org/yihui/rmarkdown/rmarkdown-site.html"
                          >R Markdown static site builder</a
                        >.
                      </li>
                      <li>
                        The <code>blogdown</code> package is used to build the
                        site within RStudio.
                      </li>
                      <li>
                        The <a href="_site.yml">_site.yml</a> file specifies the
                        website build parameters. The website builds into the
                        <em>docs</em> directory for compatibility with
                        <a
                          href="https://bookdown.org/yihui/rmarkdown/rmarkdown-site.html"
                          >GitHub Pages</a
                        >.
                      </li>
                      <li>
                        The static site builder uses
                        <a href="https://getbootstrap.com/">Bootstrap</a>, and
                        themes come from
                        <a href="https://bootswatch.com/3/">Bootswatch</a>. For
                        more info on themes and options, see the
                        <a
                          href="https://bookdown.org/yihui/rmarkdown/html-document.html"
                          >R Markdown HTML document docs</a
                        >
                        or run <code>?rmarkdown::html_document</code>.
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
            <div id="rstudio-projects" class="section level3">
              <h3>RStudio Projects</h3>
              <p>
                <a
                  href="https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects"
                  >RStudio Projects</a
                >
                are the best way to manage code in RStudio. RStudio projects
                make it straightforward to divide your work into multiple
                contexts, each with their own working directory, workspace,
                history, and source documents. Projects allow use of version
                control with Git. Installing packages into the project locally
                with <code>renv</code> is very helpful.
              </p>
            </div>
            <div id="r-packages" class="section level3">
              <h3>R packages</h3>
              <ul>
                <li>
                  Packages are programs written for R that provide additional
                  functionality.
                  <strong>R is dependent on large numbers of packages.</strong>
                </li>
                <li>
                  <a href="https://rstudio.github.io/renv/index.html"
                    ><code>renv</code></a
                  >
                  is a package dependency management system for R projects. It
                  helps avoid problems caused by different package versions and
                  installations by giving each project its own isolated package
                  library. <code>renv</code> is separate from general package
                  managers like Homebrew used to install R and RStudio.
                </li>
                <li>
                  This project previously used
                  <a href="https://github.com/rstudio/packrat">Packrat</a>, a
                  predecessor to <code>renv</code> that is now
                  “soft-deprecated.”
                  <ul>
                    <li>
                      Migration from Packrat to <code>renv</code> is simple. Run
                      the following command in the console:
                      <code>renv::migrate("~/path/to/repo")</code>.
                    </li>
                    <li>
                      Then if packages aren’t already installed, run
                      <code>renv::restore()</code>, or in the Packages pane,
                      navigate to <code>renv</code> -&gt; Restore Library.
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
            <div id="git-and-github" class="section level3">
              <h3>Git and GitHub</h3>
              <ul>
                <li>
                  Version control is highly recommended for code of all kinds.
                  Git is a free and widely used version control system.
                </li>
                <li>
                  RStudio provides a
                  <a
                    href="https://support.rstudio.com/hc/en-us/articles/200532077-Version-Control-with-Git-and-SVN"
                    >version control guide</a
                  >
                  for linking with Git.
                </li>
                <li>
                  <strong>Git has a substantial learning curve.</strong> RStudio
                  and GitHub have helpful documentation for getting started.
                </li>
                <li>
                  The directory where the files are located must be
                  <a
                    href="https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects"
                    >formatted as an R Project</a
                  >.
                </li>
              </ul>
            </div>
            <div id="coding-practice" class="section level3">
              <h3>Coding practice</h3>
              <p><strong>Hands-on coding practice is essential.</strong></p>
              <ul>
                <li>
                  Read
                  <a href="http://cran.rstudio.com/manuals.html"
                    >An Introduction to R</a
                  >
                  and complete the sample session in Appendix A.
                </li>
                <li>
                  <a href="https://www.datacamp.com/courses/introduction-to-r"
                    >DataCamp</a
                  >
                  is extremely helpful. I took the ‘Introduction to R,’ ‘Data
                  Analysis and Statistical Inference,’ and ‘A hands-on
                  introduction to statistics with R track-introductory’ courses.
                </li>
              </ul>
              <p><a href="#top">back to top</a></p>
            </div>
          </div>
          <div id="running-code" class="section level2">
            <h2>Running code</h2>
            <ul>
              <li>
                See the
                <a
                  href="https://support.rstudio.com/hc/en-us/articles/200484448-Editing-and-Executing-Code"
                  >RStudio documentation on editing and executing code</a
                >
              </li>
              <li>
                The basic code entry window is called the
                <strong>console</strong>. It is oriented towards entering and
                running code one line at a time, and does not allow you to save
                the code.
              </li>
              <li>
                In the RStudio source editor, to execute a single line of source
                code where the cursor currently resides use
                <code>Cmd+Enter</code> (or use the <strong>Run</strong> toolbar
                button).
              </li>
              <li>
                There are three ways to execute multiple lines from within the
                editor:
                <ol style="list-style-type: decimal">
                  <li>
                    Select the lines and press <code>Cmd+Enter</code> (or use
                    the <strong>Run</strong> toolbar button)
                  </li>
                  <li>
                    After executing a selection of code, use the
                    <strong>Re-Run Previous Region</strong> command (or its
                    associated toolbar button) to run the same selection again.
                    Note that changes to the selection including additional,
                    removal, and modification of lines will be reflected in this
                    subsequent run of the selection.
                  </li>
                  <li>
                    To run the entire document press
                    <code>Cmd+option+R</code> (or use the
                    <strong>Source</strong> toolbar button). Using the
                    <strong>Knit</strong> command in R Markdown documents will
                    run the entire document.
                  </li>
                </ol>
              </li>
              <li>
                The difference between running lines from a selection and
                invoking <strong>Source</strong> is that when running a
                selection all lines are inserted directly into the console
                whereas for <strong>Source</strong> the file is saved to a
                temporary location and then sourced into the console from there
                (thereby creating less clutter in the console).
              </li>
              <li>
                <strong
                  >Source on save: Enabling source on save runs the code every
                  time you save.</strong
                >
                When editing re-usable functions (as opposed to freestanding
                lines of R) you may wish to set the Source on Save option for
                the document (available on the toolbar next to the Save icon).
                Enabling this option will cause the file to automatically be
                sourced into the global environment every time it is saved.
                Setting Source on Save ensures that the copy of a function
                within the global environment is always in sync with its source,
                and also provides a good way to arrange for frequent syntax
                checking as you develop a function.
              </li>
              <li>
                The <strong>workspace</strong> is your current R working
                environment and includes imported datasets, vectors, data
                frames, and other objects.
              </li>
              <li>
                Workspace info persists after restarting R. To clear: In
                RStudio, Session -&gt; Clear Workspace will reset the workspace
                (shown in the environment tab in RStudio). Click the broom icon
                in the environment, history and plots tabs to clear
                individually.
              </li>
            </ul>
            <p><a href="#top">back to top</a></p>
          </div>
          <div id="basic-syntax" class="section level2">
            <h2>Basic syntax</h2>
            <ul>
              <li>
                R can be used as a calculator if needed: <code>5+5</code>,
                <code>5-5</code>, <code>(5+5)/2</code>, etc)
              </li>
              <li>
                Use <code>&lt;-</code> or <code>=</code> to define a
                <strong>variable</strong> (also called an
                <strong>object</strong> in other languages like Python). For
                example, <code>x &lt;- 42</code> or <code>x = 42</code> set x
                equal to 42.
              </li>
              <li>
                <code>~</code> is similar to <code>=</code> in Python or SAS or
                a normal written equation. It means ‘versus’ or ‘as a function
                of.’
              </li>
              <li>
                Use quotation marks to enter non-numeric values, called
                ‘<strong>strings</strong>.’ Entering
                <code>'forty-two'</code> will treat forty-two as a word. Both
                single and double quotes can be used, but double quotes are
                <a
                  href="http://stat.ethz.ch/R-manual/R-devel/library/base/html/Quotes.html"
                  >preferred</a
                >.
              </li>
              <li>
                Logical Boolean values are entered with <code>TRUE</code> or
                <code>T</code>, and <code>FALSE</code> or <code>F</code>.
              </li>
              <li>
                You can check the data type of a variable by entering
                <code>class(var_name).</code>
              </li>
              <li>
                You can perform mathematical operations on variables, but only
                if they have a numerical value assigned.
              </li>
              <li>
                Print (view data) in R by simply typing the variable name. There
                is no need to type <code>print(variable)</code> like Python or
                PROC PRINT like in SAS.
              </li>
              <li>
                Hit tab when typing a function to call up a list of suggestions
                with explanations.
              </li>
              <li>
                Use <code>apropos()</code> to help identify the proper function
                to use.
              </li>
              <li>
                The help function can be called up by typing
                <code>?</code> followed by the command of interest
              </li>
              <li>
                The arrow keys can be used to scroll through previous commands
                (see p.2 of Verzani simpleR PDF)
              </li>
              <li>
                R is case-sensitive, so true and TRUE are treated differently.
              </li>
              <li>
                <strong>Adding comments or notes to your code:</strong>
                <ul>
                  <li>
                    In RStudio, you can comment and uncomment code using the
                    <strong>Edit -&gt; Comment/Uncomment Lines</strong> menu
                    item (or Cmd+Shift+C).
                  </li>
                  <li>
                    Begin comments with <code>#</code>, similar to how you would
                    use a star to begin a note and a semicolon to conclude it in
                    SAS. A note in R will begin after <code>#</code> and
                    continue until the end of the line (until you insert a line
                    break with enter).
                  </li>
                  <li>
                    You can also use a semicolon instead of a line break to
                    indicate the end of a command.
                  </li>
                </ul>
              </li>
            </ul>
            <p><a href="#top">back to top</a></p>
          </div>
          <div id="data-structures" class="section level2">
            <h2>Data structures</h2>
            <p>
              R vs. SAS from
              <a href="http://www.statmethods.net/input/index.html">Quick-R</a>:
            </p>
            <blockquote>
              <p>
                Unlike SAS, which has DATA and PROC steps, R has data structures
                (vectors, matrices, arrays, data frames) that you can operate on
                through functions that perform statistical analyses and create
                graphs. In this way, R is similar to PROC IML.
              </p>
            </blockquote>
            <p>See R in Action 2.2 for further info.</p>
            <div id="vector" class="section level3">
              <h3>Vector</h3>
              <ul>
                <li>
                  <p>
                    The vector is a way to store data in R. Vectors are created
                    with the combine function <code>c()</code>. Enter values or
                    variable names separated by commas:
                  </p>
                  <div class="sourceCode" id="cb7">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>Vector_name &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">45</span>, <span class="dv">6545</span>, <span class="dv">453</span>, <span class="dv">3453</span>, etc)</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    You can also name the elements of a vector, and create a
                    variable with the vector names for ease of use.
                  </p>
                  <div class="sourceCode" id="cb8">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw">names</span>(vector_name) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;name1&#39;</span>, <span class="st">&#39;name2&#39;</span>)</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    Select parts of a vector with brackets. For example, to
                    create a new vector with a specific portion of another
                    vector, type
                  </p>
                  <div class="sourceCode" id="cb9">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>new_vector_name &lt;-<span class="st"> </span>old_vector_name[vector_element]</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    R stores vectors as strings of data. To put the data in
                    table format, use a <strong>data frame</strong> (see below).
                  </p>
                </li>
              </ul>
            </div>
            <div id="matrix" class="section level3">
              <h3>Matrix</h3>
              <p>From DataCamp introduction to R course:</p>
              <blockquote>
                <p>
                  In R, a matrix is a collection of elements of the same data
                  type (numeric, character, or logical) arranged into a fixed
                  number of rows and columns. Since you are only working with
                  rows and columns, a matrix is called two-dimensional. You can
                  construct a matrix in R with the matrix function, for example:
                </p>
                <div class="sourceCode" id="cb10">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">matrix</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">9</span>, <span class="dt">byrow=</span><span class="ot">TRUE</span>, <span class="dt">nrow=</span><span class="dv">3</span>)</span></code></pre>
                </div>
                <p>
                  The first argument is the collection of elements that R will
                  arrange into the rows and columns of the matrix. Here, we use
                  <code>1:9</code> which constructs the vector
                  <code>c(1,2,3,4,5,6,7,8,9)</code>. The argument
                  <code>byrow</code> indicates that the matrix is filled by the
                  rows. If we want the vector to be filled by the columns, we
                  just place <code>bycol=TRUE</code> or
                  <code>byrow=FALSE</code>. The third argument<code>nrow</code>
                  indicates that the matrix should have three rows.
                </p>
              </blockquote>
              <ul>
                <li>
                  <p>
                    You can use vectors to name rows and columns of a matrix,
                    for example:
                  </p>
                  <div class="sourceCode" id="cb11">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="kw">colnames</span>(star_wars_matrix) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;US&#39;</span>,<span class="st">&#39;non-US&#39;</span>);</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw">rownames</span>(star_wars_matrix) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;A New Hope&#39;</span>,<span class="st">&#39;The Empire Strikes Back&#39;</span>,<span class="st">&#39;Return of the Jedi&#39;</span>)</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    The <code>rowSums(matrix_name)</code> and
                    <code>colSums(matrix_name)</code> functions will sum the
                    matrix (the first S is capital).
                  </p>
                </li>
                <li>
                  <p>
                    Use the <code>cbind()</code> function to merge vectors and
                    matrices.
                  </p>
                </li>
                <li>
                  <p>
                    Similar to vectors, use square brackets <code>[ ]</code> to
                    select elements from a matrix. Whereas vectors have one
                    dimension, matrices have two dimensions. You should
                    therefore use a comma to separate that which you want to
                    select from the rows from that which you want to select from
                    the columns. For example:
                  </p>
                  <ul>
                    <li>
                      <code>my_matrix[1,2]</code> first row, second element.
                    </li>
                    <li>
                      <code>my_matrix[1:3,2:4]</code> rows 1,2,3 and columns
                      2,3,4.
                    </li>
                    <li>
                      If you want to select all elements of a row or a column,
                      no number is needed before or after the comma:
                      <ul>
                        <li>
                          <code>my_matrix[,1]</code> all elements of the first
                          column.
                        </li>
                        <li>
                          <code>my_matrix[1,]</code> all elements of the first
                          row.
                        </li>
                      </ul>
                    </li>
                    <li>
                      Just like <code>2*my_matrix</code> multiplied every
                      element of my_matrix by two,
                      <code>my_matrix1 - my_matrix2</code> creates a matrix
                      where each element is the product of the corresponding
                      elements in <code>my_matrix1</code> and
                      <code>my_matrix2</code>.
                    </li>
                  </ul>
                </li>
                <li>
                  <p>
                    Similar to vectors, the standard operators like `+ - /
                    <em
                      ><code
                        >etc. work in an element-wise way on matrices in R. For
                        example:</code
                      >2</em
                    >my_matrix<code>multiplies each element of</code>my_matrix`
                    by two.
                  </p>
                </li>
              </ul>
            </div>
            <div id="factor" class="section level3">
              <h3>Factor</h3>
              <p>
                <strong
                  >The factor is an R function used to store categorical
                  variables.</strong
                >
              </p>
              <ul>
                <li>
                  <p>
                    To create factors in R, use the function
                    <code>factor()</code>. First, create a vector that contains
                    the observations to be converted to a factor. For example,
                    <code>gender_vector</code> contains the sex of 5 different
                    individuals and <code>participants_1</code> contains
                    discrete numbers representing deidentified study
                    participants:
                  </p>
                  <div class="sourceCode" id="cb12">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>gender_vector =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;Male&#39;</span>,<span class="st">&#39;Female&#39;</span>,<span class="st">&#39;Female&#39;</span>,<span class="st">&#39;Male&#39;</span>,<span class="st">&#39;Male&#39;</span>)</span>
<span id="cb12-2"><a href="#cb12-2"></a>participants_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">7</span>,<span class="dv">11</span>,<span class="dv">13</span>,<span class="dv">17</span>)</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    When you first get a data set, you will often notice that it
                    contains factors with specific factor levels. However,
                    sometimes you will want to change the names of these levels
                    for clarity or other reasons. R allows you to do this with
                    the function <code>levels()</code>:
                  </p>
                  <div class="sourceCode" id="cb13">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw">levels</span>(factor_vector) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;name1&#39;</span>,<span class="st">&#39;name2&#39;</span>,...)</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    To create an <strong>ordered factor</strong>, you have to
                    add two additional arguments:
                  </p>
                  <div class="sourceCode" id="cb14">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="kw">factor</span>(some_vector,<span class="dt">ordered=</span><span class="ot">TRUE</span>,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&#39;Level 1&#39;</span>,<span class="st">&#39;Level 2&#39;</span>...))</span></code></pre>
                  </div>
                  <ul>
                    <li>
                      <p>
                        By setting the argument <code>ordered=TRUE</code> in the
                        function <code>factor()</code>, you indicate that the
                        factor is ordered. With the argument
                        <code>levels</code> you give the values of the factor in
                        the correct order:
                      </p>
                      <div class="sourceCode" id="cb15">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>levels =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;Low&#39;</span>,<span class="st">&#39;Medium&#39;</span>,<span class="st">&#39;High&#39;</span>)</span></code></pre>
                      </div>
                    </li>
                  </ul>
                </li>
              </ul>
              <p>
                Here’s an example within a functional R Markdown code block:
              </p>
              <div class="sourceCode" id="cb16">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co"># Create a vector of temperature observations</span></span>
<span id="cb16-2"><a href="#cb16-2"></a>temperature_vector &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;High&#39;</span>, <span class="st">&#39;Low&#39;</span>, <span class="st">&#39;High&#39;</span>, <span class="st">&#39;Low&#39;</span>, <span class="st">&#39;Medium&#39;</span>)</span>
<span id="cb16-3"><a href="#cb16-3"></a></span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="co"># Specify ordinal variables with given levels</span></span>
<span id="cb16-5"><a href="#cb16-5"></a>factor_temperature_vector &lt;-<span class="st"> </span><span class="kw">factor</span>(temperature_vector,</span>
<span id="cb16-6"><a href="#cb16-6"></a>                                    <span class="dt">order =</span> <span class="ot">TRUE</span>,</span>
<span id="cb16-7"><a href="#cb16-7"></a>                                    <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&#39;Low&#39;</span>, <span class="st">&#39;Medium&#39;</span>, <span class="st">&#39;High&#39;</span>))</span>
<span id="cb16-8"><a href="#cb16-8"></a></span>
<span id="cb16-9"><a href="#cb16-9"></a><span class="co"># Display the vector</span></span>
<span id="cb16-10"><a href="#cb16-10"></a>factor_temperature_vector</span></code></pre>
              </div>
              <pre><code>## [1] High   Low    High   Low    Medium
## Levels: Low &lt; Medium &lt; High</code></pre>
            </div>
            <div id="data-frame" class="section level3">
              <h3>Data frame</h3>
              <ul>
                <li>
                  <p>
                    <strong
                      >A data frame is a combination of vectors that functions
                      like a spreadsheet within R.</strong
                    >
                  </p>
                  <div class="sourceCode" id="cb18">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="kw">data.frame</span>()</span></code></pre>
                  </div>
                </li>
                <li><p>Variables are columns, observations are rows.</p></li>
                <li>
                  <p>
                    Different types of vectors can be combined into a data
                    frame, as long as the vectors have the same length.
                  </p>
                </li>
                <li><p>This is also how datasets are organized in SAS.</p></li>
              </ul>
              <p>Example data frame <code>mtcars</code>:</p>
              <div class="sourceCode" id="cb19">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="kw">names</span>(mtcars) <span class="co"># names of columns</span></span></code></pre>
              </div>
              <pre><code>##  [1] &quot;mpg&quot;  &quot;cyl&quot;  &quot;disp&quot; &quot;hp&quot;   &quot;drat&quot; &quot;wt&quot;   &quot;qsec&quot; &quot;vs&quot;   &quot;am&quot;   &quot;gear&quot; &quot;carb&quot;</code></pre>
              <div class="sourceCode" id="cb21">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="kw">head</span>(mtcars)  <span class="co"># first observations</span></span></code></pre>
              </div>
              <pre><code>##                    mpg cyl disp  hp drat    wt  qsec vs am gear carb
## Mazda RX4         21.0   6  160 110 3.90 2.620 16.46  0  1    4    4
## Mazda RX4 Wag     21.0   6  160 110 3.90 2.875 17.02  0  1    4    4
## Datsun 710        22.8   4  108  93 3.85 2.320 18.61  1  1    4    1
## Hornet 4 Drive    21.4   6  258 110 3.08 3.215 19.44  1  0    3    1
## Hornet Sportabout 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2
## Valiant           18.1   6  225 105 2.76 3.460 20.22  1  0    3    1</code></pre>
              <div class="sourceCode" id="cb23">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="kw">tail</span>(mtcars)  <span class="co"># last observations</span></span></code></pre>
              </div>
              <pre><code>##                 mpg cyl  disp  hp drat    wt qsec vs am gear carb
## Porsche 914-2  26.0   4 120.3  91 4.43 2.140 16.7  0  1    5    2
## Lotus Europa   30.4   4  95.1 113 3.77 1.513 16.9  1  1    5    2
## Ford Pantera L 15.8   8 351.0 264 4.22 3.170 14.5  0  1    5    4
## Ferrari Dino   19.7   6 145.0 175 3.62 2.770 15.5  0  1    5    6
## Maserati Bora  15.0   8 301.0 335 3.54 3.570 14.6  0  1    5    8
## Volvo 142E     21.4   4 121.0 109 4.11 2.780 18.6  1  1    4    2</code></pre>
              <div class="sourceCode" id="cb25">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="kw">str</span>(mtcars)   <span class="co"># structure of data frame</span></span></code></pre>
              </div>
              <pre><code>## &#39;data.frame&#39;:    32 obs. of  11 variables:
##  $ mpg : num  21 21 22.8 21.4 18.7 18.1 14.3 24.4 22.8 19.2 ...
##  $ cyl : num  6 6 4 6 8 6 8 4 4 6 ...
##  $ disp: num  160 160 108 258 360 ...
##  $ hp  : num  110 110 93 110 175 105 245 62 95 123 ...
##  $ drat: num  3.9 3.9 3.85 3.08 3.15 2.76 3.21 3.69 3.92 3.92 ...
##  $ wt  : num  2.62 2.88 2.32 3.21 3.44 ...
##  $ qsec: num  16.5 17 18.6 19.4 17 ...
##  $ vs  : num  0 0 1 1 0 1 0 1 1 1 ...
##  $ am  : num  1 1 1 0 0 0 0 0 0 0 ...
##  $ gear: num  4 4 4 3 3 3 3 4 4 4 ...
##  $ carb: num  4 4 1 1 2 1 4 2 2 4 ...</code></pre>
              <div class="sourceCode" id="cb27">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="kw">dim</span>(mtcars)   <span class="co"># dimensions</span></span></code></pre>
              </div>
              <pre><code>## [1] 32 11</code></pre>
              <ul>
                <li>
                  The function <code>str()</code> shows you the structure of
                  your dataset. For a data frame it provides:
                  <ul>
                    <li>
                      The total number of observations (e.g. 32 car types)
                    </li>
                    <li>
                      The total number of variables (e.g. 11 car features)
                    </li>
                    <li>A full list of variable names (e.g. mpg, cyl … )</li>
                    <li>
                      The data type of each variable (e.g. num for car features)
                    </li>
                    <li>The first observations.</li>
                  </ul>
                </li>
                <li>
                  Also see Verzani simpleR Appendix: A sample R session for more
                  with the built-in mtcars dataset.
                </li>
              </ul>
              <p>Example from DataCamp Introduction to R course:</p>
              <div class="figure">
                <img
                  src="img/DataCamp_planets01.png"
                  title="DataCamp planets data frame 01"
                  alt=""
                />
                <p class="caption">DataCamp planets data frame 01</p>
              </div>
              <div class="figure">
                <img
                  src="img/DataCamp_planets01.png"
                  title="DataCamp planets data frame 02"
                  alt=""
                />
                <p class="caption">DataCamp planets data frame 02</p>
              </div>
              <ul>
                <li>
                  <p>
                    <strong
                      >Remember that a data frame is essentially a combination
                      of vectors.</strong
                    >
                    The DataCamp example above created the data frame in two
                    steps for instructional purposes, but you can usually build
                    this in to the <code>data.frame</code> command.
                  </p>
                </li>
                <li>
                  <p>
                    To create a data frame on planets sorted by decreasing
                    diameter (example from DataCamp):
                  </p>
                  <div class="sourceCode" id="cb29">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>positions &lt;-<span class="st"> </span><span class="kw">order</span>(planets_df<span class="op">$</span>diameter, <span class="dt">decreasing=</span><span class="ot">TRUE</span>)</span>
<span id="cb29-2"><a href="#cb29-2"></a><span class="co"># this does not order the data yet, for some reason</span></span>
<span id="cb29-3"><a href="#cb29-3"></a>largest_first_df &lt;-<span class="st"> </span>planets_df[positions,]</span></code></pre>
                  </div>
                </li>
              </ul>
              <div id="selecting-parts-of-data-frames" class="section level4">
                <h4>Selecting parts of data frames</h4>
                <ul>
                  <li>
                    Similar to vectors and matrices, select parts of a data
                    frame with square brackets <code>[ ]</code> in the form
                    <code>my_data_frame[rows,columns]</code>.
                    <ul>
                      <li>
                        <code>my_data_frame[1,2]</code> selects the second
                        element from the first row in
                        <code>my_data_frame</code>.
                      </li>
                      <li>
                        <code>my_data_frame[1:3,2:4]</code> selects rows 1,2,3
                        and columns 2,3,4 in <code>my_data_frame</code>.
                      </li>
                      <li>
                        <code>my_data_frame[1,]</code> selects all elements of
                        the first row.
                      </li>
                    </ul>
                  </li>
                  <li>
                    Selecting part of a data frame column:
                    <ul>
                      <li>
                        Maybe you want to select only the first 3 elements of
                        the variable <code>type</code> from the
                        <code>planets</code> data frame.
                      </li>
                      <li>
                        One way to do this is: <code>planets_df[1:3,1]</code>.
                        You have to know (or look up) the position of the
                        variable <code>type</code>, which gets hard if you have
                        a lot of variables.
                      </li>
                      <li>
                        It is often easier to just make use of the variable
                        name, e.g. <code>planets_df[1:3,'type']</code>.
                      </li>
                    </ul>
                  </li>
                  <li>
                    Selecting an entire data frame column:
                    <ul>
                      <li>
                        If you want to select all elements of the variable
                        <code>rings</code>, both <code>planets_df[,5]</code> and
                        <code>planets_df[,'rings']</code> work.
                      </li>
                      <li>
                        Use the <code>$</code> sign to tell R that it only has
                        to look up all the elements of the variable behind the
                        sign: <code>df_name$var_name</code>. You can combine
                        this with the element section above:
                        <code>df_name$var_name[rows]</code>. Note that you do
                        not need to specify a column because it is already
                        specified by the variable name.
                      </li>
                    </ul>
                  </li>
                  <li>
                    Selecting parts of a data frame conditionally:
                    <ul>
                      <li>
                        <p>
                          The <code>subset()</code> function selects a subset of
                          a data frame according to conditions you specify.
                          <code
                            >subset(df_name, some.condition== 'condition')</code
                          >. The first argument of
                          <code>subset()</code> specifies the dataset for which
                          you want a subset. By adding the second argument, you
                          give R the necessary information and conditions to
                          select the correct subset. Examples:
                        </p>
                        <div class="sourceCode" id="cb30">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="kw">subset</span>(planets_df, <span class="dt">subset=</span>(planets_df<span class="op">$</span>rings <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>))</span>
<span id="cb30-2"><a href="#cb30-2"></a><span class="kw">subset</span>(planets_df, <span class="dt">subset=</span>(planets_df<span class="op">$</span>diameter<span class="op">&lt;</span><span class="dv">1</span>))</span>
<span id="cb30-3"><a href="#cb30-3"></a>very_good =<span class="st"> </span><span class="kw">subset</span>(cdc,cdc<span class="op">$</span>genhlth<span class="op">==</span><span class="st">&#39;very good&#39;</span>)</span>
<span id="cb30-4"><a href="#cb30-4"></a>age_gt50 =<span class="st"> </span><span class="kw">subset</span>(cdc,cdc<span class="op">$</span>age<span class="op">&gt;</span><span class="dv">50</span>)</span>
<span id="cb30-5"><a href="#cb30-5"></a>red_wine &lt;-<span class="st"> </span><span class="kw">subset</span>(wine_data, wine_data[, <span class="dv">1</span>]<span class="op">==</span><span class="st">&#39;Red&#39;</span>)</span>
<span id="cb30-6"><a href="#cb30-6"></a>red_usa &lt;-<span class="st"> </span><span class="kw">subset</span>(red_wine_data, red_wine_data<span class="op">$</span>condition <span class="op">==</span><span class="st"> &#39;USA&#39;</span>)</span></code></pre>
                        </div>
                      </li>
                      <li>
                        <p>
                          The second <code>subset=</code> inside the parentheses
                          may not be necessary.
                        </p>
                      </li>
                      <li>
                        <p>
                          Conditions can be used with logical operators
                          <code>&amp;</code> and <code>|</code>.
                        </p>
                        <ul>
                          <li>
                            The <code>&amp;</code> is read ‘and’ so that
                            <code
                              >subset(cdc, cdc$gender == 'f' &amp; cdc$age &gt;
                              30)</code
                            >
                            will give you the data for women over the age of 30.
                            <code
                              >under23_and_smoke = subset(cdc,cdc$age&lt;23
                              &amp; cdc$smoke100==T)</code
                            >
                            will select respondents who are under the age of 23
                            and have smoked &gt;100 cigarettes.
                          </li>
                          <li>
                            The<code>|</code> character is read ‘or’ so that
                            <code
                              >subset(cdc, cdc$gender == 'f' | cdc$age &gt;
                              30)</code
                            >
                            will select people who are women or over the age of
                            30.
                          </li>
                          <li>
                            In principle, you may use as many ‘and’ and ‘or’
                            clauses as you like in a subset.
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
              <div id="sorting-data-frames" class="section level4">
                <h4>Sorting data frames</h4>
                <ul>
                  <li>
                    <p>Sort data frames with <code>order()</code>.</p>
                    <ul>
                      <li>
                        <code>order()</code> is a function that gives you the
                        position of each element when it is applied on a
                        variable. Let us look at the vector a:
                        <code>a &lt;- c(100,9,101)</code>. Now
                        <code>order(a)</code> returns 2,1,3. Since 100 is the
                        second largest element of the vector, 9 is the smallest
                        element and 101 is the largest element.
                      </li>
                      <li>
                        Subsequently, <code>a[order(a)]</code> will give you the
                        ordered vector 9,100,101, since it first picks the
                        second element of a, then the first and then the last.
                      </li>
                    </ul>
                  </li>
                  <li>
                    <p>
                      Example from my
                      <a href="https://github.com/br3ndonland/R-proteomics-Nrf1"
                        >postdoc Nrf1 proteomics experiments</a
                      >:
                    </p>
                    <div class="sourceCode" id="cb31">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a><span class="co"># Groups in alphabetical order</span></span>
<span id="cb31-2"><a href="#cb31-2"></a>groups &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;lacZ&#39;</span>,</span>
<span id="cb31-3"><a href="#cb31-3"></a>            <span class="st">&#39;Nrf1-HA bort&#39;</span>,</span>
<span id="cb31-4"><a href="#cb31-4"></a>            <span class="st">&#39;Nrf1-HA chol&#39;</span>,</span>
<span id="cb31-5"><a href="#cb31-5"></a>            <span class="st">&#39;Nrf1-HA chow&#39;</span>)</span>
<span id="cb31-6"><a href="#cb31-6"></a>groups[<span class="kw">order</span>(groups)]</span></code></pre>
                    </div>
                    <pre><code>## [1] &quot;lacZ&quot;         &quot;Nrf1-HA bort&quot; &quot;Nrf1-HA chol&quot; &quot;Nrf1-HA chow&quot;</code></pre>
                    <div class="sourceCode" id="cb33">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="co"># Put groups in my preferred order</span></span>
<span id="cb33-2"><a href="#cb33-2"></a>ordered =<span class="st"> </span><span class="kw">factor</span>(groups,</span>
<span id="cb33-3"><a href="#cb33-3"></a>                 <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&#39;lacZ&#39;</span>,</span>
<span id="cb33-4"><a href="#cb33-4"></a>                            <span class="st">&#39;Nrf1-HA chow&#39;</span>,</span>
<span id="cb33-5"><a href="#cb33-5"></a>                            <span class="st">&#39;Nrf1-HA chol&#39;</span>,</span>
<span id="cb33-6"><a href="#cb33-6"></a>                            <span class="st">&#39;Nrf1-HA bort&#39;</span>))</span>
<span id="cb33-7"><a href="#cb33-7"></a><span class="co"># Verify desired order of groups</span></span>
<span id="cb33-8"><a href="#cb33-8"></a>groups[<span class="kw">order</span>(ordered)]</span></code></pre>
                    </div>
                    <pre><code>## [1] &quot;lacZ&quot;         &quot;Nrf1-HA chow&quot; &quot;Nrf1-HA chol&quot; &quot;Nrf1-HA bort&quot;</code></pre>
                    <ul>
                      <li>
                        Make sure you refer to the new sorted variable (<code
                          >Orderedgroups</code
                        >
                        in this case) in subsequent code.
                      </li>
                    </ul>
                  </li>
                  <li>
                    <p>
                      Example:
                      <a
                        href="http://www.cookbook-r.com/Manipulating_data/Reordering_the_columns_in_a_data_frame/"
                        >Reordering columns in a data frame from Cookbook for
                        R</a
                      >:
                    </p>
                    <div class="sourceCode" id="cb35">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a><span class="co"># A sample data frame</span></span>
<span id="cb35-2"><a href="#cb35-2"></a>data &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb35-3"><a href="#cb35-3"></a>                   <span class="dt">text =</span> <span class="st">&#39;</span></span>
<span id="cb35-4"><a href="#cb35-4"></a><span class="st">  id weight   size</span></span>
<span id="cb35-5"><a href="#cb35-5"></a><span class="st">  1   20  small</span></span>
<span id="cb35-6"><a href="#cb35-6"></a><span class="st">  2   27  large</span></span>
<span id="cb35-7"><a href="#cb35-7"></a><span class="st">  3   24 medium</span></span>
<span id="cb35-8"><a href="#cb35-8"></a><span class="st">&#39;</span>)</span>
<span id="cb35-9"><a href="#cb35-9"></a></span>
<span id="cb35-10"><a href="#cb35-10"></a><span class="co"># Reorder by column number</span></span>
<span id="cb35-11"><a href="#cb35-11"></a>data[<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">2</span>)]</span></code></pre>
                    </div>
                    <pre><code>##   id   size weight
## 1  1  small     20
## 2  2  large     27
## 3  3 medium     24</code></pre>
                    <div class="sourceCode" id="cb37">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a><span class="co"># Reorder by column name</span></span>
<span id="cb37-2"><a href="#cb37-2"></a>data[<span class="kw">c</span>(<span class="st">&#39;size&#39;</span>, <span class="st">&#39;id&#39;</span>, <span class="st">&#39;weight&#39;</span>)]</span></code></pre>
                    </div>
                    <pre><code>##     size id weight
## 1  small  1     20
## 2  large  2     27
## 3 medium  3     24</code></pre>
                    <div class="sourceCode" id="cb39">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a><span class="co"># Overwrite data variable with new info:</span></span>
<span id="cb39-2"><a href="#cb39-2"></a><span class="co"># data &lt;- data[c(1,3,2)]</span></span></code></pre>
                    </div>
                    <blockquote>
                      <p>
                        The above examples index into the data frame by treating
                        it as a list (a data frame is essentially a list of
                        vectors). You can also use matrix-style indexing, as in
                        data[row, col], where row is left blank.
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb40">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a>data[, <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">2</span>)]</span></code></pre>
                    </div>
                    <pre><code>##   id   size weight
## 1  1  small     20
## 2  2  large     27
## 3  3 medium     24</code></pre>
                    <blockquote>
                      <p>
                        The drawback to matrix indexing is that it gives
                        different results when you specify just one column. In
                        these cases, the returned object is a vector, not a data
                        frame. Because the returned data type isn’t always
                        consistent with matrix indexing, it’s generally safer to
                        use list-style indexing, or the drop=FALSE option:
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb42">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="co"># List-style indexing of one column</span></span>
<span id="cb42-2"><a href="#cb42-2"></a>data[<span class="dv">2</span>]</span></code></pre>
                    </div>
                    <pre><code>##   weight
## 1     20
## 2     27
## 3     24</code></pre>
                    <div class="sourceCode" id="cb44">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a><span class="co"># Matrix-style indexing of one column</span></span>
<span id="cb44-2"><a href="#cb44-2"></a><span class="co"># drops dimension to become a vector</span></span>
<span id="cb44-3"><a href="#cb44-3"></a>data[, <span class="dv">2</span>]</span></code></pre>
                    </div>
                    <pre><code>## [1] 20 27 24</code></pre>
                    <div class="sourceCode" id="cb46">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a><span class="co"># Matrix-style indexing with drop=FALSE</span></span>
<span id="cb46-2"><a href="#cb46-2"></a><span class="co"># preserves dimension to remain data frame</span></span>
<span id="cb46-3"><a href="#cb46-3"></a>data[, <span class="dv">2</span>, drop =<span class="st"> </span><span class="ot">FALSE</span>]</span></code></pre>
                    </div>
                    <pre><code>##   weight
## 1     20
## 2     27
## 3     24</code></pre>
                  </li>
                  <li>
                    <p>
                      <a
                        href="https://stackoverflow.com/questions/1296646/how-to-sort-a-dataframe-by-columns"
                        >Another example from Stack Overflow</a
                      >
                    </p>
                    <ul>
                      <li>
                        <p>Create the data frame</p>
                        <div class="sourceCode" id="cb48">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>dd &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</span>
<span id="cb48-2"><a href="#cb48-2"></a>  <span class="dt">b =</span> <span class="kw">factor</span>(</span>
<span id="cb48-3"><a href="#cb48-3"></a>    <span class="kw">c</span>(<span class="st">&#39;Hi&#39;</span>, <span class="st">&#39;Med&#39;</span>, <span class="st">&#39;Hi&#39;</span>, <span class="st">&#39;Low&#39;</span>),</span>
<span id="cb48-4"><a href="#cb48-4"></a>    <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&#39;Low&#39;</span>, <span class="st">&#39;Med&#39;</span>, <span class="st">&#39;Hi&#39;</span>),</span>
<span id="cb48-5"><a href="#cb48-5"></a>    <span class="dt">ordered =</span> <span class="ot">TRUE</span></span>
<span id="cb48-6"><a href="#cb48-6"></a>  ),</span>
<span id="cb48-7"><a href="#cb48-7"></a>  <span class="dt">x =</span> <span class="kw">c</span>(<span class="st">&#39;A&#39;</span>, <span class="st">&#39;D&#39;</span>, <span class="st">&#39;A&#39;</span>, <span class="st">&#39;C&#39;</span>),</span>
<span id="cb48-8"><a href="#cb48-8"></a>  <span class="dt">y =</span> <span class="kw">c</span>(<span class="dv">8</span>, <span class="dv">3</span>, <span class="dv">9</span>, <span class="dv">9</span>),</span>
<span id="cb48-9"><a href="#cb48-9"></a>  <span class="dt">z =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">2</span>)</span>
<span id="cb48-10"><a href="#cb48-10"></a>)</span>
<span id="cb48-11"><a href="#cb48-11"></a>dd</span></code></pre>
                        </div>
                        <pre><code>##     b x y z
## 1  Hi A 8 1
## 2 Med D 3 1
## 3  Hi A 9 1
## 4 Low C 9 2</code></pre>
                      </li>
                      <li>
                        <p>
                          Sort the frame by column z (descending) then by column
                          b (ascending).
                        </p>
                        <ul>
                          <li>
                            <strong
                              >You don’t have to put
                              <code>decreasing=TRUE</code> like the DataCamp
                              examples, you can just put a negative sign before
                              the column name (-z).</strong
                            >
                          </li>
                          <li>
                            <blockquote>
                              <p>
                                You can use the <code>order()</code> function
                                directly without resorting to add-on tools – see
                                this simpler answer which uses a trick right
                                from the top of the example(order) code:
                              </p>
                            </blockquote>
                            <div class="sourceCode" id="cb50">
                              <pre
                                class="sourceCode r"
                              ><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a>R &lt;-<span class="st"> </span>dd[<span class="kw">with</span>(dd, <span class="kw">order</span>(<span class="op">-</span>z, b)), ]</span></code></pre>
                            </div>
                          </li>
                          <li>
                            <blockquote>
                              <p>
                                Edit some 2+ years later: It was just asked how
                                to do this by column index. The answer is to
                                simply pass the desired sorting column(s) to the
                                order() function, rather than using the name of
                                the column (and <code>with()</code> for
                                easier/more direct access).
                              </p>
                            </blockquote>
                            <div class="sourceCode" id="cb51">
                              <pre
                                class="sourceCode r"
                              ><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>R &lt;-<span class="st"> </span>dd[ <span class="kw">order</span>(<span class="op">-</span>dd[,<span class="dv">4</span>], dd[,<span class="dv">1</span>]), ]</span></code></pre>
                            </div>
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
              <div id="merging-data-frames" class="section level4">
                <h4>Merging data frames</h4>
                <ul>
                  <li>
                    Adding columns
                    <ul>
                      <li>
                        <p>
                          To merge two data frames (datasets) horizontally, use
                          the <code>merge</code> function. In most cases, you
                          join two data frames by one or more common key
                          variables (i.e., an inner join).
                        </p>
                        <div class="sourceCode" id="cb52">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a><span class="co"># merge two data frames by ID</span></span>
<span id="cb52-2"><a href="#cb52-2"></a>total &lt;-<span class="st"> </span><span class="kw">merge</span>(data frameA,data frameB,<span class="dt">by=</span><span class="st">&#39;ID&#39;</span>)</span>
<span id="cb52-3"><a href="#cb52-3"></a><span class="co"># merge two data frames by ID and Country</span></span>
<span id="cb52-4"><a href="#cb52-4"></a>total &lt;-<span class="st"> </span><span class="kw">merge</span>(data frameA,data frameB,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&#39;ID&#39;</span>,<span class="st">&#39;Country&#39;</span>))</span></code></pre>
                        </div>
                      </li>
                    </ul>
                  </li>
                  <li>
                    Adding rows
                    <ul>
                      <li>
                        <p>
                          To join two data frames (datasets) vertically, use the
                          <code>rbind</code> function. The two data frames must
                          have the same variables, but they do not have to be in
                          the same order.
                        </p>
                        <div class="sourceCode" id="cb53">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a>total &lt;-<span class="st"> </span><span class="kw">rbind</span>(data_frame_A, data_frame_B)</span></code></pre>
                        </div>
                      </li>
                      <li>
                        <p>
                          If <code>data_frame_A</code> has variables that
                          <code>data_frame_B</code> does not, then either:
                        </p>
                        <ul>
                          <li>Delete the extra variables in data frameA or</li>
                          <li>
                            Create the additional variables in data frameB and
                            <a
                              href="http://www.statmethods.net/input/missingdata.html"
                              >set them to NA (missing)</a
                            >
                            before joining them with <code>rbind( )</code>.
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
              <div id="manipulating-data-frames" class="section level4">
                <h4>Manipulating data frames</h4>
                <ul>
                  <li>
                    <p>
                      Stack and unstack using the PlantGrowth dataset included
                      with R:
                    </p>
                    <div class="sourceCode" id="cb54">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a><span class="kw">unstack</span>(PlantGrowth)</span></code></pre>
                    </div>
                    <ul>
                      <li>
                        <code>Unstack</code> separates out the groups into a
                        data frame.
                      </li>
                    </ul>
                  </li>
                  <li>
                    <p>To create a boxplot of the three groups:</p>
                    <div class="sourceCode" id="cb55">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a><span class="kw">boxplot</span>(<span class="kw">unstack</span>(PlantGrowth))</span></code></pre>
                    </div>
                    <ul>
                      <li>
                        General boxplots are made with
                        <code>boxplot(independent_var~dependent_var)</code>. See
                        the <a href="#plots">Plots section</a>.
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
            <div id="list" class="section level3">
              <h3>List</h3>
              <ul>
                <li>
                  <p>
                    A list in R gathers a variety of objects under one name
                    (that is, the name of the list) in an ordered way. These
                    objects can be matrices, vectors, data frames, other lists,
                    etc. The objects do not need to be related to each other.
                  </p>
                </li>
                <li>
                  <p>
                    To construct a list you use the function
                    <code>list()</code>:
                  </p>
                  <div class="sourceCode" id="cb56">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a>my_list &lt;-<span class="st"> </span><span class="kw">list</span>(component1, component2...)</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    The arguments to the list function are the list components.
                    To name items in the list:
                    <code>list(name1=your.component1, name2=component2,…)</code>
                  </p>
                </li>
              </ul>
              <div id="selecting-elements-from-a-list" class="section level4">
                <h4>Selecting elements from a list</h4>
                <ul>
                  <li>
                    <p>
                      Lists will often be built out of numerous elements and
                      components. Getting a single element, multiple elements,
                      or a component out of the list is not always
                      straightforward.
                    </p>
                  </li>
                  <li>
                    <p>
                      List components can be selected with square brackets as
                      usual, like <code>list[1]</code>.
                    </p>
                  </li>
                  <li>
                    <p>
                      <code>length()</code> returns the total number of
                      components in the list. For example, to identify the last
                      actor in a list of actors in the movie The Shining:
                    </p>
                    <div class="sourceCode" id="cb57">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a>shining_list<span class="op">$</span>actors[<span class="kw">length</span>(shining_list<span class="op">$</span>actors)]</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p>
                      As with data frames, components can be referenced by name:
                    </p>
                    <div class="sourceCode" id="cb58">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a>shining_list<span class="op">$</span>reviews</span>
<span id="cb58-2"><a href="#cb58-2"></a>shining_list[[<span class="st">&#39;reviews&#39;</span>]]</span></code></pre>
                    </div>
                  </li>
                </ul>
              </div>
              <div id="adding-elements-to-a-list" class="section level4">
                <h4>Adding elements to a list</h4>
                <ul>
                  <li>
                    The concatenate function <code>c()</code> is used to add
                    elements to a list: <code>c(list1,some.object)</code>
                  </li>
                  <li>
                    If you want to give the new list item a name, you just add
                    this to the function:
                    <code>c(list1,new.item.name=some.object)</code>
                  </li>
                </ul>
                <p><a href="#top">back to top</a></p>
              </div>
            </div>
          </div>
          <div id="importing-data-into-r" class="section level2">
            <h2>Importing data into R</h2>
            <p>
              In RStudio, datasets can be loaded from CSV, from web using
              <strong>Tools -&gt; Import dataset</strong> or by clicking
              <strong>Import dataset</strong> in the environment pane.
            </p>
            <div id="csv-file" class="section level3">
              <h3>CSV file</h3>
              <p>
                This is one of the best ways to read in a dataset. Note that the
                file path should be in UNIX format. UNIX file paths have forward
                slashes, Windows file paths have backslashes (‘whacks’).
              </p>
              <div class="sourceCode" id="cb59">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a>mydata &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;c:/mydata.csv&#39;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">sep=</span><span class="st">&#39;,&#39;</span>)</span></code></pre>
              </div>
              <ul>
                <li>First row contains variable names, comma is separator</li>
                <li>Note the / instead of  on mswindows systems</li>
              </ul>
              <p>
                There are many other parameters that can be specified when
                reading in a dataset. See R help.
              </p>
            </div>
            <div id="excel-file" class="section level3">
              <h3>Excel file</h3>
              <p>
                Export the excel file to a CSV and follow instructions above, or
                use the xlsx package:
              </p>
              <div class="sourceCode" id="cb60">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1"></a><span class="co"># read in the first worksheet from the workbook myexcel.xlsx</span></span>
<span id="cb60-2"><a href="#cb60-2"></a><span class="co"># first row contains variable names</span></span>
<span id="cb60-3"><a href="#cb60-3"></a><span class="kw">library</span>(xlsx)</span>
<span id="cb60-4"><a href="#cb60-4"></a>mydata &lt;-<span class="st"> </span><span class="kw">read.xlsx</span>(<span class="st">&#39;c:/myexcel.xlsx&#39;</span>, <span class="dv">1</span>)</span>
<span id="cb60-5"><a href="#cb60-5"></a></span>
<span id="cb60-6"><a href="#cb60-6"></a><span class="co"># read in the worksheet named mysheet</span></span>
<span id="cb60-7"><a href="#cb60-7"></a>mydata &lt;-<span class="st"> </span><span class="kw">read.xlsx</span>(<span class="st">&#39;c:/myexcel.xlsx&#39;</span>, <span class="dt">sheetName =</span> <span class="st">&#39;mysheet&#39;</span>)</span></code></pre>
              </div>
            </div>
            <div id="website" class="section level3">
              <h3>Website</h3>
              <div class="sourceCode" id="cb61">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a><span class="kw">load</span>(<span class="kw">url</span>(<span class="st">&#39;dataset_url&#39;</span>))</span>
<span id="cb61-2"><a href="#cb61-2"></a><span class="co"># or</span></span>
<span id="cb61-3"><a href="#cb61-3"></a>Dataset_name=<span class="kw">read.table</span>(<span class="st">&#39;dataset_url&#39;</span>)</span></code></pre>
              </div>
              <p>URL must be enclosed in quotes to make it a string.</p>
            </div>
            <div id="apis" class="section level3">
              <h3>APIs</h3>
              <ul>
                <li>
                  R is less flexible than Python. Accessing Application
                  Programming Interfaces (APIs) is one example of this.
                </li>
                <li>
                  <a href="https://www.programmableweb.com">ProgrammableWeb</a>
                  shared a helpful
                  <a
                    href="https://www.programmableweb.com/news/how-to-access-any-restful-api-using-r-language/how-to/2017/07/21"
                    >guide</a
                  >
                  to accessing APIs with R.
                </li>
              </ul>
            </div>
            <div id="manually" class="section level3">
              <h3>Manually</h3>
              <ul>
                <li>
                  Create vectors with the dataset numbers and then create a data
                  frame with those vectors (see above).
                </li>
                <li>
                  the <code>scan()</code> function can be used to import a
                  vector or list.
                </li>
                <li>
                  See
                  <a
                    href="http://cran.rstudio.com/doc/manuals/r-release/R-data.pdf"
                    >R-data</a
                  >
                  CRAN document, and Verzani simpleR Appendix: Entering Data
                  into R for more info.
                </li>
              </ul>
              <p><a href="#top">back to top</a></p>
            </div>
          </div>
          <div id="working-with-data" class="section level2">
            <h2>Working with data</h2>
            <div id="cleaning-data" class="section level3">
              <h3>Cleaning data</h3>
              <ul>
                <li>
                  The <code>dplyr</code> package is one of the most popular for
                  working with data in R. It is similar to Python Pandas. See
                  <a
                    href="https://gist.github.com/TomAugspurger/6e052140eaa5fdb6e8c0"
                    >Tom Augspurger’s Gist on dplyr and Pandas</a
                  >.
                </li>
                <li>
                  <code>na.omit(dataset_name$optional_var_name)</code> will omit
                  any missing values (usually labeled ‘NA’)
                </li>
              </ul>
            </div>
            <div id="attaching-data" class="section level3">
              <h3>Attaching data</h3>
              <div class="sourceCode" id="cb62">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1"></a><span class="kw">attach</span>(dataset_name, options)</span></code></pre>
              </div>
              <ul>
                <li>
                  Datasets are usually attached so that variables can be named
                  without reference to the original dataset.
                </li>
                <li>
                  Note that if the code is run repeatedly without detaching the
                  dataset each time, a message in the console will explain that
                  the variables are masked by the previously input variables of
                  the same name.
                </li>
                <li>
                  You can also detach() the dataset when you are done. Empty
                  parentheses detach all variables in the workspace.
                </li>
                <li>
                  From R help (article: ‘Attach Set of R Objects to Search
                  Path’): ‘By attaching a data frame (or list) to the search
                  path it is possible to refer to the variables in the data
                  frame by their names alone, rather than as components of the
                  data frame (e.g., in the example below, height rather than
                  women$height).’
                </li>
              </ul>
            </div>
            <div id="exporting-data-frames" class="section level3">
              <h3>Exporting data frames</h3>
              <ul>
                <li>
                  <p>Exporting to a delimited text file:</p>
                  <div class="sourceCode" id="cb63">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1"></a><span class="kw">write.table</span>(x, outfile, <span class="dt">sep=</span>delimiter, <span class="dt">quote=</span><span class="ot">TRUE</span>, <span class="dt">na=</span><span class="st">&#39;NA&#39;</span>)</span></code></pre>
                  </div>
                  <ul>
                    <li>
                      Where <code>x</code> is the object and
                      <code>outfile</code> is the target file. For example, the
                      statement
                      <code>write.table(mydata, 'mydata.txt', sep=',')</code>
                      would save the dataset <code>mydata</code> to a
                      comma-delimited file named mydata.txt in the current
                      working directory. Include a path (for example,
                      <code>'c:/myprojects/mydata. txt'</code>) to save the
                      output file elsewhere. Replacing<code>sep=','</code> with
                      <code>sep='\t'</code> would save the data in a
                      tab-delimited file. By default, strings are enclosed in
                      quotes (’’) and missing values are written as
                      <code>NA</code>.
                    </li>
                  </ul>
                </li>
                <li>
                  <p>Exporting to an Excel workbook using the xlsx package:</p>
                  <div class="sourceCode" id="cb64">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1"></a><span class="kw">library</span>(xlsx) <span class="kw">write.xlsx</span>(x, outfile, <span class="dt">col.Names=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="ot">TRUE</span>, <span class="dt">sheetName=</span><span class="st">&#39;Sheet 1&#39;</span>, <span class="dt">append=</span><span class="ot">FALSE</span>)</span></code></pre>
                  </div>
                </li>
              </ul>
              <p><a href="#top">back to top</a></p>
            </div>
          </div>
          <div id="statistical-models" class="section level2">
            <h2>Statistical models</h2>
            <div
              id="helpful-resources-for-statistical-modeling"
              class="section level3"
            >
              <h3>Helpful resources for statistical modeling</h3>
              <ul>
                <li>
                  <strong
                    >The
                    <a href="https://br3ndonland.github.io/R-guide/code"
                      >R sample code page</a
                    >.</strong
                  >
                </li>
                <li>
                  <a
                    href="https://www.manning.com/books/r-in-action-second-edition"
                    >R in Action</a
                  >
                </li>
                <li>
                  <a
                    href="http://proquest.safaribooksonline.com/book/programming/r/9781449363086"
                    >R Graphics Cookbook</a
                  >
                  by Winston Chang (from Cookbook for R creator)
                </li>
                <li>
                  <a href="https://www.r-bloggers.com/tag/r-tutorial-series/"
                    >R-bloggers R tutorial series</a
                  >
                </li>
                <li>
                  <a href="http://www.rdocumentation.org/">R documentation</a>
                  and <a href="http://www.statmethods.net/">Quick-R</a> for
                  reference
                </li>
                <li>
                  <a href="http://www.ats.ucla.edu/stat/r/"
                    >UCLA IDRE website</a
                  >
                </li>
                <li>
                  Chapter 11: Statistical Models in R from An Introduction to R
                </li>
                <li>Linear models in R by Faraway</li>
                <li>The R book by Crawley</li>
                <li>
                  Guidelines for selection of statistical tests from
                  <a
                    href="http://www.ats.ucla.edu/stat/mult_pkg/whatstat/choosestat.html"
                    >UCLA</a
                  >
                  and
                  <a href="http://bama.ua.edu/~jleeper/627/choosestat.html"
                    >University of Alabama</a
                  >
                </li>
              </ul>
            </div>
            <div id="types-of-variables" class="section level3">
              <h3>Types of variables</h3>
              <ul>
                <li>
                  Quantitative
                  <ul>
                    <li>
                      Interval/continuous: A measured value where the distance
                      between each value is the same (length, width, weight,
                      blood cholesterol level). A subtype of continuous variable
                      that has a true zero value (age, time, temperature in
                      degrees K) is sometimes called a ratio variable.
                    </li>
                    <li>
                      Discrete: having only counted integer values (# cells, #
                      people)
                    </li>
                  </ul>
                </li>
                <li>
                  Qualitative
                  <ul>
                    <li>
                      Nominal: also called categorical. Variable with discrete
                      names for levels (color, diet group)
                    </li>
                    <li>
                      Ordinal: variable with names ranked and organized into
                      levels (low/medium/high)
                    </li>
                  </ul>
                </li>
                <li>
                  Also see
                  <ul>
                    <li>
                      Stevens SS <em>Science</em> 1946, DOI:
                      <a href="https://doi.org/10.1126/science.103.2684.677"
                        >10.1126/science.103.2684.677</a
                      >
                    </li>
                    <li>
                      DataCamp introduction to statistics course lesson 1 video
                    </li>
                    <li>
                      My course notes
                      <ul>
                        <li>
                          <a
                            href="https://github.com/br3ndonland/R-guide/tree/master/lfs/CPSC440-Lecture-Ch01-annotated.url"
                            >CPSC 440 Lecture Notes Chapter 01-03</a
                          >
                          slide 5
                        </li>
                        <li>
                          <a
                            href="https://github.com/br3ndonland/R-guide/tree/master/lfs/CPSC440-Lab-01-slides-annotated.url"
                            >CPSC 440 lab 1</a
                          >
                          slide 9
                        </li>
                        <li>
                          <a
                            href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-05A-annotated.url"
                            >PATH 591 Lecture 05 A</a
                          >
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
            <div id="basic-statistics" class="section level3">
              <h3>Basic statistics</h3>
              <ul>
                <li>
                  <p>Mean: <code>mean()</code></p>
                  <ul>
                    <li>
                      <p>
                        Groups of means can be compared with the
                        <code>by()</code> function:
                      </p>
                      <div class="sourceCode" id="cb65">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1"></a><span class="kw">by</span>(df_name<span class="op">$</span>var_name1,df_name<span class="op">$</span>var_name2,mean)</span></code></pre>
                      </div>
                    </li>
                  </ul>
                </li>
                <li>
                  <p>Median: <code>median()</code></p>
                </li>
                <li>
                  <p>Variance: <code>variance()</code></p>
                </li>
                <li>
                  <p>Absolute value: <code>abs(x)</code></p>
                </li>
                <li>
                  <p>Square root: <code>sqrt(x)</code></p>
                </li>
                <li>
                  <p>Log: <code>log(x)</code></p>
                </li>
                <li>
                  <p>
                    Computing summary statistics of data subsets:
                    <code>aggregate(model_formula, df_name, mean)</code>. This
                    function is useful if you want to calculate means of each
                    group in a dataset.
                  </p>
                </li>
                <li>
                  <p>
                    Frequency table (to count the number of times each category
                    occurs in a data frame): <code>table(df_name)</code>
                  </p>
                  <ul>
                    <li>
                      <p>
                        For a specific variable:
                        <code>table(df_name$var_name)</code>
                      </p>
                    </li>
                    <li>
                      <p>
                        The <code>table</code> command can also be used to
                        combine multiple vectors of different types into a
                        table:
                      </p>
                      <div class="sourceCode" id="cb66">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1"></a>smokes =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;Y&#39;</span>,<span class="st">&#39;N&#39;</span>,<span class="st">&#39;N&#39;</span>,<span class="st">&#39;Y&#39;</span>,<span class="st">&#39;N&#39;</span>,<span class="st">&#39;Y&#39;</span>,<span class="st">&#39;Y&#39;</span>,<span class="st">&#39;Y&#39;</span>,<span class="st">&#39;N&#39;</span>,<span class="st">&#39;Y&#39;</span>)</span>
<span id="cb66-2"><a href="#cb66-2"></a>amount =<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)</span>
<span id="cb66-3"><a href="#cb66-3"></a><span class="kw">table</span>(smokes,amount)</span></code></pre>
                      </div>
                    </li>
                  </ul>
                </li>
                <li>
                  <p>Relative frequency table:</p>
                  <div class="sourceCode" id="cb67">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1"></a><span class="kw">table</span>(df_name<span class="op">$</span>var_name1)<span class="op">/</span>number_observations</span></code></pre>
                  </div>
                </li>
                <li>
                  <p>
                    The <code>summary()</code> function gives you an overview
                    description of a variable (minimum, first quartile, median,
                    mean, third quartile, and maximum). The
                    <code>describe()</code> function is similar, but includes
                    more information on the distribution (sd, range, skew,
                    kurtosis). Remember that skew is classified by the direction
                    of the data tail.
                  </p>
                </li>
                <li>
                  <p>
                    Using <strong>comparisons</strong> is a good way to
                    facilitate data analysis.
                  </p>
                  <ul>
                    <li>
                      <p>
                        To select all elements of a vector that are greater than
                        zero, type
                        <code>new_vector_name &lt;- vector_name &gt;0</code>
                      </p>
                    </li>
                    <li>
                      <p>
                        To compare vectors/variables (can also use
                        <code>&lt;</code> or <code>==</code>):
                      </p>
                      <div class="sourceCode" id="cb68">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1"></a>df_name<span class="op">$</span>var_name1<span class="op">&gt;</span>df_name<span class="op">$</span>var_name2</span></code></pre>
                      </div>
                    </li>
                    <li>
                      <p>To compare multiple elements of a vector:</p>
                      <div class="sourceCode" id="cb69">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1"></a>vector_name[element1]<span class="op">&gt;</span>vector_name[element2]</span></code></pre>
                      </div>
                    </li>
                    <li>
                      <p>To create a ratio:</p>
                      <div class="sourceCode" id="cb70">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1"></a>ratio_name &lt;-<span class="st"> </span>df_name<span class="op">$</span>var_name1<span class="op">/</span>df_name<span class="op">$</span>var_name2</span></code></pre>
                      </div>
                    </li>
                    <li>
                      <p>Proportion:</p>
                      <div class="sourceCode" id="cb71">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1"></a>proportion_name &lt;-<span class="st"> </span>df_name<span class="op">$</span>var_name1<span class="op">/</span>(df_name<span class="op">$</span>var_name1<span class="op">+</span>df_name<span class="op">$</span>var_name2)</span></code></pre>
                      </div>
                    </li>
                    <li>
                      <p>
                        You can calculate a proportion based on a subset of the
                        data. Example:
                      </p>
                      <div class="sourceCode" id="cb72">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1"></a><span class="co"># The &#39;atheism&#39; data frame has already been loaded</span></span>
<span id="cb72-2"><a href="#cb72-2"></a><span class="co"># Your code to create the &#39;us12&#39; subset:</span></span>
<span id="cb72-3"><a href="#cb72-3"></a>us12 =<span class="st"> </span><span class="kw">subset</span>(atheism, atheism<span class="op">$</span>nationality <span class="op">==</span><span class="st"> &#39;United States&#39;</span> <span class="op">&amp;</span><span class="st"> </span>atheism<span class="op">$</span>year <span class="op">==</span><span class="st">&#39;2012&#39;</span>)</span>
<span id="cb72-4"><a href="#cb72-4"></a><span class="co"># Calculate the proportion of atheist responses:</span></span>
<span id="cb72-5"><a href="#cb72-5"></a>proportion =<span class="st"> </span><span class="kw">nrow</span>(<span class="kw">subset</span>(us12,response <span class="op">==</span><span class="st"> &#39;atheist&#39;</span>))<span class="op">/</span><span class="kw">nrow</span>(us12)</span></code></pre>
                      </div>
                    </li>
                  </ul>
                </li>
                <li>
                  <p>
                    Sum vectors with the <code>sum()</code> function, and
                    compare sums with <code>var_name&gt;other_var_name</code>.
                  </p>
                </li>
                <li>
                  <p>
                    The <code>scale()</code> function converts values to z
                    scores.
                  </p>
                </li>
              </ul>
              <div id="confidence-intervals" class="section level4">
                <h4>Confidence intervals</h4>
                <ul>
                  <li>
                    <p>
                      95% confidence for a sample size n means that 95% of
                      random samples of size n will yield confidence intervals
                      that contain the true average value.
                    </p>
                  </li>
                  <li>
                    <p>
                      The margin or error is equal to the radius (or half the
                      width) of the confidence interval.
                    </p>
                  </li>
                  <li>
                    <p>Sample code:</p>
                    <div class="sourceCode" id="cb73">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1"></a><span class="co"># Calculate standard error</span></span>
<span id="cb73-2"><a href="#cb73-2"></a>se =<span class="st"> </span><span class="kw">sd</span>(vector_name)<span class="op">/</span><span class="kw">sqrt</span>(<span class="co"># observations)</span></span>
<span id="cb73-3"><a href="#cb73-3"></a></span>
<span id="cb73-4"><a href="#cb73-4"></a><span class="co"># Calculate lower and upper CI bounds</span></span>
<span id="cb73-5"><a href="#cb73-5"></a><span class="dt">lower =</span> sample_mean <span class="op">-</span><span class="st"> </span><span class="fl">1.96</span> <span class="op">-</span><span class="st"> </span>se</span>
<span id="cb73-6"><a href="#cb73-6"></a><span class="dt">upper =</span> sample_mean <span class="op">+</span><span class="st"> </span><span class="fl">1.96</span> <span class="op">-</span><span class="st"> </span>se</span>
<span id="cb73-7"><a href="#cb73-7"></a><span class="kw">c</span>(lower, upper)</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p>
                      Calculating the critical value for a confidence interval:
                      <code>qnorm(0.95)</code>
                    </p>
                  </li>
                  <li>
                    <p>Margin of error:</p>
                    <blockquote>
                      <p>
                        Imagine you’ve set out to survey 1000 people on two
                        questions: are you female? and are you left-handed?
                        Since both of these sample proportions were calculated
                        from the same sample size, they should have the same
                        margin of error, right? Wrong! While the margin of error
                        does change with sample size, it is also affected by the
                        proportion.
                      </p>
                      <p>
                        The first step is to make a vector <code>p</code> that
                        is a sequence from 0 to 1 with each number separated by
                        0.01:
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb74">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1"></a>n =<span class="st"> </span><span class="dv">1000</span></span>
<span id="cb74-2"><a href="#cb74-2"></a>p =<span class="st"> </span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.01</span>)</span></code></pre>
                    </div>
                    <blockquote>
                      <p>
                        We then create a vector of the margin of error
                        (<code>me</code>) associated with each of these values
                        of <code>p</code> using the familiar approximate formula
                        (ME = 2 X SE):
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb75">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1"></a>me =<span class="st"> </span><span class="fl">1.96</span> <span class="op">-</span><span class="st"> </span><span class="kw">sqrt</span>(p <span class="op">-</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>p) <span class="op">/</span><span class="st"> </span>n)</span></code></pre>
                    </div>
                    <pre><code>## Warning in sqrt(p - (1 - p)/n): NaNs produced</code></pre>
                    <blockquote>
                      <p>
                        Finally, plot the two vectors against each other to
                        reveal their relationship:
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb77">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1"></a><span class="kw">plot</span>(me <span class="op">~</span><span class="st"> </span>p)</span></code></pre>
                    </div>
                    <p>
                      <img
                        src="index_files/figure-html/index-stats-confidence-intervals-03-1.png"
                        width="672"
                      />
                    </p>
                  </li>
                </ul>
              </div>
              <div id="probability" class="section level4">
                <h4>Probability</h4>
                <ul>
                  <li>
                    <p>Simulating a coin flip with <code>sample()</code>:</p>
                    <div class="sourceCode" id="cb78">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1"></a>outcomes =<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;heads&#39;</span>, <span class="st">&#39;tails&#39;</span>)</span>
<span id="cb78-2"><a href="#cb78-2"></a><span class="kw">sample</span>(vector_name, <span class="dt">size=</span><span class="dv">100</span>, <span class="dt">replace=</span><span class="ot">TRUE</span>)</span></code></pre>
                    </div>
                    <ul>
                      <li>
                        <p>
                          The code above simulates 100 flips of a fair coin.
                        </p>
                      </li>
                      <li>
                        <p>Add different probability with</p>
                        <div class="sourceCode" id="cb79">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1"></a><span class="kw">sample</span>(vector_name, <span class="dt">size=</span><span class="dv">100</span>, <span class="dt">replace=</span><span class="ot">TRUE</span>,<span class="dt">prob=</span><span class="kw">c</span>(<span class="fl">0.2</span>,<span class="fl">0.8</span>))</span></code></pre>
                        </div>
                      </li>
                      <li>
                        <p>
                          The sample can also be saved as a vector for further
                          analysis (such as calculating the mean, or other
                          statistics)
                        </p>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
              <div
                id="for-loops-bootstrapping-and-inference"
                class="section level4"
              >
                <h4>For loops, bootstrapping, and inference</h4>
                <blockquote>
                  <p>
                    The <strong>for loop</strong> is a cornerstone of computer
                    programming. The idea behind the for loop is
                    <strong>iteration</strong>: it allows you to execute code as
                    many times as you want without having to type out every
                    iteration.
                  </p>
                </blockquote>
                <ul>
                  <li>
                    <p>
                      Example for loop (from DataCamp data analysis and
                      statistical inference course, also see Verzani simpleR
                      p.48):
                    </p>
                    <blockquote>
                      <p>
                        In this example, we want to iterate the two lines of
                        code inside the curly braces that take a random sample
                        of size 50 from area then save the mean of that sample
                        into the sample means50 vector. Without the for loop,
                        this would be painful:
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb80">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1"></a>sample_means50 &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">5000</span>)</span>
<span id="cb80-2"><a href="#cb80-2"></a>samp &lt;-<span class="st"> </span><span class="kw">sample</span>(area, <span class="dv">50</span>)</span>
<span id="cb80-3"><a href="#cb80-3"></a>sample_means50[<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="kw">mean</span>(samp)</span>
<span id="cb80-4"><a href="#cb80-4"></a>samp &lt;-<span class="st"> </span><span class="kw">sample</span>(area, <span class="dv">50</span>)</span>
<span id="cb80-5"><a href="#cb80-5"></a>sample_means50[<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="kw">mean</span>(samp)</span>
<span id="cb80-6"><a href="#cb80-6"></a>samp &lt;-<span class="st"> </span><span class="kw">sample</span>(area, <span class="dv">50</span>)</span>
<span id="cb80-7"><a href="#cb80-7"></a>sample_means50[<span class="dv">3</span>] &lt;-<span class="st"> </span><span class="kw">mean</span>(samp)</span>
<span id="cb80-8"><a href="#cb80-8"></a>samp &lt;-<span class="st"> </span><span class="kw">sample</span>(area, <span class="dv">50</span>)</span>
<span id="cb80-9"><a href="#cb80-9"></a>sample_means50[<span class="dv">4</span>] &lt;-<span class="st"> </span><span class="kw">mean</span>(samp)</span></code></pre>
                    </div>
                    <blockquote>
                      <p>
                        and so on… With the for loop, these thousands of lines
                        of code are compressed into a handful of lines:
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb81">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1"></a><span class="co"># The &#39;ames&#39; data frame and &#39;area&#39; and &#39;price&#39; objects are already loaded</span></span>
<span id="cb81-2"><a href="#cb81-2"></a><span class="co"># Set up an empty vector of 5000 NAs to store sample means:</span></span>
<span id="cb81-3"><a href="#cb81-3"></a>sample_means50 =<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">5000</span>)</span>
<span id="cb81-4"><a href="#cb81-4"></a><span class="co"># Take 5000 samples of size 50 of &#39;area&#39; and store in &#39;sample_means50&#39;.</span></span>
<span id="cb81-5"><a href="#cb81-5"></a><span class="co"># Also print &#39;i&#39; (the index counter):</span></span>
<span id="cb81-6"><a href="#cb81-6"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">5000</span>)</span>
<span id="cb81-7"><a href="#cb81-7"></a>{</span>
<span id="cb81-8"><a href="#cb81-8"></a>  samp =<span class="st"> </span><span class="kw">sample</span>(area, <span class="dv">50</span>)</span>
<span id="cb81-9"><a href="#cb81-9"></a>  sample_means50[i] =<span class="st"> </span><span class="kw">mean</span>(samp)</span>
<span id="cb81-10"><a href="#cb81-10"></a>  <span class="kw">print</span>(i)</span>
<span id="cb81-11"><a href="#cb81-11"></a>}</span></code></pre>
                    </div>
                    <blockquote>
                      <p>
                        In the first line we initialize a vector. In this case,
                        we created a vector of 5000 NAs called
                        <code>sample means50</code>. This vector will store
                        values generated within the for loop. NA means not
                        available, and in this case they’re used as placeholders
                        until we fill in the values with actual sample means. NA
                        is also often used for missing data in R.
                      </p>
                      <p>
                        The second line calls the for loop itself. The syntax
                        can be loosely read as, ‘for every element
                        <code>i</code> from 1 to 5000, run the following lines
                        of code’. You can think of <code>i</code> as the counter
                        that keeps track of which loop you’re on. Therefore,
                        more precisely, the loop will run once when
                        <code>i</code>=1, then once when <code>i</code>=2, and
                        so on up to <code>i</code>=5000.
                      </p>
                      <p>
                        The body of the for loop is the part inside the curly
                        braces, and this set of code is run for each value of
                        <code>i</code>. Here, on every loop, we take a random
                        sample of size 50 from <code>area</code> (note: the
                        sample size can be previously defined by creating a
                        vector), take its mean, and store it as the ith element
                        of <code>sample_means50</code>. In order to display that
                        this is really happening, we asked R to print
                        <code>i</code> at each iteration. This line of code is
                        optional and is only used for displaying what’s going on
                        while the for loop is running.
                      </p>
                      <p>
                        The for loop allows us to not just run the code 5000
                        times, but to neatly package the results, element by
                        element, into the empty vector that we initialized at
                        the outset. You can also run multiple distributions
                        through the same for loop:
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb82">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1"></a><span class="co"># The &#39;ames&#39; data frame and &#39;area&#39; and &#39;price&#39; objects are already loaded</span></span>
<span id="cb82-2"><a href="#cb82-2"></a></span>
<span id="cb82-3"><a href="#cb82-3"></a><span class="co"># Initialize the sample distributions:</span></span>
<span id="cb82-4"><a href="#cb82-4"></a>sample_means10 =<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">5000</span>)</span>
<span id="cb82-5"><a href="#cb82-5"></a>sample_means100 =<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">5000</span>)</span>
<span id="cb82-6"><a href="#cb82-6"></a></span>
<span id="cb82-7"><a href="#cb82-7"></a><span class="co"># Run the for loop:</span></span>
<span id="cb82-8"><a href="#cb82-8"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">5000</span>) {</span>
<span id="cb82-9"><a href="#cb82-9"></a>  samp =<span class="st"> </span><span class="kw">sample</span>(area, <span class="dv">10</span>)</span>
<span id="cb82-10"><a href="#cb82-10"></a>  sample_means10[i] =<span class="st"> </span><span class="kw">mean</span>(samp)</span>
<span id="cb82-11"><a href="#cb82-11"></a>  samp =<span class="st"> </span><span class="kw">sample</span>(area, <span class="dv">100</span>)</span>
<span id="cb82-12"><a href="#cb82-12"></a>  sample_means100[i] =<span class="st"> </span><span class="kw">mean</span>(samp)</span>
<span id="cb82-13"><a href="#cb82-13"></a>}</span></code></pre>
                    </div>
                    <blockquote>
                      <p>
                        Don’t worry about the fact that <code>samp</code> is
                        used for the name of two different objects. In the
                        second command of the for loop, the mean of
                        <code>samp</code> is saved to the relevant place in the
                        vector <code>sample_means10</code>. With the mean saved,
                        we’re now free to overwrite the object samp with a new
                        sample, this time of size 100. In general, anytime you
                        create an object using a name that is already in use,
                        the old object will get replaced with the new one,
                        i.e. R will write over the existing object with the new
                        one, which is something you want to be careful about if
                        you don’t intend to do so.’
                      </p>
                      <p>
                        Another example: Obtain a random sample, calculate the
                        sample’s mean and standard deviation, use these
                        statistics to calculate a confidence interval, repeat
                        steps (1)-(3) 50 times.
                      </p>
                    </blockquote>
                    <div class="sourceCode" id="cb83">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1"></a><span class="co"># The &#39;ames&#39; data frame is already loaded</span></span>
<span id="cb83-2"><a href="#cb83-2"></a></span>
<span id="cb83-3"><a href="#cb83-3"></a><span class="co"># Initialize &#39;samp_mean&#39;, &#39;samp_sd&#39; and &#39;n&#39;:</span></span>
<span id="cb83-4"><a href="#cb83-4"></a>samp_mean =<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">50</span>)</span>
<span id="cb83-5"><a href="#cb83-5"></a>samp_sd =<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dv">50</span>)</span>
<span id="cb83-6"><a href="#cb83-6"></a>n =<span class="st"> </span><span class="dv">60</span></span>
<span id="cb83-7"><a href="#cb83-7"></a></span>
<span id="cb83-8"><a href="#cb83-8"></a><span class="co"># For loop goes here:</span></span>
<span id="cb83-9"><a href="#cb83-9"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>)</span>
<span id="cb83-10"><a href="#cb83-10"></a>{</span>
<span id="cb83-11"><a href="#cb83-11"></a>  samp =<span class="st"> </span><span class="kw">sample</span>(population, n)</span>
<span id="cb83-12"><a href="#cb83-12"></a>  samp_mean[i] =<span class="st"> </span><span class="kw">mean</span>(samp)</span>
<span id="cb83-13"><a href="#cb83-13"></a>  samp_sd[i] =<span class="st"> </span><span class="kw">sd</span>(samp)</span>
<span id="cb83-14"><a href="#cb83-14"></a>}</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p>
                      For loops can also be used for
                      <strong>bootstrapping</strong>.
                    </p>
                    <ul>
                      <li>
                        <p>
                          Bootstrapping is a way of amplifying your data for
                          estimation.
                        </p>
                      </li>
                      <li>
                        <p>Definition (from Google):</p>
                        <blockquote>
                          <p>
                            a technique of loading a program into a computer by
                            means of a few initial instructions that enable the
                            introduction of the rest of the program from an
                            input device.
                          </p>
                        </blockquote>
                      </li>
                      <li>
                        <p>
                          There is also a <code>boot</code> package available
                          for R
                        </p>
                      </li>
                      <li>
                        <p>
                          <strong
                            >Sampling vs. bootstrapping: The sampling
                            distribution is calculated by resampling from the
                            population, the bootstrap distribution is calculated
                            by resampling from the sample.</strong
                          >
                        </p>
                      </li>
                      <li>
                        <p>
                          Example: bootstrap confidence interval for the average
                          weight gained by all mothers during pregnancy. (From
                          Datacamp data analysis and statistical inference
                          course lab 4)
                        </p>
                        <div class="sourceCode" id="cb84">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1"></a><span class="co"># The &#39;nc&#39; data frame is already loaded</span></span>
<span id="cb84-2"><a href="#cb84-2"></a></span>
<span id="cb84-3"><a href="#cb84-3"></a><span class="co"># Initialize the &#39;boot_means&#39; object:</span></span>
<span id="cb84-4"><a href="#cb84-4"></a>boot_means=<span class="kw">rep</span>(<span class="ot">NA</span>,<span class="dv">100</span>)</span>
<span id="cb84-5"><a href="#cb84-5"></a></span>
<span id="cb84-6"><a href="#cb84-6"></a><span class="co"># Insert your for loop:</span></span>
<span id="cb84-7"><a href="#cb84-7"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>) {</span>
<span id="cb84-8"><a href="#cb84-8"></a>  boot_sample =<span class="st"> </span><span class="kw">sample</span>(gained_clean, n, <span class="dt">replace=</span>T)</span>
<span id="cb84-9"><a href="#cb84-9"></a>  boot_means[i] =<span class="st"> </span><span class="kw">mean</span>(boot_sample)</span>
<span id="cb84-10"><a href="#cb84-10"></a>}</span>
<span id="cb84-11"><a href="#cb84-11"></a></span>
<span id="cb84-12"><a href="#cb84-12"></a><span class="co"># Make a histogram of &#39;boot_means&#39;:</span></span>
<span id="cb84-13"><a href="#cb84-13"></a><span class="kw">hist</span>(boot_means)</span></code></pre>
                        </div>
                      </li>
                    </ul>
                  </li>
                </ul>
                <div id="the-inference-function" class="section level5">
                  <h5>The inference function</h5>
                  <ul>
                    <li>
                      <p>
                        Writing a for loop every time you want to calculate a
                        bootstrap interval or run a randomization test is
                        cumbersome. This is a custom function that automates the
                        process. By default the <code>inference</code> function
                        takes 10,000 bootstrap samples (instead of the 100
                        you’ve taken above), creates a bootstrap distribution,
                        and calculates the confidence interval, using the
                        percentile method.
                      </p>
                    </li>
                    <li>
                      <p>Example</p>
                      <div class="sourceCode" id="cb85">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1"></a><span class="co"># The &#39;nc&#39; data frame is already loaded into the workspace</span></span>
<span id="cb85-2"><a href="#cb85-2"></a><span class="co"># Load the &#39;inference&#39; function:</span></span>
<span id="cb85-3"><a href="#cb85-3"></a><span class="kw">load</span>(<span class="kw">url</span>(<span class="st">&#39;http://s3.amazonaws.com/assets.datacamp.com/course/dasi/inference.Rdata&#39;</span>))</span>
<span id="cb85-4"><a href="#cb85-4"></a></span>
<span id="cb85-5"><a href="#cb85-5"></a><span class="co"># Run the inference function:</span></span>
<span id="cb85-6"><a href="#cb85-6"></a><span class="kw">inference</span>(nc<span class="op">$</span>gained, <span class="dt">type=</span><span class="st">&#39;ci&#39;</span>, <span class="dt">method=</span><span class="st">&#39;simulation&#39;</span>,</span>
<span id="cb85-7"><a href="#cb85-7"></a>          <span class="dt">conflevel=</span><span class="fl">0.9</span>, <span class="dt">est=</span><span class="st">&#39;mean&#39;</span>, <span class="dt">boot_method=</span><span class="st">&#39;perc&#39;</span></span>
<span id="cb85-8"><a href="#cb85-8"></a>          )</span></code></pre>
                      </div>
                      <ul>
                        <li>
                          <code>method='se'</code> will perform the method based
                          on the standard error
                        </li>
                        <li>
                          <code>est='median'</code> will create an interval for
                          the median instead of the mean
                        </li>
                      </ul>
                    </li>
                    <li>
                      <p>
                        Example
                        <em
                          >(Datacamp data analysis and statistical inference
                          course lab 4)</em
                        >:
                      </p>
                      <blockquote>
                        <p>
                          use the inference function for evaluating whether
                          there is a difference between the average birth
                          weights of babies born to smoker and non-smoker
                          mothers:
                        </p>
                      </blockquote>
                      <div class="sourceCode" id="cb86">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1"></a><span class="co"># The &#39;nc&#39; data frame is already loaded into the workspace</span></span>
<span id="cb86-2"><a href="#cb86-2"></a><span class="kw">inference</span>(<span class="dt">y =</span> nc<span class="op">$</span>weight, <span class="dt">x =</span> nc<span class="op">$</span>habit,</span>
<span id="cb86-3"><a href="#cb86-3"></a>          <span class="dt">est =</span> <span class="st">&#39;mean&#39;</span>, <span class="dt">type =</span> <span class="st">&#39;ci&#39;</span>, <span class="dt">null =</span> <span class="dv">0</span>,</span>
<span id="cb86-4"><a href="#cb86-4"></a>          <span class="dt">alternative =</span> <span class="st">&#39;twosided&#39;</span>, <span class="dt">method =</span> <span class="st">&#39;theoretical&#39;</span>,</span>
<span id="cb86-5"><a href="#cb86-5"></a>          <span class="dt">order=</span><span class="kw">c</span>(<span class="st">&#39;smoker&#39;</span>,<span class="st">&#39;nonsmoker&#39;</span>)</span>
<span id="cb86-6"><a href="#cb86-6"></a>          )</span></code></pre>
                      </div>
                      <blockquote>
                        <p>
                          The first argument is y, which is the response
                          variable that we are interested in:
                          <code>nc$weight</code>.
                        </p>
                        <p>
                          The second argument is the grouping variable
                          <code>x</code>, which is the explanatory variable
                          multiplied by the grouping variable across the levels
                          of which we’re comparing the average value for the
                          response variable, smokers and non-smokers:
                          <code>nc$habit</code>.
                        </p>
                        <p>
                          The third argument <code>est</code> is the parameter
                          we’re interested in: ‘mean’ (other options are
                          ‘median’, or ‘proportion’.)
                        </p>
                        <p>
                          Next we decide on the type of inference we want: a
                          hypothesis test (‘ht’) or a confidence interval(‘ci’).
                        </p>
                        <p>
                          When performing a hypothesis test, we also need to
                          supply the null value, which in this case is 0, since
                          the null hypothesis sets the two population means
                          equal to each other.
                        </p>
                        <p>
                          The alternative hypothesis can be ‘less’, ‘greater’,
                          or ‘twosided’.
                        </p>
                        <p>
                          The method of inference can be ‘theoretical’ or
                          ‘simulation’ based.
                        </p>
                        <p>
                          The order function changes the order of the variable
                          levels.
                        </p>
                        <p>
                          You can also add <code>success=</code>, which defines
                          a success for survey data-type questions.
                        </p>
                      </blockquote>
                    </li>
                    <li>
                      <p>Another example:</p>
                      <blockquote>
                        <p>
                          Take a subset <code>spain</code> of
                          <code>atheism</code> with only the respondents from
                          Spain. Calculate the proportion of atheists within
                          this subset. Use the <code>inference</code> function
                          on this subset, grouping them by year.
                          <em
                            >(From Datacamp data analysis and statistical
                            inference course lab 4)</em
                          >
                        </p>
                      </blockquote>
                      <div class="sourceCode" id="cb87">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1"></a><span class="co"># Take the &#39;spain&#39; subset:</span></span>
<span id="cb87-2"><a href="#cb87-2"></a>spain =<span class="st"> </span><span class="kw">subset</span>(atheism, atheism<span class="op">$</span>nationality <span class="op">==</span><span class="st"> &#39;Spain&#39;</span>)</span>
<span id="cb87-3"><a href="#cb87-3"></a><span class="co"># Calculate the proportion of atheists from the &#39;spain&#39; subset:</span></span>
<span id="cb87-4"><a href="#cb87-4"></a>proportion =<span class="st"> </span><span class="kw">nrow</span>(<span class="kw">subset</span>(spain,response <span class="op">==</span><span class="st"> &#39;atheist&#39;</span>))<span class="op">/</span><span class="kw">nrow</span>(spain)</span>
<span id="cb87-5"><a href="#cb87-5"></a><span class="co"># Use the inference function:</span></span>
<span id="cb87-6"><a href="#cb87-6"></a><span class="kw">inference</span>(spain<span class="op">$</span>response, spain<span class="op">$</span>year, <span class="dt">est =</span> <span class="st">&#39;proportion&#39;</span>,</span>
<span id="cb87-7"><a href="#cb87-7"></a>  <span class="dt">type =</span> <span class="st">&#39;ci&#39;</span>, <span class="dt">method =</span> <span class="st">&#39;theoretical&#39;</span>, <span class="dt">success =</span> <span class="st">&#39;atheist&#39;</span></span>
<span id="cb87-8"><a href="#cb87-8"></a>  )</span></code></pre>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div id="t-test" class="section level3">
              <h3>T-test</h3>
              <div class="sourceCode" id="cb88">
                <pre
                  class="sourceCode r"
                ><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1"></a><span class="kw">t.test</span>(y<span class="op">~</span>x, <span class="dt">var.equal =</span> <span class="ot">TRUE</span>)</span>
<span id="cb88-2"><a href="#cb88-2"></a><span class="co"># when both variables are numeric</span></span>
<span id="cb88-3"><a href="#cb88-3"></a><span class="kw">t.test</span>(y1,y2, <span class="dt">var.equal =</span> <span class="ot">TRUE</span>)</span>
<span id="cb88-4"><a href="#cb88-4"></a><span class="co"># one-sample t-test, with null hypothesis mean mu=3.</span></span>
<span id="cb88-5"><a href="#cb88-5"></a><span class="kw">t.test</span>(y,<span class="dt">mu=</span><span class="dv">2</span>, <span class="dt">var.equal =</span> <span class="ot">TRUE</span>)</span>
<span id="cb88-6"><a href="#cb88-6"></a><span class="co"># paired t-test</span></span>
<span id="cb88-7"><a href="#cb88-7"></a><span class="kw">t.test</span>(y<span class="op">~</span>x, <span class="dt">paired=</span><span class="ot">TRUE</span>, <span class="dt">var.equal =</span> <span class="ot">TRUE</span>)</span>
<span id="cb88-8"><a href="#cb88-8"></a><span class="co"># one-tailed t-test</span></span>
<span id="cb88-9"><a href="#cb88-9"></a><span class="kw">t.test</span>(y<span class="op">~</span>x, <span class="dt">alternative =</span> <span class="st">&#39;less&#39;</span>, <span class="dt">var.equal =</span> <span class="ot">TRUE</span>)</span>
<span id="cb88-10"><a href="#cb88-10"></a><span class="co"># one-tailed t-test</span></span>
<span id="cb88-11"><a href="#cb88-11"></a><span class="kw">t.test</span>(y<span class="op">~</span>x, <span class="dt">alternative =</span> <span class="st">&#39;greater&#39;</span>, <span class="dt">var.equal =</span> <span class="ot">TRUE</span>)</span></code></pre>
              </div>
              <ul>
                <li>
                  Default t-test assumes heterogeneity of variance and applies a
                  Welch/Satterthwaite df modification. Include
                  <code>var.equal = TRUE</code> to indicate homogeneity of
                  variance.
                </li>
                <li>
                  <a href="http://www.statmethods.net/stats/nonparametric.html"
                    >Nonparametric</a
                  >
                  and
                  <a href="http://www.statmethods.net/stats/resampling.html"
                    >resampling</a
                  >
                  alternatives to t-tests are available.
                </li>
                <li>
                  Also see
                  <ul>
                    <li>
                      <a href="http://www.statmethods.net/stats/ttest.html"
                        >Quick-R</a
                      >
                    </li>
                    <li>Verzani simpleR p.63</li>
                    <li>
                      <a
                        href="http://www.stat.columbia.edu/~martin/W2024/R2.pdf"
                        >T-tests Columbia</a
                      >
                    </li>
                    <li>
                      <a
                        href="https://github.com/br3ndonland/R-guide/tree/master/lfs/CPSC440-Lecture-Ch06-annotated.url"
                        >CPSC 440 Lecture Notes Chapter 06</a
                      >
                      p.16
                    </li>
                    <li>
                      <a
                        href="https://github.com/br3ndonland/R-guide/tree/master/lfs/CPSC440-Lab-07-handout-annotated.url"
                        >CPSC 440 lab 07 notes</a
                      >
                      p.5
                    </li>
                  </ul>
                </li>
              </ul>
              <div id="t-test-assumptions" class="section level4">
                <h4>T-test assumptions</h4>
                <div id="normality" class="section level5">
                  <h5>Normality</h5>
                  <ul>
                    <li>
                      <p>
                        Shapiro-Wilk W test (W&gt;0.9 desired, see flowchart
                        from
                        <a
                          href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                          >PATH 591 lecture 03</a
                        >
                        p.11 in
                        <a href="#linear-regression"
                          >linear regression section</a
                        >):
                      </p>
                      <div class="sourceCode" id="cb89">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1"></a><span class="kw">shapiro.test</span>(dependent_var, options)</span></code></pre>
                      </div>
                    </li>
                    <li>
                      <p>Normal quantile-quantile plot</p>
                      <div class="sourceCode" id="cb90">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1"></a><span class="kw">qqnorm</span>(dependent_var, options)</span>
<span id="cb90-2"><a href="#cb90-2"></a><span class="kw">qqline</span>(dependent_var, options)</span></code></pre>
                      </div>
                      <ul>
                        <li>
                          Note that ggplot2 does not have features for easy
                          creation of q-q plots.
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
                <div id="equality-of-variances" class="section level5">
                  <h5>Equality of variances</h5>
                  <div class="sourceCode" id="cb91">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1"></a><span class="kw">var.test</span>(dependent_var<span class="op">~</span>independent_var)</span></code></pre>
                  </div>
                  <ul>
                    <li>
                      This is identical to the method used in SAS proc ttest,
                      and simply divides one variance by the other.
                    </li>
                    <li>
                      As explained above, if variances are not equal, the
                      Welch/Satterthwaite modification should be used.
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div id="linear-regression" class="section level3">
              <h3>Linear regression</h3>
              <p><em>See R in action ch.8.</em></p>
              <div id="model-syntax-and-functions" class="section level4">
                <h4>Model syntax and functions</h4>
                <div class="sourceCode" id="cb92">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1"></a>model.name &lt;-<span class="st"> </span><span class="kw">lm</span>(model, data, ...)</span></code></pre>
                </div>
                <p>See R in Action tables 8.2 and 8.3 for full syntax info.</p>
                <p>
                  <em
                    >Linear regression formula symbols. Adapted from R in
                    Action.</em
                  >
                </p>
                <ul>
                  <li>
                    <code>~</code>: dependent variable on left, independent
                    variables on right, like <code>y~a+b</code>.
                    -<code>+</code>: Separates independent variables.
                  </li>
                  <li>
                    <code>:</code> Interaction, such as <code>y~a:b</code>,
                    meaning interaction of independent variables a and b.
                  </li>
                  <li>
                    <code>*</code>: All possible combinations of the variables.
                  </li>
                  <li>
                    <code>^</code>: Restricts interactions to a certain degree.
                    The formula <code>y ~ (a + b + c)^2</code> would be read by
                    R as y ~ a + b + c + a:b + a:c + b:c.
                  </li>
                  <li>
                    <code>.</code>: References all independent variables in the
                    data frame.
                  </li>
                  <li><code>-</code>: Subtracts a variable from the model.</li>
                  <li><code>-1</code>: Suppresses the intercept.</li>
                  <li>
                    <code>I()</code>: Allows use of arithmetic within the
                    parentheses.
                  </li>
                </ul>
                <p>
                  <em
                    >Linear regression functions that can be used after
                    specifying the model. Adapted from R in Action.</em
                  >
                </p>
                <ul>
                  <li>
                    <code>aic()</code> or <code>bic()</code>: Prints Aikake’s or
                    Bayes’ Information Criterion.
                  </li>
                  <li><code>anova()</code>: ANOVA table.</li>
                  <li>
                    <code>coefficients()</code>: Shows intercept and slopes for
                    the regression model.
                  </li>
                  <li>
                    <code>confint()</code>: Confidence intervals (95 percent by
                    default).
                  </li>
                  <li><code>fitted()</code>: Lists predicted values.</li>
                  <li>
                    <code>plot()</code>: Plots for evaluating model fit.
                    <em>To select specific plots, add ‘which=’.</em>
                  </li>
                  <li>
                    <code>residuals()</code>: Residuals are basically the
                    deviation from the model fit.
                    <ul>
                      <li>
                        <p>
                          <code>rstandard()</code> for standardized residuals
                          (divided by standard deviation, fitted including the
                          current data point).
                        </p>
                      </li>
                      <li>
                        <p>
                          <code>rstudent()</code> for studentized residuals
                          (divided by standard deviation, but fitted ignoring
                          the current data point).
                        </p>
                      </li>
                      <li>
                        <p>
                          <code>Stdres()</code> and <code>studres()</code> can
                          also be used but require the MASS package (Modern
                          Applied Statistics with S).
                        </p>
                      </li>
                      <li>
                        <p>
                          See
                          <a
                            href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                            >PATH 591 lecture 03</a
                          >
                          for a clear visual explanation of residuals.
                        </p>
                        <div class="figure">
                          <img
                            src="img/residuals.png"
                            title="Visual explanation of residuals"
                            alt=""
                          />
                          <p class="caption">
                            Visual explanation of residuals from
                            <a
                              href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                              >PATH 591 lecture 03</a
                            >
                            p.1
                          </p>
                        </div>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <code>summary()</code>: Summary of the regression model.
                    <em
                      >To view a specific part, such as just the R<sup>2</sup>,
                      add <code>$r.squared</code> or
                      <code>$adj.r.squared</code></em
                    >
                  </li>
                  <li>
                    <code>vcov()</code>: Lists the covariance matrix for model
                    parameters.
                  </li>
                </ul>
              </div>
              <div id="selection-of-regression-models" class="section level4">
                <h4>Selection of regression models</h4>
                <ul>
                  <li>Begin with full model</li>
                  <li>
                    Hierarchical elimination of variables in multiple linear
                    regression: eliminate non-significant variables from the
                    full model, or until there is no further improvement in the
                    adjusted r<sup>2</sup>.
                  </li>
                  <li>
                    Information criteria, such as Akaike’s Information Criterion
                    (AIC) or Bayes Information Criterion (BIC), can be used to
                    select models and covariance stuctures. These criteria rate
                    linear models based on fit and model size. The model with
                    the lowest AIC and/or BIC has the best balance between fit
                    and model size.
                  </li>
                  <li>
                    Further information:
                    <ul>
                      <li>R in Action 8.6</li>
                      <li>Linear Models with R 8.3</li>
                      <li>
                        <a
                          href="https://github.com/br3ndonland/R-guide/tree/master/lfs/CPSC542-Lecture-10-annotated.url"
                          >CPSC 542 Lecture 10</a
                        >
                        pp. 4-7
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
              <div id="linear-regression-assumptions" class="section level4">
                <h4>Linear regression assumptions</h4>
                <div id="normality-of-residuals" class="section level5">
                  <h5>Normality of residuals</h5>
                  <ul>
                    <li>
                      <p>
                        Shapiro-Wilk W statistic: based on mathematics of normal
                        probability plots. Compares variance of distribution to
                        variance expected under normal distribution.
                        Shapiro-Wilk W statistic trumps p value. Normality
                        testing workflow from
                        <a
                          href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                          >PATH 591 lecture 03</a
                        >
                        p.11:
                      </p>
                      <div class="figure">
                        <img
                          src="img/normality-testing-workflow.png"
                          title="Normality testing workflow"
                          alt=""
                        />
                        <p class="caption">
                          Normality testing workflow from
                          <a
                            href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                            >PATH 591 lecture 03</a
                          >
                          p.11
                        </p>
                      </div>
                      <ul>
                        <li>
                          <strong
                            >Make sure you are testing residuals and not raw
                            data. It is acceptable to test normality using
                            studentized residuals</strong
                          >.
                        </li>
                        <li>
                          See SAS for mixed models section 10.2.3 p.418, and
                          ‘Linear Mixed Models: A Practical Guide Using
                          Statistical Software’ by West Welch and Galecki p.105.
                        </li>
                      </ul>
                    </li>
                    <li>
                      <p>
                        Graphical evaluation by normal quantile-quantile plot
                      </p>
                    </li>
                    <li>
                      <p>
                        Skewness: measure of asymmetry in distribution. A
                        perfect normal distribution has skewness=0. Positively
                        skewed distributions=tail to the right, negatively
                        skewed=tail to the left.
                      </p>
                    </li>
                    <li>
                      <p>
                        Kurtosis: another measure of asymmetry in distribution.
                        Leptokurtic distributions (kurtosis&lt;0) will have more
                        observations at the center, platykurtic distributions
                        (kurtosis&gt;0) are spread evenly.
                      </p>
                    </li>
                  </ul>
                </div>
                <div id="independence-of-residuals" class="section level5">
                  <h5>Independence of residuals</h5>
                  <ul>
                    <li>
                      Serial correlation: Durbin-Watson D.
                      <ul>
                        <li>
                          Found in the <code>car</code> and
                          <code>lmtest</code> packages.
                        </li>
                        <li>
                          Use <code>acf(lm_name$residuals)</code>,
                          <code>acf(rstudent(lm_name))</code>, or
                          <code>acf(studres(lm_name))</code>.
                        </li>
                        <li>
                          <code>studres</code> (in the MASS package) will
                          produce an autocorrelation plot. The dashed lines
                          indicate the 95% confidence interval, and correlations
                          ideally lie within the lines.
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
                <div
                  id="homogeneity-of-residual-variances"
                  class="section level5"
                >
                  <h5>Homogeneity of residual variances</h5>
                  <ul>
                    <li>
                      Plots of residuals vs. fitted/predicted values, residuals
                      vs. any independent variable (dot or box plots); absolute
                      value or square root of absolute value of residuals
                      vs. fitted values.
                    </li>
                    <li>
                      Plots of residuals vs predicted values in datasets with
                      HOV would look like a cloud/ball with no correlation. If
                      the plot looks like a megaphone, it suggests a missing
                      term in the model (see ‘strategies to address model
                      violations’ below), or the need to use an unstructured
                      covariance model.
                    </li>
                    <li>Residuals should also sum to zero</li>
                    <li>
                      Spearman rank order correlation between the absolute
                      values of the residuals and the predicted values
                    </li>
                    <li>
                      See below under ‘Variance assessment for multivariate,
                      factorial, and repeated measures designs’ for information
                      on how to assess HOV in complex designs using fit
                      statistics like aic() and bic().
                    </li>
                  </ul>
                </div>
                <div id="linearity-and-additivity" class="section level5">
                  <h5>Linearity and additivity</h5>
                  <ul>
                    <li>
                      Linearity and additivity of the relationship between
                      dependent and independent variables.
                    </li>
                    <li>
                      Assessed with graphical evaluation by plot of residuals vs
                      predicted values.
                    </li>
                  </ul>
                </div>
                <div
                  id="lack-of-influential-observations"
                  class="section level5"
                >
                  <h5>Lack of influential observations</h5>
                  <ul>
                    <li>
                      <p>
                        Cook’s D is a distance statistic that measures change in
                        regression parameter estimates when an observation is
                        deleted.
                      </p>
                      <ul>
                        <li>
                          Cook’s D values are sometimes compared with a critical
                          F value F<sub>0.50</sub>(k,n-k-1) (see
                          <a
                            href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                            >PATH 591 lecture 03</a
                          >
                          p.15).
                        </li>
                        <li>
                          According to R in Action 8.4.3, Cook’s D values
                          greater than 4/(n-k-1), where n is the sample size and
                          k is the number of predictor variables, indicate
                          influential observations, but 1 can also be an
                          appropriate cutoff.
                        </li>
                      </ul>
                    </li>
                    <li>
                      <p>
                        A Bonferroni outlier test can also be performed with the
                        <code>outlierTest()</code> function in the
                        <code>car</code> package.
                      </p>
                    </li>
                    <li>
                      <p>
                        Plots of Cook’s D values and residuals vs leverage allow
                        graphical assessment of influential observations. Add a
                        critical line to the Cook’s Distance plot with:
                      </p>
                      <div class="sourceCode" id="cb93">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1"></a>cookscritical &lt;-<span class="st"> </span><span class="dv">4</span><span class="op">/</span>(<span class="kw">nrow</span>(dataframe)<span class="op">-</span><span class="kw">length</span>(lm1<span class="op">$</span>coefficients)<span class="op">-</span><span class="dv">2</span>)</span>
<span id="cb93-2"><a href="#cb93-2"></a><span class="kw">plot</span>(lmlog, <span class="dt">which=</span><span class="dv">4</span>, <span class="dt">cook.levels=</span>cookscritical)</span>
<span id="cb93-3"><a href="#cb93-3"></a><span class="kw">abline</span>(<span class="dt">h=</span>cookscritical, <span class="dt">lty=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="st">&#39;red&#39;</span>)</span></code></pre>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
              <div
                id="global-validation-of-linear-model-assumptions-with-the-gvlma-package"
                class="section level4"
              >
                <h4>
                  Global validation of linear model assumptions with the
                  <code>gvlma</code> package
                </h4>
                <ul>
                  <li>
                    This is a new and improved approach to validating the
                    assumptions of linear models.
                    <ul>
                      <li>
                        Normally, individual tests for specific assumptions are
                        performed. This increases the type I error rate and
                        masks combinatorial effects of multiple assumption
                        violations (results of one assumption test may be
                        impacted by violations of other assumptions).
                      </li>
                      <li>Assessment by graphical methods is subjective.</li>
                      <li>
                        gvlma provides an integrated global test of all four
                        major assumptions (normality of residuals, independence
                        of residuals, homogeneity of residual variances,
                        linearity) that controls type I error rate, and also
                        provides several plots.
                      </li>
                    </ul>
                  </li>
                  <li>
                    <code>gvlma</code> returns five summary values:
                    <ol style="list-style-type: decimal">
                      <li>
                        Global stat: combination of the statistics for skewness,
                        kurtosis, link function, and heteroscedasticity,
                        representing the global test of all model assumptions.
                      </li>
                      <li>Skewness</li>
                      <li>Kurtosis</li>
                      <li>
                        Link Function: the link function is a feature of all
                        linear models that links the linear model to the mean of
                        the response variable. ‘Identity’ is the link function
                        for a normal distribution. A misspecified link function,
                        meaning that the link function does not match the
                        distribution of the data, will result in a large and
                        significant value for this statistic.
                      </li>
                      <li>
                        Heteroscedasticity: tests the assumption of homogeneity
                        of residual variances.
                      </li>
                    </ol>
                  </li>
                  <li>
                    Further information:
                    <ul>
                      <li>
                        Peña EA, Slate EH. Global Validation of Linear Model
                        Assumptions. <em>J. Am. Stat. Assoc.</em> 101:341
                        (2006). doi:
                        <a href="https://dx.doi.org/10.1198/016214505000000637"
                          >10.1198/016214505000000637</a
                        >.
                      </li>
                      <li>
                        R in Action 8.3, 13, and
                        <a
                          href="http://www.statmethods.net/stats/rdiagnostics.html"
                          >Quick-R regression diagnostics</a
                        >.
                      </li>
                      <li>
                        <a
                          href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                          >PATH 591 lecture 03</a
                        >
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
              <div
                id="strategies-for-addressing-model-violations"
                class="section level4"
              >
                <h4>Strategies for addressing model violations</h4>
                <ul>
                  <li>
                    <strong
                      >It is not appropriate to simply throw out influential
                      data points unless they are due to measurement error or
                      some other problem.</strong
                    >
                  </li>
                  <li>
                    Addition of variables to the model to account for residual
                    variance
                  </li>
                  <li>
                    Transformation of independent or dependent variable
                    <ul>
                      <li>
                        Normalizing transformations (+1 used to avoid undefined
                        values for zeros, log(0+1)=1)
                        <ul>
                          <li>
                            Log: log(y+1) (usually log10 but can also be log2 or
                            loge/ln)
                          </li>
                          <li>Square root:</li>
                          <li>
                            Reciprocal: 1/(y+1) (particularly effective at
                            reducing skewness)
                          </li>
                          <li>
                            See
                            <a
                              href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                              >PATH 591 lecture 03</a
                            >
                            pp.11-15.
                          </li>
                        </ul>
                      </li>
                      <li>
                        Variance-stabilizing transformations
                        <ul>
                          <li>
                            Square root, log, reciprocal (in order of increasing
                            heterogeneity of variance)
                          </li>
                        </ul>
                      </li>
                      <li>
                        Linearizing transformations for curvilinear
                        relationships
                        <ul>
                          <li>
                            Exponential function: linearized by log
                            transformation of y
                          </li>
                          <li>
                            Logarithmic function: linearized by log
                            transformation of x
                          </li>
                          <li>
                            Power function: linearized by log transformation of
                            both x and y
                          </li>
                          <li>
                            See
                            <a
                              href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-04C-annotated.url"
                              >PATH 591 lecture 04 C</a
                            >
                          </li>
                        </ul>
                      </li>
                      <li>
                        Arcsin(y) can correct non-normality or
                        heteroscedasticity for variables that are proportions
                        (1-100%). See
                        <a
                          href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-ProblemSet-04.url"
                          >PATH 591 problem set 4</a
                        >.
                      </li>
                      <li>
                        The independent variable can also be transformed if it
                        is continuous. Categorical independent variables should
                        not be transformed.
                      </li>
                    </ul>
                  </li>
                  <li>
                    The Spearman rank order correlation between the independent
                    and dependent variable is another type of correlation that
                    is more widely valid, especially in situations where F test
                    assumptions are violated.
                  </li>
                  <li>
                    Curvilinearity in a plot of resids vs. preds can be
                    addressed by fitting a polynomial regression function.
                  </li>
                  <li>
                    Weighted least squares regression can be used to address
                    serial correlation or influential observations. See
                    <a
                      href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                      >PATH 591 lecture 03</a
                    >
                    p.14. It allows influential observations to be re-weighted
                    to minimize interference with the model. SAS provides a
                    ‘reweight’ statement for this purpose.
                  </li>
                  <li>
                    Link function violations: as with violations of other
                    assumptions, transformation of the data set, addition of
                    variables to the model, or use of a different model may be
                    necessary. Different link functions, such as binomial or
                    Poisson, can usually be specified with a generalized linear
                    mixed models approach. If a simple correlation is desired,
                    the Spearman test can be used instead.
                  </li>
                </ul>
              </div>
              <div id="reporting-regression-results" class="section level4">
                <h4>Reporting regression results</h4>
                <p>
                  <em>F<sub>0.05</sub>(k, n-1 df)=, r<sup>2</sup>, p&lt;</em>
                </p>
                <p>
                  This information is usually adequate for publication purposes.
                  See
                  <a
                    href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-04A-annotated.url"
                    >PATH 591 lecture 04 A</a
                  >
                  p.8.
                </p>
              </div>
            </div>
            <div id="anova" class="section level3">
              <h3>ANOVA</h3>
              <p>
                <strong
                  >ANOVA is a type of linear regression analysis with
                  categorical independent variables.</strong
                >
                R in Action ch.9 contains an excellent section on ANOVA.
              </p>
              <div id="anova-model-syntax-and-functions" class="section level4">
                <h4>ANOVA model syntax and functions</h4>
                <p><strong>Options for basic ANOVA:</strong></p>
                <ol style="list-style-type: decimal">
                  <li><code>aov(model, data=data.frame)</code></li>
                  <li><code>lm()</code> <code>anova()</code></li>
                  <li><code>glm()</code></li>
                </ol>
                <p>
                  <em
                    >See R in Action table 9.5 for more syntax info and model
                    examples.</em
                  >
                </p>
                <ul>
                  <li>
                    Model syntax is similar to linear regression model syntax.
                  </li>
                  <li>
                    <strong
                      >A regression approach with the <code>lm()</code> function
                      is more flexible than a basic ANOVA analysis with the
                      <code>aov()</code> function.</strong
                    >
                    It allows inclusion of both categorical and quantitative
                    independent variables and enables prediction.
                  </li>
                  <li>
                    The <code>aov()</code> function uses Type I sums of squares,
                    which is not appropriate for imbalanced designs (groups of
                    different sizes). The <code>Anova()</code> function (with a
                    capital A) in the <code>car</code> package allows use of
                    Type II or III sums of squares.
                  </li>
                  <li>
                    The order of terms in the model can sometimes influence the
                    result in certain situations, such as imbalanced designs and
                    designs including covariates. It may also depend on how the
                    variance is partitioned (Type I-III sums of squares). See
                    <a
                      href="https://cran.r-project.org/doc/FAQ/R-FAQ.html#Why-does-the-output-from-anova_0028_0029-depend-on-the-order-of-factors-in-the-model_003f"
                      >CRAN 7.18</a
                    >.
                    <ul>
                      <li>
                        Note on Type III sums of squares from Muenchen R for SAS
                        and SPSS users 23.12 p.431: &gt; The R community has a
                        very strongly held belief that tests based on partial
                        sums of squares can be misleading. One problem with them
                        is that they test the main effect after supposedly
                        partialing out significant interactions. In many
                        circumstances, that does not make much sense. See the
                        paper,
                        <a
                          href="http://www.stats.ox.ac.uk/pub/MASS3/Exegeses.pdf"
                          >Exegesis on Linear Models</a
                        >, by W.N. Venables, for details [39].
                      </li>
                    </ul>
                  </li>
                  <li>
                    <code>glm()</code> uses iteratively reweighted least squares
                    (IWLS).
                    <em>Further information in R in Action chapter 13.</em>
                  </li>
                  <li>
                    If the model contains categorical variables with &gt;2
                    levels, R creates ‘indicator’ or dummy variables for each
                    level.
                  </li>
                </ul>
              </div>
              <div id="anova-assumptions" class="section level4">
                <h4>ANOVA assumptions</h4>
                <div id="normality-of-residuals-1" class="section level5">
                  <h5>Normality of residuals</h5>
                  <ul>
                    <li>
                      <p>
                        Shapiro-Wilk W test (W&gt;0.9 desired, see flowchart
                        from
                        <a
                          href="https://github.com/br3ndonland/R-guide/tree/master/lfs/PATH591-Lecture-03-annotated.url"
                          >PATH 591 lecture 03</a
                        >
                        p.11 in
                        <a href="#linear-regression"
                          >linear regression section</a
                        >):
                      </p>
                      <div class="sourceCode" id="cb94">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1"></a><span class="kw">shapiro.test</span>(dependent_var, options)</span></code></pre>
                      </div>
                    </li>
                    <li>
                      <p>Normal quantile-quantile plot</p>
                      <div class="sourceCode" id="cb95">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1"></a><span class="kw">qqnorm</span>(dependent_var, options)</span>
<span id="cb95-2"><a href="#cb95-2"></a><span class="kw">qqline</span>(dependent_var, options)</span></code></pre>
                      </div>
                    </li>
                  </ul>
                </div>
                <div
                  id="homogeneity-of-residual-variances-1"
                  class="section level5"
                >
                  <h5>Homogeneity of residual variances</h5>
                  <ul>
                    <li>
                      <p>
                        Bartlett test: <code>bartlett.test()</code>. Bartlett
                        test is sensitive to deviations from normality.
                      </p>
                    </li>
                    <li>
                      <p>Levene test</p>
                      <div class="sourceCode" id="cb96">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1"></a><span class="co"># Download and install the car package first</span></span>
<span id="cb96-2"><a href="#cb96-2"></a><span class="kw">library</span>(car)</span>
<span id="cb96-3"><a href="#cb96-3"></a><span class="kw">leveneTest</span>(y<span class="op">~</span>a, <span class="dt">location=</span> <span class="st">&#39;mean&#39;</span>)</span></code></pre>
                      </div>
                    </li>
                    <li>
                      <p>Brown-Forsythe test</p>
                      <div class="sourceCode" id="cb97">
                        <pre
                          class="sourceCode r"
                        ><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1"></a><span class="co"># Download and install the car package first</span></span>
<span id="cb97-2"><a href="#cb97-2"></a><span class="kw">library</span>(car)</span>
<span id="cb97-3"><a href="#cb97-3"></a><span class="kw">leveneTest</span>(y<span class="op">~</span>a)</span></code></pre>
                      </div>
                      <ul>
                        <li>
                          The Brown-Forsythe test evaluates deviations from
                          group medians, and thus is robust to violations of
                          normality.
                        </li>
                      </ul>
                    </li>
                    <li>
                      <p>
                        <strong
                          >These HOV tests are only appropriate for one-way
                          designs. A different approach is required for more
                          complex designs.</strong
                        >
                      </p>
                    </li>
                    <li>
                      <p>
                        <strong>Factorial designs</strong> can be converted to
                        one-way by splitting each treatment combination into a
                        separate group and identifying it as a single level in
                        one independent variable, calling the one new
                        independent variable ‘treatment,’ and then performing
                        the Brown-Forsythe test. This does not work for repeated
                        measures data because values for each group at one
                        timepoint are correlated to values at other timepoints.
                      </p>
                    </li>
                    <li>
                      <p>
                        <strong
                          >Variance assessment for multivariate, factorial, and
                          repeated measures designs</strong
                        >
                      </p>
                      <ul>
                        <li>
                          Graphical assessment as described for linear
                          regression above:
                          <ul>
                            <li>Plot of residuals vs predicted values</li>
                            <li>
                              Plot of residuals vs independent variables (dot or
                              box plots)
                            </li>
                          </ul>
                        </li>
                        <li>
                          Numerical assessment using fit statistics
                          <ul>
                            <li>
                              <p>
                                <em
                                  >I am more familiar with this in SAS.
                                  Explanation below.</em
                                >
                              </p>
                            </li>
                            <li>
                              <p>
                                In general, you want to use the simplest model
                                possible that minimizes AIC, AICC, and BIC while
                                still meeting the assumptions of the model. Mind
                                the assumptions of each covariance structure.
                                See
                                <a
                                  href="https://github.com/br3ndonland/R-guide/tree/master/lfs/CPSC542-Lecture-10-annotated.url"
                                  >CPSC 542 Lecture 10</a
                                >
                                pp. 4-7.
                              </p>
                            </li>
                            <li>
                              <p>
                                Donald Bullock (Illinois CPSC statistics
                                professor):
                              </p>
                              <blockquote>
                                <p>
                                  You select an error structure with
                                  heterogeneous variances [like unstructured]
                                  and compare that to a model with everything
                                  the same with the exception of homogeneous
                                  variances [like compound symmetry]. The
                                  differences in the log likelihoods will be a
                                  chi-square with degrees of freedom equal to
                                  the differences in the number of estimates in
                                  the two models.
                                </p>
                              </blockquote>
                            </li>
                            <li>
                              <p>
                                This is also described in the Littell ‘SAS for
                                mixed models’ text p. 172.
                              </p>
                            </li>
                            <li>
                              <p>
                                In other words, run the code using the
                                unstructured model, and copy/paste the ‘Fit
                                Statistics’ table and the number of covariance
                                parameter estimates into Excel. The number of
                                estimates (also called covariance parameters) is
                                given under ‘Dimensions’ in the output. Also
                                look at the ‘-2 Res Log Likelihood’ number under
                                ‘Fit Statistics’ in the output.
                              </p>
                            </li>
                            <li>
                              <p>
                                Then run the code several more times, using
                                different covariance structures each time (cs,
                                ar(1), arh(1), toep, and toeph), and paste the
                                fit statistics and estimates into excel also.
                                Then, for each test other than unstructured,
                                type =chidist(x, df), using the difference
                                between the ‘-2 Res Log Likelihood’ for
                                unstructured and the other test as the ‘x’ or
                                chi square, and the difference in #parameters
                                between unstructured and the other test as the
                                df.
                              </p>
                            </li>
                            <li>
                              <p>
                                The answer will be a p value. A significant p
                                indicates that the unstructured model provides
                                superior fit.
                              </p>
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
                <div id="independence-of-residuals-1" class="section level5">
                  <h5>Independence of residuals</h5>
                  <p>
                    Independence of residuals is required. Any pairing in the
                    data, such as in repeated measures designs, must be
                    appropriately specified in the model.
                  </p>
                </div>
              </div>
              <div id="anova-multiple-comparisons" class="section level4">
                <h4>ANOVA multiple comparisons</h4>
                <div id="all-pairwise-comparisons" class="section level5">
                  <h5>All pairwise comparisons</h5>
                  <div class="sourceCode" id="cb98">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1"></a><span class="co"># note that this is adjusted for multiple comparisons</span></span>
<span id="cb98-2"><a href="#cb98-2"></a><span class="kw">TukeyHSD</span>(model)</span>
<span id="cb98-3"><a href="#cb98-3"></a><span class="kw">pairwise.t.test</span>(y, a)</span></code></pre>
                  </div>
                  <p>or</p>
                  <div class="sourceCode" id="cb99">
                    <pre
                      class="sourceCode r"
                    ><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1"></a><span class="kw">library</span>(multcomp)</span>
<span id="cb99-2"><a href="#cb99-2"></a><span class="co"># load RColorBrewer for additional color palettes.</span></span>
<span id="cb99-3"><a href="#cb99-3"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb99-4"><a href="#cb99-4"></a>tuk &lt;-<span class="st"> </span><span class="kw">glht</span>(aov1, <span class="dt">linfct=</span><span class="kw">mcp</span>(<span class="dt">var_name=</span><span class="st">&#39;Tukey&#39;</span>))</span>
<span id="cb99-5"><a href="#cb99-5"></a><span class="kw">cld</span>(tuk)</span>
<span id="cb99-6"><a href="#cb99-6"></a><span class="kw">plot</span>(<span class="kw">cld</span>(tuk, <span class="dt">level=</span>.<span class="dv">05</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>,<span class="st">&#39;Pastel1&#39;</span>))</span>
<span id="cb99-7"><a href="#cb99-7"></a><span class="co"># &#39;I&#39;m a very vivacious man... Earth tones are so passe.</span></span>
<span id="cb99-8"><a href="#cb99-8"></a><span class="co"># I&#39;m a pastel man.&#39; Larry David</span></span></code></pre>
                  </div>
                </div>
                <div
                  id="specific-pairwise-comparisons-with-contrasts-and-estimates"
                  class="section level5"
                >
                  <h5>
                    Specific pairwise comparisons with contrasts and estimates
                  </h5>
                  <ul>
                    <li>
                      This is similar to the estimate capability in SAS PROC
                      MIXED and GLIMMIX. Also see R in Action 9.4, Linear Models
                      with R p.180.
                      <ul>
                        <li>
                          <p>
                            Download and install the
                            <code>multcomp</code> package.
                          </p>
                        </li>
                        <li>
                          <p>
                            First, print the covariance matrix with
                            <code>model.matrix(model_name)</code> to identify
                            coefficients for contrasts.
                          </p>
                        </li>
                        <li>
                          <p>
                            Next, use the numbers from error matrix to indicate
                            group comparisons of interest.
                          </p>
                          <div class="sourceCode" id="cb100">
                            <pre
                              class="sourceCode r"
                            ><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1"></a><span class="kw">library</span>(multcomp)</span>
<span id="cb100-2"><a href="#cb100-2"></a>contrast &lt;-<span class="st"> </span><span class="kw">rbind</span>(<span class="st">&#39;contrast name&#39;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>))</span>
<span id="cb100-3"><a href="#cb100-3"></a><span class="kw">summary</span>(<span class="kw">glht</span>(model_name, <span class="dt">linfct=</span><span class="kw">mcp</span>(<span class="dt">x=</span>contrast)))</span></code></pre>
                          </div>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </div>
              <div id="anova-interaction-effects" class="section level4">
                <h4>ANOVA Interaction effects</h4>
                <p>Options for creation of interaction plots:</p>
                <div class="sourceCode" id="cb101">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1"></a><span class="kw">interaction.plot</span>(variable1, variable2, options)</span></code></pre>
                </div>
                <div class="sourceCode" id="cb102">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1"></a><span class="kw">library</span>(effects)</span>
<span id="cb102-2"><a href="#cb102-2"></a><span class="kw">plot</span>(<span class="kw">effect</span>(factorial_term, model_name, xlevels),</span>
<span id="cb102-3"><a href="#cb102-3"></a>     <span class="dt">multiline=</span><span class="ot">TRUE</span></span>
<span id="cb102-4"><a href="#cb102-4"></a>     )</span></code></pre>
                </div>
                <div class="sourceCode" id="cb103">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1"></a><span class="kw">library</span>(gplots)</span>
<span id="cb103-2"><a href="#cb103-2"></a><span class="kw">plotmeans</span>(y <span class="op">~</span><span class="st"> </span><span class="kw">interaction</span>(variable1, variable2, <span class="dt">sep=</span><span class="st">&#39; &#39;</span>),</span>
<span id="cb103-3"><a href="#cb103-3"></a>          <span class="dt">connect=</span><span class="kw">list</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">5</span>),<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>)),</span>
<span id="cb103-4"><a href="#cb103-4"></a>          <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkgreen&#39;</span>),</span>
<span id="cb103-5"><a href="#cb103-5"></a>          <span class="dt">main =</span> <span class="st">&#39;Interaction Plot with 95% CIs&#39;</span>,</span>
<span id="cb103-6"><a href="#cb103-6"></a>          <span class="dt">xlab=</span><span class="st">&#39;Treatment and Dose Combination&#39;</span></span>
<span id="cb103-7"><a href="#cb103-7"></a>          )</span></code></pre>
                </div>
                <div class="sourceCode" id="cb104">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1"></a><span class="kw">library</span>(HH)</span>
<span id="cb104-2"><a href="#cb104-2"></a><span class="kw">interaction2wt</span>(model)</span></code></pre>
                </div>
              </div>
              <div id="covariates-and-repeated-measures" class="section level4">
                <h4>Covariates and repeated measures</h4>
                <p>
                  Using <code>aov()</code> or <code>lm()</code> is the simplest
                  method, but also the least flexible. It assumes sphericity,
                  and thus is not suitable for most real-world datasets. A mixed
                  models approach is required to appropriately model covariance.
                  R in Action does not cover mixed models in depth.
                </p>
                <p>A few options:</p>
                <ul>
                  <li><code>Anova()</code> in the <code>car</code> package</li>
                  <li>
                    <code>lmer()</code> in the <code>lme4</code> package. See
                    <a
                      href="https://stat.ethz.ch/pipermail/r-help/2006-May/094765.html"
                      >CRAN 7.35</a
                    >
                    for discussion of p values in this function.
                  </li>
                  <li>
                    <code>lme()</code> or <code>gls()</code> in the
                    <code>nlme</code> package
                  </li>
                </ul>
                <p>
                  <strong
                    >Analysis of covariance includes both fixed effects
                    (independent variables) and random effects (covariates
                    included to model random variation). If the effect can be
                    reproduced, it’s fixed.</strong
                  >
                  With fixed effects you are interested in the means, but with
                  random effects you are interested in variances. Blocks could
                  be either fixed or random; for example, if you are doing a
                  test on trees and you want to block for forest, setting blocks
                  as random would indicate that you want to block for just all
                  forests everywhere in general and these forests are
                  representative of all forests, whereas setting blocks as fixed
                  would indicate that you want to block for these specific
                  forests you’re evaluating. See
                  <a
                    href="https://github.com/br3ndonland/R-guide/tree/master/lfs/CPSC440-Lecture-Ch08-annotated.url"
                    >CPSC 440 Lecture Notes Chapter 08 ANOVA</a
                  >.
                </p>
                <p>
                  <strong
                    >The covariance structure must be appropriately
                    specified.</strong
                  >
                  Fit statistics and information criteria can be used to
                  identify an appropriate covariance structure:
                  <code>AIC(model)</code>. See ‘Selection of regression models’
                  in the Linear regression section, and ‘Variance assessment for
                  multivariate, factorial, and repeated measures designs’ in the
                  ANOVA section above.
                </p>
              </div>
            </div>
            <div id="non-parametric-tests" class="section level3">
              <h3>Non-parametric tests</h3>
              <div id="fishers-exact-test" class="section level4">
                <h4>Fisher’s exact test</h4>
                <ul>
                  <li>
                    <p>
                      Useful when both dependent and independent variables are
                      categorical, or when dependent variable is a count.
                    </p>
                  </li>
                  <li>
                    <p>
                      It evaluates the association, or contingency, between two
                      kinds of classifications.
                    </p>
                  </li>
                  <li>
                    <p>Data must first be put into a contingency table:</p>
                    <div class="sourceCode" id="cb105">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1"></a>table_name &lt;-<span class="st"> </span><span class="kw">xtabs</span>(formula, <span class="dt">data=</span>df_name)</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p>Next, run the test and reference the table</p>
                    <div class="sourceCode" id="cb106">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1"></a><span class="kw">fisher.test</span>(mytable)</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p>
                      See R in action ch. 7.2, Crawley The R Book ch. 14-15,
                      <a
                        href="http://www.pmc.ucsc.edu/~mclapham/Rtips/categorical.htm"
                        >UCSC R tips</a
                      >.
                    </p>
                  </li>
                  <li>
                    <p>
                      Example from my grad school lab: Prostate cancer incidence
                      data from TRAMP Tomato Soy paper
                    </p>
                    <ul>
                      <li>
                        Zuniga KE, Clinton SK, Erdman JW. The interactions of
                        dietary tomato powder and soy germ on prostate
                        carcinogenesis in the TRAMP model.
                        <em>Cancer Prev. Res.</em> (2013). DOI:
                        <a href="https://doi.org/10.1158/1940-6207.CAPR-12-0443"
                          >10.1158/1940-6207.CAPR-12-0443</a
                        >
                      </li>
                      <li>
                        Code located in
                        <a
                          href="https://github.com/br3ndonland/R-guide/code.html"
                          >R-sample-code/fishers-exact</a
                        >.
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
              <div id="wilcoxon-rank-sum" class="section level4">
                <h4>Wilcoxon rank-sum</h4>
                <div class="sourceCode" id="cb107">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1"></a><span class="kw">wilcox.test</span>(y<span class="op">~</span>x, data)</span></code></pre>
                </div>
                <ul>
                  <li>Also known as the Mann-Whitney U test</li>
                  <li>
                    Used to compare two non-parametric independent samples
                  </li>
                  <li>
                    Note that, although the Wilcoxon rank sum test does not
                    require normality, it does require homogeneity of variances
                    and independence of random samples.
                  </li>
                  <li>See R in Action 7.5</li>
                </ul>
              </div>
              <div id="wilcoxon-signed-rank" class="section level4">
                <h4>Wilcoxon signed rank</h4>
                <div class="sourceCode" id="cb108">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1"></a><span class="kw">wilcox.test</span>(y<span class="op">~</span>x, data, <span class="dt">paired=</span><span class="ot">TRUE</span>)</span></code></pre>
                </div>
                <ul>
                  <li>Used to compare two non-parametric paired samples</li>
                  <li>See R in Action 7.5, Verzani simpleR p.67</li>
                </ul>
              </div>
              <div id="kruskal-wallis" class="section level4">
                <h4>Kruskal-Wallis</h4>
                <div class="sourceCode" id="cb109">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1"></a><span class="kw">kruskal_test</span>(y<span class="op">~</span>a, data)</span></code></pre>
                </div>
                <p>
                  Non-parametric one-way ANOVA, where <code>y</code> is a
                  dependent variable and <code>a</code> is a categorical
                  independent variable with ≥2 levels.
                </p>
                <div id="kruskal-wallis-assumptions" class="section level5">
                  <h5>Kruskal-Wallis assumptions</h5>
                  <div id="non-normality" class="section level6">
                    <h6>Non-normality</h6>
                    <p>
                      A non-parametric Kruskal-Wallis test is used when data
                      fail the assumption of normality, and are not amenable to
                      transformation. It is still worthwhile to perform the
                      usual normality testing to demonstrate this.
                    </p>
                  </div>
                  <div id="homogeneity-of-variances" class="section level6">
                    <h6>Homogeneity of variances</h6>
                    <p>
                      HOV is still required. Use a test that is not affected by
                      non-normality, such as Brown-Forsythe.
                    </p>
                  </div>
                </div>
                <div
                  id="kruskal-wallis-multiple-comparisons-post-hoc-tests"
                  class="section level5"
                >
                  <h5>Kruskal-Wallis multiple comparisons post-hoc tests</h5>
                  <div id="pairwise-wilcoxon-testing" class="section level6">
                    <h6>Pairwise Wilcoxon testing</h6>
                    <div class="sourceCode" id="cb110">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1"></a><span class="kw">pairwise.wilcox.test</span>(y, x, <span class="dt">p.adjust.method=</span> <span class="st">&#39;holm&#39;</span>)</span></code></pre>
                    </div>
                    <ul>
                      <li>
                        <p>
                          This function allows control of the experiment-wide
                          error rate and is the best non-parametric multiple
                          comparisons method available in R.
                        </p>
                      </li>
                      <li>
                        <p>Adjustment options:</p>
                        <ul>
                          <li><code>holm</code></li>
                          <li><code>hochberg</code></li>
                          <li><code>hommel</code></li>
                          <li><code>bonferroni</code></li>
                          <li><code>BH</code></li>
                          <li><code>BY</code></li>
                          <li><code>fdr</code></li>
                          <li><code>none</code></li>
                        </ul>
                      </li>
                      <li>
                        <p>From R Documentation:</p>
                        <blockquote>
                          <p>
                            The first four methods are designed to give strong
                            control of the family-wise error rate. There seems
                            no reason to use the unmodified Bonferroni
                            correction because it is dominated by Holm’s method,
                            which is also valid under arbitrary assumptions.
                          </p>
                        </blockquote>
                      </li>
                    </ul>
                  </div>
                  <div id="nemenyi-test" class="section level6">
                    <h6>Nemenyi test</h6>
                    <div class="sourceCode" id="cb111">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1"></a><span class="kw">library</span>(PMCMRplus)</span>
<span id="cb111-2"><a href="#cb111-2"></a><span class="kw">kwAllPairsNemenyiTest</span>(y, x, <span class="dt">method=</span> <span class="kw">c</span>(<span class="st">&#39;Tukey&#39;</span>, <span class="st">&#39;Chisq&#39;</span>))</span></code></pre>
                    </div>
                    <ul>
                      <li>
                        Requires the <code>PMCMRplus</code> package (Pairwise
                        Multiple Comparisons of Mean Rank Sums). Note that the
                        original <code>PMCMR</code> package has been deprecated.
                      </li>
                    </ul>
                  </div>
                  <div id="nparcomp-package" class="section level6">
                    <h6><code>nparcomp</code> package</h6>
                    <div class="sourceCode" id="cb112">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1"></a><span class="kw">library</span>(nparcomp)</span>
<span id="cb112-2"><a href="#cb112-2"></a><span class="kw">nparcomp</span>(y<span class="op">~</span>a, data, <span class="dt">type=</span>)</span></code></pre>
                    </div>
                  </div>
                  <div id="npmc-package" class="section level6">
                    <h6><code>npmc</code> package</h6>
                    <p>
                      There used to be an npmc package that is referred to in
                      some documentation. It has been deprecated.
                    </p>
                  </div>
                </div>
              </div>
              <div id="friedman" class="section level4">
                <h4>Friedman</h4>
                <div class="sourceCode" id="cb113">
                  <pre
                    class="sourceCode r"
                  ><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1"></a><span class="kw">friedman.test</span>(y<span class="op">~</span>a <span class="op">|</span><span class="st"> </span>b, data)</span></code></pre>
                </div>
                <ul>
                  <li>
                    where <code>y</code> is the numeric outcome variable,
                    <code>A</code> is a grouping variable, and <code>B</code> is
                    a blocking variable that identifies matched observations. In
                    both cases, <code>data</code> is an option argument
                    specifying a matrix or data frame containing the variables.
                  </li>
                  <li>
                    Useful for non-parametric non-independent data with ≥2
                    groups (such as repeated measures designs).
                  </li>
                  <li>May only be valid with balanced designs.</li>
                </ul>
                <div
                  id="non-parametric-factorial-datasets"
                  class="section level5"
                >
                  <h5>Non-parametric factorial datasets</h5>
                  <p>
                    Try the <code>nlme</code>, <code>MultNonParam</code>,
                    <code>npmlreg</code>, or <code>nparLD</code> packages.
                  </p>
                </div>
              </div>
            </div>
            <div id="power-analysis" class="section level3">
              <h3>Power analysis</h3>
              <p>See R in Action ch.10</p>
            </div>
            <div id="principal-component-analysis" class="section level3">
              <h3>Principal Component Analysis</h3>
              <p>See R in Action ch.14</p>
            </div>
            <div id="bioconductor" class="section level3">
              <h3>Bioconductor</h3>
              <p>
                Bioconductor is a repository of R packages for bioinformatic
                analysis.
              </p>
              <div id="basics" class="section level4">
                <h4>Basics</h4>
                <ul>
                  <li>
                    <p>
                      Install core packages (run the commands sequentially).
                      These commands can also be used to update packages.
                    </p>
                    <div class="sourceCode" id="cb114">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1"></a><span class="kw">source</span>(<span class="st">&#39;http://bioconductor.org/biocLite.R&#39;</span>)</span>
<span id="cb114-2"><a href="#cb114-2"></a><span class="kw">biocLite</span>()</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p>Install specific packages:</p>
                    <div class="sourceCode" id="cb115">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1"></a><span class="kw">biocLite</span>(<span class="kw">c</span>(<span class="st">&#39;GenomicFeatures&#39;</span>, <span class="st">&#39;AnnotationDbi&#39;</span>))</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p>
                      Troubleshoot installations to flag packages that are
                      either out-of-date or too new for your version of
                      Bioconductor. The output suggests ways to solve identified
                      problems, and the help page <code>?biocValid</code> lists
                      arguments influencing the behavior of the function.
                    </p>
                    <div class="sourceCode" id="cb116">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1"></a><span class="kw">library</span>(BiocInstaller)</span>
<span id="cb116-2"><a href="#cb116-2"></a><span class="kw">biocValid</span>() <span class="co"># R version 3.0 or later</span></span></code></pre>
                    </div>
                  </li>
                </ul>
              </div>
              <div id="ddct" class="section level4">
                <h4>ddCt</h4>
                <ul>
                  <li>
                    <p>
                      The ddCt package for Bioconductor performs automated ΔΔCt
                      calculations on qPCR data. Install with
                    </p>
                    <div class="sourceCode" id="cb117">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1"></a><span class="kw">source</span>(<span class="st">&#39;http://bioconductor.org/biocLite.R&#39;</span>)</span>
<span id="cb117-2"><a href="#cb117-2"></a><span class="kw">biocLite</span>(<span class="st">&#39;ddCt&#39;</span>)</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p>
                      It seems to require a very specific format for the files
                      output from SDS, and I was not able to get it to work with
                      my data. It would be far easier to create a VBA macro in
                      Excel.
                    </p>
                  </li>
                </ul>
              </div>
              <div id="ppistats" class="section level4">
                <h4>ppiStats</h4>
                <ul>
                  <li>
                    ppiStats calculates summary statistics and makes graphs for
                    AP-MS or hybrid assay data. Not very useful.
                  </li>
                </ul>
              </div>
              <div id="proteomics" class="section level4">
                <h4>Proteomics</h4>
                <ul>
                  <li>
                    I walked through the
                    <a
                      href="https://master.bioconductor.org/packages/release/workflows/html/proteomics.html"
                      >‘Mass spectrometry and proteomics data analysis’</a
                    >
                    workflow in 2014. Running the code (at least all at once)
                    was only partially successful.
                  </li>
                </ul>
                <p><a href="#top">back to top</a></p>
              </div>
            </div>
          </div>
          <div id="plots" class="section level2">
            <h2>Plots</h2>
            <div id="helpful-resources-for-plotting" class="section level3">
              <h3>Helpful resources for plotting</h3>
              <ul>
                <li><strong>R in Action</strong></li>
                <li>
                  <a
                    href="http://proquest.safaribooksonline.com/book/programming/r/9781449363086"
                    >R Graphics Cookbook</a
                  >
                  by Winston Chang (mostly for ggplot2).
                  <em
                    >CC0 1.0 license (public domain dedication, no
                    copyright).</em
                  >
                </li>
              </ul>
            </div>
            <div id="base-graphics" class="section level3">
              <h3>Base graphics</h3>
              <div id="plot-types" class="section level4">
                <h4>Plot types</h4>
                <ul>
                  <li>
                    <p>Scatterplots: <code>plot(x=,y=,options)</code></p>
                  </li>
                  <li>
                    <p>
                      If your data frame only has two variables, you only need
                      to specify the data frame: <code>plot(data.frame)</code>
                    </p>
                  </li>
                  <li>
                    <p>
                      To select specific variables from a data frame to display
                      on the graph, use
                      <code
                        >plot(x=data.frame$variable, y=
                        data.frame$variable)</code
                      >
                    </p>
                  </li>
                  <li>
                    <p>Line chart: <code>plot(x,y,type='l')</code></p>
                  </li>
                  <li>
                    <p>
                      Plotting &gt;2 variables:
                      <code>plot(data.frame[rows,columns])</code>
                    </p>
                  </li>
                  <li>
                    <p><strong>Boxplot</strong>: <code>boxplot()</code></p>
                    <ul>
                      <li>
                        <p>See the multiple comparisons ANOVA section above</p>
                      </li>
                      <li>
                        <p>
                          Plotting one variable as a function of another is very
                          similar to creating a linear model:
                          <code>boxplot(y ~ a)</code>
                        </p>
                        <ul>
                          <li>
                            As usual, if you <code>attach()</code> the dataset
                            you do not need to specify the data frame name here.
                          </li>
                          <li>
                            <strong
                              >I found base graphics actually work better for
                              boxplots than ggplot2.</strong
                            >
                          </li>
                        </ul>
                      </li>
                      <li>
                        <p>
                          New variables can also be calculated and plotted even
                          if they are not in the original data frame:
                        </p>
                        <div class="sourceCode" id="cb118">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1"></a>new_calculation=vector<span class="op">+</span>vector</span>
<span id="cb118-2"><a href="#cb118-2"></a><span class="kw">boxplot</span>(new_calculation<span class="op">~</span>df_name<span class="op">$</span>var_name)</span></code></pre>
                        </div>
                      </li>
                      <li>
                        <p>
                          Setting line widths for individual parts of the
                          boxplot:
                        </p>
                        <ul>
                          <li>box outline: <code>lwd=</code></li>
                          <li>median line: <code>medlwd=</code></li>
                          <li>outlier points: <code>outlwd=</code></li>
                        </ul>
                      </li>
                      <li>
                        <p>Adding symbols for group means:</p>
                        <ul>
                          <li>
                            <p>
                              After creating the boxplot, create a new vector
                              and use the <code>aggregate()</code> function:
                              `means &lt;- aggregate(model, data.frame, mean)
                            </p>
                          </li>
                          <li>
                            <p>
                              Next, create a separate
                              <code>points()</code> function:
                            </p>
                            <div class="sourceCode" id="cb119">
                              <pre
                                class="sourceCode r"
                              ><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1"></a><span class="kw">points</span>(means<span class="op">$</span>dependent_var, <span class="dt">cex=</span><span class="dv">2</span>, <span class="dt">pch=</span><span class="dv">23</span>, <span class="dt">bg=</span><span class="dv">0</span>, <span class="dt">lwd=</span><span class="dv">4</span>)</span></code></pre>
                            </div>
                          </li>
                          <li>
                            <p>
                              <strong
                                >Note that, as with other functions in R,
                                variables will be sorted alphabetically or
                                numerically. To change the order, you must
                                create a separate vector and use
                                <code>order()</code> or <code>factor()</code> to
                                specify the order. See the section on data
                                frames above, and previous code.</strong
                              >
                            </p>
                          </li>
                        </ul>
                      </li>
                      <li>
                        <p>
                          Plotting factorial datasets: Sample code from the R
                          Graphics Cookbook ch.2 ‘Creating a box plot’.
                        </p>
                        <div class="sourceCode" id="cb120">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1"></a><span class="co"># Formula syntax</span></span>
<span id="cb120-2"><a href="#cb120-2"></a><span class="kw">boxplot</span>(len <span class="op">~</span><span class="st"> </span>supp, <span class="dt">data =</span> ToothGrowth)</span></code></pre>
                        </div>
                        <p>
                          <img
                            src="index_files/figure-html/index-plots-boxplot-01-1.png"
                            width="672"
                          />
                        </p>
                        <div class="sourceCode" id="cb121">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1"></a><span class="co"># Put interaction of two variables on x-axis</span></span>
<span id="cb121-2"><a href="#cb121-2"></a><span class="kw">boxplot</span>(len <span class="op">~</span><span class="st"> </span>supp <span class="op">+</span><span class="st"> </span>dose, <span class="dt">data =</span> ToothGrowth)</span></code></pre>
                        </div>
                        <p>
                          <img
                            src="index_files/figure-html/index-plots-boxplot-01-2.png"
                            width="672"
                          />
                        </p>
                        <p>
                          With the ggplot2 package, you can get a similar result
                          using qplot() (Figure 2-11), with geom=‘boxplot’:
                        </p>
                        <div class="sourceCode" id="cb122">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb122-2"><a href="#cb122-2"></a><span class="kw">qplot</span>(ToothGrowth<span class="op">$</span>supp, ToothGrowth<span class="op">$</span>len, <span class="dt">geom =</span> <span class="st">&#39;boxplot&#39;</span>)</span></code></pre>
                        </div>
                        <p>
                          <img
                            src="index_files/figure-html/index-plots-ggplot2-01-1.png"
                            width="672"
                          />
                        </p>
                        <p>
                          If the two vectors are already in the same data frame,
                          you can use the following syntax:
                        </p>
                        <div class="sourceCode" id="cb123">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1"></a><span class="kw">qplot</span>(supp, len, <span class="dt">data =</span> ToothGrowth, <span class="dt">geom =</span> <span class="st">&#39;boxplot&#39;</span>)</span></code></pre>
                        </div>
                        <p>
                          <img
                            src="index_files/figure-html/index-plots-qplot-01-1.png"
                            width="672"
                          />
                        </p>
                        <p>This is equivalent to:</p>
                        <div class="sourceCode" id="cb124">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1"></a><span class="kw">ggplot</span>(ToothGrowth, <span class="kw">aes</span>(<span class="dt">x =</span> supp, <span class="dt">y =</span> len)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()</span></code></pre>
                        </div>
                        <p>
                          <img
                            src="index_files/figure-html/index-plots-ggplot2-02-1.png"
                            width="672"
                          />
                        </p>
                        <p>
                          It’s also possible to make box plots for multiple
                          variables, by combining the variables with
                          interaction(), as in Figure 2-11. In this case, the
                          dose variable is numeric, so we must convert it to a
                          factor to use it as a grouping variable:
                        </p>
                        <div class="sourceCode" id="cb125">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1"></a><span class="co"># Using three separate vectors</span></span>
<span id="cb125-2"><a href="#cb125-2"></a><span class="kw">qplot</span>(<span class="kw">interaction</span>(ToothGrowth<span class="op">$</span>supp, ToothGrowth<span class="op">$</span>dose),</span>
<span id="cb125-3"><a href="#cb125-3"></a>      ToothGrowth<span class="op">$</span>len,</span>
<span id="cb125-4"><a href="#cb125-4"></a>      <span class="dt">geom =</span> <span class="st">&#39;boxplot&#39;</span>)</span></code></pre>
                        </div>
                        <p>
                          <img
                            src="index_files/figure-html/index-plots-qplot-02-1.png"
                            width="672"
                          />
                        </p>
                        <p>
                          Alternatively, get the columns from the data frame:
                        </p>
                        <div class="sourceCode" id="cb126">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1"></a><span class="kw">qplot</span>(<span class="kw">interaction</span>(supp, dose), len, <span class="dt">data =</span> ToothGrowth, <span class="dt">geom =</span> <span class="st">&#39;boxplot&#39;</span>)</span></code></pre>
                        </div>
                        <p>
                          <img
                            src="index_files/figure-html/index-plots-qplot-03-1.png"
                            width="672"
                          />
                        </p>
                        <p>This is equivalent to:</p>
                        <div class="sourceCode" id="cb127">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1"></a><span class="kw">ggplot</span>(ToothGrowth, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">interaction</span>(supp, dose), <span class="dt">y =</span> len)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()</span></code></pre>
                        </div>
                        <p>
                          <img
                            src="index_files/figure-html/index-plots-ggplot2-03-1.png"
                            width="672"
                          />
                        </p>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <p>Dot plot</p>
                    <ul>
                      <li>See R Graphics Cookbook Ch. 6.</li>
                      <li>
                        Can make dot plots with box plots underlaid or alongside
                        using ggplot2.
                      </li>
                    </ul>
                  </li>
                  <li>
                    <p>Histogram: <code>hist()</code></p>
                    <ul>
                      <li>
                        <p>
                          Histograms are useful for inspecting the distribution
                          of the dataset.
                        </p>
                      </li>
                      <li>
                        <p>
                          To control histogram divisions and limits of x axis:
                        </p>
                        <div class="sourceCode" id="cb128">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1"></a><span class="kw">hist</span>(df_name<span class="op">$</span>var_name,<span class="dt">breaks=</span>number,<span class="dt">xlim=</span>number or variable you created)</span></code></pre>
                        </div>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <p>
                      Combined boxplot and histogram:
                      <code>simple.hist.and.boxplot()</code>
                    </p>
                  </li>
                  <li>
                    <p>Bar chart: <code>barplot()</code></p>
                    <ul>
                      <li>
                        You can nest the creation of a table into the barplot
                        function: `barplot(table(data.frame$variable))
                      </li>
                    </ul>
                  </li>
                  <li>
                    <p>Pie chart: <code>pie(data.frame)</code></p>
                  </li>
                  <li>
                    <p>
                      Mosaic plot (useful for showing how different categories
                      are related):
                    </p>
                    <div class="sourceCode" id="cb129">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1"></a><span class="kw">mosaicplot</span>(data.frame<span class="op">$</span>variable1<span class="op">~</span>data.frame<span class="op">$</span>variable2)</span></code></pre>
                    </div>
                  </li>
                  <li>
                    <p><strong>Volcano plots</strong></p>
                    <ul>
                      <li>
                        <a
                          href="http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html"
                          >Volcano plot example using genetics data</a
                        >
                      </li>
                      <li>
                        <a
                          href="https://github.com/br3ndonland/R-proteomics-Nrf1"
                          >Volcano plot example with unpublished proteomics data
                          from my postdoc</a
                        >
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
              <div id="graphical-parameters" class="section level4">
                <h4>Graphical parameters</h4>
                <p>
                  See
                  <a href="http://www.statmethods.net/advgraphs/parameters.html"
                    >Quick-R</a
                  >, and of course R in Action, for thorough documentation.
                </p>
                <ul>
                  <li>
                    <p>
                      Many graphical options are specified with the
                      <code>par()</code> function.
                    </p>
                    <ul>
                      <li>Size of text and symbols: <code>cex=</code></li>
                      <li>Symbol style: <code>pch=</code></li>
                      <li>Line type: <code>lty=</code></li>
                      <li>Line width: <code>lwd=</code></li>
                    </ul>
                  </li>
                  <li>
                    <p>Colors</p>
                    <ul>
                      <li>
                        <p>
                          Colors can be specified by index number, name,
                          hexadecimal, or RGB.
                        </p>
                      </li>
                      <li>
                        <p>
                          See Earl Glynn’s color chart for a visual listing of
                          colors in R by index number.
                        </p>
                        <div class="figure">
                          <img
                            src="http://research.stowers.org/mcm/efg/R/Color/Chart/ColorsChart1.jpg"
                            title="R colors"
                            alt=""
                          />
                          <p class="caption">R colors</p>
                        </div>
                      </li>
                      <li>
                        <p>
                          View all available color names: <code>color()</code>
                        </p>
                      </li>
                      <li>
                        <p>
                          The basic palette consists of 8 colors. Other color
                          vectors with n contiguous colors can be created with
                          <code>gray(n)</code>, <code>rainbow(n)</code>,
                          <code>heat.colors(n)</code>,
                          <code>terrain.colors(n)</code>,
                          <code>topo.colors(n)</code>, and
                          <code>cm.colors(n)</code>.
                        </p>
                      </li>
                      <li>
                        <p>
                          The <code>RColorBrewer</code> package can also be used
                          for color palettes. See
                          <a
                            href="http://colorbrewer2.org/#type=sequential&amp;scheme=BuGn&amp;n=3"
                            >website</a
                          >
                          for more info.
                        </p>
                        <ul>
                          <li>
                            <p>
                              To make palettes available in R:
                              <code
                                >palette.name &lt;- brewer.pal(n, 'name')</code
                              >. Next, when selecting a color palette for a
                              chart, use the vector name (<code
                                >palette.name</code
                              >
                              in this example).
                            </p>
                          </li>
                          <li>
                            <p>
                              There are 3 types of palettes, sequential,
                              diverging, and qualitative.
                            </p>
                            <ol style="list-style-type: decimal">
                              <li>
                                Sequential palettes are suited to ordered data
                                that progress from low to high. Lightness steps
                                dominate the look of these schemes, with light
                                colors for low data values to dark colors for
                                high data values. The sequential palette names
                                are: Blues BuGn BuPu GnBu Greens Greys Oranges
                                OrRd PuBu PuBuGn PuRd Purples RdPu Reds YlGn
                                YlGnBu YlOrBr YlOrRd. All the sequential
                                palettes are available in variations from 3
                                different values up to 9 different values.
                              </li>
                              <li>
                                Diverging palettes put equal emphasis on
                                mid-range critical values and extremes at both
                                ends of the data range. The critical class or
                                break in the middle of the legend is emphasized
                                with light colors and low and high extremes are
                                emphasized with dark colors that have
                                contrasting hues. The diverging palettes are
                                BrBG PiYG PRGn PuOr RdBu RdGy RdYlBu RdYlGn
                                Spectral. All the diverging palettes are
                                available in variations from 3 different values
                                up to 11 different values.
                              </li>
                              <li>
                                Qualitative palettes do not imply magnitude
                                differences between legend classes, and hues are
                                used to create the primary visual differences
                                between classes. Qualitative schemes are best
                                suited to representing nominal or categorical
                                data. The qualitative palettes are: Accent,
                                Dark2, Paired, Pastel1, Pastel2, Set1, Set2,
                                Set3.
                              </li>
                            </ol>
                          </li>
                          <li>
                            <p>To view all palettes as a plot:</p>
                            <div class="sourceCode" id="cb130">
                              <pre
                                class="sourceCode r"
                              ><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1"></a><span class="kw">display.brewer.all</span>(<span class="dt">n=</span><span class="ot">NULL</span>, <span class="dt">type=</span><span class="st">&#39;all&#39;</span>, <span class="dt">select=</span><span class="ot">NULL</span>, <span class="dt">exact.n=</span><span class="ot">TRUE</span>, <span class="dt">colorblindFriendly=</span><span class="ot">FALSE</span>)</span></code></pre>
                            </div>
                          </li>
                          <li>
                            <p>
                              Use bracketed numbers to select specific colors or
                              a range from within a palette:
                              <code>col=set1[5:9]</code>. Note that colors will
                              be repeated if necessary. To start from a specific
                              color in the palette, but still retain all colors
                              in the palette, type the number for the starting
                              color and then add that to the total number of
                              colors in the palette.
                            </p>
                          </li>
                          <li>
                            <p>For Set1: <code>col=set1[5:14]</code></p>
                          </li>
                          <li>
                            <p>
                              To color a plot by category (from
                              <a
                                href="http://stackoverflow.com/questions/7466023/how-to-give-color-to-each-class-in-scatter-plot-in-r"
                                >Stackoverflow</a
                              >)
                            </p>
                            <div class="sourceCode" id="cb131">
                              <pre
                                class="sourceCode r"
                              ><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1"></a>DF &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dt">y=</span><span class="kw">rnorm</span>(<span class="dv">10</span>)<span class="op">+</span><span class="dv">5</span>, <span class="dt">z=</span><span class="kw">sample</span>(letters[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], <span class="dv">10</span>, <span class="dt">replace=</span><span class="ot">TRUE</span>))</span>
<span id="cb131-2"><a href="#cb131-2"></a><span class="kw">attach</span>(DF); <span class="kw">plot</span>(x, y, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&#39;red&#39;</span>,<span class="st">&#39;blue&#39;</span>,<span class="st">&#39;green&#39;</span>)[z]); <span class="kw">detach</span>(DF)</span></code></pre>
                            </div>
                            <ul>
                              <li>
                                The method for this may vary depending on what
                                is being created (scatter plot, figure legend,
                                boxplot) and how categories have been specified.
                              </li>
                            </ul>
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                  <li><p>Fonts: See Quick-R.</p></li>
                  <li>
                    <p>Legend: <code>legend()</code>. See Quick-R.</p>
                  </li>
                  <li>
                    <p>
                      <a
                        href="https://www.rdocumentation.org/packages/base/versions/3.4.1/topics/jitter"
                        >Adding jitter</a
                      >
                      when there are too many points sitting on top of each
                      other:
                      <code>plot(evals$score ~ jitter(evals$bty_avg))</code> or
                      <code>plot(jitter(evals$score) ~ evals$bty_avg)</code>
                    </p>
                  </li>
                  <li>
                    <p>
                      Margins: <code>mar</code> for margin size in lines,
                      <code>mai</code> for margin size in inches. For complete
                      information on margins, see
                      <a
                        href="http://research.stowers.org/mcm/efg/R/Graphics/Basics/mar-oma/index.htm"
                        >Earl Glynn’s margins tutorial</a
                      >.
                    </p>
                  </li>
                  <li>
                    <p>Axes</p>
                    <ul>
                      <li>
                        <strong
                          >You cannot change axis tick mark thickness within the
                          default graphing capabilities.</strong
                        >
                        Instead:
                        <ul>
                          <li>Suppress all axes with axes=FALSE</li>
                          <li>
                            Re-create the axes with the axis() command. Each y
                            and x axis will get its own command. lwd= will
                            control line width, or thickness, for both the axis
                            line and the tick marks.
                          </li>
                          <li>
                            This will only create partial axes that encompass
                            the data. To make the axes meet at 0, also create a
                            box around the graph with
                            <code>box(which = 'plot', bty = 'l')</code>
                          </li>
                          <li>
                            The default axes will usually detect categorical
                            data and label axes accordingly, but categories will
                            need to be specified for axes created separately.
                          </li>
                          <li>
                            See info from Quick-R and R in Action Ch. 3.4 for
                            more info.
                          </li>
                        </ul>
                      </li>
                      <li>
                        Adding symbols to axis titles (such as mu):
                        <code
                          >plot([options],
                          ylab=expression(bold(paste('Thickness, ',mu,
                          'm'))))</code
                        >
                      </li>
                      <li>Axis labels: <code>xlab=, ylab=</code></li>
                      <li>
                        If a custom axis is created, the automatically generated
                        axis should be suppressed.
                        <code>xaxt='n'</code> suppresses the x axis,
                        <code>yaxt='n'</code> suppresses the y axis, and
                        <code>axes=FALSE</code> suppresses both x and y axes.
                      </li>
                      <li>
                        Rotated axis labels in base graphics
                        <ul>
                          <li>
                            To rotate axis labels (using base graphics), you
                            need to use <code>text()</code>, rather than
                            <code>mtext()</code>, as the latter does not support
                            <code>par('srt')</code>.
                          </li>
                          <li>
                            When plotting the x axis labels, we use
                            <code>srt = 45</code> for text rotation angle,
                            <code>adj = 1</code> to place the right end of text
                            at the tick marks, and <code>xpd = TRUE</code> to
                            allow for text outside the plot region. You can
                            adjust the value of the <code>0.25offset</code> as
                            required to move the axis labels up or down relative
                            to the x axis. See <code>?par</code> for more
                            information.
                          </li>
                          <li>
                            Also see Figure 1 and associated code in Paul
                            Murrell (2003), ‘Integrating grid Graphics Output
                            with Base Graphics Output’, *R News,- 3/2, 7–12.
                          </li>
                        </ul>
                        <div class="sourceCode" id="cb132">
                          <pre
                            class="sourceCode r"
                          ><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1"></a><span class="co"># Increase bottom margin to make room for rotated labels</span></span>
<span id="cb132-2"><a href="#cb132-2"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">7</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="fl">0.1</span>)</span>
<span id="cb132-3"><a href="#cb132-3"></a><span class="co"># Create plot with no x axis and no x axis label</span></span>
<span id="cb132-4"><a href="#cb132-4"></a><span class="kw">plot</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">8</span>, <span class="dt">xaxt =</span> <span class="st">&#39;n&#39;</span>,  <span class="dt">xlab =</span> <span class="st">&#39;&#39;</span>)</span>
<span id="cb132-5"><a href="#cb132-5"></a><span class="co"># Set up x axis with tick marks alone</span></span>
<span id="cb132-6"><a href="#cb132-6"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">labels =</span> <span class="ot">FALSE</span>)</span>
<span id="cb132-7"><a href="#cb132-7"></a><span class="co"># Create some text labels</span></span>
<span id="cb132-8"><a href="#cb132-8"></a>labels &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&#39;Label&#39;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>, <span class="dt">sep =</span> <span class="st">&#39; &#39;</span>)</span>
<span id="cb132-9"><a href="#cb132-9"></a><span class="co"># Plot x axis labels at default tick marks</span></span>
<span id="cb132-10"><a href="#cb132-10"></a><span class="kw">text</span>(</span>
<span id="cb132-11"><a href="#cb132-11"></a>  <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>,</span>
<span id="cb132-12"><a href="#cb132-12"></a>  <span class="kw">par</span>(<span class="st">&#39;usr&#39;</span>)[<span class="dv">3</span>] <span class="op">-</span><span class="st"> </span><span class="fl">0.25</span>,</span>
<span id="cb132-13"><a href="#cb132-13"></a>  <span class="dt">srt =</span> <span class="dv">45</span>,</span>
<span id="cb132-14"><a href="#cb132-14"></a>  <span class="dt">adj =</span> <span class="dv">1</span>,</span>
<span id="cb132-15"><a href="#cb132-15"></a>  <span class="dt">labels =</span> labels,</span>
<span id="cb132-16"><a href="#cb132-16"></a>  <span class="dt">xpd =</span> <span class="ot">TRUE</span></span>
<span id="cb132-17"><a href="#cb132-17"></a>)</span>
<span id="cb132-18"><a href="#cb132-18"></a><span class="co"># Plot x axis label at line 6 (of 7)</span></span>
<span id="cb132-19"><a href="#cb132-19"></a><span class="kw">mtext</span>(<span class="dv">1</span>, <span class="dt">text =</span> <span class="st">&#39;X Axis Label&#39;</span>, <span class="dt">line =</span> <span class="dv">6</span>)</span></code></pre>
                        </div>
                        <img
                          src="index_files/figure-html/index-plots-params-01-1.png"
                          width="672"
                        />
                      </li>
                    </ul>
                  </li>
                  <li>
                    <p>Create <strong>figure panels</strong> with</p>
                    <div class="sourceCode" id="cb133">
                      <pre
                        class="sourceCode r"
                      ><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1"></a><span class="kw">layout</span>()</span>
<span id="cb133-2"><a href="#cb133-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span>)</span>
<span id="cb133-3"><a href="#cb133-3"></a><span class="kw">par</span>(mfcol)</span>
<span id="cb133-4"><a href="#cb133-4"></a>split.screen</span></code></pre>
                    </div>
                    <ul>
                      <li>
                        See R help <code>help(par)</code>, An Introduction to R
                        12.5.4 p.73, Muenchen_2009_R for SAS and SPSS Users Ch.
                        21.3 for more info.
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
            <div id="ggplot2" class="section level3">
              <h3>ggplot2</h3>
              <p>
                The ggplot2 package can generate advanced graphs (see ggplot2
                documentation
                <a href="http://ggplot2.tidyverse.org/reference/">here</a> and
                <a
                  href="https://www.rdocumentation.org/packages/ggplot2/versions/2.2.1"
                  >here</a
                >).
              </p>
              <ul>
                <li>Basic quick plot: <code>qplot()</code></li>
                <li>
                  Build a plot progressively by repeatedly calling
                  <code>ggplot()</code>.
                </li>
                <li>
                  Other comments
                  <ul>
                    <li>
                      As described above, I prefer to use base graphics to
                      create boxplots. For example, boxplots in ggplot2 do not
                      have endcaps on the whiskers. It is possible to add them,
                      but they are abnormally wide.
                    </li>
                  </ul>
                </li>
              </ul>
              <p>
                <a href="http://www.ggobi.org/docs/">ggobi</a> is a data
                visualization program that interfaces with R.
              </p>
              <p>
                <a href="https://plot.ly/d3-js-for-r-and-shiny-charts/"
                  >plotly</a
                >
                is an online tool that can interface with R and Shiny. Shiny is
                a nice tool for creating interactive plots and presentations.
              </p>
              <p><a href="#top">back to top</a></p>
            </div>
          </div>
        </div>
      </div>
    </div>

    <script>
      // add bootstrap table styles to pandoc tables
      function bootstrapStylePandocTables() {
        $("tr.header")
          .parent("thead")
          .parent("table")
          .addClass("table table-condensed")
      }
      $(document).ready(function () {
        bootstrapStylePandocTables()
      })
    </script>

    <!-- tabsets -->

    <script>
      $(document).ready(function () {
        window.buildTabsets("TOC")
      })

      $(document).ready(function () {
        $(".tabset-dropdown > .nav-tabs > li").click(function () {
          $(this).parent().toggleClass("nav-tabs-open")
        })
      })
    </script>

    <!-- code folding -->

    <script>
      $(document).ready(function () {
        // move toc-ignore selectors from section div to header
        $("div.section.toc-ignore")
          .removeClass("toc-ignore")
          .children("h1,h2,h3,h4,h5")
          .addClass("toc-ignore")

        // establish options
        var options = {
          selectors: "h1,h2,h3",
          theme: "bootstrap3",
          context: ".toc-content",
          hashGenerator: function (text) {
            return text.replace(/[.\\/?&!#<>]/g, "").replace(/\s/g, "_")
          },
          ignoreSelector: ".toc-ignore",
          scrollTo: 0,
        }
        options.showAndHide = true
        options.smoothScroll = true

        // tocify
        var toc = $("#TOC").tocify(options).data("toc-tocify")
      })
    </script>

    <!-- dynamically load mathjax for compatibility with self-contained -->
    <script>
      ;(function () {
        var script = document.createElement("script")
        script.type = "text/javascript"
        script.src =
          "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"
        document.getElementsByTagName("head")[0].appendChild(script)
      })()
    </script>
  </body>
</html>
